// @splunk/themes/prisma/light.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _tinycolor = _interopRequireDefault(require("tinycolor2"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
/**
 * ## Background colors
 * Background colors should be used only for backgrounds of higher level sections & containers of a UI.
 *
 * @colorSet verbose
 */

var backgroundColors = {
  backgroundColorPopup: '#ffffff',
  backgroundColorSection: '#ffffff',
  backgroundColorSidebar: '#f8f8f8',
  backgroundColorPage: '#f9f9f9',
  backgroundColorNavigation: '#f7f7f7',
  backgroundColorFloating: '#000000',
  backgroundColorDialog: '#ffffff',
  backgroundColorScrim: 'rgba(255, 255, 255, 0.75)'
};

/**
 * ## Content colors
 * Content colors should be used for text, icons and dividers.
 *
 * @colorSet verbose
 */

var contentColors = {
  contentColorActive: '#2c2c2c',
  contentColorDefault: '#4d4d4d',
  contentColorDisabled: '#878787',
  contentColorInverted: '#ffffff',
  contentColorMuted: '#6b6b6b'
};

/**
 * ## Interactive colors
 * Interactive colors are specifically chosen for borders and backgrounds of controls and other interactive content.
 * "Overlay" colors are intended to be placed over the default background color, such as interactiveColorPrimary.
 * If the default background color is not transparent, the `blend` mixin can be used to create a new color that combines the two.
 *
 * @colorSet verbose
 */

var interactiveColors = {
  interactiveColorPrimary: '#0264d7',
  interactiveColorBorder: 'rgba(0, 0, 0, 0.48)',
  interactiveColorBorderActive: 'rgba(0, 0, 0, 0.5)',
  interactiveColorBorderHover: 'rgba(0, 0, 0, 0.6)',
  interactiveColorBorderDisabled: 'rgba(0, 0, 0, 0.3)',
  interactiveColorOverlaySelected: 'rgba(0, 0, 0, 0.04)',
  interactiveColorOverlayHover: 'rgba(0, 0, 0, 0.03)',
  interactiveColorOverlayActive: 'rgba(0, 0, 0, 0.07)',
  interactiveColorOverlayDrag: 'rgba(2, 100, 215, 0.16)',
  interactiveColorBackground: '#eeeeee',
  interactiveColorBackgroundDisabled: 'rgba(0, 0, 0, 0.1)'
};

/**
 * ## Neutral colors
 * Neutrals are used for dividers and as backup colors that can sparingly be used for cases, when the other defined colors are not enough.
 *
 * @colorSet verbose
 */

var neutralColors = {
  white: '#ffffff',
  neutral100: '#f0f0f0',
  neutral200: '#e6e6e6',
  neutral300: '#dddddd',
  neutral400: '#cacaca',
  neutral500: '#b8b8b8',
  black: '#000000'
};

/**
 * ## Status colors
 * Status colors are reserved for communicating urgency and severity associated with data objects and to categorize the visual communication of system responses.
 *
 * @colorSet verbose
 */
var statusColors = {
  statusColorInfo: '#006be5',
  statusColorInfoWeak: (0, _tinycolor["default"])('#006be5').lighten(10).toHexString(),
  statusColorInfoStrong: (0, _tinycolor["default"])('#006be5').darken(10).toHexString(),
  statusColorNormal: '#407a06',
  statusColorNormalWeak: (0, _tinycolor["default"])('#407a06').lighten(10).toHexString(),
  statusColorNormalStrong: (0, _tinycolor["default"])('#407a06').darken(10).toHexString(),
  statusColorLow: '#155a4e',
  statusColorLowWeak: (0, _tinycolor["default"])('#155a4e').lighten(10).toHexString(),
  statusColorLowStrong: (0, _tinycolor["default"])('#155a4e').darken(10).toHexString(),
  statusColorMedium: '#c97705',
  statusColorMediumWeak: (0, _tinycolor["default"])('#c97705').lighten(10).toHexString(),
  statusColorMediumStrong: (0, _tinycolor["default"])('#c97705').darken(10).toHexString(),
  statusColorHigh: '#e00000',
  statusColorHighWeak: (0, _tinycolor["default"])('#e00000').lighten(10).toHexString(),
  statusColorHighStrong: (0, _tinycolor["default"])('#e00000').darken(10).toHexString(),
  statusColorCritical: '#9e1534',
  statusColorCriticalWeak: (0, _tinycolor["default"])('#9e1534').lighten(10).toHexString(),
  statusColorCriticalStrong: (0, _tinycolor["default"])('#9e1534').darken(10).toHexString()
};

/**
 * ## Accent colors
 * Accent colors aid and categorize the visual communication of the system response.
 *
 * @colorSet verbose
 */
var accentColors = {
  accentColorPositive: statusColors.statusColorNormal,
  accentColorWarning: statusColors.statusColorMedium,
  accentColorAlert: '#c6400d',
  accentColorNegative: statusColors.statusColorHigh
};

/**
 * ## Elevation shadows
 *
 * @shadowSet
 *
 */
var elevationShadows = {
  embossShadow: ' 0px 1px 5px rgba(0, 0, 0, 0.07), 0px 0px 1px rgba(0, 0, 0, 0.07)',
  overlayShadow: '0px 26px 103px rgba(0, 0, 0, 0.13), 0px 11px 18px rgba(0, 0, 0, 0.06), 0px 3px 6px rgba(0, 0, 0, 0.06)',
  dragShadow: '0px 26px 103px rgba(0, 0, 0, 0.13), 0px 11px 18px rgba(0, 0, 0, 0.06), 0px 3px 6px rgba(0, 0, 0, 0.06)',
  modalShadow: '0px 50px 200px rgba(0, 0, 0, 0.3), 0px 29px 66px rgba(0, 0, 0, 0.08), 0px 29px 47px rgba(0, 0, 0, 0.08), 0px 5px 10px rgba(0, 0, 0, 0.03)'
};

/**
 * ## Syntax colors
 * Syntax colors are used only for code blocks.
 *
 * @colorSet verbose alphabetical
 */
var syntaxColors = {
  syntaxBlue: '#0f7594',
  syntaxBrown: '#9f6404',
  syntaxGray: '#6b6b6b',
  syntaxGreen: '#5c780c',
  syntaxOrange: '#ba4f08',
  syntaxPink: '#cc15ae',
  syntaxPurple: '#7b4df5',
  syntaxRed: '#db1e47',
  syntaxTeal: '#1d7c6b'
};
var theme = _objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread({}, accentColors), statusColors), elevationShadows), backgroundColors), contentColors), neutralColors), interactiveColors), syntaxColors);
var _default = theme;
exports["default"] = _default;
