// @splunk/react-ui/Accordion.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var t = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, t) => {
            /******/ for (var r in t) {
                /******/ if (e.o(t, r) && !e.o(n, r)) {
                    /******/ Object.defineProperty(n, r, {
                        enumerable: true,
                        get: t[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        Panel: () => /* reexport */ w,
        default: () => /* reexport */ Q
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var r =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var a =  e.n(o);
    // CONCATENATED MODULE: external "lodash/has"
    const i = require("lodash/has");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "lodash/keys"
    const u = require("lodash/keys");
    var c =  e.n(u);
    // CONCATENATED MODULE: external "lodash/omit"
    const s = require("lodash/omit");
    var p =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/react-ui/CollapsiblePanel"
    const f = require("@splunk/react-ui/CollapsiblePanel");
    var d =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/themes"
    const y = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Accordion/AccordionContext.tsx
    var v =  (0, t.createContext)({});
    v.displayName = "Accordion";
    /* harmony default export */    const b = v;
    // CONCATENATED MODULE: ./src/Accordion/Panel.tsx
    function m() {
        return m = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, m.apply(null, arguments);
    }
    function h(e, n) {
        if (null == e) return {};
        var t, r, o = g(e, n);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (r = 0; r < a.length; r++) {
                t = a[r], n.includes(t) || {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function g(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (n.includes(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    var P = {
        description: a().string,
        elementRef: a().oneOfType([ a().func, a().object ]),
        inset: a().bool,
        panelId: a().any.isRequired,
        title: a().node.isRequired
    };
    var O = function e(n, t) {
        var r = {
            enterprise: {
                compact: "8px",
                comfortable: "10px"
            },
            prisma: {
                compact: "12px 20px",
                comfortable: "20px"
            }
        };
        return r[n][t];
    };
    var j = {
        flexShrink: 1
    };
    /**
 * `Accordion.Panel` operates as a container component for content in an `Accordion`.
 */    function C(e) {
        var n = e.panelId, o = e.inset, a = h(e, [ "panelId", "inset" ]);
        // @docs-props-type AccordionPanelPropsBase
                var i = (0, y.useSplunkTheme)(), l = i.family, u = i.density;
        var c = (0, t.useContext)(v), s = c.onChange, p = c.openPanelId, f = c.inset;
        var b = p === n;
        var g = o !== undefined ? o : f;
        // Props are different based on controlled/uncontrolled
        // For controlled, onChange required prop for controlled cases and must be defined.
                var P = s ? {
            onChange: s,
            open: b
        } : {
            onChange: undefined
        };
        var C = function e() {
            if (g) {
                var n = O(l, u);
                return {
                    padding: n
                };
            }
            return undefined;
        };
        
        return r().createElement(d(), m({
            panelId: n,
            "data-test": "panel",
            innerBodyStyles: C(),
            "data-test-panel-id": n,
            style: j
        }, a, P));
    }
    C.propTypes = P;
    /* harmony default export */    const w = C;
    // CONCATENATED MODULE: external "styled-components"
    const S = require("styled-components");
    var I =  e.n(S);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const x = require("@splunk/react-ui/Box");
    var _ =  e.n(x);
    // CONCATENATED MODULE: ./src/Accordion/AccordionStyles.ts
    var k = I()(_()).withConfig({
        displayName: "AccordionStyles__StyledBox",
        componentId: "m39hyh-0"
    })([ "", ";flex-direction:column;margin-bottom:0;", "" ], y.mixins.reset("flex"), (0, 
    y.pick)({
        enterprise: (0, S.css)([ "background:", ";" ], y.variables.backgroundColor)
    }));
    // CONCATENATED MODULE: ./src/Accordion/Accordion.tsx
    function E(e) {
        "@babel/helpers - typeof";
        return E = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, E(e);
    }
    function T() {
        return T = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, T.apply(null, arguments);
    }
    function q(e, n) {
        if (!(e instanceof n)) throw new TypeError("Cannot call a class as a function");
    }
    function R(e, n) {
        for (var t = 0; t < n.length; t++) {
            var r = n[t];
            r.enumerable = r.enumerable || !1, r.configurable = !0, "value" in r && (r.writable = !0), 
            Object.defineProperty(e, G(r.key), r);
        }
    }
    function B(e, n, t) {
        return n && R(e.prototype, n), t && R(e, t), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function A(e, n) {
        if ("function" != typeof n && null !== n) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(n && n.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), n && M(e, n);
    }
    function M(e, n) {
        return M = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, n) {
            return e.__proto__ = n, e;
        }, M(e, n);
    }
    function N(e) {
        var n = V();
        return function() {
            var t, r = z(e);
            if (n) {
                var o = z(this).constructor;
                t = Reflect.construct(r, arguments, o);
            } else t = r.apply(this, arguments);
            return D(this, t);
        };
    }
    function D(e, n) {
        if (n && ("object" == E(n) || "function" == typeof n)) return n;
        if (void 0 !== n) throw new TypeError("Derived constructors may only return object or undefined");
        return U(e);
    }
    function U(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function V() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return (V = function n() {
            return !!e;
        })();
    }
    function z(e) {
        return z = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, z(e);
    }
    function F(e, n, t) {
        return (n = G(n)) in e ? Object.defineProperty(e, n, {
            value: t,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[n] = t, e;
    }
    function G(e) {
        var n = H(e, "string");
        return "symbol" == E(n) ? n : n + "";
    }
    function H(e, n) {
        if ("object" != E(e) || !e) return e;
        var t = e[Symbol.toPrimitive];
        if (void 0 !== t) {
            var r = t.call(e, n || "default");
            if ("object" != E(r)) return r;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === n ? String : Number)(e);
    }
    /** @public */    var J = {
        children: a().node,
        defaultOpenPanelId: a().any,
        elementRef: a().oneOfType([ a().func, a().object ]),
        inset: a().bool,
        onChange: a().func,
        openPanelId: a().any
    };
    var K = {
        inset: false
    };
    var L =  function(e) {
        A(o, e);
        var n = N(o);
        // @docs-props-type AccordionPropsBase
                function o(e) {
            var t;
            q(this, o);
            t = n.call(this, e);
            F(U(t), "controlledExternally", void 0);
            F(U(t), "getCurrentOpenPanel", (function() {
                var e = t.isControlled() ? t.props.openPanelId : t.state.openPanelId;
                return e;
            }));
            F(U(t), "handleChange", (function(e, n) {
                var r, o;
                var a = n.panelId, i = n.reason;
                var l = t.getCurrentOpenPanel();
                var u = l === a ? undefined : a;
                if (!t.isControlled()) {
                    t.setState({
                        openPanelId: u
                    });
                }
                (r = (o = t.props).onChange) === null || r === void 0 ? void 0 : r.call(o, e, {
                    event: e,
                    panelId: u,
                    reason: i
                });
            }));
            t.controlledExternally = l()(e, "openPanelId");
            if (!t.isControlled()) {
                t.state = {
                    openPanelId: e.defaultOpenPanelId
                };
            }
            if (false) {}
            return t;
        }
        B(o, [ {
            key: "componentDidUpdate",
            value: function e(n) {
                if (false) {}
                if (false) {}
            }
        }, {
            key: "isControlled",
            value: function e() {
                return this.controlledExternally;
            }
        }, {
            key: "render",
            value: function e() {
                var n = this.props, a = n.children, i = n.elementRef, l = n.inset;
                var u = this.isControlled() ? this.props.openPanelId : this.state.openPanelId;
                var s = t.Children.toArray(a).filter(t.isValidElement);
 // remove false, null, 0, etc
                                
                return r().createElement(k, T({
                    "data-test": "accordion",
                    "data-test-open-panel-id": u,
                    elementRef: i
                }, p()(this.props, c()(o.propTypes))),  r().createElement(b.Provider, {
                    // eslint-disable-next-line react/jsx-no-constructed-context-values
                    value: {
                        onChange: this.handleChange,
                        openPanelId: u,
                        inset: l
                    }
                }, s));
            }
        } ]);
        return o;
    }(t.Component);
    F(L, "propTypes", J);
    F(L, "defaultProps", K);
    F(L, "Panel", w);
    /* harmony default export */    const Q = L;
    // CONCATENATED MODULE: ./src/Accordion/index.ts
    module.exports = n;
    /******/})();
