// @splunk/react-ui/Animation.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        Spring: () => /* reexport */ p,
        Transition: () => /* reexport */ y
    });
    // CONCATENATED MODULE: external "react-spring"
    const t = require("react-spring");
    // CONCATENATED MODULE: external "@splunk/react-ui/AnimationToggle"
    const n = require("@splunk/react-ui/AnimationToggle");
    // CONCATENATED MODULE: ./src/Animation/Animation.tsx
    function o(e) {
        "@babel/helpers - typeof";
        return o = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, o(e);
    }
    function i(e, r) {
        var t = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            r && (n = n.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }))), t.push.apply(t, n);
        }
        return t;
    }
    function u(e) {
        for (var r = 1; r < arguments.length; r++) {
            var t = null != arguments[r] ? arguments[r] : {};
            r % 2 ? i(Object(t), !0).forEach((function(r) {
                c(e, r, t[r]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : i(Object(t)).forEach((function(r) {
                Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
            }));
        }
        return e;
    }
    function c(e, r, t) {
        return (r = l(r)) in e ? Object.defineProperty(e, r, {
            value: t,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = t, e;
    }
    function l(e) {
        var r = a(e, "string");
        return "symbol" == o(r) ? r : r + "";
    }
    function a(e, r) {
        if ("object" != o(e) || !e) return e;
        var t = e[Symbol.toPrimitive];
        if (void 0 !== t) {
            var n = t.call(e, r || "default");
            if ("object" != o(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    function f(e, r) {
        if (null == e) return {};
        var t, n, o = b(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                t = i[n], r.includes(t) || {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function b(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (r.includes(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    /**
 * Wrappers to use in place of `Spring` and `Transition` from `react-spring`.
 * Uses `useAnimationToggle` to offer animation control.
 */    function p(e) {
        var r = e.children, o = e.immediate, i = f(e, [ "children", "immediate" ]);
        var c = (0, n.useAnimationToggle)();
        return r((0, t.useSpring)(u({
            immediate: o || c !== "on"
        }, i)));
    }
    function y(e) {
        var r = e.items, o = e.children, i = e.immediate, c = f(e, [ "items", "children", "immediate" ]);
        var l = (0, n.useAnimationToggle)();
        return (0, t.useTransition)(r, u({
            immediate: i || l !== "on"
        }, c))(o);
    }
    // CONCATENATED MODULE: ./src/Animation/index.ts
    module.exports = r;
    /******/})();
