// @splunk/react-ui/Box.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var t = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, t) => {
            /******/ for (var r in t) {
                /******/ if (e.o(t, r) && !e.o(n, r)) {
                    /******/ Object.defineProperty(n, r, {
                        enumerable: true,
                        get: t[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        default: () => /* reexport */ v
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var r =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const l = require("prop-types");
    var i =  e.n(l);
    // CONCATENATED MODULE: external "styled-components"
    const a = require("styled-components");
    var o =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/themes"
    const d = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Box/BoxStyles.ts
    var f = o().div.withConfig({
        displayName: "BoxStyles__Styled",
        componentId: "sc-1h4b5f6-0"
    })([ "", ";align-items:stretch;align-content:stretch;flex-flow:row nowrap;flex-grow:0;flex-shrink:0;justify-content:flex-start;width:auto;max-width:100%;@media all and (-ms-high-contrast:none){max-width:calc(100% - 0.01px);}&[data-inline]{display:inline-block;vertical-align:middle;}&[data-flex]{display:flex;}&[data-flex][data-inline]{display:inline-flex;}" ], d.mixins.reset("block"));
    // CONCATENATED MODULE: ./src/Box/Box.tsx
    function s() {
        return s = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, s.apply(null, arguments);
    }
    function c(e, n) {
        if (null == e) return {};
        var t, r, l = u(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                t = i[r], n.includes(t) || {}.propertyIsEnumerable.call(e, t) && (l[t] = e[t]);
            }
        }
        return l;
    }
    function u(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (n.includes(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    var p = {
        children: i().node,
        elementRef: i().oneOfType([ i().func, i().object ]),
        flex: i().bool,
        inline: i().bool
    };
    /**
 * Box is a utility component to normalize and abstract common layout styles.
 */    function y(e) {
        var n = e.children, t = e.elementRef, l = e.flex, i = l === void 0 ? false : l, a = e.tag, o = a === void 0 ? "div" : a, d = e.inline, u = d === void 0 ? false : d, p = c(e, [ "children", "elementRef", "flex", "tag", "inline" ]);
        
        return r().createElement(f, s({
            as: o,
            ref: t
        }, p, {
            "data-inline": u || undefined,
            "data-flex": i || undefined
        }), n);
    }
    y.propTypes = p;
    /* harmony default export */    const v = y;
    // CONCATENATED MODULE: ./src/Box/index.ts
    module.exports = n;
    /******/})();
