// @splunk/react-ui/Breadcrumbs.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var n = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, n) => {
            /******/ for (var t in n) {
                /******/ if (e.o(n, t) && !e.o(r, t)) {
                    /******/ Object.defineProperty(r, t, {
                        enumerable: true,
                        get: n[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        BreadcrumbsContext: () => /* reexport */ _,
        Item: () => /* reexport */ C,
        default: () => /* reexport */ E
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var t =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var l =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const o = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "styled-components"
    const i = require("styled-components");
    var c =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/themes"
    const u = require("@splunk/themes");
    // CONCATENATED MODULE: external "@splunk/react-ui/Link"
    const s = require("@splunk/react-ui/Link");
    var d =  e.n(s);
    // CONCATENATED MODULE: ./src/Breadcrumbs/ItemStyles.ts
    var f = c()(d()).withConfig({
        displayName: "ItemStyles__StyledLink",
        componentId: "sc-1ywtawf-0"
    })([ "display:flex;align-items:center;gap:", ";white-space:nowrap;color:", ";&:where(:hover,:focus):not([disabled],[aria-disabled='true']){color:", ";}" ], u.variables.spacingXSmall, u.variables.contentColorDefault, u.variables.contentColorActive);
    // CONCATENATED MODULE: ./src/Breadcrumbs/Item.tsx
    function b() {
        return b = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, b.apply(null, arguments);
    }
    function p(e, r) {
        if (null == e) return {};
        var n, t, a = m(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                n = l[t], r.includes(n) || {}.propertyIsEnumerable.call(e, n) && (a[n] = e[n]);
            }
        }
        return a;
    }
    function m(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (r.includes(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    var v = Object.freeze({
        allowDisabledLink: true,
        disabled: true,
        to: ""
    });
    var y = {
        elementRef: l().oneOfType([ l().func, l().object ]),
        enableCurrentPage: l().bool,
        endAdornment: l().node,
        isCurrent: l().bool,
        label: l().string.isRequired,
        onClick: l().func,
        startAdornment: l().node,
        to: l().string.isRequired
    };
    function g(e) {
        var r = e.enableCurrentPage, a = e.endAdornment, l = e.isCurrent, o = e.label, i = e.onClick, c = e.startAdornment, u = e.to, s = p(e, [ "enableCurrentPage", "endAdornment", "isCurrent", "label", "onClick", "startAdornment", "to" ]);
        // @docs-props-type ItemPropsBase
                var d = (0, n.useContext)(_), m = d.onClick;
        var y = {
            to: u
        };
        var g = (0, n.useCallback)((function(e) {
            i === null || i === void 0 ? void 0 : i(e, {
                label: o,
                to: u
            });
            m === null || m === void 0 ? void 0 : m(e, {
                label: o,
                to: u
            });
        }), [ i, m, o, u ]);
        if (l) {
            y["aria-current"] = "page";
            if (!r) {
                Object.assign(y, v);
            }
        }
        var C = i || m ? {
            onClick: g
        } : {};
        
        return t().createElement(f, b({
            "data-test": "item"
        }, C, y, s), c && c, o, a && a);
    }
    g.propTypes = y;
    /* harmony default export */    const C = g;
    // CONCATENATED MODULE: ./src/Breadcrumbs/BreadcrumbsStyles.ts
    var h = c().ol.withConfig({
        displayName: "BreadcrumbsStyles__StyledList",
        componentId: "sc-1maeyfk-0"
    })([ "", " flex-wrap:wrap;" ], u.mixins.reset("flex"));
    var O = c().li.withConfig({
        displayName: "BreadcrumbsStyles__StyledListItem",
        componentId: "sc-1maeyfk-1"
    })([ "display:flex;min-width:max-content;a[aria-current]{font-weight:", ";color:", ";cursor:default;}" ], u.variables.fontWeightSemiBold, u.variables.contentColorActive);
    var w = c().span.withConfig({
        displayName: "BreadcrumbsStyles__StyledSeparator",
        componentId: "sc-1maeyfk-2"
    })([ "cursor:default;padding:0 ", ";color:", ";" ], u.variables.spacingSmall, u.variables.contentColorDefault);
    // CONCATENATED MODULE: ./src/Breadcrumbs/Breadcrumbs.tsx
    function S() {
        return S = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, S.apply(null, arguments);
    }
    function k(e, r) {
        if (null == e) return {};
        var n, t, a = P(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                n = l[t], r.includes(n) || {}.propertyIsEnumerable.call(e, n) && (a[n] = e[n]);
            }
        }
        return a;
    }
    function P(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (r.includes(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    /** @public */    var j = {
        children: l().node.isRequired,
        elementRef: l().oneOfType([ l().func, l().object ]),
        enableCurrentPage: l().bool,
        onClick: l().func
    };
    var _ =  t().createContext({});
    function x(e) {
        var r = e.children, a = e.elementRef, l = e.enableCurrentPage, i = e.onClick, c = k(e, [ "children", "elementRef", "enableCurrentPage", "onClick" ]);
        // @docs-props-type BreadcrumbsPropsBase
                var u;
        var s = n.Children.toArray(r).filter(n.isValidElement).map((function(e, a) {
            if (false) {}
            if (false) {}
            // Set the last child as the current Breadcrumbs.Item
                        var o = a === n.Children.count(r) - 1;
            var i =  (0, n.cloneElement)(e, {
                isCurrent: o,
                enableCurrentPage: l
            });
            
            return t().createElement(O, {
                key: e.props.to
            }, i, !o &&  t().createElement(w, {
                "aria-hidden": "true"
            }, "/"));
        }));
        var d = (0, n.useMemo)((function() {
            return {
                onClick: i
            };
        }), [ i ]);
        
        return t().createElement(_.Provider, {
            value: d
        },  t().createElement("nav", S({
            "data-test": "breadcrumbs",
            "aria-label": (0, o._)("Breadcrumbs"),
            ref: a
        }, c),  t().createElement(h, null, s)));
    }
    x.propTypes = j;
    x.Item = C;
    /* harmony default export */    const E = x;
    // CONCATENATED MODULE: ./src/Breadcrumbs/index.ts
    module.exports = r;
    /******/})();
