// @splunk/react-ui/ButtonSimple.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var r = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ r.n = e => {
            /******/ var a = e && e.__esModule ? 
            /******/ () => e["default"]
            /******/ : () => e
            /******/;
            r.d(a, {
                a
            });
            /******/            return a;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ r.d = (e, a) => {
            /******/ for (var o in a) {
                /******/ if (r.o(a, o) && !r.o(e, o)) {
                    /******/ Object.defineProperty(e, o, {
                        enumerable: true,
                        get: a[o]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ r.o = (r, e) => Object.prototype.hasOwnProperty.call(r, e)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ r.r = r => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(r, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(r, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var e = {};
    // ESM COMPAT FLAG
        r.r(e);
    // EXPORTS
        r.d(e, {
        default: () => /* reexport */ br
    });
    // CONCATENATED MODULE: external "react"
    const a = require("react");
    var o =  r.n(a);
    // CONCATENATED MODULE: external "prop-types"
    const i = require("prop-types");
    var t =  r.n(i);
    // CONCATENATED MODULE: external "@splunk/themes"
    const n = require("@splunk/themes");
    // CONCATENATED MODULE: external "styled-components"
    const l = require("styled-components");
    var s =  r.n(l);
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const c = require("@splunk/react-ui/Clickable");
    var b =  r.n(c);
    // CONCATENATED MODULE: external "lodash/merge"
    const d = require("lodash/merge");
    var v =  r.n(d);
    // CONCATENATED MODULE: ./src/ButtonSimple/mixin.ts
    function u(r) {
        "@babel/helpers - typeof";
        return u = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(r) {
            return typeof r;
        } : function(r) {
            return r && "function" == typeof Symbol && r.constructor === Symbol && r !== Symbol.prototype ? "symbol" : typeof r;
        }, u(r);
    }
    var p = [ "primary", "secondary", "destructive", "subtle" ];
    function f(r) {
        var e = "transparent";
        var a = "transparent";
        var o = (0, n.pick)({
            enterprise: {
                light: n.variables.gray96,
                dark: n.variables.gray30
            },
            prisma: n.variables.interactiveColorBackgroundDisabled
        });
        var i = (0, n.pick)({
            enterprise: {
                dark: n.variables.gray22,
                light: n.variables.gray92
            },
            prisma: n.mixins.overlayColors(n.variables.interactiveColorBackgroundDisabled, n.variables.interactiveColorOverlaySelected)
        });
        var t = "transparent";
        var l = "transparent";
        var s;
        var c;
        var b = n.variables.contentColorActive;
        var d = n.variables.contentColorDisabled;
        switch (r) {
          case "primary":
            {
                e = (0, n.pick)({
                    enterprise: "#1A8929",
                    prisma: n.variables.interactiveColorPrimary
                });
                a = (0, n.pick)({
                    enterprise: n.variables.brandColorD30,
                    prisma: n.mixins.overlayColors(n.variables.interactiveColorPrimary, n.variables.interactiveColorOverlayActive)
                });
                t = (0, n.pick)({
                    enterprise: n.variables.brandColorD50,
                    prisma: n.mixins.overlayColors(n.variables.interactiveColorPrimary, n.variables.interactiveColorOverlayHover)
                });
                l = (0, n.pick)({
                    enterprise: "#235823",
                    prisma: n.mixins.overlayColors(n.variables.interactiveColorPrimary, n.variables.interactiveColorOverlaySelected)
                });
                b = n.variables.contentColorInverted;
                break;
            }

          case "destructive":
            {
                e = n.variables.accentColorNegative;
                a = (0, n.pick)({
                    enterprise: n.variables.errorColorD40,
                    prisma: n.mixins.overlayColors(n.variables.accentColorNegative, n.variables.interactiveColorOverlayActive)
                });
                t = (0, n.pick)({
                    enterprise: n.variables.errorColorD30,
                    prisma: n.mixins.overlayColors(n.variables.accentColorNegative, n.variables.interactiveColorOverlayHover)
                });
                l = (0, n.pick)({
                    enterprise: n.variables.errorColorD30,
                    prisma: n.mixins.overlayColors(n.variables.accentColorNegative, n.variables.interactiveColorOverlaySelected)
                });
                b = n.variables.contentColorInverted;
                break;
            }

          case "secondary":
            s = n.variables.interactiveColorBorder;
            c = n.variables.interactiveColorBorderDisabled;
            e = (0, n.pick)({
                enterprise: {
                    light: n.variables.gray98,
                    dark: n.variables.gray45
                },
                prisma: n.variables.interactiveColorBackground
            });
            a = (0, n.pick)({
                enterprise: {
                    light: n.variables.gray92,
                    dark: n.variables.gray22
                },
                prisma: n.variables.interactiveColorOverlayActive
            });
            t = (0, n.pick)({
                enterprise: {
                    light: "#ebeeef",
                    dark: n.variables.gray30
                },
                prisma: n.variables.interactiveColorOverlayHover
            });
            l = (0, n.pick)({
                enterprise: {
                    light: n.variables.gray92,
                    dark: n.variables.gray22
                },
                prisma: n.variables.interactiveColorOverlaySelected
            });
            break;

          case "subtle":
            // TODO: SUI-6304 this should be interactiveColorAccent once that variable is introduced in SUI 5
            b = n.variables.interactiveColorPrimary;
            a = (0, n.pick)({
                enterprise: {
                    light: n.variables.gray92,
                    dark: n.variables.gray22
                },
                prisma: n.variables.interactiveColorOverlayActive
            });
            t = (0, n.pick)({
                enterprise: {
                    light: n.variables.linkColor,
                    dark: n.variables.white
                },
                prisma: n.variables.interactiveColorOverlayHover
            });
            l = (0, n.pick)({
                enterprise: "transparent",
                prisma: n.variables.interactiveColorOverlaySelected
            });
            break;

          default:
        }
        var v = {
            background: e,
            backgroundActive: a,
            backgroundDisabled: o,
            backgroundDisabledSelected: i,
            backgroundHover: t,
            backgroundSelected: l,
            borderColor: s,
            borderColorDisabled: c,
            color: b,
            colorDisabled: d
        };
        return v;
    }
    function g(r, e) {
        var a = typeof r === "string" && p.includes(r) ? r : undefined;
        var o = {};
        if (a && e !== undefined) {
            o = e;
        } else if (a === undefined && u(r) === "object" && e === undefined) {
            o = r;
        }
        var i = f(a);
        var t = v()(i, o), s = t.background, c = t.backgroundActive, b = t.backgroundDisabled, d = t.backgroundDisabledSelected, g = t.backgroundHover, y = t.backgroundSelected, k = t.borderColor, h = t.borderColorDisabled, m = t.color, C = t.colorDisabled;
        var x = s != null && s !== "transparent";
        return function() {
            return (0, l.css)([ "border-radius:", ";", ";", " ", " cursor:pointer;position:relative;&:focus,&:hover{z-index:3;}&:not([disabled],[aria-disabled='true']){background-color:", ";color:", ";transition:background-color 0.2s,box-shadow 0.2s,text-decoration 0.2s;&:hover{background-color:", ";}&:active{background-color:", ";transition:none;}", "}&:not([disabled]){&:focus{box-shadow:", ";}}&[disabled],&[aria-disabled='true']{border-color:", ";color:", ";", "}" ], n.variables.borderRadius, (function() {
                return k && (0, l.css)([ "border:1px solid ", ";" ], k);
            }), (function(r) {
                var e = r.$append;
                return e && (0, l.css)([ "border-top-right-radius:0.1px;border-bottom-right-radius:0.1px;border-right:none;" ]);
            }), (function(r) {
                var e = r.$prepend;
                return e && (0, l.css)([ "border-top-left-radius:0.1px;border-bottom-left-radius:0.1px;" ]);
            }), s, m, g, c, (function(r) {
                var e = r.$selected;
                return e && (0, l.css)([ "background-color:", ";" ], y);
            }), n.variables.focusShadow, h, C, (function(r) {
                var e = r.$selected;
                return x && (0, l.css)([ "background-color:", ";" ], e ? d : b);
            }));
        };
    }
    /* harmony default export */    const y = g;
    // CONCATENATED MODULE: ./src/ButtonSimple/prismaStyles.ts
    var k = (0, l.css)([ "&:not([disabled],[aria-disabled='true']){&:focus{background-color:", ";}&:hover{border-color:", ";", "}", " ", "}", " &[disabled],&[aria-disabled='true']{", "}" ], n.variables.interactiveColorOverlayHover, n.variables.interactiveColorBorderHover, (0, 
    n.pickVariant)("$variant", {
        secondary: (0, l.css)([ "box-shadow:", ";" ], n.variables.hoverShadow),
        default: (0, l.css)([ "box-shadow:", ";" ], n.variables.hoverShadow),
        subtle: (0, l.css)([ "box-shadow:", ";" ], n.variables.hoverShadow)
    }), (function(r) {
        var e = r.$selected;
        return e && (0, l.css)([ "border-color:", ";&:hover{background-color:", ";}" ], n.variables.interactiveColorBorderDisabled, n.mixins.overlayColors(n.variables.interactiveColorOverlaySelected, n.variables.interactiveColorOverlayHover));
    }), (function(r) {
        var e = r.$error;
        return e && (0, l.css)([ "border-color:", ";&:hover{border-color:", ";}" ], n.variables.accentColorNegative, n.variables.accentColorNegative);
    }), (0, n.pickVariant)("$variant", {
        secondary: (0, l.css)([ "", " font-weight:", ";" ], y("subtle", {
            color: n.variables.contentColorActive
        }), n.variables.fontWeightSemiBold),
        subtle: (0, l.css)([ "", " font-weight:", ";" ], y("subtle", {
            color: n.variables.contentColorActive
        }), n.variables.fontWeightSemiBold),
        default: (0, l.css)([ "", " font-weight:", ";" ], y("secondary", {
            background: "transparent"
        }), n.variables.fontWeightSemiBold),
        toggle: (0, l.css)([ "", " font-weight:normal;" ], y("secondary", {
            background: "transparent"
        }))
    }), (function(r) {
        var e = r.$selected;
        return e && (0, l.css)([ "background-color:", ";" ], n.mixins.overlayColors(n.variables.interactiveColorBackgroundDisabled, n.variables.interactiveColorOverlaySelected));
    }));
    var h = (0, l.css)([ "&:not([disabled],[aria-disabled='true']){&:focus{background-color:", ";}&:hover{box-shadow:", ";}}", " font-weight:", ";" ], (0, 
    n.pickVariant)("$variant", {
        primary: n.mixins.overlayColors(n.variables.interactiveColorPrimary, n.variables.interactiveColorOverlayHover),
        destructive: n.mixins.overlayColors(n.variables.accentColorNegative, n.variables.interactiveColorOverlayHover),
        flat: n.mixins.overlayColors(n.variables.interactiveColorBackground, n.variables.interactiveColorOverlayHover)
    }), n.variables.hoverShadow, (0, n.pickVariant)("$variant", {
        primary: y("primary", {
            // this is intentional - the disabled background had always been applying the $selected style
            // since beccfde78f787861217353e9149ce213bee0270 so keeping this for visual consistency
            backgroundDisabled: n.mixins.overlayColors(n.variables.interactiveColorBackgroundDisabled, n.variables.interactiveColorOverlaySelected)
        }),
        destructive: y("destructive", {
            // this is intentional - the disabled background had always been applying the $selected style
            // since beccfde78f787861217353e9149ce213bee0270 so keeping this for visual consistency
            backgroundDisabled: n.mixins.overlayColors(n.variables.interactiveColorBackgroundDisabled, n.variables.interactiveColorOverlaySelected)
        }),
        flat: y({
            // this is intentional - the background had always been applying the $selected style
            // since beccfde78f787861217353e9149ce213bee0270 so keeping this for visual consistency
            background: n.mixins.overlayColors(n.variables.interactiveColorBackground, n.variables.interactiveColorOverlaySelected),
            backgroundActive: n.mixins.overlayColors(n.variables.interactiveColorBackground, n.variables.interactiveColorOverlayActive),
            // this is intentional - the disabled background had always been applying the $selected style
            // since beccfde78f787861217353e9149ce213bee0270 so keeping this for visual consistency
            backgroundDisabled: n.mixins.overlayColors(n.variables.interactiveColorBackgroundDisabled, n.variables.interactiveColorOverlaySelected),
            backgroundHover: n.mixins.overlayColors(n.variables.interactiveColorBackground, n.variables.interactiveColorOverlayHover),
            backgroundSelected: n.mixins.overlayColors(n.variables.interactiveColorBackground, n.variables.interactiveColorOverlaySelected)
        })
    }), n.variables.fontWeightSemiBold);
    // CONCATENATED MODULE: ./src/ButtonSimple/enterpriseStyles.ts
    var m = "#1A8929";
 // SUI-2439 to meet WCAG AA compliance
        var C = "#235823";
 // SUI-2439 to meet WCAG AA compliance
        var x = (0, l.css)([ "border-radius:", ";cursor:pointer;position:relative;&:focus{z-index:3;}" ], n.variables.borderRadius);
    var w = (0, l.css)([ "", " ", "" ], (function(r) {
        var e = r.$append;
        return e && (0, l.css)([ "border-top-right-radius:0;border-bottom-right-radius:0;border-right:none;" ]);
    }), (function(r) {
        var e = r.$prepend;
        return e && (0, l.css)([ "border-top-left-radius:0;border-bottom-left-radius:0;" ]);
    }));
    var S = "inset 0 -1px 0 rgba(0, 0, 0, 0.1)";
    var O = "inset 0 1px 0 rgba(0, 0, 0, 0.1)";
    var D = (0, l.css)([ "", ";border:1px solid;border-color:", ";", ";font-weight:", ";&:not([disabled],[aria-disabled='true']){background-color:", ";color:", ";transition:background-image 0.2s,border 0.2s,box-shadow 0.2s,text-decoration 0.2s;", ";", " &:hover{background-color:", ";", ";}&:active{transition:none;background-color:", ";border-color:", ";", ";}&:focus{", " ", "}}&:not([disabled]){&:focus{", "}}&[disabled],&[aria-disabled='true']{color:", ";background-color:", ";border-color:", ";", "}" ], x, (0, 
    n.pick)({
        light: n.variables.gray60,
        dark: n.variables.borderColor
    }), w, (0, n.pickVariant)("$variant", {
        default: "normal",
        secondary: n.variables.fontWeightSemiBold
    }), (0, n.pick)({
        light: n.variables.gray98,
        dark: n.variables.gray45
    }), (0, n.pick)({
        light: n.variables.gray45,
        dark: n.variables.white
    }), (0, n.pick)({
        light: (0, l.css)([ "box-shadow:", ";" ], S),
        dark: (0, l.css)([ "box-shadow:inset 0 -1px 0 ", ";" ], n.variables.gray30)
    }), (function(r) {
        var e = r.$selected;
        return e && (0, l.css)([ "", ";background-color:", ";border-color:", ";" ], (0, 
        n.pick)({
            light: (0, l.css)([ "box-shadow:", ";" ], O),
            dark: (0, l.css)([ "box-shadow:inset 0 1px 0 ", ";" ], n.variables.black)
        }), (0, n.pick)({
            light: n.variables.gray92,
            dark: n.variables.gray22
        }), (0, n.pick)({
            light: n.variables.gray60,
            dark: n.variables.gray20
        }));
    }), (0, n.pick)({
        light: "#ebeeef",
        dark: n.variables.gray30
    }), (0, n.pick)({
        light: (0, l.css)([ "box-shadow:", ";" ], S),
        dark: (0, l.css)([ "box-shadow:inset 0 -1px 0 ", ";" ], n.variables.gray25)
    }), (0, n.pick)({
        light: n.variables.gray92,
        dark: n.variables.gray22
    }), (0, n.pick)({
        light: n.variables.gray60,
        dark: n.variables.gray20
    }), (0, n.pick)({
        dark: (0, l.css)([ "box-shadow:inset 0 -1px 0 ", ";" ], n.variables.gray22)
    }), (function(r) {
        var e = r.$append;
        return e && (0, n.pick)({
            light: (0, l.css)([ "box-shadow:", " ", ",inset -1px 0 0 ", ";" ], S, n.variables.focusShadow, n.variables.borderColor),
            dark: (0, l.css)([ "box-shadow:inset 0 -1px 0 ", " ", ",inset -1px 0 0 ", ";" ], n.variables.gray30, n.variables.focusShadow, n.variables.borderColor)
        });
    }), (function(r) {
        var e = r.$append, a = r.$selected;
        return e && a && (0, n.pick)({
            light: (0, l.css)([ "box-shadow:", ",", ",", ",inset -1px 0 0 ", ";" ], S, O, n.variables.focusShadow, n.variables.borderColor),
            dark: (0, l.css)([ "box-shadow:inset 0 1px 0 ", ",", ",inset -1px 0 0 ", ";" ], n.variables.black, n.variables.focusShadow, n.variables.borderColor)
        });
    }), (0, n.pick)({
        light: (0, l.css)([ "box-shadow:", ",", ";" ], S, n.variables.focusShadow),
        dark: (0, l.css)([ "box-shadow:inset 0 -1px 0 ", ",", ";" ], n.variables.gray30, n.variables.focusShadow)
    }), n.variables.textDisabledColor, (0, n.pick)({
        light: n.variables.gray96,
        dark: n.variables.gray30
    }), (0, n.pick)({
        light: n.variables.borderLightColor,
        dark: n.variables.gray30
    }), (function(r) {
        var e = r.$selected;
        return e && (0, n.pick)({
            light: (0, l.css)([ "box-shadow:", ";background-color:", ";border-color:", ";" ], O, n.variables.gray92, n.variables.gray80),
            dark: (0, l.css)([ "box-shadow:inset 0 1px 0 ", ";background-color:", ";border-color:", ";" ], n.variables.black, n.variables.gray22, n.variables.gray20)
        });
    }));
    var $ = "inset 0 -2px 0 rgba(0, 0, 0, 0.1)";
    var j = "inset 0 2px 0 rgba(0, 0, 0, 0.1)";
    var B = (0, l.css)([ "", ";font-weight:", ";", " &:not([disabled],[aria-disabled='true']){background-color:", ";color:", ";box-shadow:", ";transition:background-image 0.2s,border 0.2s,box-shadow 0.2s,text-decoration 0.2s;", " ", " &:hover{background-color:", ";}&:active{background-color:", ";transition:none;}&:focus{&:active{background-color:", ";transition:none;}}}&:not([disabled]){&:focus{box-shadow:", ",", ";}}&[disabled],&[aria-disabled='true']{color:", ";background-color:", ";", "}" ], x, n.variables.fontWeightSemiBold, w, m, n.variables.white, $, (function(r) {
        var e = r.$selected;
        return e && (0, l.css)([ "box-shadow:", ";background-color:", ";" ], j, C);
    }), (function(r) {
        var e = r.$prepend;
        return e && (0, l.css)([ "border-left:1px solid ", ";" ], m);
    }), n.variables.brandColorD50, n.variables.brandColorD30, n.variables.brandColorD30, $, n.variables.focusShadow, n.variables.brandColorL30, n.variables.brandColorL10, (function(r) {
        var e = r.$selected;
        return e && (0, l.css)([ "box-shadow:", ";background-color:", ";" ], j, n.variables.brandColorD20);
    }));
    var P = (0, l.css)([ "", ";font-weight:", ";", " &:not([disabled],[aria-disabled='true']){background-color:", ";color:", ";box-shadow:", ";transition:background-image 0.2s,border 0.2s,box-shadow 0.2s,text-decoration 0.2s;", " ", " &:hover{background-color:", ";}&:active{background-color:", ";transition:none;}&:focus{&:active{background-color:", ";transition:none;}}}&:not([disabled]){&:focus{box-shadow:", ",", ";}}&[disabled],&[aria-disabled='true']{color:", ";background-color:", ";", "}" ], x, n.variables.fontWeightSemiBold, w, n.variables.errorColorD10, n.variables.white, $, (function(r) {
        var e = r.$selected;
        return e && (0, l.css)([ "box-shadow:", ";background-color:", ";" ], j, n.variables.errorColorD30);
    }), (function(r) {
        var e = r.$prepend;
        return e && (0, l.css)([ "border-left:1px solid ", ";" ], n.variables.errorColorD30);
    }), n.variables.errorColorD30, n.variables.errorColorD40, n.variables.errorColorD40, $, n.variables.focusShadow, n.variables.errorColorL10, n.variables.errorColorD10, (function(r) {
        var e = r.$selected;
        return e && (0, l.css)([ "box-shadow:", ";background-color:", ";" ], j, n.variables.errorColorD40);
    }));
    var A = (0, l.css)([ "", ";color:", ";border:1px solid transparent;&:not([disabled],[aria-disabled='true']){transition:background-color 0.2s,border 0.2s,box-shadow 0.2s,text-decoration 0.2s;", " &:hover{color:", ";background-color:", ";border-color:", ";}", " &:focus{color:", ";}&:active,&[aria-expanded='true']{background-color:", ";transition:none;}}&:not([disabled]){&:focus{box-shadow:", ";}}&[disabled],&[aria-disabled='true']{color:", ";", "}" ], x, (0, 
    n.pick)({
        light: n.variables.gray45,
        dark: n.variables.white
    }), (function(r) {
        var e = r.$selected;
        return e && (0, l.css)([ "border-color:", ";" ], n.variables.accentColor);
    }), (0, n.pick)({
        light: n.variables.linkColor,
        dark: n.variables.white
    }), n.variables.backgroundColorHover, (0, n.pick)({
        light: n.variables.gray60,
        dark: n.variables.borderColor
    }), (function(r) {
        var e = r.$error;
        return e && (0, l.css)([ "&,&:hover{color:", ";}" ], n.variables.errorColor);
    }), (0, n.pick)({
        light: n.variables.linkColor,
        dark: n.variables.white
    }), (0, n.pick)({
        light: n.variables.gray92,
        dark: n.variables.gray22
    }), n.variables.focusShadow, n.variables.textDisabledColor, (function(r) {
        var e = r.$selected;
        return e && (0, l.css)([ "border-color:", ";" ], n.variables.borderLightColor);
    }));
    // CONCATENATED MODULE: ./src/ButtonSimple/ButtonSimpleStyles.ts
    var H = s()(b()).withConfig({
        displayName: "ButtonSimpleStyles__StyledClickable",
        componentId: "vlarwe-0"
    })([ "", " ", "" ], (function(r) {
        var e = r.to;
        return e ? (0, l.css)([ "", " align-items:center;justify-content:center;" ], n.mixins.reset("flex")) : (0, 
        l.css)([ "", "" ], n.mixins.reset("block"));
    }), (0, n.pick)({
        enterprise: (0, n.pickVariant)("$variant", {
            // the four main styles
            primary: B,
            secondary: D,
            destructive: P,
            subtle: A,
            // same as "secondary"
            default: D,
            // "pill" is an Enterprise-only style
            pill: A,
            // fallbacks for styles unsupported by Enterprise
            toggle: D,
            flat: D
        }),
        prisma: (0, n.pickVariant)("$variant", {
            // the four main styles
            primary: h,
            secondary: k,
            destructive: h,
            subtle: k,
            // same as "secondary"
            default: k,
            // "toggle" and "flat" are Prisma-only styles
            toggle: k,
            flat: h,
            // fallback for "pill" which is not supported by Prisma
            pill: k
        })
    }));
    // CONCATENATED MODULE: ./src/ButtonSimple/ButtonSimple.tsx
    function T(r) {
        "@babel/helpers - typeof";
        return T = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(r) {
            return typeof r;
        } : function(r) {
            return r && "function" == typeof Symbol && r.constructor === Symbol && r !== Symbol.prototype ? "symbol" : typeof r;
        }, T(r);
    }
    function _() {
        return _ = Object.assign ? Object.assign.bind() : function(r) {
            for (var e = 1; e < arguments.length; e++) {
                var a = arguments[e];
                for (var o in a) {
                    ({}).hasOwnProperty.call(a, o) && (r[o] = a[o]);
                }
            }
            return r;
        }, _.apply(null, arguments);
    }
    function N(r, e) {
        if (null == r) return {};
        var a, o, i = I(r, e);
        if (Object.getOwnPropertySymbols) {
            var t = Object.getOwnPropertySymbols(r);
            for (o = 0; o < t.length; o++) {
                a = t[o], e.includes(a) || {}.propertyIsEnumerable.call(r, a) && (i[a] = r[a]);
            }
        }
        return i;
    }
    function I(r, e) {
        if (null == r) return {};
        var a = {};
        for (var o in r) {
            if ({}.hasOwnProperty.call(r, o)) {
                if (e.includes(o)) continue;
                a[o] = r[o];
            }
        }
        return a;
    }
    function V(r, e) {
        if (!(r instanceof e)) throw new TypeError("Cannot call a class as a function");
    }
    function E(r, e) {
        for (var a = 0; a < e.length; a++) {
            var o = e[a];
            o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), 
            Object.defineProperty(r, J(o.key), o);
        }
    }
    function W(r, e, a) {
        return e && E(r.prototype, e), a && E(r, a), Object.defineProperty(r, "prototype", {
            writable: !1
        }), r;
    }
    function q(r, e) {
        if ("function" != typeof e && null !== e) throw new TypeError("Super expression must either be null or a function");
        r.prototype = Object.create(e && e.prototype, {
            constructor: {
                value: r,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(r, "prototype", {
            writable: !1
        }), e && M(r, e);
    }
    function M(r, e) {
        return M = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(r, e) {
            return r.__proto__ = e, r;
        }, M(r, e);
    }
    function R(r) {
        var e = U();
        return function() {
            var a, o = F(r);
            if (e) {
                var i = F(this).constructor;
                a = Reflect.construct(o, arguments, i);
            } else a = o.apply(this, arguments);
            return L(this, a);
        };
    }
    function L(r, e) {
        if (e && ("object" == T(e) || "function" == typeof e)) return e;
        if (void 0 !== e) throw new TypeError("Derived constructors may only return object or undefined");
        return z(r);
    }
    function z(r) {
        if (void 0 === r) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return r;
    }
    function U() {
        try {
            var r = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (r) {}
        return (U = function e() {
            return !!r;
        })();
    }
    function F(r) {
        return F = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(r) {
            return r.__proto__ || Object.getPrototypeOf(r);
        }, F(r);
    }
    function G(r, e, a) {
        return (e = J(e)) in r ? Object.defineProperty(r, e, {
            value: a,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : r[e] = a, r;
    }
    function J(r) {
        var e = K(r, "string");
        return "symbol" == T(e) ? e : e + "";
    }
    function K(r, e) {
        if ("object" != T(r) || !r) return r;
        var a = r[Symbol.toPrimitive];
        if (void 0 !== a) {
            var o = a.call(r, e || "default");
            if ("object" != T(o)) return o;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === e ? String : Number)(r);
    }
    function Q(r) {
        return rr(r) || Z(r) || Y(r) || X();
    }
    function X() {
        throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function Y(r, e) {
        if (r) {
            if ("string" == typeof r) return er(r, e);
            var a = {}.toString.call(r).slice(8, -1);
            return "Object" === a && r.constructor && (a = r.constructor.name), "Map" === a || "Set" === a ? Array.from(r) : "Arguments" === a || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(a) ? er(r, e) : void 0;
        }
    }
    function Z(r) {
        if ("undefined" != typeof Symbol && null != r[Symbol.iterator] || null != r["@@iterator"]) return Array.from(r);
    }
    function rr(r) {
        if (Array.isArray(r)) return er(r);
    }
    function er(r, e) {
        (null == e || e > r.length) && (e = r.length);
        for (var a = 0, o = Array(e); a < e; a++) {
            o[a] = r[a];
        }
        return o;
    }
    var ar = [ "default", "secondary", "primary", "destructive", "subtle" ];
    var or = [].concat(ar, [ "pill" ]);
    var ir = [].concat(ar, [ "toggle", "flat" ]);
    // Array.from(new Set... is used to de-duplicate the appearances
        var tr = Array.from(new Set([].concat(Q(or), Q(ir))));
    var nr = {
        action: t().string,
        appearance: t().oneOf([ "default", "secondary", "primary", "destructive", "subtle", "pill", "toggle", "flat" ]),
        append: t().bool,
        children: t().node,
        disabled: t().oneOfType([ t().bool, t().oneOf([ "dimmed" ]) ]),
        elementRef: t().oneOfType([ t().func, t().object ]),
        error: t().bool,
        inline: t().bool,
        openInNewContext: t().bool,
        prepend: t().bool,
        selected: t().bool,
        splunkTheme: t().object,
        to: t().string
    };
    var lr = {
        appearance: "default",
        append: false,
        disabled: false,
        error: false,
        inline: true,
        openInNewContext: false,
        prepend: false
    };
    var sr =  function(r) {
        q(a, r);
        var e = R(a);
        function a() {
            var r;
            V(this, a);
            for (var o = arguments.length, i = new Array(o), t = 0; t < o; t++) {
                i[t] = arguments[t];
            }
            r = e.call.apply(e, [ this ].concat(i));
            G(z(r), "component", null);
            G(z(r), "handleMount", (function(e) {
                r.component = e;
            }));
            return r;
        }
 // @docs-props-type ButtonSimplePropsBase
                W(a, [ {
            key: "focus",
            /**
     * Places focus on the button.
     */
            // eslint-disable-next-line react/no-unused-class-component-methods
            value: function r() {
                var e;
                (e = this.component) === null || e === void 0 ? void 0 : e.focus();
            }
        }, {
            key: "getVariant",
            value: function r() {
                var e = this.props, a = e.appearance, o = e.error, i = e.splunkTheme;
                var t = a;
                if (i.family === "enterprise") {
                    if (o) {
                        t = "destructive";
                    } else if (!or /* the readonly string[] cast is needed due to https://github.com/microsoft/TypeScript/issues/26255 */ .includes(a)) {
                        t = "default";
                    }
                } else if (i.family === "prisma" && !ir /* the readonly string[] cast is needed due to https://github.com/microsoft/TypeScript/issues/26255 */ .includes(a)) {
                    t = "secondary";
                }
                return t;
            }
        }, {
            key: "render",
            value: function r() {
                var e = this.props, a = e.append, i = e.children, t = e.error, n = e.prepend, l = e.selected, s = N(e, [ "append", "children", "error", "prepend", "selected" ]);
                
                return o().createElement(H, _({
                    "aria-invalid": t,
                    "aria-pressed": l,
                    "data-test": "button-simple",
                    ref: this.handleMount,
                    $variant: this.getVariant(),
                    $append: a,
                    $prepend: n,
                    $selected: l || false,
                    $error: t
                }, s), i);
            }
        } ]);
        return a;
    }(a.Component);
    G(sr, "defaultProps", lr);
    G(sr, "propTypes", nr);
    var cr = (0, n.withSplunkTheme)(sr);
    cr.propTypes = sr.propTypes;
    /* harmony default export */    const br = cr;
    // only for styled-components that wrap ButtonSimple
        // CONCATENATED MODULE: ./src/ButtonSimple/index.ts
    module.exports = e;
    /******/})();
