// @splunk/react-ui/ButtonGroup.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var n = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, n) => {
            /******/ for (var t in n) {
                /******/ if (e.o(n, t) && !e.o(r, t)) {
                    /******/ Object.defineProperty(r, t, {
                        enumerable: true,
                        get: n[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ m
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var t =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var l =  e.n(o);
    // CONCATENATED MODULE: external "styled-components"
    const a = require("styled-components");
    var i =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const u = require("@splunk/react-ui/Box");
    var c =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/themes"
    const s = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/ButtonGroup/ButtonGroupStyles.ts
    var p = i()(c()).withConfig({
        displayName: "ButtonGroupStyles__StyledBox",
        componentId: "sc-14fs2xs-0"
    })([ "position:relative;[data-inline] + &{margin-left:", ";}" ], s.variables.spacingSmall);
    // CONCATENATED MODULE: ./src/ButtonGroup/ButtonGroup.tsx
    function f() {
        return f = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, f.apply(null, arguments);
    }
    function d(e, r) {
        if (null == e) return {};
        var n, t, o = y(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                n = l[t], r.includes(n) || {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function y(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (r.includes(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    var v = {
        children: l().node,
        elementRef: l().oneOfType([ l().func, l().object ])
    };
    function b(e) {
        var r = e.children, o = d(e, [ "children" ]);
        // @docs-props-type ButtonGroupPropsBase
                var l = n.Children.toArray(r).filter(n.isValidElement).map((function(e, r, t) {
            var o = t.length;
            
            return (0, n.cloneElement)(e, {
                prepend: r > 0,
                append: r < o - 1,
                inline: false
            });
        }));
        
        return t().createElement(p, f({
            inline: true,
            flex: true,
            "data-test": "button-group",
            role: "group"
        }, o), l);
    }
    b.propTypes = v;
    /* harmony default export */    const m = b;
    // CONCATENATED MODULE: ./src/ButtonGroup/index.ts
    module.exports = r;
    /******/})();
