// @splunk/react-ui/Button.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var n = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, n) => {
            /******/ for (var r in n) {
                /******/ if (e.o(n, r) && !e.o(t, r)) {
                    /******/ Object.defineProperty(t, r, {
                        enumerable: true,
                        get: n[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        default: () => /* reexport */ Y
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var r =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const i = require("prop-types");
    var a =  e.n(i);
    // CONCATENATED MODULE: external "lodash/isString"
    const o = require("lodash/isString");
    var l =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/react-ui/Dropdown"
    const s = require("@splunk/react-ui/Dropdown");
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const c = require("@splunk/react-ui/ScreenReaderContent");
    var p =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const u = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/ui-utils/style"
    const f = require("@splunk/ui-utils/style");
    // CONCATENATED MODULE: external "styled-components"
    const d = require("styled-components");
    var m =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/react-ui/ButtonSimple"
    const v = require("@splunk/react-ui/ButtonSimple");
    var y =  e.n(v);
    // CONCATENATED MODULE: external "@splunk/themes"
    const b = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Button/ButtonStyles.ts
    var h = m().span.withConfig({
        displayName: "ButtonStyles__StyledIcon",
        componentId: "eqxqs2-0"
    })([ "display:flex;align-items:center;flex-shrink:0;font-size:", ";", "" ], (0, 
    b.pick)({
        enterprise: "inherit",
        prisma: {
            comfortable: "21px",
            compact: "16px"
        }
    }), (0, b.pickVariant)("$isStartIcon", {
        true: {
            enterprise: (0, d.css)([ "padding-right:3px;" ]),
            prisma: (0, d.css)([ "padding-right:8px;" ])
        }
    }));
    var x = m()(y()).withConfig({
        displayName: "ButtonStyles__StyledButtonSimple",
        componentId: "eqxqs2-1"
    })([ "vertical-align:middle;text-decoration:none;white-space:nowrap;flex-grow:1;min-width:0;max-width:100%;line-height:", ";min-height:", ";padding:", ";", " ", " &[data-inline]{:where(:not(a)){display:inline-block;}:where(a){display:inline-flex;}", "}[data-inline] + &&[data-inline]{margin-bottom:0;", "}" ], b.variables.lineHeight, b.variables.inputHeight, (0, 
    b.pick)({
        enterprise: {
            comfortable: "5px 14px",
            compact: "2px 14px"
        },
        prisma: {
            comfortable: function e(t) {
                var n = t.$hasPrismaIcon;
                return n ? "7px 18px" : "9px 18px";
            },
            compact: function e(t) {
                var n = t.$isMenu;
                return n ? "5px 8px 5px 16px" : "5px 16px";
            }
        }
    }), (function(e) {
        var t = e.$inline, n = e.append, r = e.prepend;
        return !t && !n && !r && (0, d.css)([ "width:100%;" ]);
    }), (function(e) {
        var t = e.$isIconOnly;
        return t && (0, d.css)([ "box-sizing:border-box;", ";width:", ";min-width:", ";" ], (0, 
        b.pick)({
            enterprise: {
                comfortable: (0, d.css)([ "padding:0 5px;" ]),
                compact: (0, d.css)([ "padding:0 2px;" ])
            },
            prisma: {
                comfortable: (0, d.css)([ "padding:0 calc(", " / 2 - 1px);" ], b.variables.inputHeight),
                compact: (0, d.css)([ "padding:0 calc(", " / 2 - 1px);" ], b.variables.inputHeight)
            }
        }), b.variables.inputHeight, b.variables.inputHeight);
    }), (function(e) {
        var t = e.$isIconOnly;
        return t ? null : (0, d.css)([ "width:auto;" ]);
    }), (function(e) {
        var t = e.prepend;
        return !t && (0, d.css)([ "margin-left:", ";" ], b.variables.spacingSmall);
    }));
    var g = m().span.withConfig({
        displayName: "ButtonStyles__StyledContentWrapper",
        componentId: "eqxqs2-2"
    })([ "display:flex;flex-direction:row;justify-content:center;align-items:center;flex:1 0 auto;max-width:100%;box-sizing:border-box;padding:", ";" ], (0, 
    b.pick)({
        enterprise: function e(t) {
            var n = t.$error;
            if (n) {
                return "1px";
            }
            return (0, b.pickVariant)("$appearance", {
                default: "0",
                secondary: "0",
                destructive: "1px",
                primary: "1px",
                pill: "0",
                toggle: "0",
                // === default
                flat: "0",
                // === default
                subtle: "0"
            });
        },
        prisma: (0, b.pickVariant)("$appearance", {
            default: 0,
            secondary: "1px 0",
            primary: "1px 0",
            destructive: "1px 0",
            pill: "1px 0",
            // === secondary
            toggle: "0",
            flat: "1px 0",
            subtle: "1px 0"
        })
    }));
    var w = m().span.withConfig({
        displayName: "ButtonStyles__StyledLabel",
        componentId: "eqxqs2-3"
    })([ "text-overflow:ellipsis;overflow:hidden;flex:0 1 auto;&:not(:last-child){padding-right:", ";}", "" ], (0, 
    b.pick)({
        enterprise: "3px",
        prisma: "10px"
    }), (function(e) {
        var t = e.$isMenu;
        return t && (0, d.css)([ "flex:1 1 auto;text-align:left;" ]);
    }));
    var S = m().span.withConfig({
        displayName: "ButtonStyles__StyledCaretWrapper",
        componentId: "eqxqs2-4"
    })([ "display:inline-flex;", "" ], (0, b.pickVariant)("$disabled", {
        false: {
            prisma: (0, b.pickVariant)("$primary", {
                true: (0, d.css)([ "color:", ";" ], b.variables.contentColorInverted),
                false: (0, d.css)([ "color:", ";", ":hover > * > &,", ":active > * > &,{color:", ";}" ], b.variables.contentColorDefault, 
                /* sc-sel */
                x, 
                /* sc-sel */
                x, b.variables.contentColorDefault)
            })
        }
    }));
    // CONCATENATED MODULE: external "@splunk/react-icons/CaretSmallDown"
    const k = require("@splunk/react-icons/CaretSmallDown");
    var O =  e.n(k);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/Caret"
    const C = require("@splunk/react-icons/enterprise/Caret");
    var q =  e.n(C);
    // CONCATENATED MODULE: ./src/Button/icons/CaretSmallDown.tsx
    var $ = function e() {
        var t = (0, b.useSplunkTheme)(), n = t.isEnterprise, i = t.isCompact;
        var a = i ? "20px" : "24px";
        return n ?  r().createElement(q(), {
            screenReaderText: null,
            hideDefaultTooltip: true,
            size: .5
        }) :  r().createElement(O(), {
            height: a,
            width: a
        });
    };
    /* harmony default export */    const _ = $;
    // CONCATENATED MODULE: external "@splunk/react-icons/ArrowSquareTopRightInset"
    const I = require("@splunk/react-icons/ArrowSquareTopRightInset");
    var E =  e.n(I);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/External"
    const P = require("@splunk/react-icons/enterprise/External");
    var j =  e.n(P);
    // CONCATENATED MODULE: ./src/Button/icons/External.tsx
    function N() {
        var e = (0, b.useSplunkTheme)(), t = e.isEnterprise;
        return t ?  r().createElement(j(), {
            screenReaderText: null
        }) :  r().createElement(E(), null);
    }
    /* harmony default export */    const T = N;
    // CONCATENATED MODULE: ./src/Button/Button.tsx
    function M(e) {
        "@babel/helpers - typeof";
        if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") {
            M = function e(t) {
                return typeof t;
            };
        } else {
            M = function e(t) {
                return t && typeof Symbol === "function" && t.constructor === Symbol && t !== Symbol.prototype ? "symbol" : typeof t;
            };
        }
        return M(e);
    }
    function R() {
        R = Object.assign || function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var n = arguments[t];
                for (var r in n) {
                    if (Object.prototype.hasOwnProperty.call(n, r)) {
                        e[r] = n[r];
                    }
                }
            }
            return e;
        };
        return R.apply(this, arguments);
    }
    function B(e, t) {
        if (!(e instanceof t)) {
            throw new TypeError("Cannot call a class as a function");
        }
    }
    function D(e, t) {
        for (var n = 0; n < t.length; n++) {
            var r = t[n];
            r.enumerable = r.enumerable || false;
            r.configurable = true;
            if ("value" in r) r.writable = true;
            Object.defineProperty(e, r.key, r);
        }
    }
    function H(e, t, n) {
        if (t) D(e.prototype, t);
        if (n) D(e, n);
        return e;
    }
    function V(e, t) {
        if (typeof t !== "function" && t !== null) {
            throw new TypeError("Super expression must either be null or a function");
        }
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: true,
                configurable: true
            }
        });
        if (t) z(e, t);
    }
    function z(e, t) {
        z = Object.setPrototypeOf || function e(t, n) {
            t.__proto__ = n;
            return t;
        };
        return z(e, t);
    }
    function W(e) {
        var t = F();
        return function n() {
            var r = G(e), i;
            if (t) {
                var a = G(this).constructor;
                i = Reflect.construct(r, arguments, a);
            } else {
                i = r.apply(this, arguments);
            }
            return A(this, i);
        };
    }
    function A(e, t) {
        if (t && (M(t) === "object" || typeof t === "function")) {
            return t;
        }
        return L(e);
    }
    function L(e) {
        if (e === void 0) {
            throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        }
        return e;
    }
    function F() {
        if (typeof Reflect === "undefined" || !Reflect.construct) return false;
        if (Reflect.construct.sham) return false;
        if (typeof Proxy === "function") return true;
        try {
            Date.prototype.toString.call(Reflect.construct(Date, [], (function() {})));
            return true;
        } catch (e) {
            return false;
        }
    }
    function G(e) {
        G = Object.setPrototypeOf ? Object.getPrototypeOf : function e(t) {
            return t.__proto__ || Object.getPrototypeOf(t);
        };
        return G(e);
    }
    function J(e, t, n) {
        if (t in e) {
            Object.defineProperty(e, t, {
                value: n,
                enumerable: true,
                configurable: true,
                writable: true
            });
        } else {
            e[t] = n;
        }
        return e;
    }
    var K = {
        action: a().string,
        appearance: a().oneOf([ "default", "secondary", "primary", "destructive", "pill", "toggle", "flat", "subtle" ]),
        append: a().bool,
        children: a().node,
        /** @private An additional className to add to the button. */
        classNamePrivate: a().string,
        disabled: a().oneOfType([ a().bool, a().oneOf([ "dimmed" ]) ]),
        elementRef: a().oneOfType([ a().func, a().object ]),
        error: a().bool,
        icon: a().node,
        inline: a().bool,
        isMenu: a().bool,
        label: a().node,
        onClick: a().func,
        openInNewContext: a().oneOfType([ a().bool, a().string ]),
        prepend: a().bool,
        selected: a().bool,
        to: a().string,
        value: a().any
    };
    var Q = {
        appearance: "default",
        append: false,
        disabled: false,
        error: false,
        inline: true,
        isMenu: false,
        openInNewContext: false,
        prepend: false
    };
    var U = (0, u._)("(Opens new window)");
    var X =  function(e) {
        V(n, e);
        var t = W(n);
        // @docs-props-type ButtonPropsBase
                function n(e) {
            var r;
            B(this, n);
            r = t.call(this, e);
            J(L(r), "component", null);
            J(L(r), "handleClick", (function(e) {
                var t = r.props, n = t.label, i = t.icon, a = t.value, o = t.action, l = t.onClick;
                l === null || l === void 0 ? void 0 : l(e, {
                    label: n,
                    icon: i,
                    value: a,
                    action: o
                });
            }));
            J(L(r), "handleMount", (function(e) {
                r.component = e;
            }));
            if (false) {}
            return r;
        }
        H(n, [ {
            key: "focus",
            /**
     * Places focus on the button.
     */
            value: function e() {
                var t;
                (t = this.component) === null || t === void 0 ? void 0 : t.focus();
            }
        }, {
            key: "render",
            value: function e() {
                var t = this.props, n = t.action, i = t.appearance, a = t.append, o = t.className, s = t.classNamePrivate, c = t.disabled, u = t.error, d = t.icon, m = t.inline, v = t.isMenu, y = t.onClick, b = t.openInNewContext, k = t.prepend, O = t.to, C = t.value;
                var q = this.props, $ = q.children, I = q.label;
                if (!I && l()($)) {
                    I = $;
                    $ = null;
                }
                var E = false;
                var P;
                if (b) {
                    E = true;
                    P = typeof b === "string" ? b : U;
                }
                var j = !!(d && !I && !v && !$);
                var N = O && E || v || !!d;
                
                return r().createElement(x, R({
                    "aria-haspopup": v || undefined,
                    "aria-invalid": u || undefined,
                    "data-test": "button"
                }, this.props, {
                    className: (0, f.toClassName)(o, s),
                    "data-action": n,
                    "data-inline": m || null,
                    disabled: c,
                    append: a,
                    prepend: k,
                    $inline: m,
                    $hasPrismaIcon: N,
                    $isIconOnly: j,
                    $isMenu: v,
                    value: C,
                    onClick: y ? this.handleClick : undefined,
                    ref: this.handleMount,
                    openInNewContext: E,
                    appearance: i
                }),  r().createElement(g, {
                    $appearance: i,
                    $error: u
                }, d &&  r().createElement(h, {
                    $isStartIcon: !j
                }, d), I &&  r().createElement(w, {
                    "data-test": "label",
                    $isMenu: v
                }, I), $, v &&  r().createElement(S, {
                    $disabled: c,
                    $primary: i === "primary"
                },  r().createElement(_, null)), b &&  r().createElement(h, {
                    $isStartIcon: false
                },  r().createElement(T, null),  r().createElement(p(), null, P))));
            }
        } ]);
        return n;
    }(n.Component);
    J(X, "propTypes", K);
    J(X, "defaultProps", Q);
    J(X, s.legacyRefMode, true);
    /* harmony default export */    const Y = X;
    // CONCATENATED MODULE: ./src/Button/index.ts
    module.exports = t;
    /******/})();
