// @splunk/react-ui/Calendar.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = a => {
            /******/ var r = a && a.__esModule ? 
            /******/ () => a["default"]
            /******/ : () => a
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (a, r) => {
            /******/ for (var t in r) {
                /******/ if (e.o(r, t) && !e.o(a, t)) {
                    /******/ Object.defineProperty(a, t, {
                        enumerable: true,
                        get: r[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, a) => Object.prototype.hasOwnProperty.call(e, a)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var a = {};
    // ESM COMPAT FLAG
        e.r(a);
    // EXPORTS
        e.d(a, {
        calendarActiveSelector: () => /* reexport */ Me,
        default: () => /* reexport */ Oe,
        focusActiveDay: () => /* reexport */ Ce
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var t =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const n = require("prop-types");
    var l =  e.n(n);
    // CONCATENATED MODULE: external "moment"
    const o = require("moment");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const c = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "lodash/times"
    const u = require("lodash/times");
    var s =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/themes"
    const d = require("@splunk/themes");
    // CONCATENATED MODULE: external "@splunk/ui-utils/keyboard"
    const v = require("@splunk/ui-utils/keyboard");
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const f = require("@splunk/react-ui/ScreenReaderContent");
    var p =  e.n(f);
    // CONCATENATED MODULE: external "styled-components"
    const y = require("styled-components");
    var h =  e.n(y);
    // CONCATENATED MODULE: ./src/Calendar/DayStyles.ts
    /* Element used to prevent vertical shift on firefox when the font-weight is increased with $selected */
    var Y = h().div.withConfig({
        displayName: "DayStyles__StyledValue",
        componentId: "sc-1nryhyt-0"
    })([ "left:0;line-height:", ";position:absolute;top:0;width:100%;" ], (0, d.pick)({
        enterprise: "2em",
        prisma: "36px "
    }));
    var m = h().div.withConfig({
        displayName: "DayStyles__StyledInner",
        componentId: "sc-1nryhyt-1"
    })([ "border:", ";border-radius:", ";color:", ";height:", ";position:relative;text-align:center;width:", ";" ], (0, 
    d.pick)({
        enterprise: "1px solid transparent",
        prisma: "none"
    }), (0, d.pick)({
        enterprise: d.variables.borderRadius,
        prisma: "50%"
    }), (0, d.pick)({
        enterprise: {
            light: d.variables.gray45,
            dark: d.variables.gray98
        },
        prisma: d.variables.contentColorActive
    }), (0, d.pick)({
        enterprise: "2em",
        prisma: "36px "
    }), (0, d.pick)({
        enterprise: "calc(2em - 2px)",
        prisma: "36px"
    }));
    var b = h().td.withConfig({
        displayName: "DayStyles__StyledCell",
        componentId: "sc-1nryhyt-2"
    })([ "", ";cursor:pointer;", ";", ";" ], d.mixins.reset("table-cell"), (0, d.pickVariant)("$today", {
        true: {
            enterprise: (0, y.css)([ "& > ", "{background-color:", ";}&:focus{& > ", "{box-shadow:", ";}}" ], /* sc-sel */ m, (0, 
            d.pick)({
                light: d.variables.gray92,
                dark: d.variables.gray45
            }), /* sc-sel */ m, d.variables.focusShadow),
            prisma: (0, y.css)([ "& > ", "{box-shadow:inset 0 0 0 1px ", ";}&:focus{& > ", "{box-shadow:inset 0 0 0 1px ", ",", ";}}" ], /* sc-sel */ m, d.variables.interactiveColorBorder, /* sc-sel */ m, d.variables.interactiveColorBorder, d.variables.focusShadow)
        },
        false: (0, y.css)([ "&:focus{& > ", "{box-shadow:", ";}}" ], /* sc-sel */ m, d.variables.focusShadow)
    }), (0, d.pickVariant)("$selected", {
        true: (0, d.pick)({
            enterprise: (0, y.css)([ "& > ", "{border-color:", ";}" ], /* sc-sel */ m, d.variables.linkColor),
            prisma: (0, y.css)([ "& > ", "{background-color:", ";font-weight:", ";}" ], /* sc-sel */ m, d.variables.interactiveColorOverlaySelected, d.variables.fontWeightBold)
        }),
        false: (0, d.pick)({
            enterprise: (0, y.css)([ "&:hover{& > ", "{background-color:", ";color:", ";}}" ], /* sc-sel */ m, d.variables.backgroundColorHover, d.variables.linkColor),
            prisma: (0, y.css)([ "&:hover{& > ", "{background-color:", ";}}" ], /* sc-sel */ m, d.variables.interactiveColorOverlayHover)
        })
    }));
    // CONCATENATED MODULE: ./src/Calendar/Day.tsx
    function g() {
        return g = Object.assign ? Object.assign.bind() : function(e) {
            for (var a = 1; a < arguments.length; a++) {
                var r = arguments[a];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, g.apply(null, arguments);
    }
    function M(e, a) {
        if (null == e) return {};
        var r, t, n = D(e, a);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                r = l[t], a.includes(r) || {}.propertyIsEnumerable.call(e, r) && (n[r] = e[r]);
            }
        }
        return n;
    }
    function D(e, a) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (a.includes(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var k = {
        active: l().bool,
        locale: l().string,
        onClick: l().func,
        onKeydown: l().func,
        selected: l().bool,
        today: l().bool,
        value: l().string
    };
    function w(e) {
        var a = e.active, n = a === void 0 ? false : a, l = e.locale, o = l === void 0 ? "en_US" : l, c = e.onClick, u = e.onKeyDown, s = e.selected, d = s === void 0 ? false : s, v = e.today, f = v === void 0 ? false : v, p = e.value, y = M(e, [ "active", "locale", "onClick", "onKeyDown", "selected", "today", "value" ]);
        var h = (0, r.useCallback)((function(e) {
            e.preventDefault();
            c === null || c === void 0 ? void 0 : c(e, {
                value: p
            });
        }), [ c, p ]);
        var D = (0, r.useCallback)((function(e) {
            e.preventDefault();
            u === null || u === void 0 ? void 0 : u(e, {
                value: p
            });
        }), [ u, p ]);
        var k = p && i()(p, "YYYY-MM-DD", o).format("D");
        
        return t().createElement(b, g({
            tabIndex: n ? 0 : -1,
            "aria-selected": d,
            "data-test-day": k,
            "data-test": "day-of-month",
            "data-active-day": n,
            "data-is-today": f,
            onClick: h,
            onKeyDown: D,
            $today: f,
            $selected: d
        }, y),  t().createElement(m, null,  t().createElement(Y, null, k)));
    }
    w.propTypes = k;
    /* harmony default export */    const C = w;
    // CONCATENATED MODULE: ./src/Calendar/DateTableStyles.ts
    var S = h().table.withConfig({
        displayName: "DateTableStyles__Styled",
        componentId: "sc-1lbrjq0-0"
    })([ "", ";table-layout:fixed;border-spacing:", ";width:", ";margin:", ";" ], d.mixins.reset("table"), (0, 
    d.pick)({
        enterprise: "3px",
        prisma: d.variables.spacingSmall
    }), (0, d.pick)({
        enterprise: "100%",
        prisma: "calc(100% + 16px)"
    }), (0, d.pick)({
        enterprise: "0",
        prisma: "-8px"
    }));
    var O = h().th.withConfig({
        displayName: "DateTableStyles__StyledTableHeader",
        componentId: "sc-1lbrjq0-1"
    })([ "", ";text-align:center;padding-bottom:", ";color:", ";font-size:", ";" ], d.mixins.reset("table-cell"), (0, 
    d.pick)({
        enterprise: "2px",
        prisma: d.variables.spacingXSmall
    }), d.variables.contentColorMuted, (0, d.pick)({
        enterprise: d.variables.fontSizeSmall,
        prisma: d.variables.fontSize
    }));
    // CONCATENATED MODULE: ./src/Calendar/DateTable.tsx
    function x() {
        return x = Object.assign ? Object.assign.bind() : function(e) {
            for (var a = 1; a < arguments.length; a++) {
                var r = arguments[a];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, x.apply(null, arguments);
    }
    function E(e, a) {
        if (null == e) return {};
        var r, t, n = I(e, a);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                r = l[t], a.includes(r) || {}.propertyIsEnumerable.call(e, r) && (n[r] = e[r]);
            }
        }
        return n;
    }
    function I(e, a) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (a.includes(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var _ = {
        displayValue: l().string.isRequired,
        highlightToday: l().bool,
        labelledBy: l().string,
        locale: l().string,
        onChange: l().func,
        onInternalChange: l().func,
        selectedValue: l().string.isRequired,
        todayValue: l().string
    };
    function j(e) {
        return s()(e, (function(e) {
            
            return t().createElement("td", {
                key: "".concat(e)
            });
        }));
    }
    function q(e) {
        var a = e.displayValue, r = e.endDate, n = e.highlightToday, l = e.locale, o = e.onChange, c = e.onKeyDown, u = e.selectedDate, s = e.selectedValue, d = e.startDate, v = e.todayValue;
        var f = i()(d).locale(l).startOf("month");
        var p = [];
        while (f.isSameOrBefore(r)) {
            var y = f.format("YYYY-MM-DD");
            var h = y === v;
            var Y = y === s;
            var m = y === a;
            p.push( t().createElement(C, {
                key: y,
                locale: l,
                onClick: o,
                onKeyDown: c,
                active: u ? m : h,
                today: n && h,
                value: y,
                selected: Y
            }));
            f.add(1, "day");
        }
        return p;
    }
    function T(e) {
        return e.reduce((function(e, a, r) {
            var t = Math.floor(r / 7);
            e[t].push(a);
            return e;
        }), s()(7, (function() {
            return [];
        }))).map((function(e, a) {
            
            // eslint-disable-next-line react/no-array-index-key
            return t().createElement("tr", {
                key: "$week-".concat(a)
            }, e);
        }));
    }
    function P(e, a) {
        
        return t().createElement("thead", null,  t().createElement("tr", null, s()(7, (function(r) {
            var n = i()().locale(e).weekday(r).format(a);
            var l = i()().locale(e).weekday(r).format("dddd");
            
            return t().createElement(O, {
                key: n
            },  t().createElement(p(), null, l),  t().createElement("span", {
                "aria-hidden": true
            }, n));
        }))));
    }
    function V(e) {
        var a = e.displayValue, r = e.highlightToday, n = r === void 0 ? false : r, l = e.labelledBy, o = e.locale, c = o === void 0 ? "en_US" : o, u = e.onChange, s = e.onInternalChange, f = e.selectedValue, p = e.todayValue, y = E(e, [ "displayValue", "highlightToday", "labelledBy", "locale", "onChange", "onInternalChange", "selectedValue", "todayValue" ]);
        var h = (0, d.useSplunkTheme)(), Y = h.family;
        var m = Y === "prisma" ? "dd" : "ddd";
        var b = i()(a, "YYYY-MM-DD", c);
        var g = i()(f, "YYYY-MM-DD", c);
        var M = i()(b).startOf("month");
        var D = i()(b).endOf("month").startOf("day");
        var k = function e(a, r) {
            a.preventDefault();
            u === null || u === void 0 ? void 0 : u(a, r);
        };
        var w = function e(a, r) {
            var t = r.value;
            a.preventDefault();
            s === null || s === void 0 ? void 0 : s(a, {
                value: i()(t, "YYYY-MM-DD").add(-1, "d").format("YYYY-MM-DD")
            });
        };
        var C = function e(a, r) {
            var t = r.value;
            a.preventDefault();
            s === null || s === void 0 ? void 0 : s(a, {
                value: i()(t, "YYYY-MM-DD").add(1, "d").format("YYYY-MM-DD")
            });
        };
        var O = function e(a, r) {
            var t = r.value;
            a.preventDefault();
            s === null || s === void 0 ? void 0 : s(a, {
                value: i()(t, "YYYY-MM-DD").add(-1, "w").format("YYYY-MM-DD")
            });
        };
        var I = function e(a, r) {
            var t = r.value;
            a.preventDefault();
            s === null || s === void 0 ? void 0 : s(a, {
                value: i()(t, "YYYY-MM-DD").add(1, "w").format("YYYY-MM-DD")
            });
        };
        var _ = function e(a, r) {
            var t = r.value;
            a.preventDefault();
            s === null || s === void 0 ? void 0 : s(a, {
                value: i()(t, "YYYY-MM-DD").add(-1, "M").format("YYYY-MM-DD")
            });
        };
        var V = function e(a, r) {
            var t = r.value;
            a.preventDefault();
            s === null || s === void 0 ? void 0 : s(a, {
                value: i()(t, "YYYY-MM-DD").add(1, "M").format("YYYY-MM-DD")
            });
        };
        var B = function e(a, r) {
            var t = r.value;
            a.preventDefault();
            s === null || s === void 0 ? void 0 : s(a, {
                value: i()(t, "YYYY-MM-DD").add(-1, "y").format("YYYY-MM-DD")
            });
        };
        var R = function e(a, r) {
            var t = r.value;
            a.preventDefault();
            s === null || s === void 0 ? void 0 : s(a, {
                value: i()(t, "YYYY-MM-DD").add(1, "y").format("YYYY-MM-DD")
            });
        };
        var z = function e(a, r) {
            var t = r.value;
            a.preventDefault();
            s === null || s === void 0 ? void 0 : s(a, {
                value: i()(t, "YYYY-MM-DD").startOf("week").format("YYYY-MM-DD")
            });
        };
        var A = function e(a, r) {
            var t = r.value;
            a.preventDefault();
            s === null || s === void 0 ? void 0 : s(a, {
                value: i()(t, "YYYY-MM-DD").endOf("week").format("YYYY-MM-DD")
            });
        };
        var N = function e(a, r) {
            var t = (0, v.keycode)(a.nativeEvent);
            switch (t) {
              case "enter":
              case "space":
                k(a, r);
                break;

              case "up":
                O(a, r);
                break;

              case "down":
                I(a, r);
                break;

              case "left":
                w(a, r);
                break;

              case "right":
                C(a, r);
                break;

              case "page up":
                if (a.shiftKey) {
                    B(a, r);
                } else {
                    _(a, r);
                }
                break;

              case "page down":
                if (a.shiftKey) {
                    R(a, r);
                } else {
                    V(a, r);
                }
                break;

              case "home":
                z(a, r);
                break;

              case "end":
                A(a, r);
                break;

              default:
 // do nothing
                        }
        };
        var K = j(M.format("e")).concat(q({
            displayValue: a,
            endDate: D,
            highlightToday: n,
            locale: c,
            onChange: u,
            onKeyDown: N,
            selectedDate: g,
            selectedValue: f,
            startDate: M,
            todayValue: p
        })).concat(j(6 - D.format("e")));
        
        return t().createElement(S, x({
            role: "grid",
            "aria-labelledby": l
        }, y), P(c, m),  t().createElement("tbody", null, T(K)));
    }
    V.propTypes = _;
    /* harmony default export */    const B = V;
    // CONCATENATED MODULE: external "lodash/omit"
    const R = require("lodash/omit");
    var z =  e.n(R);
    // CONCATENATED MODULE: external "@splunk/react-icons/ChevronLeft"
    const A = require("@splunk/react-icons/ChevronLeft");
    var N =  e.n(A);
    // CONCATENATED MODULE: external "@splunk/react-icons/ChevronRight"
    const K = require("@splunk/react-icons/ChevronRight");
    var H =  e.n(K);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/ChevronLeft"
    const $ = require("@splunk/react-icons/enterprise/ChevronLeft");
    var U =  e.n($);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/ChevronRight"
    const L = require("@splunk/react-icons/enterprise/ChevronRight");
    var W =  e.n(L);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const X = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const F = require("@splunk/react-ui/Button");
    var G =  e.n(F);
    // CONCATENATED MODULE: ./src/Calendar/MonthHeaderStyles.ts
    var J = h().div.withConfig({
        displayName: "MonthHeaderStyles__StyledBox",
        componentId: "sc-18bmbh7-0"
    })([ "display:flex;position:relative;line-height:", ";padding:0 2px;margin-bottom:10px;" ], d.variables.inputHeight);
    var Q = h().span.withConfig({
        displayName: "MonthHeaderStyles__StyledHeading",
        componentId: "sc-18bmbh7-1"
    })([ "display:block;text-align:center;text-transform:", ";flex:1 0 1px;", ";" ], (0, 
    d.pick)({
        enterprise: "capitalize",
        prisma: "uppercase"
    }), (0, d.pick)({
        prisma: (0, y.css)([ "font-size:20px;font-weight:bold;color:", ";" ], d.variables.contentColorActive)
    }));
    var Z = h()(G()).withConfig({
        displayName: "MonthHeaderStyles__StyledButton",
        componentId: "sc-18bmbh7-2"
    })([ "flex:0 0 auto;" ]);
    // CONCATENATED MODULE: ./src/Calendar/MonthHeader.tsx
    function ee(e, a) {
        if (null == e) return {};
        var r, t, n = ae(e, a);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                r = l[t], a.includes(r) || {}.propertyIsEnumerable.call(e, r) && (n[r] = e[r]);
            }
        }
        return n;
    }
    function ae(e, a) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (a.includes(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var re = {
        headingId: l().string,
        locale: l().string,
        onChange: l().func,
        value: l().string.isRequired
    };
    var te = {
        default: "MMMM YYYY",
        ja: "YYYY年MMM",
        ko: "YYYY년 MMMM",
        "zh-cn": "YYYY年MMM",
        "zh-tw": "YYYY年MMM"
    };
    var ne = {
        default: "MMM YYYY",
        ja: "YYYY年 MMM",
        ko: "YYYY년 MMMM",
        "zh-cn": "YYYY年 MMM",
        "zh-tw": "YYYY年 MMM"
    };
    function le(e) {
        var a = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;
        var r = e.toLowerCase().replace("_", "-");
        return a ? ne[r] || 
        // full locale match
        ne[r.substr(0, 2)] || 
        // language match
        ne["default"] : te[r] || 
        // full locale match
        te[r.substr(0, 2)] || 
        // language match
        te["default"];
 // default
        }
    function oe(e) {
        var a = e.headingId, n = e.locale, l = n === void 0 ? "en_US" : n, o = e.onChange, c = e.value, u = ee(e, [ "headingId", "locale", "onChange", "value" ]);
        var s = (0, d.useSplunkTheme)(), v = s.isEnterprise, f = s.isPrisma, p = s.density;
        var y = (0, r.useCallback)((function(e, a) {
            e.stopPropagation();
            var r = i()(c, "YYYY-MM-DD").add(a, "M").format("YYYY-MM-DD");
            o === null || o === void 0 ? void 0 : o(e, {
                value: r
            });
        }), [ o, c ]);
        var h = (0, r.useCallback)((function(e) {
            y(e, -1);
        }), [ y ]);
        var Y = (0, r.useCallback)((function(e) {
            y(e, 1);
        }), [ y ]);
        var m = i()(c, "YYYY-MM-DD", l);
        var b = m.format(le(l, f));
        var g = (0, X._)("Previous month");
        var M = (0, X._)("Next month");
        
        return t().createElement(d.SplunkThemeProvider, {
            density: f ? "comfortable" : p
        },  t().createElement(J, z()(u, "onChange"),  t().createElement(Z, {
            appearance: "subtle",
            icon: v ?  t().createElement(U(), {
                hideDefaultTooltip: true,
                screenReaderText: g,
                size: 1
            }) :  t().createElement(N(), {
                "aria-label": g,
                height: "24",
                width: "24"
            }),
            "data-test": "previous-month",
            onClick: h
        }),  t().createElement(Q, {
            id: a,
            "data-test": "header-label"
        }, b),  t().createElement(Z, {
            appearance: "subtle",
            icon: v ?  t().createElement(W(), {
                hideDefaultTooltip: true,
                screenReaderText: M,
                size: 1
            }) :  t().createElement(H(), {
                "aria-label": M,
                height: "24",
                width: "24"
            }),
            "data-test": "next-month",
            onClick: Y
        })));
    }
    oe.propTypes = re;
    /* harmony default export */    const ie = oe;
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const ce = require("@splunk/react-ui/Box");
    var ue =  e.n(ce);
    // CONCATENATED MODULE: ./src/Calendar/CalendarStyles.ts
    var se = h()(ue()).withConfig({
        displayName: "CalendarStyles__StyledBox",
        componentId: "sc-18925wt-0"
    })([ "width:", ";padding:", ";" ], (0, d.pick)({
        enterprise: "220px",
        prisma: "332px"
    }), (0, d.pick)({
        enterprise: "8px",
        prisma: d.variables.spacingLarge
    }));
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function de(e, a) {
        if (e) {
            if (typeof e === "function") {
                e(a);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = a;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Calendar/Calendar.tsx
    function ve() {
        return ve = Object.assign ? Object.assign.bind() : function(e) {
            for (var a = 1; a < arguments.length; a++) {
                var r = arguments[a];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, ve.apply(null, arguments);
    }
    function fe(e, a) {
        return me(e) || Ye(e, a) || ye(e, a) || pe();
    }
    function pe() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function ye(e, a) {
        if (e) {
            if ("string" == typeof e) return he(e, a);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? he(e, a) : void 0;
        }
    }
    function he(e, a) {
        (null == a || a > e.length) && (a = e.length);
        for (var r = 0, t = Array(a); r < a; r++) {
            t[r] = e[r];
        }
        return t;
    }
    function Ye(e, a) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var t, n, l, o, i = [], c = !0, u = !1;
            try {
                if (l = (r = r.call(e)).next, 0 === a) {
                    if (Object(r) !== r) return;
                    c = !1;
                } else for (;!(c = (t = l.call(r)).done) && (i.push(t.value), i.length !== a); c = !0) {
                }
            } catch (e) {
                u = !0, n = e;
            } finally {
                try {
                    if (!c && null != r["return"] && (o = r["return"](), Object(o) !== o)) return;
                } finally {
                    if (u) throw n;
                }
            }
            return i;
        }
    }
    function me(e) {
        if (Array.isArray(e)) return e;
    }
    function be(e, a) {
        if (null == e) return {};
        var r, t, n = ge(e, a);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                r = l[t], a.includes(r) || {}.propertyIsEnumerable.call(e, r) && (n[r] = e[r]);
            }
        }
        return n;
    }
    function ge(e, a) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (a.includes(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var Me = '[data-active-day="true"]';
    var De = {
        describedBy: l().string,
        highlightToday: l().bool,
        labelledBy: l().string,
        locale: l().string,
        mockDate: l().string,
        onChange: l().func,
        /** @private. */
        required: l().bool,
        value: l().string,
        elementRef: l().oneOfType([ l().func, l().object ])
    };
    var ke = function e(a) {
        return i()().locale(a).format("YYYY-MM-DD");
    };
    var we = function e(a) {
        if (!moment(a, "YYYY-MM-DD").isValid()) {
            return new Error("Invalid value prop ".concat(a, " passed to Calendar. value must be a string date formatted 'YYYY-MM-DD'."));
        }
        return null;
    };
    /**
 * Place focus on the active day.
 */    var Ce = function e(a) {
        var r;
        var t = a === null || a === void 0 ? void 0 : (r = a.current) === null || r === void 0 ? void 0 : r.querySelector(Me);
        t === null || t === void 0 ? void 0 : t.focus();
    };
    function Se(e) {
        var a = e.describedBy, n = e.elementRef, l = e.highlightToday, o = e.labelledBy, u = e.locale, s = u === void 0 ? "en_US" : u, d = e.mockDate, v = e.onChange, f = e.required, p = e.value, y = p === void 0 ? i()().format("YYYY-MM-DD") : p, h = be(e, [ "describedBy", "elementRef", "highlightToday", "labelledBy", "locale", "mockDate", "onChange", "required", "value" ]);
        var Y = (0, r.useState)(y), m = fe(Y, 2), b = m[0], g = m[1];
        var M = (0, r.useState)(d || ke(s)), D = fe(M, 2), k = D[0], w = D[1];
        var C = (0, r.useState)(null), S = fe(C, 2), O = S[0], x = S[1];
        var E = (0, r.useRef)((0, c.createDOMID)("header"));
        var I = (0, r.useRef)();
        (0, r.useEffect)((function() {
            if (false) {}
        }), [ y ]);
        (0, r.useEffect)((function() {
            if (!d) {
                var e = setInterval((function() {
                    var e = ke(s);
                    if (e !== k) {
                        w(e);
                    }
                }), 1e3);
                return function() {
                    if (e !== undefined) {
                        clearInterval(e);
                    }
                };
            }
            return undefined;
        }), [ s, d, k ]);
        (0, r.useEffect)((function() {
            if (O === "day") {
                Ce(I);
            }
        }), [ b, O ]);
        var _ = (0, r.useCallback)((function(e, a) {
            var r = a.value;
            g(r);
            x("month");
        }), []);
        var j = (0, r.useCallback)((function(e, a) {
            var r = a.value;
            g(r);
            x("day");
        }), []);
        var q = (0, r.useCallback)((function(e) {
            de(I, e);
            de(n, e);
        }), [ n ]);
        
        return t().createElement(se, ve({
            inline: true,
            "aria-labelledby": o,
            "aria-describedby": a,
            "aria-required": f,
            "data-test": "calendar"
        }, h, {
            elementRef: q
        }),  t().createElement(ie, {
            headingId: E.current,
            locale: s,
            onChange: _,
            value: b
        }),  t().createElement(B, {
            displayValue: b,
            highlightToday: l,
            labelledBy: E.current,
            locale: s,
            onChange: v,
            onInternalChange: j,
            selectedValue: y,
            todayValue: k
        }));
    }
    Se.propTypes = De;
    /* harmony default export */    const Oe = Se;
    // CONCATENATED MODULE: ./src/Calendar/index.ts
    module.exports = a;
    /******/})();
