// @splunk/react-ui/Card.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        Body: () => /* reexport */ m,
        Footer: () => /* reexport */ k,
        Header: () => /* reexport */ Q,
        default: () => /* reexport */ qe
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var n =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "styled-components"
    const a = require("styled-components");
    var l =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const c = require("@splunk/react-ui/Box");
    var s =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/themes"
    const u = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Card/BodyStyles.ts
    var d = l()(s()).withConfig({
        displayName: "BodyStyles__StyledBox",
        componentId: "sc-1130udp-0"
    })([ "flex:1 1 auto;overflow:auto;height:100%;", ";", "" ], (function(e) {
        var r = e.$inset;
        return r ? (0, a.css)([ "", "" ], (0, u.pick)({
            enterprise: (0, a.css)([ "padding:", ";&:not(:first-child){padding-top:0;}" ], u.variables.spacingLarge),
            prisma: {
                comfortable: (0, a.css)([ "padding:12px 24px;&:first-child{padding-top:28px;}&:last-child{padding-bottom:28px;}" ]),
                compact: (0, a.css)([ "padding:8px 24px;&:first-child{padding-top:20px;}&:last-child{padding-bottom:20px;}" ])
            }
        })) : (0, a.css)([ "padding:0;" ]);
    }), (0, u.pick)({
        prisma: {
            comfortable: (0, a.css)([ "border-radius:", ";color:", ";" ], u.variables.borderRadius, u.variables.contentColorMuted),
            compact: (0, a.css)([ "border-radius:", ";color:", ";" ], u.variables.borderRadius, u.variables.contentColorMuted)
        }
    }));
    // CONCATENATED MODULE: ./src/Card/Body.tsx
    function p() {
        return p = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, p.apply(null, arguments);
    }
    function f(e, r) {
        if (null == e) return {};
        var t, n, o = b(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                t = i[n], r.includes(t) || {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function b(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (r.includes(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    var v = {
        children: i().node,
        elementRef: i().oneOfType([ i().func, i().object ]),
        inset: i().bool
    };
    /**
 * A styled container for `Card` body content.
 */    function y(e) {
        var r = e.children, t = e.inset, o = t === void 0 ? true : t, i = f(e, [ "children", "inset" ]);
        // @docs-props-type BodyPropsBase
                
        return n().createElement(d, p({
            "data-test": "body",
            $inset: o
        }, i), r);
    }
    y.propTypes = v;
    /* harmony default export */    const m = y;
    // CONCATENATED MODULE: ./src/Card/FooterStyles.ts
    var h = l()(s()).withConfig({
        displayName: "FooterStyles__StyledBox",
        componentId: "sc-1yu3r4s-0"
    })([ "", ";text-align:right;color:", ";", "" ], (0, u.pick)({
        enterprise: (0, a.css)([ "padding:", ";&:first-child{border-top:none;}" ], u.variables.spacingLarge),
        prisma: {
            comfortable: (0, a.css)([ "padding:24px;" ]),
            compact: "padding: 18px 24px"
        }
    }), u.variables.contentColorMuted, (function(e) {
        var r = e.$showTopBorder;
        return r && (0, u.pick)({
            enterprise: (0, a.css)([ "border-top:1px solid ", ";" ], u.variables.gray92)
        });
    }));
    // CONCATENATED MODULE: ./src/Card/Footer.tsx
    function g() {
        return g = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, g.apply(null, arguments);
    }
    function w(e, r) {
        if (null == e) return {};
        var t, n, o = O(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                t = i[n], r.includes(t) || {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function O(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (r.includes(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    var x = {
        children: i().node,
        elementRef: i().oneOfType([ i().func, i().object ]),
        showBorder: i().bool
    };
    /**
 * A styled container for `Card` footer content.
 */    function S(e) {
        var r = e.children, t = e.showBorder, o = t === void 0 ? true : t, i = w(e, [ "children", "showBorder" ]);
        // @docs-props-type FooterPropsBase
                if (false) {}
        
        return n().createElement(h, g({
            "data-test": "footer",
            $showTopBorder: o
        }, i), r);
    }
    S.propTypes = x;
    /* harmony default export */    const k = S;
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const j = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/react-icons/DotsThreeVertical"
    const C = require("@splunk/react-icons/DotsThreeVertical");
    var P =  e.n(C);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/MoreVertical"
    const _ = require("@splunk/react-icons/enterprise/MoreVertical");
    var E =  e.n(_);
    // CONCATENATED MODULE: external "@splunk/react-ui/Anchor"
    const T = require("@splunk/react-ui/Anchor");
    var B =  e.n(T);
    // CONCATENATED MODULE: external "@splunk/react-ui/Dropdown"
    const I = require("@splunk/react-ui/Dropdown");
    var R =  e.n(I);
    // CONCATENATED MODULE: external "@splunk/react-ui/Tooltip"
    const q = require("@splunk/react-ui/Tooltip");
    var N =  e.n(q);
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const H = require("@splunk/react-ui/Button");
    var $ =  e.n(H);
    // CONCATENATED MODULE: ./src/Card/HeaderStyles.ts
    var A = l().div.withConfig({
        displayName: "HeaderStyles__StyledTitle",
        componentId: "eqi6uk-0"
    })([ "", " ", " margin:0;overflow-wrap:break-word;padding:0;" ], u.mixins.reset("block"), (0, 
    u.pick)({
        enterprise: (0, a.css)([ "font-weight:", ";font-size:", ";line-height:1.2;" ], u.variables.fontWeightSemiBold, u.variables.fontSizeLarge),
        prisma: (0, a.css)([ "font-weight:", ";color:", ";" ], u.variables.fontWeightBold, u.variables.contentColorActive)
    }));
    var W = l().div.withConfig({
        displayName: "HeaderStyles__StyledSubtitle",
        componentId: "eqi6uk-1"
    })([ "", " ", " margin:0;overflow-wrap:break-word;padding:0;font-weight:normal;" ], u.mixins.reset("block"), (0, 
    u.pick)({
        enterprise: (0, a.css)([ "font-size:", ";color:", ";" ], u.variables.fontSizeSmall, u.variables.textGray),
        prisma: (0, a.css)([ "color:", ";" ], u.variables.contentColorDefault)
    }));
    var D = l().div.withConfig({
        displayName: "HeaderStyles__StyledTitleContainer",
        componentId: "eqi6uk-2"
    })([ "", " flex:0 1 auto;", " &:not(:last-child){margin-right:", ";}" ], u.mixins.reset("block"), (function(e) {
        var r = e.$truncateTitle;
        return r && (0, a.css)([ "overflow:hidden;& > ", ",& > ", "{overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}" ], /* sc-sel */ A, /* sc-sel */ W);
    }), u.variables.spacingSmall);
    var M = l().div.withConfig({
        displayName: "HeaderStyles__StyledActionContainer",
        componentId: "eqi6uk-3"
    })([ "", " align-items:center;display:flex;flex:0 0 auto;justify-content:space-between;", ";" ], u.mixins.reset("block"), (0, 
    u.pick)({
        prisma: {
            comfortable: (0, a.css)([ "margin-right:-12px;margin-top:-4px;" ]),
            compact: (0, a.css)([ "margin-right:-12px;margin-top:0;" ])
        }
    }));
    var F = l()($()).withConfig({
        displayName: "HeaderStyles__StyledSecondaryAction",
        componentId: "eqi6uk-4"
    })([ "margin-left:10px;" ]);
    var L = l()(s()).withConfig({
        displayName: "HeaderStyles__StyledBox",
        componentId: "eqi6uk-5"
    })([ "align-items:start;display:flex;flex:0 0 auto;justify-content:space-between;position:relative;min-height:30px;", " ", " >:not(", "):not(", "){flex:1 0 auto;}" ], (0, 
    u.pick)({
        enterprise: (0, a.css)([ "padding:", ";" ], u.variables.spacingLarge),
        prisma: {
            comfortable: (0, a.css)([ "padding:16px 24px;&:last-child{padding-bottom:32px;}" ]),
            compact: (0, a.css)([ "padding:12px 24px;&:last-child{padding-bottom:24px;}" ])
        }
    }), (0, u.pick)({
        prisma: (0, a.css)([ "border-top-left-radius:", ";border-top-right-radius:", ";" ], u.variables.borderRadius, u.variables.borderRadius)
    }), /* sc-sel */ D, /* sc-sel */ M);
    // CONCATENATED MODULE: ./src/Card/Header.tsx
    function z(e, r) {
        if (null == e) return {};
        var t, n, o = V(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                t = i[n], r.includes(t) || {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function V(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (r.includes(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    function G() {
        return G = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, G.apply(null, arguments);
    }
    var U = {
        actionPrimary: i().node,
        actionsSecondary: i().node,
        anchor: i().string,
        children: i().node,
        elementRef: i().oneOfType([ i().func, i().object ]),
        subtitle: i().node,
        title: i().node,
        truncateTitle: i().bool
    };
    // The order of the interfaces is necessary to maintain proper error messages: HeaderWithActions then HeaderWithActionsRenderProp
    // If the actions comes first the error will report that actions should be undefined
    // when we want actions to be the intended use case
    // type HeaderProps = ComponentProps<HeaderPropsBase, 'div'>;
    // `forwardRef` is used to enable focus on toggle when Dropdown is closing
        var J =  n().forwardRef((function(e, r) {
        var t = (0, u.useSplunkTheme)(), o = t.family, i = t.isCompact;
        var a = i ? "20px" : "24px";
        var l = o === "prisma" ?  n().createElement(P(), {
            height: a,
            width: a
        }) :  n().createElement(E(), null);
        
        return n().createElement(N(), G({
            content: (0, j._)("Actions"),
            contentRelationship: "label"
        }, e),  n().createElement(F, {
            appearance: "secondary",
            "data-test": "actions-secondary-toggle",
            icon: l,
            elementRef: r
        }));
    }));
    /**
 * A styled container for `Card` header content.
 */    function K(e) {
        var r = e.actions, t = e.actionPrimary, o = e.actionsSecondary, i = e.anchor, a = e.children, l = e.subtitle, c = e.title, s = e.truncateTitle, d = s === void 0 ? true : s, p = z(e, [ "actions", "actionPrimary", "actionsSecondary", "anchor", "children", "subtitle", "title", "truncateTitle" ]);
        // @docs-props-type HeaderPropsBase
                var f = (0, u.useSplunkTheme)(), b = f.family;
        var v = b === "prisma" ? "right" : "below";
        if (false) {}
        var y;
        if (r) {
            y = r();
        } else if (t || o) {
            y =  n().createElement(n().Fragment, null, t, o &&  n().createElement(R(), {
                defaultPlacement: v,
                toggle:  n().createElement(J, null)
            }, o));
        }
        
        return n().createElement(L, G({
            "data-test": "header"
        }, p), (c || l) &&  n().createElement(D, {
            $truncateTitle: d
        }, c &&  n().createElement(A, null, i ?  n().createElement(B(), {
            name: i
        }, c) : c), l &&  n().createElement(W, null, l)), a, y &&  n().createElement(M, null, y));
    }
    K.propTypes = U;
    /* harmony default export */    const Q = K;
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const X = require("@splunk/react-ui/Clickable");
    var Y =  e.n(X);
    // CONCATENATED MODULE: ./src/Card/CardStyles.ts
    var Z = "0.2s";
    var ee = (0, a.css)([ "", ";flex-direction:column;align-items:stretch;min-width:100px;flex:1;vertical-align:top;", " transition:height ", ",width ", ",min-width ", ",max-width ", ",margin ", ",box-shadow ", ",border-color ", ";", " &[data-clickable='true']{cursor:pointer;", " &:hover{", "}&:focus{", "}}" ], u.mixins.reset("inline-flex"), (0, 
    u.pick)({
        enterprise: (0, a.css)([ "background-color:", ";border:1px solid transparent;" ], u.variables.backgroundColor),
        prisma: (0, a.css)([ "background-color:", ";border-radius:", ";box-shadow:", ";" ], u.variables.backgroundColorSection, u.variables.borderRadius, u.variables.embossShadow)
    }), Z, Z, Z, Z, Z, Z, Z, (function(e) {
        var r = e.$cardHasBorder;
        return r && (0, u.pick)({
            enterprise: (0, a.css)([ "border-color:", ";" ], u.variables.borderLightColor)
        });
    }), (function(e) {
        var r = e.$selected;
        return r && (0, u.pick)({
            enterprise: (0, a.css)([ "border-color:", ";" ], u.variables.focusColor),
            prisma: (0, a.css)([ "background-color:", ";" ], u.mixins.overlayColors(u.variables.backgroundColorSection, u.variables.interactiveColorOverlaySelected))
        });
    }), (0, u.pick)({
        enterprise: (0, a.css)([ "box-shadow:", ";" ], u.variables.overlayShadow),
        prisma: (0, a.css)([ "box-shadow:", ";background-color:", ";" ], u.variables.embossShadow, u.mixins.overlayColors(u.variables.backgroundColorSection, u.variables.interactiveColorOverlayHover))
    }), (0, u.pick)({
        enterprise: (0, a.css)([ "box-shadow:", ";" ], u.variables.focusShadow),
        prisma: (0, a.css)([ "box-shadow:", ",", ";background-color:", ";" ], u.variables.embossShadow, u.variables.focusShadow, u.mixins.overlayColors(u.variables.backgroundColorSection, u.variables.interactiveColorOverlayHover))
    }));
    var re = l().article.withConfig({
        displayName: "CardStyles__Styled",
        componentId: "ola3x0-0"
    })([ "", "" ], ee);
    var te = l()(Y()).withConfig({
        displayName: "CardStyles__StyledClickable",
        componentId: "ola3x0-1"
    })([ "", "" ], ee);
    /* A child element we add to <button>s to make flexbox work.
   Without this wrapper element, child elements won't flex
   in Firefox.
   This unfortunately exposes some fragility in IE (e.g., adding
   a flex: 1 style here will cause the card to collapse) so test
   across the supported browsers if making a change here.
*/    var ne = l().div.withConfig({
        displayName: "CardStyles__StyledFirefoxFlexHack",
        componentId: "ola3x0-2"
    })([ "display:flex;flex:1 1 auto;flex-direction:column;align-items:stretch;justify-content:stretch;height:100%;", "" ], (0, 
    u.pick)({
        prisma: (0, a.css)([ "border-radius:", ";" ], u.variables.borderRadius)
    }));
    // CONCATENATED MODULE: ./src/Card/Card.tsx
    function oe(e) {
        "@babel/helpers - typeof";
        return oe = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, oe(e);
    }
    function ie() {
        return ie = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, ie.apply(null, arguments);
    }
    function ae(e, r) {
        return de(e) || ue(e, r) || ce(e, r) || le();
    }
    function le() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function ce(e, r) {
        if (e) {
            if ("string" == typeof e) return se(e, r);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? se(e, r) : void 0;
        }
    }
    function se(e, r) {
        (null == r || r > e.length) && (r = e.length);
        for (var t = 0, n = Array(r); t < r; t++) {
            n[t] = e[t];
        }
        return n;
    }
    function ue(e, r) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var n, o, i, a, l = [], c = !0, s = !1;
            try {
                if (i = (t = t.call(e)).next, 0 === r) {
                    if (Object(t) !== t) return;
                    c = !1;
                } else for (;!(c = (n = i.call(t)).done) && (l.push(n.value), l.length !== r); c = !0) {
                }
            } catch (e) {
                s = !0, o = e;
            } finally {
                try {
                    if (!c && null != t["return"] && (a = t["return"](), Object(a) !== a)) return;
                } finally {
                    if (s) throw o;
                }
            }
            return l;
        }
    }
    function de(e) {
        if (Array.isArray(e)) return e;
    }
    function pe(e, r) {
        if (!(e instanceof r)) throw new TypeError("Cannot call a class as a function");
    }
    function fe(e, r) {
        for (var t = 0; t < r.length; t++) {
            var n = r[t];
            n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), 
            Object.defineProperty(e, je(n.key), n);
        }
    }
    function be(e, r, t) {
        return r && fe(e.prototype, r), t && fe(e, t), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function ve(e, r) {
        if ("function" != typeof r && null !== r) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(r && r.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), r && ye(e, r);
    }
    function ye(e, r) {
        return ye = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, r) {
            return e.__proto__ = r, e;
        }, ye(e, r);
    }
    function me(e) {
        var r = we();
        return function() {
            var t, n = Oe(e);
            if (r) {
                var o = Oe(this).constructor;
                t = Reflect.construct(n, arguments, o);
            } else t = n.apply(this, arguments);
            return he(this, t);
        };
    }
    function he(e, r) {
        if (r && ("object" == oe(r) || "function" == typeof r)) return r;
        if (void 0 !== r) throw new TypeError("Derived constructors may only return object or undefined");
        return ge(e);
    }
    function ge(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function we() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return (we = function r() {
            return !!e;
        })();
    }
    function Oe(e) {
        return Oe = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, Oe(e);
    }
    function xe(e, r) {
        var t = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            r && (n = n.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }))), t.push.apply(t, n);
        }
        return t;
    }
    function Se(e) {
        for (var r = 1; r < arguments.length; r++) {
            var t = null != arguments[r] ? arguments[r] : {};
            r % 2 ? xe(Object(t), !0).forEach((function(r) {
                ke(e, r, t[r]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : xe(Object(t)).forEach((function(r) {
                Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
            }));
        }
        return e;
    }
    function ke(e, r, t) {
        return (r = je(r)) in e ? Object.defineProperty(e, r, {
            value: t,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = t, e;
    }
    function je(e) {
        var r = Ce(e, "string");
        return "symbol" == oe(r) ? r : r + "";
    }
    function Ce(e, r) {
        if ("object" != oe(e) || !e) return e;
        var t = e[Symbol.toPrimitive];
        if (void 0 !== t) {
            var n = t.call(e, r || "default");
            if ("object" != oe(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    function Pe(e, r) {
        if (null == e) return {};
        var t, n, o = _e(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                t = i[n], r.includes(t) || {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function _e(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (r.includes(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    /** @public */    var Ee = {
        children: i().node,
        elementRef: i().oneOfType([ i().func, i().object ]),
        /** @private */
        margin: i().number,
        /** @private */
        maxWidth: i().oneOfType([ i().number, i().string ]),
        /** @private */
        minWidth: i().oneOfType([ i().number, i().string ]),
        onClick: i().func,
        openInNewContext: i().bool,
        selected: i().bool,
        showBorder: i().bool,
        to: i().string,
        value: i().any,
        tag: i().oneOf([ "article", "div", "aside", "section" ])
    };
    var Te = {
        openInNewContext: false,
        showBorder: true
    };
    function Be(e) {
        return !!e.onClick || !!e.to;
    }
    function Ie(e) {
        var r = e.margin, t = e.maxWidth, n = e.minWidth, o = e.showBorder, i = e.style, a = Pe(e, [ "margin", "maxWidth", "minWidth", "showBorder", "style" ]);
        var l = Se({
            minWidth: n,
            maxWidth: t,
            margin: r
        }, i);
        var c = {
            showBorder: o,
            "data-test": "card",
            style: l
        };
        return [ c, a ];
    }
    var Re =  function(e) {
        ve(t, e);
        var r = me(t);
        function t() {
            var e;
            pe(this, t);
            for (var n = arguments.length, o = new Array(n), i = 0; i < n; i++) {
                o[i] = arguments[i];
            }
            e = r.call.apply(r, [ this ].concat(o));
            ke(ge(e), "handleCardClick", (function(r) {
                var t, n;
                var o = e.props, i = o.selected, a = o.value;
                (t = (n = e.props).onClick) === null || t === void 0 ? void 0 : t.call(n, r, {
                    selected: i || false,
                    value: a
                });
            }));
            return e;
        }
 // @docs-props-type CardPropsBase
                be(t, [ {
            key: "render",
            value: function e() {
                if (false) {}
                if (Be(this.props)) {
                    var r = Ie(this.props), t = ae(r, 2), o = t[0], i = t[1];
                    var a = o.showBorder;
                    var l = i.children, c = i.elementRef, s = i.onClick, u = i.openInNewContext, d = i.selected, p = i.to, f = Pe(i, [ "children", "elementRef", "onClick", "openInNewContext", "selected", "to" ]);
                    
                    return n().createElement(te, ie({
                        "data-clickable": true,
                        $selected: d || false,
                        $cardHasBorder: a,
                        elementRef: c
                    }, o, f, {
                        onClick: s ? this.handleCardClick : undefined,
                        openInNewContext: u,
                        to: p || undefined
                    }),  n().createElement(ne, null, l));
                }
                var b = Ie(this.props), v = ae(b, 2), y = v[0], m = v[1];
                var h = y.showBorder;
                var g = m.children, w = m.elementRef, O = m.tag, x = Pe(m, [ "children", "elementRef", "tag" ]);
                
                return n().createElement(re, ie({
                    $selected: false,
                    $cardHasBorder: h,
                    ref: w,
                    as: O
                }, y, x, {
                    onClick: undefined
                }), g);
            }
        } ]);
        return t;
    }(t.Component);
    ke(Re, "propTypes", Ee);
    ke(Re, "defaultProps", Te);
    ke(Re, "Header", Q);
    ke(Re, "Body", m);
    ke(Re, "Footer", k);
    /* harmony default export */    const qe = Re;
    // CONCATENATED MODULE: ./src/Card/index.ts
    module.exports = r;
    /******/})();
