// @splunk/react-ui/CloseButton.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ j
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var n =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var i =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/react-icons/Cross"
    const o = require("@splunk/react-icons/Cross");
    var l =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/Close"
    const s = require("@splunk/react-icons/enterprise/Close");
    var u =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/themes"
    const c = require("@splunk/themes");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const p = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "styled-components"
    const d = require("styled-components");
    var f =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/react-ui/ButtonSimple"
    const v = require("@splunk/react-ui/ButtonSimple");
    var b =  e.n(v);
    // CONCATENATED MODULE: ./src/CloseButton/CloseButtonStyles.ts
    var y = f()(b()).withConfig({
        displayName: "CloseButtonStyles__StyledButtonSimple",
        componentId: "b3h40f-0"
    })([ "float:right;width:", ";height:", ";text-align:center;" ], c.variables.inputHeight, c.variables.inputHeight);
    // CONCATENATED MODULE: ./src/CloseButton/CloseButton.tsx
    function m() {
        return m = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, m.apply(null, arguments);
    }
    function g(e, r) {
        if (null == e) return {};
        var t, n, a = h(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                t = i[n], r.includes(t) || {}.propertyIsEnumerable.call(e, t) && (a[t] = e[t]);
            }
        }
        return a;
    }
    function h(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (r.includes(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    var O = {
        disabled: i().bool,
        elementRef: i().oneOfType([ i().func, i().object ]),
        screenReaderText: i().string
    };
    function S(e) {
        var r = e.disabled, t = r === void 0 ? false : r, a = e.screenReaderText, i = a === void 0 ? (0, 
        p._)("Close") : a, o = g(e, [ "disabled", "screenReaderText" ]);
        var s = (0, c.useSplunkTheme)(), d = s.isEnterprise, f = s.isCompact;
        var v = f ? "20px" : "24px";
        
        return n().createElement(y, m({
            appearance: "pill",
            disabled: t,
            "aria-label": i
        }, o), d ?  n().createElement(u(), {
            "data-test": "cross",
            inline: false,
            screenReaderText: null,
            size: "13px"
        }) :  n().createElement(l(), {
            "data-test": "cross",
            height: v,
            width: v
        }));
    }
    S.propTypes = O;
    /* harmony default export */    const j = S;
    // CONCATENATED MODULE: ./src/CloseButton/index.ts
    module.exports = r;
    /******/})();
