// @splunk/react-ui/CollapsiblePanel.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var n = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, n) => {
            /******/ for (var r in n) {
                /******/ if (e.o(n, r) && !e.o(t, r)) {
                    /******/ Object.defineProperty(t, r, {
                        enumerable: true,
                        get: n[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        SingleOpenPanelGroup: () => /* reexport */ ce,
        default: () => /* reexport */ je
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var r =  e.n(n);
    // CONCATENATED MODULE: external "lodash/has"
    const a = require("lodash/has");
    var o =  e.n(a);
    // CONCATENATED MODULE: external "lodash/keys"
    const i = require("lodash/keys");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "lodash/omit"
    const s = require("lodash/omit");
    var c =  e.n(s);
    // CONCATENATED MODULE: external "prop-types"
    const p = require("prop-types");
    var u =  e.n(p);
    // CONCATENATED MODULE: external "@splunk/react-ui/TransitionOpen"
    const d = require("@splunk/react-ui/TransitionOpen");
    var f =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const b = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const v = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/react-ui/usePrevious"
    const m = require("@splunk/react-ui/usePrevious");
    var y =  e.n(m);
    // CONCATENATED MODULE: external "styled-components"
    const h = require("styled-components");
    var g =  e.n(h);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const k = require("@splunk/react-ui/Box");
    var C =  e.n(k);
    // CONCATENATED MODULE: external "@splunk/react-ui/ButtonSimple"
    const x = require("@splunk/react-ui/ButtonSimple");
    var O =  e.n(x);
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const S = require("@splunk/react-ui/Clickable");
    var I =  e.n(S);
    // CONCATENATED MODULE: external "@splunk/themes"
    const P = require("@splunk/themes");
    // CONCATENATED MODULE: external "@splunk/react-icons/ChevronRight"
    const w = require("@splunk/react-icons/ChevronRight");
    var j =  e.n(w);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/ChevronRight"
    const E = require("@splunk/react-icons/enterprise/ChevronRight");
    var _ =  e.n(E);
    // CONCATENATED MODULE: ./src/CollapsiblePanel/icons/ExpandPanel.tsx
    var q = {
        className: u().string
    };
    var R = function e(t) {
        var n = t.className;
        var a = (0, P.useSplunkTheme)(), o = a.isEnterprise;
        return o ?  r().createElement(_(), {
            className: n,
            hideDefaultTooltip: true,
            screenReaderText: null
        }) :  r().createElement(j(), {
            className: n
        });
    };
    R.propTypes = q;
    /* harmony default export */    const A = R;
    // CONCATENATED MODULE: ./src/CollapsiblePanel/CollapsiblePanelStyles.ts
    var D = g()(C()).withConfig({
        displayName: "CollapsiblePanelStyles__StyledBox",
        componentId: "sc-1phikbp-0"
    })([ "display:flex;flex-direction:column;flex-shrink:1;", ";", ";& + &{margin-top:", ";}" ], (0, 
    P.pick)({
        enterprise: (0, h.css)([ "min-height:calc(", " + 12px);" ], P.variables.lineHeight),
        prisma: {
            comfortable: (0, h.css)([ "min-height:56px;" ]),
            compact: (0, h.css)([ "min-height:48px;" ])
        }
    }), (0, P.pick)({
        prisma: (0, h.css)([ "background-color:", ";" ], P.variables.backgroundColorSection)
    }), (0, P.pick)({
        enterprise: "2px",
        prisma: "1px"
    }));
    var T = g()(C()).withConfig({
        displayName: "CollapsiblePanelStyles__StyledSingleOpenPanelGroupBox",
        componentId: "sc-1phikbp-1"
    })([ "", ";flex-direction:column;margin-bottom:0;", "" ], P.mixins.reset("flex"), (0, 
    P.pick)({
        enterprise: (0, h.css)([ "background:", ";" ], P.variables.backgroundColor)
    }));
    /*
    these styles are consistent for both regular and titleWithActions, but applied to different elements
    * in regular mode, they're on the SVG's span wrapper
    * in titleWithActionsMode, they're on the button
    applying them directly to the icon wrapper doesn't work because Enterprise has 1px border around the button
*/    var N = (0, h.css)([ "min-width:", ";" ], (0, P.pick)({
        enterprise: "20px",
        prisma: "35px"
    }));
    var X = g().span.withConfig({
        displayName: "CollapsiblePanelStyles__StyledIconWrapper",
        componentId: "sc-1phikbp-2"
    })([ "display:inline-block;flex:0 auto;", "" ], N);
    var W = g().div.withConfig({
        displayName: "CollapsiblePanelStyles__StyledTitleWithActions",
        componentId: "sc-1phikbp-3"
    })([ "display:flex;flex-shrink:0;border-bottom:0;position:relative;padding:", ";transition:background-color 0.2s,border 0.2s,box-shadow 0.2s;color:", ";", " ", "" ], (0, 
    P.pick)({
        enterprise: (0, h.css)([ "", " ", " ", " ", "" ], P.variables.spacingXSmall, P.variables.spacingSmall, P.variables.spacingXSmall, P.variables.spacingXSmall),
        prisma: {
            comfortable: "8px 10px 8px 11px",
            compact: "8px 10px 8px 11px"
        }
    }), (0, P.pick)({
        enterprise: P.variables.textColor,
        prisma: P.variables.contentColorActive
    }), (0, P.pick)({
        prisma: (0, h.css)([ "background-color:", ";" ], P.variables.interactiveColorBackground),
        enterprise: {
            dark: (0, h.css)([ "background-color:", ";" ], P.variables.gray45),
            light: (0, h.css)([ "background-color:", ";" ], P.variables.gray96)
        }
    }), (function(e) {
        var t = e.$disabled;
        return t && (0, h.css)([ "cursor:default;color:", ";" ], (0, P.pick)({
            enterprise: P.variables.contentColorDefault,
            prisma: P.variables.contentColorDisabled
        }));
    }));
    var B = g()(O()).withConfig({
        displayName: "CollapsiblePanelStyles__StyledToggleButton",
        componentId: "sc-1phikbp-4"
    })([ "flex:0 auto;padding:0;", "" ], N);
    var L = g()(I()).withConfig({
        displayName: "CollapsiblePanelStyles__StyledTitleClickable",
        componentId: "sc-1phikbp-5"
    })([ "width:100%;flex-shrink:0;border-bottom:0;cursor:pointer;position:relative;display:flex;transition:background-color 0.2s,border 0.2s,box-shadow 0.2s;padding:", ";color:", ";", " ", " &[disabled]{cursor:default;color:", ";}&:focus:not([disabled]),&:active:not([disabled]){color:", ";box-shadow:", ";z-index:1;}&:hover:not([disabled]){color:", ";background-color:", ";}" ], (0, 
    P.pick)({
        enterprise: "6px 30px 6px 5px",
        prisma: {
            comfortable: "18px 20px 18px 11px",
            compact: "14px 20px 14px 11px"
        }
    }), (0, P.pick)({
        enterprise: P.variables.textColor,
        prisma: P.variables.contentColorActive
    }), (0, P.pick)({
        prisma: (0, h.css)([ "background-color:", ";" ], P.variables.interactiveColorBackground)
    }), (0, P.pickVariant)("$open", {
        false: {
            enterprise: {
                dark: (0, h.css)([ "background-color:", ";" ], P.variables.gray45),
                light: (0, h.css)([ "background-color:", ";" ], P.variables.gray96)
            }
        }
    }), (0, P.pick)({
        enterprise: P.variables.contentColorDefault,
        prisma: P.variables.contentColorDisabled
    }), (0, P.pick)({
        enterprise: P.variables.textColor,
        prisma: P.variables.contentColorActive
    }), P.variables.focusShadowInset, (0, P.pick)({
        enterprise: P.variables.textColor,
        prisma: P.variables.contentColorActive
    }), (0, P.pick)({
        enterprise: {
            light: P.variables.gray92,
            dark: P.variables.gray30
        },
        prisma: P.mixins.overlayColors(P.variables.interactiveColorBackground, P.variables.interactiveColorOverlayHover)
    }));
    var $ = g().span.withConfig({
        displayName: "CollapsiblePanelStyles__StyledHeadingContent",
        componentId: "sc-1phikbp-6"
    })([ "display:flex;justify-content:space-between;margin-left:", ";flex:1 1 auto;", " ", "" ], (0, 
    P.pick)({
        enterprise: "5px",
        prisma: "6px"
    }), (0, P.pick)({
        prisma: (0, h.css)([ "font-weight:", ";" ], P.variables.fontWeightSemiBold)
    }), (function(e) {
        var t = e.$titleWithActions;
        return t && (0, h.css)([ "min-height:", ";" ], (0, P.pick)({
            prisma: {
                comfortable: P.variables.spacingXXXLarge,
                compact: P.variables.spacingXXLarge
            }
        }));
    }));
    var M = g().span.withConfig({
        displayName: "CollapsiblePanelStyles__StyledHeadingActions",
        componentId: "sc-1phikbp-7"
    })([ "min-height:", ";flex:1 0 auto;" ], (0, P.pick)({
        prisma: {
            comfortable: P.variables.spacingXXXLarge,
            compact: P.variables.spacingXXLarge
        }
    }));
    var G = g().span.withConfig({
        displayName: "CollapsiblePanelStyles__StyledDescription",
        componentId: "sc-1phikbp-8"
    })([ "font-size:", ";margin-left:", ";" ], P.variables.fontSizeSmall, (0, P.pick)({
        enterprise: P.variables.spacingSmall,
        prisma: P.variables.spacingLarge
    }));
    var H = g().div.withConfig({
        displayName: "CollapsiblePanelStyles__StyledInsetWrapper",
        componentId: "sc-1phikbp-9"
    })([ "", ";padding:", ";" ], P.mixins.reset("block"), (0, P.pickVariant)("$inset", {
        true: (0, P.pick)({
            enterprise: {
                compact: P.variables.spacingXSmall,
                comfortable: P.variables.spacingSmall
            },
            prisma: {
                compact: P.variables.spacingMedium,
                comfortable: P.variables.spacingXLarge
            }
        }),
        false: 0
    }));
    var U = g()(A).withConfig({
        displayName: "CollapsiblePanelStyles__StyledExpandPanelIcon",
        componentId: "sc-1phikbp-10"
    })([ "transition:transform ", ";", ";", " width:100%;vertical-align:", ";" ], (0, 
    P.pick)({
        enterprise: "300ms",
        prisma: "200ms"
    }), (0, P.pick)({
        enterprise: (0, h.css)([ "[disabled] &{color:", ";}" ], (0, P.pick)({
            dark: P.variables.contentColorDefault,
            light: P.variables.contentColorDisabled
        })),
        prisma: (0, h.css)([ "color:", ";[disabled] &{color:inherit;}" ], P.variables.contentColorDefault)
    }), (function(e) {
        var t = e.open;
        return t && (0, h.css)([ "transform:rotate(90deg);" ]);
    }), (0, P.pick)({
        enterprise: "baseline",
        prisma: "text-bottom"
    }));
    // CONCATENATED MODULE: ./src/CollapsiblePanel/SingleOpenPanelGroupContext.tsx
    var z =  (0, n.createContext)({});
    /* harmony default export */    const V = z;
    // CONCATENATED MODULE: ./src/CollapsiblePanel/SingleOpenPanelGroup.tsx
    function F() {
        F = Object.assign || function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var n = arguments[t];
                for (var r in n) {
                    if (Object.prototype.hasOwnProperty.call(n, r)) {
                        e[r] = n[r];
                    }
                }
            }
            return e;
        };
        return F.apply(this, arguments);
    }
    function J(e) {
        return Y(e) || Q(e) || te(e) || K();
    }
    function K() {
        throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function Q(e) {
        if (typeof Symbol !== "undefined" && Symbol.iterator in Object(e)) return Array.from(e);
    }
    function Y(e) {
        if (Array.isArray(e)) return ne(e);
    }
    function Z(e, t) {
        return ae(e) || re(e, t) || te(e, t) || ee();
    }
    function ee() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function te(e, t) {
        if (!e) return;
        if (typeof e === "string") return ne(e, t);
        var n = Object.prototype.toString.call(e).slice(8, -1);
        if (n === "Object" && e.constructor) n = e.constructor.name;
        if (n === "Map" || n === "Set") return Array.from(e);
        if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return ne(e, t);
    }
    function ne(e, t) {
        if (t == null || t > e.length) t = e.length;
        for (var n = 0, r = new Array(t); n < t; n++) {
            r[n] = e[n];
        }
        return r;
    }
    function re(e, t) {
        if (typeof Symbol === "undefined" || !(Symbol.iterator in Object(e))) return;
        var n = [];
        var r = true;
        var a = false;
        var o = undefined;
        try {
            for (var i = e[Symbol.iterator](), l; !(r = (l = i.next()).done); r = true) {
                n.push(l.value);
                if (t && n.length === t) break;
            }
        } catch (e) {
            a = true;
            o = e;
        } finally {
            try {
                if (!r && i["return"] != null) i["return"]();
            } finally {
                if (a) throw o;
            }
        }
        return n;
    }
    function ae(e) {
        if (Array.isArray(e)) return e;
    }
    function oe(e, t) {
        if (e == null) return {};
        var n = ie(e, t);
        var r, a;
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (a = 0; a < o.length; a++) {
                r = o[a];
                if (t.indexOf(r) >= 0) continue;
                if (!Object.prototype.propertyIsEnumerable.call(e, r)) continue;
                n[r] = e[r];
            }
        }
        return n;
    }
    function ie(e, t) {
        if (e == null) return {};
        var n = {};
        var r = Object.keys(e);
        var a, o;
        for (o = 0; o < r.length; o++) {
            a = r[o];
            if (t.indexOf(a) >= 0) continue;
            n[a] = e[a];
        }
        return n;
    }
    /** @public */    var le = {
        children: u().node,
        defaultOpenPanelId: u().any,
        elementRef: u().oneOfType([ u().func, u().object ]),
        inset: u().bool,
        onChange: u().func,
        openPanelId: u().any
    };
    function se(e) {
        // @docs-props-type SingleOpenPanelGroupPropsBase
        var t = e.children, a = e.defaultOpenPanelId, i = e.elementRef, l = e.inset, s = l === void 0 ? false : l, p = e.onChange, u = e.openPanelId, d = oe(e, [ "children", "defaultOpenPanelId", "elementRef", "inset", "onChange", "openPanelId" ]);
        var f = o()(e, "openPanelId");
        var b = y()(a);
        var v = !f ? a : undefined;
        var m = (0, n.useState)(v), h = Z(m, 2), g = h[0], k = h[1];
        var C = function e() {
            var t = f ? u : g;
            return t;
        };
        var x = (0, n.useCallback)((function(e, t) {
            var n = t.panelId;
            var r = C();
            var a = r === n ? null : n;
            if (!f) {
                k(a);
            }
            p === null || p === void 0 ? void 0 : p(e, t);
        }), [ u, f, g, C, p ]);
        (0, n.useEffect)((function() {
            if (false) {}
 // In this functional component, we want the dependencies to be [], but when we do that
            // we get a linting error.
            // eslint-disable-next-line react-hooks/exhaustive-deps
                }), []);
        (0, n.useEffect)((function() {
            if (false) {}
        }), [ b, f ]);
        (0, n.useEffect)((function() {
            if (false) {}
        }), [ b, a ]);
        var O = n.Children.toArray(t).filter(n.isValidElement);
 // remove false, null, 0, etc
                var S = (0, n.useMemo)((function() {
            return {
                onChange: x,
                openPanelId: f ? u : g,
                inset: s
            };
        }), [ x, f, s, u, g ]);
        
        return r().createElement(T, F({
            "data-test": "single-open-panel-group",
            "data-test-open-panel-id": u,
            elementRef: i
        }, c().apply(void 0, [ d ].concat(J(Object.keys(le))))),  r().createElement(V.Provider, {
            value: S
        }, O));
    }
    se.propTypes = le;
    /* harmony default export */    const ce = se;
    // CONCATENATED MODULE: ./src/CollapsiblePanel/CollapsiblePanel.tsx
    function pe(e) {
        "@babel/helpers - typeof";
        if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") {
            pe = function e(t) {
                return typeof t;
            };
        } else {
            pe = function e(t) {
                return t && typeof Symbol === "function" && t.constructor === Symbol && t !== Symbol.prototype ? "symbol" : typeof t;
            };
        }
        return pe(e);
    }
    function ue() {
        ue = Object.assign || function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var n = arguments[t];
                for (var r in n) {
                    if (Object.prototype.hasOwnProperty.call(n, r)) {
                        e[r] = n[r];
                    }
                }
            }
            return e;
        };
        return ue.apply(this, arguments);
    }
    function de(e, t) {
        var n = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var r = Object.getOwnPropertySymbols(e);
            if (t) r = r.filter((function(t) {
                return Object.getOwnPropertyDescriptor(e, t).enumerable;
            }));
            n.push.apply(n, r);
        }
        return n;
    }
    function fe(e) {
        for (var t = 1; t < arguments.length; t++) {
            var n = arguments[t] != null ? arguments[t] : {};
            if (t % 2) {
                de(Object(n), true).forEach((function(t) {
                    Se(e, t, n[t]);
                }));
            } else if (Object.getOwnPropertyDescriptors) {
                Object.defineProperties(e, Object.getOwnPropertyDescriptors(n));
            } else {
                de(Object(n)).forEach((function(t) {
                    Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(n, t));
                }));
            }
        }
        return e;
    }
    function be(e, t) {
        if (!(e instanceof t)) {
            throw new TypeError("Cannot call a class as a function");
        }
    }
    function ve(e, t) {
        for (var n = 0; n < t.length; n++) {
            var r = t[n];
            r.enumerable = r.enumerable || false;
            r.configurable = true;
            if ("value" in r) r.writable = true;
            Object.defineProperty(e, r.key, r);
        }
    }
    function me(e, t, n) {
        if (t) ve(e.prototype, t);
        if (n) ve(e, n);
        return e;
    }
    function ye(e, t) {
        if (typeof t !== "function" && t !== null) {
            throw new TypeError("Super expression must either be null or a function");
        }
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: true,
                configurable: true
            }
        });
        if (t) he(e, t);
    }
    function he(e, t) {
        he = Object.setPrototypeOf || function e(t, n) {
            t.__proto__ = n;
            return t;
        };
        return he(e, t);
    }
    function ge(e) {
        var t = xe();
        return function n() {
            var r = Oe(e), a;
            if (t) {
                var o = Oe(this).constructor;
                a = Reflect.construct(r, arguments, o);
            } else {
                a = r.apply(this, arguments);
            }
            return ke(this, a);
        };
    }
    function ke(e, t) {
        if (t && (pe(t) === "object" || typeof t === "function")) {
            return t;
        }
        return Ce(e);
    }
    function Ce(e) {
        if (e === void 0) {
            throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        }
        return e;
    }
    function xe() {
        if (typeof Reflect === "undefined" || !Reflect.construct) return false;
        if (Reflect.construct.sham) return false;
        if (typeof Proxy === "function") return true;
        try {
            Date.prototype.toString.call(Reflect.construct(Date, [], (function() {})));
            return true;
        } catch (e) {
            return false;
        }
    }
    function Oe(e) {
        Oe = Object.setPrototypeOf ? Object.getPrototypeOf : function e(t) {
            return t.__proto__ || Object.getPrototypeOf(t);
        };
        return Oe(e);
    }
    function Se(e, t, n) {
        if (t in e) {
            Object.defineProperty(e, t, {
                value: n,
                enumerable: true,
                configurable: true,
                writable: true
            });
        } else {
            e[t] = n;
        }
        return e;
    }
    var Ie = {
        innerBodyStyles: u().object,
        actions: u().node,
        children: u().node,
        defaultOpen: u().bool,
        description: u().string,
        disabled: u().bool,
        elementRef: u().oneOfType([ u().func, u().object ]),
        headingLevel: u().number,
        inset: u().bool,
        onRequestClose: u().func,
        onRequestOpen: u().func,
        onChange: u().func,
        open: u().bool,
        overflow: u().string,
        panelId: u().any,
        renderChildrenWhenCollapsed: u().bool,
        title: u().node.isRequired,
        titleWithActions: u().bool
    };
    var Pe = {
        disabled: false,
        overflow: "auto",
        renderChildrenWhenCollapsed: false,
        titleWithActions: false,
        inset: false
    };
    var we =  function(e) {
        ye(n, e);
        var t = ge(n);
        // @docs-props-type CollapsiblePanelPropsBase
                function n(e) {
            var r;
            be(this, n);
            r = t.call(this, e);
            Se(Ce(r), "controlledExternally", void 0);
            Se(Ce(r), "containerId", void 0);
            Se(Ce(r), "toggleId", void 0);
            Se(Ce(r), "handleRequestClose", (function(e) {
                var t = r.props, n = t.panelId, a = t.onRequestClose;
                a === null || a === void 0 ? void 0 : a({
                    event: e,
                    panelId: n,
                    reason: "toggleClick"
                });
            }));
            Se(Ce(r), "handleRequestOpen", (function(e) {
                var t = r.props, n = t.panelId, a = t.onRequestOpen;
                a === null || a === void 0 ? void 0 : a({
                    event: e,
                    panelId: n,
                    reason: "toggleClick"
                });
            }));
            Se(Ce(r), "handleChange", (function(e) {
                var t, n;
                var a = r.props, o = a.panelId, i = a.onChange;
                var l = r.isOpen();
                if (!r.isControlled()) {
                    r.setState({
                        animating: true,
                        open: !l
                    });
                }
                var s = l ? "close" : "open";
                i === null || i === void 0 ? void 0 : i(e, {
                    panelId: o,
                    reason: "toggleClick",
                    action: s
                });
                (t = (n = r.context).onChange) === null || t === void 0 ? void 0 : t.call(n, e, {
                    panelId: o,
                    reason: "toggleClick",
                    action: s
                });
 // Maintains backwards compatibility until the `handleRequestClose` and `handleRequestOpen` handlers are removed in SUI-5
                                if (l) {
                    r.handleRequestClose(e);
                } else {
                    r.handleRequestOpen(e);
                }
            }));
            Se(Ce(r), "handleAnimationEnd", (function() {
                r.setState({
                    animating: false
                });
            }));
            r.controlledExternally = o()(e, "open");
            r.state = fe(fe({}, !r.isControlled() ? {
                open: e.defaultOpen || false
            } : undefined), {}, {
                animating: false
            });
            if (false) {}
            if (false) {}
            if (false) {}
            r.containerId = "container-".concat((0, b.createGUID)());
            r.toggleId = "toggle-".concat((0, b.createGUID)());
            return r;
        }
 // Only testing this in the scenario of a SingleOpenPanelGroup.
                me(n, [ {
            key: "componentDidMount",
            value: function e() {
                if (false) {}
            }
        }, {
            key: "componentDidUpdate",
            value: function e(t) {
                if (this.props.open !== t.open) {
                    this.setState({
                        animating: true
                    });
 // eslint-disable-line react/no-did-update-set-state
                                }
                if (false) {}
                if (false) {}
            }
            /**
     * SingleOpenPanelGroup already takes into account if it is controlled or not in determining openPanelId
     * so this.context.openPanelId is used for both the controlled and uncontrolled case.
     */        }, {
            key: "isOpen",
            value: function e() {
                var e = this.isControlled() ? this.props.open : this.state.open;
                var t = this.context.openPanelId !== undefined ? this.context.openPanelId === this.props.panelId : e;
                return t;
            }
 // Using all three handlers: `handleRequestClose` and `handleRequestOpen` and `handleChange` to test interoperability and maintain backwards compatibility until the open/close handlers are removed in SUI-5
                }, {
            key: "isControlled",
            value: function e() {
                return this.controlledExternally;
            }
        }, {
            key: "render",
            value: function e() {
                var t = this.props, a = t.innerBodyStyles, o = t.actions, i = t.children, s = t.description, p = t.disabled, u = t.elementRef, d = t.headingLevel, m = t.overflow, y = t.renderChildrenWhenCollapsed, h = t.title, g = t.titleWithActions, k = t.inset;
                var C = this.state.animating;
                var x = (0, b.createDOMID)("panel-toggle");
                var O = this.isOpen();
                var S = {
                    disabled: p,
                    onClick: this.handleChange,
                    id: this.toggleId,
                    "aria-controls": this.containerId,
                    "aria-expanded": O,
                    $open: !!O,
                    "data-test": "toggle"
                };
                
                return r().createElement(D, ue({
                    "data-test": "collapsible-panel",
                    elementRef: u
                }, c()(this.props, l()(n.propTypes))), g || o != null ?  r().createElement(W, {
                    $disabled: p
                },  r().createElement(B, ue({
                    appearance: "subtle",
                    "aria-label": (0, v._)("Panel"),
                    "aria-labelledby": x
                }, S),  r().createElement(U, {
                    open: !!O
                })),  r().createElement($, {
                    $titleWithActions: g,
                    "data-test": "title",
                    id: o != null ? x : undefined
                }, h), o != null &&  r().createElement(M, null, o)) : 
                
                /* Clickable component is not allowed to have role="heading" so it is wrapped with a span */
                r().createElement("span", {
                    "data-test": "heading",
                    role: d ? "heading" : undefined,
                    "aria-level": d
                },  r().createElement(L, S,  r().createElement(X, null,  r().createElement(U, {
                    open: !!O
                })),  r().createElement($, null,  r().createElement("span", {
                    "data-test": "title"
                }, h), s &&  r().createElement(G, {
                    "data-test": "description"
                }, s)))),  r().createElement(f(), {
                    outerId: this.containerId,
                    "aria-labelledby": this.toggleId,
                    "data-test": "body",
                    outerStyle: {
                        overflow: C ? "hidden" : m
                    },
                    innerStyle: fe({
                        tableLayout: "fixed"
                    }, a),
                    open: O,
                    onAnimationEnd: this.handleAnimationEnd,
                    role: o != null ? "region" : undefined,
                    renderChildrenWhenCollapsed: y
                },  r().createElement(H, {
                    $inset: k
                }, i)));
            }
        } ]);
        return n;
    }(n.Component);
    Se(we, "propTypes", Ie);
    Se(we, "defaultProps", Pe);
    Se(we, "contextType", V);
    /* harmony default export */    const je = we;
    // CONCATENATED MODULE: ./src/CollapsiblePanel/index.ts
    module.exports = t;
    /******/})();
