// @splunk/react-ui/ColumnLayout.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        Column: () => /* reexport */ C,
        Row: () => /* reexport */ R,
        default: () => /* reexport */ z
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const i = require("prop-types");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/themes/useSplunkTheme"
    const a = require("@splunk/themes/useSplunkTheme");
    var o =  e.n(a);
    // CONCATENATED MODULE: external "lodash/defaults"
    const s = require("lodash/defaults");
    var c =  e.n(s);
    // CONCATENATED MODULE: external "styled-components"
    const u = require("styled-components");
    var d =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/themes"
    const f = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/ColumnLayout/ColumnStyles.ts
    var p = d().div.withConfig({
        displayName: "ColumnStyles__Styled",
        componentId: "sc-1pchz98-0"
    })([ "", ";" ], f.mixins.reset("block"));
    // CONCATENATED MODULE: ./src/utils/types.ts
    // see https://github.com/DefinitelyTyped/DefinitelyTyped/issues/31245 for details
    // regarding the css prop. certain (versions of) libraries included in the compilation
    // - such as storybook->emotion - introduce the css prop using a global augmentation,
    // which causes the prop to end up in our declaration files (via JSX.IntrinsicElements).
    // it's difficult to control all nested dependencies and their types, which is why the
    // css prop is omitted in the two prop helpers above - this ensures the css prop isn't
    // erroneously included in react-ui's public api.
    // see https://github.com/microsoft/TypeScript/pull/12253
    // see https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html
    function m(e, t) {
        if (e === undefined || e === null) {
            throw new Error(t !== null && t !== void 0 ? t : "Unexpected undefined or null");
        }
    }
    // CONCATENATED MODULE: ./src/ColumnLayout/Column.tsx
    function v() {
        return v = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, v.apply(null, arguments);
    }
    function y(e, t) {
        if (null == e) return {};
        var r, n, i = h(e, t);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (n = 0; n < l.length; n++) {
                r = l[n], t.includes(r) || {}.propertyIsEnumerable.call(e, r) && (i[r] = e[r]);
            }
        }
        return i;
    }
    function h(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (t.includes(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var g = {
        children: l().node,
        elementRef: l().oneOfType([ l().func, l().object ]),
        /** @private. */
        gutter: l().number,
        /** @private. */
        isFirstChild: l().bool,
        /** @private. */
        isLastChild: l().bool,
        span: l().number,
        /** @private. */
        style: l().object
    };
    function b(e) {
        var t = e.children, r = e.elementRef, i = e.gutter, l = e.span, a = l === void 0 ? 1 : l, o = e.isFirstChild, s = e.isLastChild, u = e.style, d = y(e, [ "children", "elementRef", "gutter", "span", "isFirstChild", "isLastChild", "style" ]);
        // @docs-props-type ColumnPropsBase
                var f = i ? "calc((100% - ".concat(11 * i, "px) * ").concat(a / 12, " + (").concat(i, "px * ").concat(a - 1, "))") : "".concat(100 / 12 * a, "%");
        m(i);
        var h = {
            marginLeft: o ? undefined : i / 2,
            marginRight: s ? undefined : i / 2,
            flex: "".concat(a, " ").concat(a, " auto"),
            // may accommodate small rounding errors and dividers
            width: f
        };
        
        return n().createElement(p, v({
            "data-test": "column",
            ref: r,
            style: c()({}, u, h)
        }, d), t);
    }
    b.propTypes = g;
    /* harmony default export */    const C = b;
    // CONCATENATED MODULE: ./src/ColumnLayout/RowStyles.ts
    var O = d().div.withConfig({
        displayName: "RowStyles__Styled",
        componentId: "sc-121ntds-0"
    })([ "", ";flex-flow:row nowrap;&[data-align-items='start']{align-items:flex-start;}&[data-align-items='end']{align-items:flex-end;}&[data-align-items='center']{align-items:center;}&[data-align-items='stretch']{align-items:stretch;}" ], f.mixins.reset("flex"));
    var w = d().div.withConfig({
        displayName: "RowStyles__StyledDivider",
        componentId: "sc-121ntds-1"
    })([ "border-left:1px solid ", ";flex:0 0 1;align-self:stretch;" ], (0, f.pick)({
        prisma: f.variables.neutral200,
        enterprise: {
            light: f.variables.gray92,
            dark: f.variables.gray60
        }
    }));
    // CONCATENATED MODULE: ./src/ColumnLayout/Row.tsx
    function S() {
        return S = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, S.apply(null, arguments);
    }
    function j(e, t) {
        if (null == e) return {};
        var r, n, i = x(e, t);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (n = 0; n < l.length; n++) {
                r = l[n], t.includes(r) || {}.propertyIsEnumerable.call(e, r) && (i[r] = e[r]);
            }
        }
        return i;
    }
    function x(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (t.includes(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var P = {
        alignItems: l().oneOf([ "start", "end", "center", "stretch" ]),
        children: l().node,
        /** @private. */
        divider: l().bool,
        elementRef: l().oneOfType([ l().func, l().object ]),
        /** @private. */
        gutter: l().number,
        /** @private. */
        isFirstChild: l().bool,
        /** @private. */
        isLastChild: l().bool,
        /** @private. */
        style: l().object
    };
    function k(e) {
        var t = e.alignItems, i = t === void 0 ? "stretch" : t, l = e.children, a = e.divider, o = e.elementRef, s = e.gutter, u = e.isFirstChild, d = e.isLastChild, f = e.style, p = j(e, [ "alignItems", "children", "divider", "elementRef", "gutter", "isFirstChild", "isLastChild", "style" ]);
        // @docs-props-type RowPropsBase
                if (false) {
            var v;
        }
        var y = (0, r.useCallback)((function(e, t) {
            
            return (0, r.cloneElement)(e, {
                gutter: s,
                isFirstChild: t === 0,
                isLastChild: t === r.Children.count(l) - 1
            });
        }), [ l, s ]);
        var h = (0, r.useCallback)((function(e, t, r, i) {
            e.push(t);
            if (a && r < i.length - 1) {
                e.push( n().createElement(w, {
                    key: "".concat(r, "-divider")
                }));
            }
            return e;
        }), [ a ]);
        m(s);
        var g = {
            marginTop: u ? undefined : s / 2,
            marginBottom: d ? undefined : s / 2
        };
        var b = r.Children.toArray(l).filter(r.isValidElement).map(y).reduce(h, []);
        
        return n().createElement(O, S({
            style: c()({}, f, g),
            "data-align-items": i,
            "data-test": "row",
            ref: o
        }, p), b);
    }
    k.propTypes = P;
    /* harmony default export */    const R = k;
    // CONCATENATED MODULE: ./src/ColumnLayout/ColumnLayoutStyles.ts
    var _ = d().div.withConfig({
        displayName: "ColumnLayoutStyles__Styled",
        componentId: "sc-3eatxz-0"
    })([ "", ";flex-direction:column;" ], f.mixins.reset("flex"));
    var E = d().div.withConfig({
        displayName: "ColumnLayoutStyles__StyledDivider",
        componentId: "sc-3eatxz-1"
    })([ "border-top:1px solid ", ";height:0;" ], (0, f.pick)({
        prisma: f.variables.neutral200,
        enterprise: {
            light: f.variables.gray92,
            dark: f.variables.gray60
        }
    }));
    // CONCATENATED MODULE: ./src/ColumnLayout/ColumnLayout.tsx
    function I() {
        return I = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, I.apply(null, arguments);
    }
    function L(e, t) {
        if (null == e) return {};
        var r, n, i = T(e, t);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (n = 0; n < l.length; n++) {
                r = l[n], t.includes(r) || {}.propertyIsEnumerable.call(e, r) && (i[r] = e[r]);
            }
        }
        return i;
    }
    function T(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (t.includes(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var F = {
        children: l().node,
        divider: l().oneOf([ "none", "vertical", "horizontal" ]),
        elementRef: l().oneOfType([ l().func, l().object ]),
        gutter: l().number
    };
    /**
 * `ColumnLayout` provides simple API for laying out content into columns. It is based on flexbox
 * and has 12 columns.
 */    function q(e) {
        var t = e.children, i = e.divider, l = i === void 0 ? "none" : i, a = e.elementRef, s = e.gutter, c = L(e, [ "children", "divider", "elementRef", "gutter" ]);
        // @docs-props-type ColumnLayoutPropsBase
                var u = o()(), d = u.family, f = u.density;
        var p = (0, r.useCallback)((function(e, n) {
            var i = n === 0;
            var a = n === r.Children.count(t) - 1;
            var o = s;
            if (o === undefined) {
                if (d === "prisma") {
                    if (f === "compact") {
                        o = 16;
                    } else if (f === "comfortable") {
                        o = 24;
                    }
                } else if (d === "enterprise") {
                    o = 30;
                }
            }
            
            return (0, r.cloneElement)(e, {
                gutter: o,
                divider: l === "vertical",
                isFirstChild: i,
                isLastChild: a
            });
        }), [ t, f, l, d, s ]);
        var m = (0, r.useCallback)((function(e, t, r, i) {
            e.push(t);
            if (l === "horizontal" && r < i.length - 1) {
                e.push( n().createElement(E, {
                    key: "".concat(r, "-divider")
                }));
            }
            return e;
        }), [ l ]);
        var v = r.Children.toArray(t).filter(r.isValidElement).map(p).reduce(m, []);
        
        return n().createElement(_, I({
            "data-test": "column-layout",
            ref: a
        }, c), v);
    }
    q.propTypes = F;
    q.Row = R;
    q.Column = C;
    /* harmony default export */    const z = q;
    // CONCATENATED MODULE: ./src/ColumnLayout/index.ts
    module.exports = t;
    /******/})();
