// @splunk/react-ui/ComboBox.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var n = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, n) => {
            /******/ for (var o in n) {
                /******/ if (e.o(n, o) && !e.o(t, o)) {
                    /******/ Object.defineProperty(t, o, {
                        enumerable: true,
                        get: n[o]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        Divider: () => /* reexport */ S.Divider,
        Heading: () => /* reexport */ S.Heading,
        Option: () => /* reexport */ G,
        default: () => /* reexport */ ge
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var o =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const r = require("prop-types");
    var i =  e.n(r);
    // CONCATENATED MODULE: external "lodash/extend"
    const a = require("lodash/extend");
    var l =  e.n(a);
    // CONCATENATED MODULE: external "lodash/has"
    const u = require("lodash/has");
    var c =  e.n(u);
    // CONCATENATED MODULE: external "lodash/memoize"
    const s = require("lodash/memoize");
    var p =  e.n(s);
    // CONCATENATED MODULE: external "lodash/omit"
    const f = require("lodash/omit");
    var d =  e.n(f);
    // CONCATENATED MODULE: external "lodash/pick"
    const v = require("lodash/pick");
    var b =  e.n(v);
    // CONCATENATED MODULE: external "lodash/some"
    const h = require("lodash/some");
    var y =  e.n(h);
    // CONCATENATED MODULE: external "@splunk/ui-utils/filter"
    const m = require("@splunk/ui-utils/filter");
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const g = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const O = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/ui-utils/keyboard"
    const w = require("@splunk/ui-utils/keyboard");
    // CONCATENATED MODULE: external "@splunk/react-ui/Menu"
    const S = require("@splunk/react-ui/Menu");
    // CONCATENATED MODULE: external "@splunk/react-ui/Popover"
    const I = require("@splunk/react-ui/Popover");
    var P =  e.n(I);
    // CONCATENATED MODULE: external "@splunk/react-ui/ResultsMenu"
    const j = require("@splunk/react-ui/ResultsMenu");
    var C =  e.n(j);
    // CONCATENATED MODULE: external "@splunk/react-ui/Text"
    const k = require("@splunk/react-ui/Text");
    var x =  e.n(k);
    // CONCATENATED MODULE: ./src/ComboBox/Option.tsx
    function E(e) {
        "@babel/helpers - typeof";
        return E = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, E(e);
    }
    function M() {
        return M = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var n = arguments[t];
                for (var o in n) {
                    ({}).hasOwnProperty.call(n, o) && (e[o] = n[o]);
                }
            }
            return e;
        }, M.apply(null, arguments);
    }
    function V(e, t) {
        if (null == e) return {};
        var n, o, r = R(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (o = 0; o < i.length; o++) {
                n = i[o], t.includes(n) || {}.propertyIsEnumerable.call(e, n) && (r[n] = e[n]);
            }
        }
        return r;
    }
    function R(e, t) {
        if (null == e) return {};
        var n = {};
        for (var o in e) {
            if ({}.hasOwnProperty.call(e, o)) {
                if (t.includes(o)) continue;
                n[o] = e[o];
            }
        }
        return n;
    }
    function D(e, t) {
        if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function");
    }
    function q(e, t) {
        for (var n = 0; n < t.length; n++) {
            var o = t[n];
            o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), 
            Object.defineProperty(e, K(o.key), o);
        }
    }
    function _(e, t, n) {
        return t && q(e.prototype, t), n && q(e, n), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function B(e, t) {
        if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), t && T(e, t);
    }
    function T(e, t) {
        return T = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, t) {
            return e.__proto__ = t, e;
        }, T(e, t);
    }
    function A(e) {
        var t = H();
        return function() {
            var n, o = N(e);
            if (t) {
                var r = N(this).constructor;
                n = Reflect.construct(o, arguments, r);
            } else n = o.apply(this, arguments);
            return L(this, n);
        };
    }
    function L(e, t) {
        if (t && ("object" == E(t) || "function" == typeof t)) return t;
        if (void 0 !== t) throw new TypeError("Derived constructors may only return object or undefined");
        return F(e);
    }
    function F(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function H() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return (H = function t() {
            return !!e;
        })();
    }
    function N(e) {
        return N = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, N(e);
    }
    function z(e, t, n) {
        return (t = K(t)) in e ? Object.defineProperty(e, t, {
            value: n,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = n, e;
    }
    function K(e) {
        var t = W(e, "string");
        return "symbol" == E(t) ? t : t + "";
    }
    function W(e, t) {
        if ("object" != E(e) || !e) return e;
        var n = e[Symbol.toPrimitive];
        if (void 0 !== n) {
            var o = n.call(e, t || "default");
            if ("object" != E(o)) return o;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    var U = {
        /** @private */
        active: i().bool,
        description: i().string,
        descriptionPosition: i().oneOf([ "right", "bottom" ]),
        disabled: i().bool,
        icon: i().node,
        label: i().string,
        matchRanges: i().arrayOf(i().shape({
            start: i().number.isRequired,
            end: i().number.isRequired
        })),
        /** @private */
        onClick: i().func,
        truncate: i().bool,
        value: i().string.isRequired
    };
    var X = {
        active: false,
        descriptionPosition: "bottom",
        disabled: false,
        truncate: false
    };
    /**
 * An option within a `ComboBox`. This inherits from
 * [PureComponent](https://reactjs.org/docs/react-api.html#reactpurecomponent)
 * so any elements passed to it must also be pure.
 */    var Y =  function(e) {
        B(n, e);
        var t = A(n);
        function n() {
            var e;
            D(this, n);
            for (var o = arguments.length, r = new Array(o), i = 0; i < o; i++) {
                r[i] = arguments[i];
            }
            e = t.call.apply(t, [ this ].concat(r));
            z(F(e), "item", null);
            z(F(e), "handleClick", (function(t) {
                var n = e.props, o = n.disabled, r = n.onClick, i = n.value;
                if (!o) {
                    r === null || r === void 0 ? void 0 : r(t, {
                        value: i
                    });
                }
            }));
            return e;
        }
 // @docs-props-type OptionPropsBase
                _(n, [ {
            key: "scrollIntoViewIfNeeded",
            // eslint-disable-next-line react/no-unused-class-component-methods
            value: function e() {
                var t;
                (t = this.item) === null || t === void 0 ? void 0 : t.scrollIntoViewIfNeeded();
            }
        }, {
            key: "render",
            value: function e() {
                var t = this;
                // eslint-disable-next-line no-unused-vars
                                var n = this.props, r = n.value, i = n.label, a = V(n, [ "value", "label" ]);
                var l = i === undefined ? r : i;
                
                return o().createElement(S.Item, M({
                    ref: function e(n) {
                        t.item = n;
                    },
                    "data-test": "option",
                    "data-test-value": r
                }, a, {
                    onClick: this.handleClick,
                    role: "option"
                }), l);
            }
        } ]);
        return n;
    }(n.PureComponent);
    z(Y, "propTypes", U);
    z(Y, "defaultProps", X);
    /* harmony default export */    const G = Y;
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function J(e, t) {
        if (e) {
            if (typeof e === "function") {
                e(t);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = t;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/ComboBox/ComboBox.tsx
    function Q(e) {
        "@babel/helpers - typeof";
        return Q = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, Q(e);
    }
    function Z() {
        return Z = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var n = arguments[t];
                for (var o in n) {
                    ({}).hasOwnProperty.call(n, o) && (e[o] = n[o]);
                }
            }
            return e;
        }, Z.apply(null, arguments);
    }
    function $(e, t) {
        var n = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            t && (o = o.filter((function(t) {
                return Object.getOwnPropertyDescriptor(e, t).enumerable;
            }))), n.push.apply(n, o);
        }
        return n;
    }
    function ee(e) {
        for (var t = 1; t < arguments.length; t++) {
            var n = null != arguments[t] ? arguments[t] : {};
            t % 2 ? $(Object(n), !0).forEach((function(t) {
                pe(e, t, n[t]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(n)) : $(Object(n)).forEach((function(t) {
                Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(n, t));
            }));
        }
        return e;
    }
    function te(e, t) {
        if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function");
    }
    function ne(e, t) {
        for (var n = 0; n < t.length; n++) {
            var o = t[n];
            o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), 
            Object.defineProperty(e, fe(o.key), o);
        }
    }
    function oe(e, t, n) {
        return t && ne(e.prototype, t), n && ne(e, n), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function re(e, t) {
        if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), t && ie(e, t);
    }
    function ie(e, t) {
        return ie = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, t) {
            return e.__proto__ = t, e;
        }, ie(e, t);
    }
    function ae(e) {
        var t = ce();
        return function() {
            var n, o = se(e);
            if (t) {
                var r = se(this).constructor;
                n = Reflect.construct(o, arguments, r);
            } else n = o.apply(this, arguments);
            return le(this, n);
        };
    }
    function le(e, t) {
        if (t && ("object" == Q(t) || "function" == typeof t)) return t;
        if (void 0 !== t) throw new TypeError("Derived constructors may only return object or undefined");
        return ue(e);
    }
    function ue(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function ce() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return (ce = function t() {
            return !!e;
        })();
    }
    function se(e) {
        return se = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, se(e);
    }
    function pe(e, t, n) {
        return (t = fe(t)) in e ? Object.defineProperty(e, t, {
            value: n,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = n, e;
    }
    function fe(e) {
        var t = de(e, "string");
        return "symbol" == Q(t) ? t : t + "";
    }
    function de(e, t) {
        if ("object" != Q(e) || !e) return e;
        var n = e[Symbol.toPrimitive];
        if (void 0 !== n) {
            var o = n.call(e, t || "default");
            if ("object" != Q(o)) return o;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    /** @public */
    /** @public */
    /** @public */
    /** @public */    var ve = {
        animateLoading: i().bool,
        append: i().bool,
        children: i().node,
        controlledFilter: i().bool,
        defaultPlacement: i().oneOf([ "above", "below", "vertical" ]),
        defaultValue: i().string,
        describedBy: i().string,
        disabled: i().bool,
        elementRef: i().oneOfType([ i().func, i().object ]),
        error: i().bool,
        footerMessage: i().node,
        inline: i().bool,
        inputId: i().string,
        inputRef: i().oneOfType([ i().func, i().object ]),
        isLoadingOptions: i().bool,
        labelledBy: i().string,
        loadingMessage: i().node,
        menuStyle: i().object,
        name: i().string,
        noOptionsMessage: i().node,
        onBlur: i().func,
        onChange: i().func,
        onClose: i().func,
        onFocus: i().func,
        onKeyDown: i().func,
        onOpen: i().func,
        onScroll: i().func,
        onScrollBottom: i().func,
        onSelect: i().func,
        placeholder: i().string,
        prepend: i().bool,
        size: i().oneOf([ "small", "medium", "large" ]),
        value: i().string,
        /** @private. */
        virtualization: i().number
    };
    var be = {
        animateLoading: false,
        controlledFilter: false,
        defaultPlacement: "vertical",
        disabled: false,
        error: false,
        inline: false,
        isLoadingOptions: false,
        menuStyle: {},
        placeholder: (0, O._)("Select...")
    };
    function he(e, t) {
        var n = t.clientX, o = t.clientY;
        var r = e.getBoundingClientRect(), i = r.top, a = r.left, l = r.bottom, u = r.right;
        return n > a && n < u && o > i && o < l;
    }
    var ye = p()((function(e) {
        var t = e.anchorWidth, n = e.maxHeight, o = e.menuStyle;
        return l()({
            maxHeight: n,
            overflow: "auto",
            width: Math.max(t !== null && t !== void 0 ? t : 0, 200)
        }, o);
    }));
    /**
 * `ComboBox` allows the user to select a predefined string or enter a new value. Unlike `Select`
 * and `Multiselect`, `Option` value must always be a string.
 */    var me =  function(e) {
        re(r, e);
        var t = ae(r);
        // @docs-props-type ComboBoxPropsBase
                function r(e) {
            var i;
            te(this, r);
            i = t.call(this, e);
            pe(ue(i), "activeItemId", void 0);
            pe(ue(i), "activeValue", void 0);
            pe(ue(i), "availableOptionCount", 0);
            pe(ue(i), "controlledExternally", void 0);
            pe(ue(i), "options", void 0);
            pe(ue(i), "popoverId", void 0);
            pe(ue(i), "previousActiveIndex", null);
            pe(ue(i), "textInput", null);
            pe(ue(i), "handleInputMount", (function(e) {
                i.textInput = e;
                J(i.props.inputRef, e);
            }));
            pe(ue(i), "handleAnchorMount", (function(e) {
                i.setState({
                    anchor: e
                });
                J(i.props.elementRef, e);
            }));
            pe(ue(i), "handleActiveOptionMount", (function(e) {
                if (i.previousActiveIndex !== i.state.activeIndex) {
                    e === null || e === void 0 ? void 0 : e.scrollIntoViewIfNeeded();
                }
            }));
            pe(ue(i), "handleInputFocus", (function(e, t) {
                var n, o;
                i.open();
                (n = (o = i.props).onFocus) === null || n === void 0 ? void 0 : n.call(o, e, t);
            }));
            pe(ue(i), "handleChange", (function(e, t) {
                var n, o;
                var r = t.value, a = t.selectedOption;
                e.stopPropagation();
                var l = i.props.name;
                var u = a ? undefined : r;
                if (i.isControlled()) {
                    i.setState({
                        activeIndex: 0,
                        typedValue: u
                    });
                } else {
                    i.setState({
                        value: r,
                        activeIndex: 0,
                        typedValue: u
                    });
                }
                (n = (o = i.props).onChange) === null || n === void 0 ? void 0 : n.call(o, e, {
                    value: r,
                    name: l
                });
            }));
            pe(ue(i), "handleSelectOption", (function(e, t) {
                i.handleChange(e, ee(ee({}, t), {}, {
                    selectedOption: true
                }));
                i.focus();
                i.close();
            }));
            pe(ue(i), "handleInputKeyDown", (function(e) {
                var t = i.props, o = t.children, r = t.onKeyDown, a = t.onScrollBottom;
                var l = i.availableOptionCount;
                var u = i.activeValue;
                var c = (0, w.keycode)(e.nativeEvent);
                if (i.state.open) {
                    switch (c) {
                      case "enter":
                        {
                            if (u) {
                                i.handleSelectOption(e, {
                                    value: u
                                });
                            }
                            break;
                        }

                      case "tab":
                        i.close();
                        break;

                      case "down":
                        i.setState((function(e) {
                            return {
                                activeIndex: Math.min(e.activeIndex + 1, l - 1)
                            };
                        }));
                        if (o && a) {
                            var s = n.Children.count(o) - 2;
                            if (i.state.activeIndex === s) {
                                i.handleScrollBottom(e);
                            }
                        }
                        break;

                      case "up":
                        i.setState((function(e) {
                            return {
                                activeIndex: Math.max(e.activeIndex - 1, 0)
                            };
                        }));
                        break;

                      default:
 // do nothing
                                        }
                } else if ((0, w.addsCharacter)(e.nativeEvent) !== false || 
                // Safari 9.0 returns undefined
                c === "enter" || c === "backspace" || c === "down" || c === "up") {
                    i.open();
                }
                r === null || r === void 0 ? void 0 : r(e);
            }));
            pe(ue(i), "handleInputClick", (function(e) {
                e.stopPropagation();
                if (!i.props.disabled) {
                    i.open();
                }
            }));
            pe(ue(i), "handleRequestClose", (function(e) {
                var t = e.event, n = e.reason;
                var o = n === "offScreen" || n === "escapeKey" || n === "clickAway" && i.state.anchor && !he(i.state.anchor, t);
                if (o) {
                    i.close();
                }
            }));
            pe(ue(i), "handleScrollBottom", (function(e) {
                if (i.state.open && !i.props.isLoadingOptions) {
                    var t, n;
                    (t = (n = i.props).onScrollBottom) === null || t === void 0 ? void 0 : t.call(n, e);
                }
            }));
            pe(ue(i), "prepareOptions", (function() {
                var e = i.props, t = e.children, r = e.controlledFilter;
                var a = i.state, l = a.activeIndex, u = a.typedValue;
                var s = i.getValue();
                var p = n.Children.toArray(t).filter(n.isValidElement);
                var f = y()(p, (function(e) {
                    if (e.type === S.Heading) {
                        return false;
                    }
                    if (e.props.label === undefined) {
                        return e.props.value === s;
                    }
                    if (e.props.label !== undefined && u !== undefined) {
                        return e.props.label === s;
                    }
                    return e.props.value === s;
                }));
                if (!f && s) {
                    p.unshift( o().createElement(G, {
                        description: (0, O._)("(New value)"),
                        descriptionPosition: "right",
                        key: "currentValueOption",
                        value: s,
                        "data-test-current-value-option": s
                    }));
                }
                // Highlight Active
                                i.availableOptionCount = 0;
                i.activeValue = undefined;
                var d = (0, m.stringToKeywords)(u ? s : i.getDisplayValue());
                i.options = (r ? p : p.filter((function(e) {
                    if (e.props.label !== undefined) {
                        return (0, m.testPhrase)(e.props.label, d);
                    }
                    if (e.props.value !== undefined) {
                        return (0, m.testPhrase)(e.props.value, d);
                    }
                    return true;
 // Keep all headers and non-interactive options
                                }))).map((function(e, t) {
                    if (!c()(e.props, "active")) {
                        // ignore Headings and Dividers
                        return e;
                    }
                    var o = i.availableOptionCount === l;
                    i.availableOptionCount += 1;
                    var a = e.props.label !== undefined ? e.props.label : e.props.value;
                    var u = e.props.matchRanges;
                    var p = !r && s && (f || t > 0) ? (0, m.keywordLocations)(a, d) || undefined : undefined;
                    if (o) {
                        if (!e.props.disabled) {
                            i.activeValue = e.props.value;
                        }
                        
                        return (0, n.cloneElement)(e, {
                            ref: i.handleActiveOptionMount,
                            id: i.activeItemId,
                            onClick: i.handleSelectOption,
                            matchRanges: u || p,
                            active: true
                        });
                    }
                    
                    return (0, n.cloneElement)(e, {
                        onClick: i.handleSelectOption,
                        matchRanges: u || p
                    });
                }));
            }));
            pe(ue(i), "renderMenu", (function(e) {
                var t = e.anchorWidth, n = e.maxHeight;
                var r = i.props, a = r.isLoadingOptions, l = r.menuStyle, u = r.onScrollBottom, c = r.virtualization;
                var s = ye({
                    anchorWidth: t,
                    maxHeight: n,
                    menuStyle: l
                });
                var p = ee({
                    style: s,
                    controlledExternally: true,
                    onScrollBottom: u ? i.handleScrollBottom : undefined,
                    isLoading: a
                }, b()(i.props, "className", "noOptionsMessage", "footerMessage", "animateLoading", "loadingMessage", "onScroll"));
                if (c) {
                    
                    return o().createElement(j.VirtualizedResultsMenu, Z({
                        virtualization: c
                    }, p), i.options);
                }
                
                return o().createElement(C(), p, i.options);
            }));
            i.state = {
                activeIndex: 0,
                anchor: null,
                typedValue: undefined,
                open: false,
                value: e.defaultValue || ""
            };
            i.controlledExternally = c()(e, "value");
            i.popoverId = (0, g.createDOMID)("popover");
            i.activeItemId = (0, g.createDOMID)("active-item");
            if (false) {}
            if (false) {}
            if (false) {}
            return i;
        }
        oe(r, [ {
            key: "componentDidUpdate",
            value: function e(t, n) {
                if (false) {}
                this.previousActiveIndex = n.activeIndex;
            }
        }, {
            key: "getValue",
            value: function e() {
                return this.isControlled() ? this.props.value : this.state.value;
            }
        }, {
            key: "getDisplayValue",
            value: function e() {
                var t = this.getValue();
                var o = this.state.typedValue;
                var r = n.Children.toArray(this.props.children).filter(n.isValidElement);
                var i = !o && r.find((function(e) {
                    return e.props.label !== undefined && e.props.value === t;
                }));
                if (i) {
                    return i === null || i === void 0 ? void 0 : i.props.label;
                }
                return t;
            }
        }, {
            key: "open",
            value: function e() {
                var t = this;
                if (!this.state.open) {
                    this.setState({
                        open: true,
                        activeIndex: 0
                    }, (function() {
                        var e, n;
                        (e = (n = t.props).onOpen) === null || e === void 0 ? void 0 : e.call(n);
                    }));
                }
            }
        }, {
            key: "close",
            value: function e() {
                var t = this;
                this.setState({
                    open: false
                }, (function() {
                    var e, n;
                    t.previousActiveIndex = null;
                    (e = (n = t.props).onClose) === null || e === void 0 ? void 0 : e.call(n);
                }));
            }
            /**
     * Focus the `ComboBox`.
     */        }, {
            key: "focus",
            value: function e() {
                var t;
                (t = this.textInput) === null || t === void 0 ? void 0 : t.focus();
            }
        }, {
            key: "isControlled",
            value: function e() {
                return this.controlledExternally;
            }
        }, {
            key: "render",
            value: function e() {
                var t = this.props, n = t.defaultPlacement, r = t.inputId;
                var i = this.state, a = i.anchor, l = i.open;
                var u = this.getValue();
                var c = this.getDisplayValue();
                this.prepareOptions();
                
                return o().createElement(x(), Z({
                    autoCapitalize: "off",
                    autoComplete: "off",
                    autoCorrect: "off",
                    spellCheck: false,
                    canClear: true,
                    "data-test": "combo-box"
                }, d()(this.props, "animateLoading", "className", "controlledFilter", "defaultValue", "footerMessage", "inputId", "isLoadingOptions", "loadingMessage", "menuStyle", "noOptionsMessage", "onClose", "onOpen", "onScroll", "onScrollBottom", "spellCheck"), {
                    "data-test-popover-id": this.popoverId,
                    "data-test-label": c,
                    "data-test-value": u,
                    "data-test-open": l && !!a,
                    onFocus: this.handleInputFocus,
                    onInputClick: this.handleInputClick,
                    onChange: this.handleChange,
                    onKeyDown: this.handleInputKeyDown,
                    inputId: r,
                    inputRef: this.handleInputMount,
                    elementRef: this.handleAnchorMount,
                    role: "combobox",
                    value: c,
                    "aria-activedescendant": this.state.open && Array.isArray(this.options) && this.options.length > 0 ? this.activeItemId : undefined,
                    "aria-expanded": l,
                    "aria-controls": l ? this.popoverId : undefined
                }),  o().createElement(P(), {
                    anchor: a,
                    autoCloseWhenOffScreen: true,
                    canCoverAnchor: false,
                    defaultPlacement: n,
                    id: this.popoverId,
                    onRequestClose: this.handleRequestClose,
                    open: l && !!a,
                    repositionMode: "flip"
                }, this.renderMenu));
            }
        } ]);
        return r;
    }(n.Component);
    pe(me, "propTypes", ve);
    pe(me, "defaultProps", be);
    pe(me, "Option", G);
    pe(me, "Divider", S.Divider);
    pe(me, "Heading", S.Heading);
    /* harmony default export */    const ge = me;
    // CONCATENATED MODULE: ./src/ComboBox/index.ts
    module.exports = t;
    /******/})();
