// @splunk/react-ui/Concertina.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var n = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, n) => {
            /******/ for (var r in n) {
                /******/ if (e.o(n, r) && !e.o(t, r)) {
                    /******/ Object.defineProperty(t, r, {
                        enumerable: true,
                        get: n[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        Panel: () => /* reexport */ ze,
        default: () => /* reexport */ it
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var r =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "lodash/debounce"
    const a = require("lodash/debounce");
    var l =  e.n(a);
    // CONCATENATED MODULE: external "lodash/take"
    const c = require("lodash/take");
    var u =  e.n(c);
    // CONCATENATED MODULE: external "lodash/takeRight"
    const s = require("lodash/takeRight");
    var f =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/react-ui/EventListener"
    const p = require("@splunk/react-ui/EventListener");
    var d =  e.n(p);
    // CONCATENATED MODULE: external "lodash/keys"
    const h = require("lodash/keys");
    var b =  e.n(h);
    // CONCATENATED MODULE: external "lodash/omit"
    const y = require("lodash/omit");
    var v =  e.n(y);
    // CONCATENATED MODULE: external "styled-components"
    const m = require("styled-components");
    var g =  e.n(m);
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const w = require("@splunk/react-ui/Clickable");
    var S =  e.n(w);
    // CONCATENATED MODULE: external "@splunk/themes"
    const O = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Concertina/HeadingStyles.ts
    var k = g()(S()).withConfig({
        displayName: "HeadingStyles__StyledClickable",
        componentId: "sc-1h92oaj-0"
    })([ "display:block;width:100%;cursor:pointer;position:relative;padding:6px ", ";line-height:", ";border-radius:0;color:", ";", " &:focus:not([disabled]){box-shadow:", ";}", "" ], O.variables.spacingLarge, O.variables.lineHeight, O.variables.textColor, (0, 
    O.pickVariant)("$inner", {
        true: (0, m.css)([ "*:not(:first-child) > &&{border-top:2px solid ", ";}*:not(:last-child) > &&{border-bottom:2px solid ", ";}" ], O.variables.backgroundColor, O.variables.backgroundColor),
        false: (0, m.css)([ "& + &&{border-top:2px solid ", ";}" ], O.variables.backgroundColor)
    }), O.variables.focusShadowInset, (0, O.pickVariant)("$status", {
        normal: (0, m.css)([ "background-color:", ";&:hover:not([disabled]){background-color:", ";}" ], (0, 
        O.pick)({
            dark: O.variables.gray45,
            light: O.variables.gray96
        }), (0, O.pick)({
            dark: O.variables.gray30,
            light: O.variables.gray92
        })),
        disabled: (0, m.css)([ "background-color:", ";" ], (0, O.pick)({
            dark: O.variables.gray80,
            light: O.variables.gray96
        })),
        error: (0, m.css)([ "background-color:", ";color:", ";&:hover:not([disabled]){background-color:", ";}" ], O.variables.errorColor, O.variables.white, O.variables.errorColorD10),
        warning: (0, m.css)([ "background-color:", ";color:", ";&:hover:not([disabled]){background-color:", ";box-shadow:none;}" ], O.variables.warningColor, O.variables.white, O.variables.warningColorD10)
    }));
    var C = g().span.withConfig({
        displayName: "HeadingStyles__StyledHeadingContent",
        componentId: "sc-1h92oaj-1"
    })([ "display:flex;justify-content:space-between;" ]);
    var E = g().span.withConfig({
        displayName: "HeadingStyles__StyledDescription",
        componentId: "sc-1h92oaj-2"
    })([ "font-size:", ";" ], O.variables.fontSizeSmall);
    // CONCATENATED MODULE: ./src/Concertina/Heading.tsx
    function P(e) {
        "@babel/helpers - typeof";
        return P = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, P(e);
    }
    function j() {
        return j = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var n = arguments[t];
                for (var r in n) {
                    ({}).hasOwnProperty.call(n, r) && (e[r] = n[r]);
                }
            }
            return e;
        }, j.apply(null, arguments);
    }
    function T(e, t) {
        if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function");
    }
    function _(e, t) {
        for (var n = 0; n < t.length; n++) {
            var r = t[n];
            r.enumerable = r.enumerable || !1, r.configurable = !0, "value" in r && (r.writable = !0), 
            Object.defineProperty(e, M(r.key), r);
        }
    }
    function x(e, t, n) {
        return t && _(e.prototype, t), n && _(e, n), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function H(e, t) {
        if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), t && I(e, t);
    }
    function I(e, t) {
        return I = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, t) {
            return e.__proto__ = t, e;
        }, I(e, t);
    }
    function R(e) {
        var t = N();
        return function() {
            var n, r = A(e);
            if (t) {
                var o = A(this).constructor;
                n = Reflect.construct(r, arguments, o);
            } else n = r.apply(this, arguments);
            return q(this, n);
        };
    }
    function q(e, t) {
        if (t && ("object" == P(t) || "function" == typeof t)) return t;
        if (void 0 !== t) throw new TypeError("Derived constructors may only return object or undefined");
        return B(e);
    }
    function B(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function N() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return (N = function t() {
            return !!e;
        })();
    }
    function A(e) {
        return A = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, A(e);
    }
    function z(e, t, n) {
        return (t = M(t)) in e ? Object.defineProperty(e, t, {
            value: n,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = n, e;
    }
    function M(e) {
        var t = W(e, "string");
        return "symbol" == P(t) ? t : t + "";
    }
    function W(e, t) {
        if ("object" != P(e) || !e) return e;
        var n = e[Symbol.toPrimitive];
        if (void 0 !== n) {
            var r = n.call(e, t || "default");
            if ("object" != P(r)) return r;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    var D = {
        children: i().node.isRequired,
        description: i().node,
        /** @private */
        index: i().number.isRequired,
        /** @private */
        onClick: i().func,
        /** @private */
        panelId: i().string,
        /** @private */
        position: i().oneOf([ "top", "inner", "bottom" ]),
        status: i().oneOf([ "normal", "warning", "error", "disabled" ])
    };
    var L = {
        position: "inner"
    };
    var $ =  function(e) {
        H(n, e);
        var t = R(n);
        function n() {
            var e;
            T(this, n);
            for (var r = arguments.length, o = new Array(r), i = 0; i < r; i++) {
                o[i] = arguments[i];
            }
            e = t.call.apply(t, [ this ].concat(o));
            z(B(e), "container", null);
            z(B(e), "handleClick", (function(t) {
                var n, r;
                (n = (r = e.props).onClick) === null || n === void 0 ? void 0 : n.call(r, t, {
                    index: e.props.index,
                    panelId: e.props.panelId,
                    position: e.props.position
                });
            }));
            z(B(e), "handleMount", (function(t) {
                e.container = t;
            }));
            return e;
        }
        x(n, [ {
            key: "focus",
            // eslint-disable-next-line react/no-unused-class-component-methods
            value: function e() {
                var t;
                (t = this.container) === null || t === void 0 ? void 0 : t.focus();
            }
        }, {
            key: "render",
            value: function e() {
                var t = this.props, o = t.children, i = t.description, a = t.panelId, l = t.position, c = t.status;
                
                return r().createElement(k, j({
                    disabled: c === "disabled",
                    $inner: l === "inner",
                    $status: c,
                    onClick: this.handleClick,
                    ref: this.handleMount,
                    "data-test": "heading",
                    "data-test-panel-id": a
                }, v()(this.props, b()(n.propTypes))),  r().createElement(C, null,  r().createElement("span", {
                    "data-concertina-role": "title"
                }, o), i &&  r().createElement(E, {
                    "data-concertina-role": "description"
                }, i)));
            }
        } ]);
        return n;
    }(n.Component);
    z($, "propTypes", D);
    z($, "defaultProps", L);
    /* harmony default export */    const U = $;
    // CONCATENATED MODULE: external "lodash/throttle"
    const V = require("lodash/throttle");
    var F =  e.n(V);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const G = require("@splunk/react-ui/Box");
    var J =  e.n(G);
    // CONCATENATED MODULE: external "@splunk/react-ui/Scroll"
    const K = require("@splunk/react-ui/Scroll");
    var Q =  e.n(K);
    // CONCATENATED MODULE: ./src/Concertina/ConcertinaStyles.ts
    var X = (0, m.css)([ "position:absolute;left:0;top:0;right:0;bottom:0;" ]);
    var Y = g()(Q()).withConfig({
        displayName: "ConcertinaStyles__StyledScroll",
        componentId: "sc-1fzuh39-0"
    })([ "", "" ], X);
    var Z = g()(J()).withConfig({
        displayName: "ConcertinaStyles__StyledBox",
        componentId: "sc-1fzuh39-1"
    })([ "", " overflow:hidden;" ], X);
    var ee = g().div.withConfig({
        displayName: "ConcertinaStyles__StyledTop",
        componentId: "sc-1fzuh39-2"
    })([ "position:absolute;left:0;top:0;right:0;z-index:1;background-color:", ";&::after{content:'';position:absolute;bottom:-10px;height:10px;left:0;right:0;background-image:linear-gradient(180deg,rgba(0,0,0,0.15),rgba(0,0,0,0));}" ], O.variables.backgroundColor);
    var te = g().div.withConfig({
        displayName: "ConcertinaStyles__StyledBottom",
        componentId: "sc-1fzuh39-3"
    })([ "position:absolute;left:0;bottom:0;right:0;background-color:", ";&::before{content:'';position:absolute;top:-5px;height:5px;left:0;right:0;background-image:linear-gradient(0deg,rgba(0,0,0,0.15),rgba(0,0,0,0));}" ], O.variables.backgroundColor);
    var ne = g().div.withConfig({
        displayName: "ConcertinaStyles__StyledPanelBody",
        componentId: "sc-1fzuh39-4"
    })([ "", "" ], (0, O.pickVariant)("$status", {
        error: {
            dark: (0, m.css)([ "background-color:", ";" ], O.mixins.colorWithAlpha(O.variables.errorColor, .5)),
            light: (0, m.css)([ "background-color:", ";" ], O.variables.errorColorL50)
        },
        warning: {
            dark: (0, m.css)([ "background-color:", ";" ], O.mixins.colorWithAlpha(O.variables.warningColor, .5)),
            light: (0, m.css)([ "background-color:", ";" ], O.variables.warningColorL50)
        },
        disabled: (0, m.css)([ "display:none;" ])
    }));
    // CONCATENATED MODULE: ./src/utils/ssrDocument.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var re = {
        body: {
            appendChild: function e() {
                return [];
            }
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        activeElement: {
            blur: function e() {},
            nodeName: ""
        },
        querySelector: function e() {
            return null;
        },
        querySelectorAll: function e() {
            return [];
        },
        getElementById: function e() {
            return null;
        },
        createEvent: function e() {
            return {
                initEvent: function e() {}
            };
        },
        createElement: function e() {
            return {
                children: [],
                childNodes: [],
                style: {},
                setAttribute: function e() {},
                getElementsByTagName: function e() {
                    return [];
                }
            };
        },
        createElementNS: function e() {
            return {};
        },
        importNode: function e() {
            return null;
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        }
    };
    function oe() {
        var e = typeof document !== "undefined" ? document : re;
        return e;
    }
    var ie = oe();
    /* harmony default export */    const ae = /* unused pure expression or super */ null && ie;
    // CONCATENATED MODULE: ./src/utils/ssrWindow.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var le = {
        document: re,
        navigator: {
            userAgent: ""
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        },
        history: {
            replaceState: function e() {},
            pushState: function e() {},
            go: function e() {},
            back: function e() {}
        },
        CustomEvent: function e() {
            return this;
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        getComputedStyle: function e() {
            return {
                getPropertyValue: function e() {
                    return "";
                }
            };
        },
        Image: function e() {},
        Date: function e() {},
        screen: {},
        setTimeout: function e() {},
        clearTimeout: function e() {},
        matchMedia: function e() {
            return {};
        },
        requestAnimationFrame: function e(t) {
            if (typeof setTimeout === "undefined") {
                t();
                return null;
            }
            return setTimeout(t, 0);
        },
        cancelAnimationFrame: function e(t) {
            if (typeof setTimeout === "undefined") {
                return;
            }
            clearTimeout(t);
        }
    };
    function ce() {
        var e = typeof window !== "undefined" ? window : le;
        return e;
    }
    var ue = ce();
    /* harmony default export */    const se = ue;
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function fe(e, t) {
        if (e) {
            if (typeof e === "function") {
                e(t);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = t;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Concertina/ConcertinaContext.tsx
    var pe =  (0, n.createContext)({});
    pe.displayName = "Concertina";
    /* harmony default export */    const de = pe;
    // CONCATENATED MODULE: ./src/Concertina/Panel.tsx
    function he(e) {
        "@babel/helpers - typeof";
        return he = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, he(e);
    }
    function be() {
        return be = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var n = arguments[t];
                for (var r in n) {
                    ({}).hasOwnProperty.call(n, r) && (e[r] = n[r]);
                }
            }
            return e;
        }, be.apply(null, arguments);
    }
    function ye(e) {
        return we(e) || ge(e) || me(e) || ve();
    }
    function ve() {
        throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function me(e, t) {
        if (e) {
            if ("string" == typeof e) return Se(e, t);
            var n = {}.toString.call(e).slice(8, -1);
            return "Object" === n && e.constructor && (n = e.constructor.name), "Map" === n || "Set" === n ? Array.from(e) : "Arguments" === n || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n) ? Se(e, t) : void 0;
        }
    }
    function ge(e) {
        if ("undefined" != typeof Symbol && null != e[Symbol.iterator] || null != e["@@iterator"]) return Array.from(e);
    }
    function we(e) {
        if (Array.isArray(e)) return Se(e);
    }
    function Se(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var n = 0, r = Array(t); n < t; n++) {
            r[n] = e[n];
        }
        return r;
    }
    function Oe(e, t) {
        if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function");
    }
    function ke(e, t) {
        for (var n = 0; n < t.length; n++) {
            var r = t[n];
            r.enumerable = r.enumerable || !1, r.configurable = !0, "value" in r && (r.writable = !0), 
            Object.defineProperty(e, Re(r.key), r);
        }
    }
    function Ce(e, t, n) {
        return t && ke(e.prototype, t), n && ke(e, n), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function Ee(e, t) {
        if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), t && Pe(e, t);
    }
    function Pe(e, t) {
        return Pe = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, t) {
            return e.__proto__ = t, e;
        }, Pe(e, t);
    }
    function je(e) {
        var t = xe();
        return function() {
            var n, r = He(e);
            if (t) {
                var o = He(this).constructor;
                n = Reflect.construct(r, arguments, o);
            } else n = r.apply(this, arguments);
            return Te(this, n);
        };
    }
    function Te(e, t) {
        if (t && ("object" == he(t) || "function" == typeof t)) return t;
        if (void 0 !== t) throw new TypeError("Derived constructors may only return object or undefined");
        return _e(e);
    }
    function _e(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function xe() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return (xe = function t() {
            return !!e;
        })();
    }
    function He(e) {
        return He = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, He(e);
    }
    function Ie(e, t, n) {
        return (t = Re(t)) in e ? Object.defineProperty(e, t, {
            value: n,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = n, e;
    }
    function Re(e) {
        var t = qe(e, "string");
        return "symbol" == he(t) ? t : t + "";
    }
    function qe(e, t) {
        if ("object" != he(e) || !e) return e;
        var n = e[Symbol.toPrimitive];
        if (void 0 !== n) {
            var r = n.call(e, t || "default");
            if ("object" != he(r)) return r;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    var Be = {
        children: i().node,
        description: i().node,
        elementRef: i().oneOfType([ i().func, i().object ]),
        /** @private */
        index: i().number,
        panelId: i().string,
        status: i().oneOf([ "normal", "warning", "error", "disabled" ]),
        title: i().node.isRequired
    };
    var Ne = {
        status: "normal"
    };
    var Ae =  function(e) {
        Ee(n, e);
        var t = je(n);
        // @docs-props-type PanelPropsBase
                function n(e) {
            var r;
            Oe(this, n);
            r = t.call(this, e);
            Ie(_e(r), "lastHeight", void 0);
            Ie(_e(r), "measureHeight", (function(e) {
                var t, n, o;
                if (!r.state.containerEl || !r.state.bodyEl) {
                    return;
                }
                var i = r.state.containerEl.clientHeight;
                if (i === r.lastHeight) {
                    return;
                }
                var a = {
                    headingHeight: r.state.containerEl.offsetHeight - r.state.bodyEl.offsetHeight,
                    height: r.state.containerEl.offsetHeight,
                    index: (t = r.props.index) !== null && t !== void 0 ? t : 0,
                    panelId: r.props.panelId
                };
                r.lastHeight = i;
                (n = (o = r.context).onChange) === null || n === void 0 ? void 0 : n.call(o, e, a);
            }));
            Ie(_e(r), "handleResize", F()(r.measureHeight, 300));
            Ie(_e(r), "handleMount", (function(e) {
                r.setState({
                    containerEl: e
                });
                fe(r.props.elementRef, e);
            }));
            Ie(_e(r), "handleHeadingMount", (function(e) {
                r.setState({
                    heading: e
                });
            }));
            Ie(_e(r), "handleBodyMount", (function(e) {
                r.setState({
                    bodyEl: e
                });
            }));
            r.state = {
                bodyEl: null,
                containerEl: null,
                heading: null
            };
            return r;
        }
        Ce(n, [ {
            key: "componentDidUpdate",
            value: function e(t, n) {
                if (this.props.children !== t.children || !n.containerEl) {
                    this.measureHeight();
                }
            }
        }, {
            key: "componentWillUnmount",
            value: function e() {
                this.handleResize.cancel();
            }
        }, {
            key: "focus",
            // eslint-disable-next-line react/no-unused-class-component-methods
            value: function e() {
                var t;
                (t = this.state.heading) === null || t === void 0 ? void 0 : t.focus();
            }
        }, {
            key: "render",
            value: function e() {
                var t = this.props, o = t.children, i = t.description, a = t.index, l = t.panelId, c = t.status, u = t.style, s = t.title;
                var f = this.context.onHeadingClick;
                
                return r().createElement("div", be({
                    ref: this.handleMount,
                    "data-test": "panel",
                    "data-test-panel-id": l || "".concat(a)
                }, v()(this.props, [].concat(ye(b()(n.propTypes)), [ "style" ]))),  r().createElement(U, {
                    onClick: f,
                    ref: this.handleHeadingMount,
                    description: i,
                    index: a !== null && a !== void 0 ? a : 0,
                    panelId: l || "".concat(a),
                    position: "inner",
                    status: c
                }, s),  r().createElement(d(), {
                    target: se,
                    eventType: "resize",
                    listener: this.handleResize
                }),  r().createElement(ne, {
                    $status: c,
                    "data-test": "body",
                    style: u,
                    ref: this.handleBodyMount
                }, o));
            }
        } ]);
        return n;
    }(n.Component);
    Ie(Ae, "propTypes", Be);
    Ie(Ae, "defaultProps", Ne);
    Ie(Ae, "contextType", de);
    /* harmony default export */    const ze = Ae;
    // CONCATENATED MODULE: ./src/Concertina/Concertina.tsx
    function Me(e) {
        "@babel/helpers - typeof";
        return Me = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, Me(e);
    }
    function We() {
        return We = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var n = arguments[t];
                for (var r in n) {
                    ({}).hasOwnProperty.call(n, r) && (e[r] = n[r]);
                }
            }
            return e;
        }, We.apply(null, arguments);
    }
    function De(e, t) {
        if (null == e) return {};
        var n, r, o = Le(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                n = i[r], t.includes(n) || {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function Le(e, t) {
        if (null == e) return {};
        var n = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (t.includes(r)) continue;
                n[r] = e[r];
            }
        }
        return n;
    }
    function $e(e, t) {
        if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function");
    }
    function Ue(e, t) {
        for (var n = 0; n < t.length; n++) {
            var r = t[n];
            r.enumerable = r.enumerable || !1, r.configurable = !0, "value" in r && (r.writable = !0), 
            Object.defineProperty(e, et(r.key), r);
        }
    }
    function Ve(e, t, n) {
        return t && Ue(e.prototype, t), n && Ue(e, n), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function Fe(e, t) {
        if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), t && Ge(e, t);
    }
    function Ge(e, t) {
        return Ge = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, t) {
            return e.__proto__ = t, e;
        }, Ge(e, t);
    }
    function Je(e) {
        var t = Xe();
        return function() {
            var n, r = Ye(e);
            if (t) {
                var o = Ye(this).constructor;
                n = Reflect.construct(r, arguments, o);
            } else n = r.apply(this, arguments);
            return Ke(this, n);
        };
    }
    function Ke(e, t) {
        if (t && ("object" == Me(t) || "function" == typeof t)) return t;
        if (void 0 !== t) throw new TypeError("Derived constructors may only return object or undefined");
        return Qe(e);
    }
    function Qe(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function Xe() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return (Xe = function t() {
            return !!e;
        })();
    }
    function Ye(e) {
        return Ye = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, Ye(e);
    }
    function Ze(e, t, n) {
        return (t = et(t)) in e ? Object.defineProperty(e, t, {
            value: n,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = n, e;
    }
    function et(e) {
        var t = tt(e, "string");
        return "symbol" == Me(t) ? t : t + "";
    }
    function tt(e, t) {
        if ("object" != Me(e) || !e) return e;
        var n = e[Symbol.toPrimitive];
        if (void 0 !== n) {
            var r = n.call(e, t || "default");
            if ("object" != Me(r)) return r;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    var nt = {
        children: i().node,
        elementRef: i().oneOfType([ i().func, i().object ])
    };
    var rt = {};
    /**
 * @deprecated
 * Concertina has been deprecated and will be removed in a future major version.
 */    var ot =  function(e) {
        Fe(o, e);
        var t = Je(o);
        // @docs-props-type ConcertinaPropsBase
                function o(e) {
            var n;
            $e(this, o);
            n = t.call(this, e);
            Ze(Qe(n), "panelPositions", void 0);
            Ze(Qe(n), "handleUpdate", (function(e, t) {
                n.panelPositions[t.index] = t;
                var r = 0;
                n.panelPositions.forEach((function(e) {
                    // eslint-disable-next-line no-param-reassign
                    e.offsetTop = r;
                    r += e.height || 0;
                }));
                n.updateHeadings();
            }));
            Ze(Qe(n), "handleElementMount", (function(e) {
                n.setState({
                    scrollEl: e
                });
                // The scrollEl gets unmounted during the animation, it's necessary to cache the
                // recorded height until it is remounted.
                                if (e) {
                    n.setState({
                        innerWidth: e.clientWidth
                    });
                }
            }));
            Ze(Qe(n), "handleResize", (function() {
                n.setState((function(e) {
                    var t;
                    return {
                        innerWidth: (t = e.scrollEl) === null || t === void 0 ? void 0 : t.clientWidth
                    };
                }));
            }));
            Ze(Qe(n), "handleScroll", (function() {
                n.updateHeadings();
            }));
            Ze(Qe(n), "handleScrollComplete", (function() {
                n.setState({
                    targetTop: undefined
                });
            }));
            Ze(Qe(n), "handleClick", (function(e, t) {
                var r;
                var o = t.index, i = t.position;
                var a = n.state.scrollEl;
                if (!a) {
                    return;
                }
                var l = u()(n.panelPositions, o).map((function(e) {
                    return e.headingHeight;
                })).reduce((function(e, t) {
                    return e + t;
                }), 0);
 // sum the heights
                                var c = f()(n.panelPositions, n.panelPositions.length - o - 1).map((function(e) {
                    return e.headingHeight;
                })).reduce((function(e, t) {
                    return e + t;
                }), 0);
 // sum the heights
                                var s = a.clientHeight - l - c;
                var p = n.panelPositions[o].height;
                var d = ((r = n.panelPositions[o].offsetTop) !== null && r !== void 0 ? r : 0) - l;
                var h = d - (s - p);
                var b = s > p && (i === "bottom" || i === "inner");
                // if the entire panel is in view, do nothing
                                if (b && h < a.scrollTop) {
                    return;
                }
                var y = b ? h : d;
                // handleRest will shift focus;
                                if (i !== "inner") {
                    e.currentTarget.blur();
                }
                n.setState({
                    targetTop: y
                });
            }));
            Ze(Qe(n), "updateHeadingsImpl", (function() {
                var e = n.state.scrollEl;
                if (!e) {
                    return;
                }
                var t = e.scrollTop;
                function r(e, n) {
                    if (n.offsetTop !== undefined && n.offsetTop < t) {
                        t += n.headingHeight;
                        return e + 1;
                    }
                    return e;
                }
                var o = t + e.clientHeight;
                function i(e, t) {
                    if (t.offsetTop !== undefined && t.offsetTop + t.headingHeight > o) {
                        o -= t.headingHeight;
                        return e + 1;
                    }
                    return e;
                }
                n.setState({
                    panelsTopCount: n.panelPositions.reduce(r, 0),
                    panelsBottomCount: n.panelPositions.reduceRight(i, 0)
                });
            }));
            Ze(Qe(n), "updateHeadings", l()(n.updateHeadingsImpl, 0));
            n.state = {
                innerWidth: "100%",
                panelsBottomCount: 0,
                panelsTopCount: 0,
                scrollEl: null
            };
            n.panelPositions = [];
            return n;
        }
        Ve(o, [ {
            key: "componentWillUnmount",
            value: function e() {
                this.updateHeadings.cancel();
            }
        }, {
            key: "renderHeadings",
            value: function e(t, n) {
                var o = this;
                var i = t.map((function(e) {
                    var t;
                    
                    return r().createElement(U, {
                        onClick: o.handleClick,
                        description: e.props.description,
                        panelId: e.props.panelId,
                        index: (t = e.props.index) !== null && t !== void 0 ? t : 0,
                        position: n,
                        status: e.props.status,
                        key: e.props.panelId,
                        tabIndex: -1
                    }, e.props.title);
                }));
                return i;
            }
        }, {
            key: "render",
            value: function e() {
                if (false) {}
                var t = this.props, o = t.children, i = De(t, [ "children" ]);
                var a = n.Children.toArray(o).filter(n.isValidElement).map((function(e, t) {
                    
                    return (0, n.cloneElement)(e, {
                        index: t
                    });
                }));
                var l = u()(a, this.state.panelsTopCount);
                var c = f()(a, this.state.panelsBottomCount);
                
                return r().createElement(Z, We({
                    "data-test": "concertina"
                }, i),  r().createElement(d(), {
                    target: se,
                    eventType: "resize",
                    listener: this.handleResize
                }), this.state.panelsTopCount > 0 &&  r().createElement(ee, {
                    style: {
                        width: this.state.innerWidth
                    },
                    "data-test": "dock-top"
                }, this.renderHeadings(l, "top")),  r().createElement(Y, {
                    onScroll: this.handleScroll,
                    stopScrollPropagation: true,
                    key: "scroll-container",
                    elementRef: this.handleElementMount,
                    "data-test": "scroll",
                    top: this.state.targetTop,
                    onScrollComplete: this.handleScrollComplete
                },  r().createElement(de.Provider, {
                    // eslint-disable-next-line react/jsx-no-constructed-context-values
                    value: {
                        onHeadingClick: this.handleClick,
                        onChange: this.handleUpdate
                    }
                }, a)), this.state.panelsBottomCount > 0 &&  r().createElement(te, {
                    style: {
                        width: this.state.innerWidth
                    },
                    "data-test": "dock-bottom"
                }, this.renderHeadings(c, "bottom")));
            }
        } ]);
        return o;
    }(n.Component);
    Ze(ot, "propTypes", nt);
    Ze(ot, "defaultProps", rt);
    Ze(ot, "Panel", ze);
    /* harmony default export */    const it = ot;
    // CONCATENATED MODULE: ./src/Concertina/index.ts
    module.exports = t;
    /******/})();
