// @splunk/react-ui/Date.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var n = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, n) => {
            /******/ for (var a in n) {
                /******/ if (e.o(n, a) && !e.o(t, a)) {
                    /******/ Object.defineProperty(t, a, {
                        enumerable: true,
                        get: n[a]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        DateWithoutTheme: () => /* reexport */ U,
        default: () => /* reexport */ G,
        validateValue: () => /* reexport */ Z
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var a =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const r = require("prop-types");
    var o =  e.n(r);
    // CONCATENATED MODULE: external "lodash/has"
    const l = require("lodash/has");
    var i =  e.n(l);
    // CONCATENATED MODULE: external "lodash/omit"
    const u = require("lodash/omit");
    var s =  e.n(u);
    // CONCATENATED MODULE: external "moment"
    const c = require("moment");
    var p =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const f = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "@splunk/ui-utils/keyboard"
    const d = require("@splunk/ui-utils/keyboard");
    // CONCATENATED MODULE: external "@splunk/react-ui/Calendar"
    const v = require("@splunk/react-ui/Calendar");
    var h =  e.n(v);
    // CONCATENATED MODULE: external "@splunk/react-ui/Popover"
    const y = require("@splunk/react-ui/Popover");
    var b =  e.n(y);
    // CONCATENATED MODULE: external "@splunk/themes"
    const m = require("@splunk/themes");
    // CONCATENATED MODULE: external "styled-components"
    const g = require("styled-components");
    var C =  e.n(g);
    // CONCATENATED MODULE: external "@splunk/react-ui/Text"
    const O = require("@splunk/react-ui/Text");
    var x =  e.n(O);
    // CONCATENATED MODULE: ./src/Date/DateStyles.ts
    var k = C()(x()).withConfig({
        displayName: "DateStyles__StyledText",
        componentId: "kacgdc-0"
    })([ "cursor:", ";&[data-inline]{flex-basis:", ";width:", ";}" ], (0, m.pickVariant)("$disabled", {
        false: "text",
        true: "not-allowed"
    }), (0, m.pick)({
        enterprise: "105px",
        prisma: "150px"
    }), (0, m.pick)({
        enterprise: {
            comfortable: "105px",
            compact: "80px"
        },
        prisma: "150px"
    }));
    var w = C().div.withConfig({
        displayName: "DateStyles__IconContainer",
        componentId: "kacgdc-1"
    })([ "align-items:center;color:", ";display:flex;padding:", ";pointer-events:none;" ], (0, 
    m.pickVariant)("$disabled", {
        false: m.variables.contentColorMuted,
        true: m.variables.contentColorDisabled
    }), (0, m.pick)({
        comfortable: "0 8px",
        compact: "0 6px"
    }));
    // CONCATENATED MODULE: ./src/Date/Icon.tsx
    function T() {
        return T = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var n = arguments[t];
                for (var a in n) {
                    ({}).hasOwnProperty.call(n, a) && (e[a] = n[a]);
                }
            }
            return e;
        }, T.apply(null, arguments);
    }
    function S(e) {
        var t = (0, m.useSplunkTheme)(), n = t.density;
        var r = n === "compact" ? "20px" : "24px";
        
        return a().createElement("svg", T({
            viewBox: "0 0 24 24",
            width: r,
            height: r
        }, e),  a().createElement("path", {
            fill: "currentColor",
            fillRule: "evenodd",
            d: "M16.6001 2C17.1524 2 17.6001 2.44772 17.6001 3V4H19C20.1046 4 21 4.89543 21 6V8.5V9.5V19.5C21 19.7761 20.7761 20 20.5 20H3.5C3.22386 20 3 19.7761 3 19.5V9.5V8.5V6C3 4.89543 3.89543 4 5 4H6.6001V3C6.6001 2.44772 7.04781 2 7.6001 2C8.15238 2 8.6001 2.44772 8.6001 3V4H15.6001V3C15.6001 2.44772 16.0478 2 16.6001 2ZM6.6001 6C6.6001 6.55228 7.04781 7 7.6001 7C8.15238 7 8.6001 6.55228 8.6001 6H15.6001C15.6001 6.55228 16.0478 7 16.6001 7C17.1524 7 17.6001 6.55228 17.6001 6H19V8H5L5 6H6.6001ZM19 10H5V18H19V10Z"
        }));
    }
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function D(e, t) {
        if (e) {
            if (typeof e === "function") {
                e(t);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = t;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Date/Date.tsx
    function V(e) {
        "@babel/helpers - typeof";
        return V = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, V(e);
    }
    function I() {
        return I = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var n = arguments[t];
                for (var a in n) {
                    ({}).hasOwnProperty.call(n, a) && (e[a] = n[a]);
                }
            }
            return e;
        }, I.apply(null, arguments);
    }
    function j(e, t) {
        if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function");
    }
    function P(e, t) {
        for (var n = 0; n < t.length; n++) {
            var a = t[n];
            a.enumerable = a.enumerable || !1, a.configurable = !0, "value" in a && (a.writable = !0), 
            Object.defineProperty(e, W(a.key), a);
        }
    }
    function R(e, t, n) {
        return t && P(e.prototype, t), n && P(e, n), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function _(e, t) {
        if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), t && E(e, t);
    }
    function E(e, t) {
        return E = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, t) {
            return e.__proto__ = t, e;
        }, E(e, t);
    }
    function q(e) {
        var t = H();
        return function() {
            var n, a = K(e);
            if (t) {
                var r = K(this).constructor;
                n = Reflect.construct(a, arguments, r);
            } else n = a.apply(this, arguments);
            return M(this, n);
        };
    }
    function M(e, t) {
        if (t && ("object" == V(t) || "function" == typeof t)) return t;
        if (void 0 !== t) throw new TypeError("Derived constructors may only return object or undefined");
        return F(e);
    }
    function F(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function H() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return (H = function t() {
            return !!e;
        })();
    }
    function K(e) {
        return K = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, K(e);
    }
    function B(e, t, n) {
        return (t = W(t)) in e ? Object.defineProperty(e, t, {
            value: n,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = n, e;
    }
    function W(e) {
        var t = A(e, "string");
        return "symbol" == V(t) ? t : t + "";
    }
    function A(e, t) {
        if ("object" != V(e) || !e) return e;
        var n = e[Symbol.toPrimitive];
        if (void 0 !== n) {
            var a = n.call(e, t || "default");
            if ("object" != V(a)) return a;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    /** @public */
    /** @public */
    /** @public */    var Y = {
        defaultValue: o().string,
        describedBy: o().string,
        disabled: o().bool,
        elementRef: o().oneOfType([ o().func, o().object ]),
        error: o().bool,
        highlightToday: o().bool,
        inline: o().bool,
        inputId: o().string,
        inputOnly: o().bool,
        labelledBy: o().string,
        locale: o().string,
        name: o().string,
        onBlur: o().func,
        onChange: o().func,
        onClick: o().func,
        onFocus: o().func,
        onKeyDown: o().func,
        /** @private */
        splunkTheme: o().object,
        value: o().string
    };
    var $ = {
        disabled: false,
        error: false,
        highlightToday: false,
        inline: true,
        inputOnly: false,
        locale: "en_US"
    };
    /**
 * This static value can be used to convert a moment date to a compatible string
 * to set the `value` prop.
 * ```
 * moment().format(Date.momentFormat);
 * ```
 * @public
 * @name momentFormat
 * @memberOf Date
 * @type string
 */    var N = "YYYY-MM-DD";
    var Z = function e(t) {
        if (t === "") {
            return;
        }
        if (!p()(t, N).isValid()) {
            // eslint-disable-next-line no-console
            console.error('Invalid date string value "'.concat(t, '" provided to Moment.'));
        }
    };
    var U =  function(e) {
        _(n, e);
        var t = q(n);
        // @docs-props-type DatePropsBase
                function n(e) {
            var r;
            j(this, n);
            r = t.call(this, e);
            B(F(r), "controlledExternally", void 0);
            B(F(r), "popoverId", void 0);
            B(F(r), "calendarRef",  a().createRef());
            B(F(r), "openWithFocus", false);
            B(F(r), "applyTextChange", (function(e) {
                var t = r.state.tempTextInputDate;
                var n = p()(t, "l", r.props.locale);
                if (t === "") {
                    r.handleDateChange(e, {
                        value: ""
                    }, "textInput");
                } else if (n.isValid()) {
                    r.handleDateChange(e, {
                        value: n.format(N)
                    }, "textInput");
                } else {
                    r.setState({
                        calendarOpen: false,
                        tempTextInputDate: undefined
                    });
                }
            }));
            B(F(r), "handleDateChange", (function(e, t, n) {
                var a = t.value;
                if (n !== "textInput") {
                    r.focus();
                }
                if (r.getValue() !== a) {
                    var o, l;
                    r.setState({
                        calendarOpen: false,
                        tempTextInputDate: undefined,
                        value: r.isControlled() ? undefined : a
                    });
                    var i = r.props.name;
                    (o = (l = r.props).onChange) === null || o === void 0 ? void 0 : o.call(l, e, {
                        value: a,
                        name: i
                    });
                } else {
                    r.setState({
                        calendarOpen: false
                    });
                }
            }));
            B(F(r), "handleFocus", (function(e, t) {
                var n, a;
                if (!r.props.inputOnly) {
                    r.setState({
                        calendarOpen: true
                    });
                }
                (n = (a = r.props).onFocus) === null || n === void 0 ? void 0 : n.call(a, e, t);
            }));
            B(F(r), "handleClick", (function(e) {
                var t, n;
                if (!r.props.inputOnly && !r.state.calendarOpen) {
                    r.setState({
                        calendarOpen: true
                    });
                }
                (t = (n = r.props).onClick) === null || t === void 0 ? void 0 : t.call(n, e);
            }));
            B(F(r), "handleInputChange", (function(e, t) {
                var n = t.value;
                r.setState({
                    tempTextInputDate: n,
                    calendarOpen: !r.props.inputOnly
                });
            }));
            B(F(r), "handleKeyDown", (function(e) {
                var t, n;
                var a = (0, d.keycode)(e.nativeEvent);
                if (a === "enter" || a === "tab") {
                    r.applyTextChange(e);
                } else if (!r.props.inputOnly && a === "down") {
                    e.preventDefault();
                    if (!r.state.calendarOpen) {
                        r.openWithFocus = true;
                        r.setState({
                            calendarOpen: true
                        });
                    } else if (r.calendarRef) {
                        (0, v.focusActiveDay)(r.calendarRef);
                    }
                } else if (a === "esc") {
                    r.setState({
                        calendarOpen: false,
                        tempTextInputDate: undefined
                    });
                    r.focus();
                }
                (t = (n = r.props).onKeyDown) === null || t === void 0 ? void 0 : t.call(n, e);
            }));
            B(F(r), "handleMount", (function(e) {
                r.setState({
                    anchor: e
                });
                D(r.props.elementRef, e);
            }));
            B(F(r), "handleInputMount", (function(e) {
                r.setState({
                    input: e
                });
            }));
            B(F(r), "handleRequestClose", (function(e) {
                var t = e.event, n = e.reason;
                if ((n === "clickAway" || n === "escapeKey") && t !== undefined && t.target !== r.state.anchor) {
                    r.applyTextChange(t);
                    if (n === "escapeKey") {
                        r.focus();
                    }
                }
            }));
            r.controlledExternally = i()(e, "value");
            /**
     * TODO(SUI-7412): defaultValue will be set to current if date gets empty string.
     * Changing this would be a breaking change but this should be changed in epic/five
     * to have consistent behavior for empty-string between controlled and uncontrolled
     */            var o = r.isControlled() ? e.value : e.defaultValue || p()().locale(e.locale).format(N);
            r.state = {
                anchor: null,
                calendarOpen: false,
                input: null,
                value: r.isControlled() ? undefined : o
            };
            r.popoverId = (0, f.createDOMID)("calendar");
            if (false) {}
            if (false) {}
            return r;
        }
        R(n, [ {
            key: "componentDidUpdate",
            value: function e(t, n) {
                if (false) {}
                if (false) {}
                if (false) {}
                /**
       * If `openWithFocus` and `calendarOpen` are set the user is attempting to navigate to the dropdown, but it's closed.
       * On the first update following the state change set focus to the active day within the calendar.
       */                if (this.openWithFocus && this.state.calendarOpen && n.calendarOpen !== this.state.calendarOpen) {
                    this.openWithFocus = false;
                    (0, v.focusActiveDay)(this.calendarRef);
                }
            }
        }, {
            key: "getValue",
            value: function e() {
                return this.isControlled() ? this.props.value : this.state.value;
            }
        }, {
            key: "getTextInputValue",
            value: function e() {
                var t = this.state.tempTextInputDate;
                if (t === undefined) {
                    var n = this.getValue();
                    if (n === undefined || n === "") {
                        return "";
                    }
                    return p()(this.getValue(), N, this.props.locale).format("l");
                }
                return t;
            }
        }, {
            key: "isControlled",
            value: function e() {
                return this.controlledExternally;
            }
            /**
     * Place focus on the input.
     */        }, {
            key: "focus",
            value: function e() {
                var t;
                (t = this.state.input) === null || t === void 0 ? void 0 : t.focus();
            }
        }, {
            key: "render",
            value: function e() {
                var t = this.props, n = t.disabled, r = t.highlightToday, o = t.inputOnly, l = t.locale, i = t.splunkTheme;
                var u = i.isPrisma;
                var c = this.getValue();
                var p = u && !o &&  a().createElement(w, {
                    "aria-hidden": true,
                    $disabled: n
                },  a().createElement(S, null));
                var f;
                var d = {};
                if (!o) {
                    d.role = "combobox";
                    d["aria-controls"] = this.state.calendarOpen ? this.popoverId : undefined;
                    d["aria-expanded"] = n ? false : this.state.calendarOpen;
                    d["aria-haspopup"] = true;
                    f =  a().createElement(b(), {
                        anchor: this.state.anchor,
                        id: this.popoverId,
                        open: n ? false : this.state.calendarOpen,
                        onRequestClose: this.handleRequestClose
                    },  a().createElement(h(), {
                        elementRef: this.calendarRef,
                        highlightToday: r,
                        locale: l,
                        onChange: this.handleDateChange,
                        value: c === "" ? undefined : c
                    }));
                }
                
                return a().createElement(k, I({
                    autoCapitalize: "off",
                    autoComplete: "off",
                    autoCorrect: "off",
                    spellCheck: false,
                    "data-test": "date",
                    "data-test-value": c,
                    "data-test-popover-id": this.popoverId,
                    elementRef: this.handleMount,
                    inputRef: this.handleInputMount,
                    onChange: this.handleInputChange,
                    onClick: this.handleClick,
                    onFocus: this.handleFocus,
                    onKeyDown: this.handleKeyDown,
                    value: this.getTextInputValue(),
                    startAdornment: p,
                    $disabled: n
                }, d, s()(this.props, "defaultValue", "elementRef", "inputOnly", "multiline", "locale", "onChange", "onClick", "onFocus", "onKeyDown", "onSelect", "spellCheck", "value")), f);
            }
        } ]);
        return n;
    }(n.Component);
    B(U, "propTypes", Y);
    B(U, "defaultProps", $);
    var z = (0, m.withSplunkTheme)(U);
    var L = z;
    L.propTypes = U.propTypes;
    L.momentFormat = N;
    /* harmony default export */    const G = L;
    // only for styled components using Date
        // CONCATENATED MODULE: ./src/Date/index.ts
    module.exports = t;
    /******/})();
