// @splunk/react-ui/DefinitionList.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var n = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, n) => {
            /******/ for (var t in n) {
                /******/ if (e.o(n, t) && !e.o(r, t)) {
                    /******/ Object.defineProperty(r, t, {
                        enumerable: true,
                        get: n[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        Description: () => /* reexport */ m,
        Term: () => /* reexport */ S,
        default: () => /* reexport */ k
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var t =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const i = require("prop-types");
    var o =  e.n(i);
    // CONCATENATED MODULE: external "styled-components"
    const l = require("styled-components");
    var a =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/themes"
    const c = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/DefinitionList/DescriptionStyles.ts
    var s = a().dd.withConfig({
        displayName: "DescriptionStyles__Styled",
        componentId: "sc-1jglixb-0"
    })([ "", ";", ";min-height:", ";padding-left:5px;" ], c.mixins.reset("block"), (0, 
    c.pick)({
        prisma: c.mixins.typography("body", {
            color: "active"
        }),
        enterprise: c.mixins.typography("body")
    }), c.variables.lineHeight);
    // CONCATENATED MODULE: ./src/DefinitionList/Description.tsx
    function f() {
        return f = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, f.apply(null, arguments);
    }
    function p(e, r) {
        if (null == e) return {};
        var n, t, i = u(e, r);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (t = 0; t < o.length; t++) {
                n = o[t], r.includes(n) || {}.propertyIsEnumerable.call(e, n) && (i[n] = e[n]);
            }
        }
        return i;
    }
    function u(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (r.includes(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    var d = {
        children: o().node,
        elementRef: o().oneOfType([ o().func, o().object ])
    };
    /**
 * Container component for a `DefinitionList` description.
 */    function y(e) {
        var r = e.children, n = e.elementRef, i = p(e, [ "children", "elementRef" ]);
        // @docs-props-type DescriptionPropsBase
                
        return t().createElement(s, f({
            "data-test": "description",
            ref: n
        }, i), r);
    }
    y.propTypes = d;
    /* harmony default export */    const m = y;
    // CONCATENATED MODULE: ./src/DefinitionList/DefinitionListStyles.ts
    var h = a().dl.withConfig({
        displayName: "DefinitionListStyles__Styled",
        componentId: "sc-1pqfrgm-0"
    })([ "", ";", ";" ], c.mixins.reset("block"), c.mixins.typography("body"));
    // CONCATENATED MODULE: ./src/DefinitionList/TermStyles.ts
    var b = a().dt.withConfig({
        displayName: "TermStyles__Styled",
        componentId: "sc-1ao97j9-0"
    })([ "", ";", " float:left;width:120px;overflow:hidden;overflow-x:hidden;white-space:nowrap;word-wrap:normal;", "" ], c.mixins.reset("block"), (0, 
    c.pick)({
        prisma: c.mixins.typography("body", {
            color: "muted"
        }),
        enterprise: c.mixins.typography("body")
    }), (0, c.pick)({
        enterprise: (0, l.css)([ "&::after{content:'", "';}" ], " ".concat(".".repeat(100)))
    }));
    // CONCATENATED MODULE: ./src/DefinitionList/Term.tsx
    function v() {
        return v = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, v.apply(null, arguments);
    }
    function g(e, r) {
        if (null == e) return {};
        var n, t, i = O(e, r);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (t = 0; t < o.length; t++) {
                n = o[t], r.includes(n) || {}.propertyIsEnumerable.call(e, n) && (i[n] = e[n]);
            }
        }
        return i;
    }
    function O(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (r.includes(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    var w = {
        children: o().node,
        elementRef: o().oneOfType([ o().func, o().object ])
    };
    /**
 * Container component for a `DefinitionList` term.
 */    function j(e) {
        var r = e.children, n = e.elementRef, i = g(e, [ "children", "elementRef" ]);
        // @docs-props-type TermPropsBase
                
        return t().createElement(b, v({
            "data-test": "term",
            ref: n
        }, i), r);
    }
    j.propTypes = w;
    /* harmony default export */    const S = j;
    // CONCATENATED MODULE: ./src/DefinitionList/DefinitionList.tsx
    function P() {
        return P = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, P.apply(null, arguments);
    }
    function x(e, r) {
        if (null == e) return {};
        var n, t, i = T(e, r);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (t = 0; t < o.length; t++) {
                n = o[t], r.includes(n) || {}.propertyIsEnumerable.call(e, n) && (i[n] = e[n]);
            }
        }
        return i;
    }
    function T(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (r.includes(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    var _ = {
        children: o().node,
        elementRef: o().oneOfType([ o().func, o().object ]),
        termWidth: o().oneOfType([ o().number, o().string ])
    };
    function E(e) {
        var r = e.children, t = e.termWidth;
        var i = n.Children.toArray(r).filter(n.isValidElement);
        return i.map((function(e) {
            if (e.type === S) {
                
                return (0, n.cloneElement)(e, {
                    style: {
                        width: t
                    }
                });
            }
            if (e.type === m) {
                
                return (0, n.cloneElement)(e, {
                    style: {
                        marginLeft: t
                    }
                });
            }
            return e;
        }));
    }
    function R(e) {
        var r = e.children, n = e.elementRef, i = e.termWidth, o = i === void 0 ? 120 : i, l = x(e, [ "children", "elementRef", "termWidth" ]);
        // @docs-props-type DefinitionListPropsBase
                
        return t().createElement(h, P({
            "data-test": "definition-list",
            ref: n
        }, l), E({
            children: r,
            termWidth: o
        }));
    }
    R.propTypes = _;
    R.Description = m;
    R.Term = S;
    /* harmony default export */    const k = R;
    // CONCATENATED MODULE: ./src/DefinitionList/index.ts
    module.exports = r;
    /******/})();
