// @splunk/react-ui/File.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/    var e = {
        /***/ 6165: 
        /***/ (e, r, t) => {
            // EXPORTS
            t.d(r, {
                default: () => /* reexport */ b
            });
            // EXTERNAL MODULE: external "react"
                        var n = t(9497);
            var a =  t.n(n);
            // EXTERNAL MODULE: external "prop-types"
                        var i = t(23);
            var o =  t.n(i);
            // EXTERNAL MODULE: external "styled-components"
                        var l = t(232);
            var s =  t.n(l);
            // EXTERNAL MODULE: external "@splunk/themes"
                        var c = t(3563);
            // CONCATENATED MODULE: ./src/ScreenReaderContent/ScreenReaderContentStyles.ts
            var p = s().span.withConfig({
                displayName: "ScreenReaderContentStyles__Styled",
                componentId: "sc-1lnohwp-0"
            })([ "", ";" ], c.mixins.screenReaderContent());
            // CONCATENATED MODULE: ./src/ScreenReaderContent/ScreenReaderContent.tsx
            function u() {
                return u = Object.assign ? Object.assign.bind() : function(e) {
                    for (var r = 1; r < arguments.length; r++) {
                        var t = arguments[r];
                        for (var n in t) {
                            ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                        }
                    }
                    return e;
                }, u.apply(null, arguments);
            }
            function d(e, r) {
                if (null == e) return {};
                var t, n, a = v(e, r);
                if (Object.getOwnPropertySymbols) {
                    var i = Object.getOwnPropertySymbols(e);
                    for (n = 0; n < i.length; n++) {
                        t = i[n], r.includes(t) || {}.propertyIsEnumerable.call(e, t) && (a[t] = e[t]);
                    }
                }
                return a;
            }
            function v(e, r) {
                if (null == e) return {};
                var t = {};
                for (var n in e) {
                    if ({}.hasOwnProperty.call(e, n)) {
                        if (r.includes(n)) continue;
                        t[n] = e[n];
                    }
                }
                return t;
            }
            var f = {
                children: o().node.isRequired,
                elementRef: o().oneOfType([ o().func, o().object ])
            };
            /**
 * The screen reader text is used to wrap content that
 * is only accessible through screen readers.
 */            function m(e) {
                var r = e.children, t = e.elementRef, n = d(e, [ "children", "elementRef" ]);
                // @docs-props-type ScreenReaderContentPropsBase
                                
                return a().createElement(p, u({
                    "data-test": "screen-reader-content",
                    ref: t
                }, n), r);
            }
            m.propTypes = f;
            /* harmony default export */            const b = m;
        } // CONCATENATED MODULE: ./src/ScreenReaderContent/index.ts
        /***/ ,
        /***/ 3563: 
        /***/ e => {
            e.exports = require("@splunk/themes");
            /***/        },
        /***/ 23: 
        /***/ e => {
            e.exports = require("prop-types");
            /***/        },
        /***/ 9497: 
        /***/ e => {
            e.exports = require("react");
            /***/        },
        /***/ 232: 
        /***/ e => {
            e.exports = require("styled-components");
            /***/
            /******/        }
    };
    /************************************************************************/
    /******/ // The module cache
    /******/    var r = {};
    /******/
    /******/ // The require function
    /******/    function t(n) {
        /******/ // Check if module is in cache
        /******/ var a = r[n];
        /******/        if (a !== undefined) {
            /******/ return a.exports;
            /******/        }
        /******/ // Create a new module (and put it into the cache)
        /******/        var i = r[n] = {
            /******/ // no module.id needed
            /******/ // no module.loaded needed
            /******/ exports: {}
            /******/        };
        /******/
        /******/ // Execute the module function
        /******/        e[n](i, i.exports, t);
        /******/
        /******/ // Return the exports of the module
        /******/        return i.exports;
        /******/    }
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ t.n = e => {
            /******/ var r = e && e.__esModule ? 
            /******/ () => e["default"]
            /******/ : () => e
            /******/;
            t.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ t.d = (e, r) => {
            /******/ for (var n in r) {
                /******/ if (t.o(r, n) && !t.o(e, n)) {
                    /******/ Object.defineProperty(e, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ t.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ t.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // This entry need to be wrapped in an IIFE because it need to be isolated against other modules in the chunk.
        (() => {
        // ESM COMPAT FLAG
        t.r(n);
        // EXPORTS
                t.d(n, {
            Item: () => /* reexport */ se,
            default: () => /* reexport */ Ye
        });
        // EXTERNAL MODULE: external "react"
                var e = t(9497);
        var r =  t.n(e);
        // EXTERNAL MODULE: external "prop-types"
                var a = t(23);
        var i =  t.n(a);
        // CONCATENATED MODULE: external "lodash/debounce"
        const o = require("lodash/debounce");
        var l =  t.n(o);
        // CONCATENATED MODULE: external "lodash/omit"
        const s = require("lodash/omit");
        var c =  t.n(s);
        // CONCATENATED MODULE: external "lodash/toUpper"
        const p = require("lodash/toUpper");
        var u =  t.n(p);
        // CONCATENATED MODULE: external "lodash/isString"
        const d = require("lodash/isString");
        var v =  t.n(d);
        // CONCATENATED MODULE: external "@splunk/react-ui/EventListener"
        const f = require("@splunk/react-ui/EventListener");
        var m =  t.n(f);
        // EXTERNAL MODULE: external "@splunk/themes"
                var b = t(3563);
        // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
        const h = require("@splunk/ui-utils/i18n");
        // CONCATENATED MODULE: external "@splunk/ui-utils/id"
        const y = require("@splunk/ui-utils/id");
        // CONCATENATED MODULE: external "lodash/isUndefined"
        const g = require("lodash/isUndefined");
        var x =  t.n(g);
        // CONCATENATED MODULE: external "@splunk/react-icons/Cross"
        const C = require("@splunk/react-icons/Cross");
        var w =  t.n(C);
        // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/Close"
        const k = require("@splunk/react-icons/enterprise/Close");
        var S =  t.n(k);
        // CONCATENATED MODULE: external "@splunk/react-ui/Progress"
        const O = require("@splunk/react-ui/Progress");
        var E =  t.n(O);
        // CONCATENATED MODULE: external "@splunk/ui-utils/format"
        const I = require("@splunk/ui-utils/format");
        // CONCATENATED MODULE: ./src/File/FileContext.tsx
        var _ =  (0, e.createContext)({
            disabled: false
        });
        _.displayName = "File";
        /* harmony default export */        const L = _;
        // CONCATENATED MODULE: external "@splunk/react-icons/SVGEnterprise"
        const R = require("@splunk/react-icons/SVGEnterprise");
        var P =  t.n(R);
        // CONCATENATED MODULE: ./src/File/PaperClip.tsx
        function q() {
            return q = Object.assign ? Object.assign.bind() : function(e) {
                for (var r = 1; r < arguments.length; r++) {
                    var t = arguments[r];
                    for (var n in t) {
                        ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                    }
                }
                return e;
            }, q.apply(null, arguments);
        }
        function T(e) {
            
            return r().createElement(P(), q({
                hideDefaultTooltip: true,
                viewBox: "0 0 16 16",
                width: "16px",
                height: "16px"
            }, e),  r().createElement("path", {
                d: "M8.624 14.7021C8.23332 15.0928 7.59989 15.0928 7.2092 14.7021L2.33962 9.83209C0.546941 8.03927 0.54695 5.1325 2.33964 3.33968C4.13162 1.54758 7.03672 1.54684 8.82962 3.33802L14.1746 8.67787C15.2682 9.77043 15.2687 11.5429 14.1756 12.636C13.083 13.7288 11.3115 13.7288 10.2189 12.636L4.87676 7.29338C4.48607 6.90266 4.48607 6.26917 4.87676 5.87845C5.26744 5.48773 5.90087 5.48773 6.29155 5.87845L11.6337 11.2211C11.9449 11.5324 12.4496 11.5324 12.7608 11.2211C13.0722 10.9097 13.0721 10.4048 12.7606 10.0936L7.41557 4.75369C6.40415 3.74325 4.76532 3.74367 3.75442 4.75463C2.74312 5.766 2.74312 7.40578 3.75441 8.41715L8.62399 13.2871C9.01468 13.6779 9.01468 14.3113 8.624 14.7021Z",
                fill: "currentColor"
            }));
        }
        // CONCATENATED MODULE: external "@splunk/react-icons/ArrowsCircularDouble"
        const D = require("@splunk/react-icons/ArrowsCircularDouble");
        var j =  t.n(D);
        // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/Refresh"
        const N = require("@splunk/react-icons/enterprise/Refresh");
        var $ =  t.n(N);
        // CONCATENATED MODULE: ./src/File/Retry.tsx
        function M() {
            var e = (0, b.useSplunkTheme)(), t = e.density, n = e.family;
            var a = t === "compact" ? "20px" : "24px";
            if (n === "enterprise") {
                
                return r().createElement($(), {
                    screenReaderText: null
                });
            }
            
            return r().createElement(j(), {
                width: a,
                height: a
            });
        }
        // CONCATENATED MODULE: ./src/File/Trash.tsx
        function A() {
            return A = Object.assign ? Object.assign.bind() : function(e) {
                for (var r = 1; r < arguments.length; r++) {
                    var t = arguments[r];
                    for (var n in t) {
                        ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                    }
                }
                return e;
            }, A.apply(null, arguments);
        }
        function F(e) {
            var t = (0, b.useSplunkTheme)(), n = t.density;
            var a = n === "compact" ? "20px" : "24px";
            
            return r().createElement(P(), A({
                hideDefaultTooltip: true,
                viewBox: "0 0 24 24",
                width: a,
                height: a
            }, e),  r().createElement("path", {
                d: "M10 4V6H14V4H10ZM16 6H17H19H20C20.5523 6 21 6.44772 21 7C21 7.55228 20.5523 8 20 8H19V20.5C19 21.3284 18.3284 22 17.5 22H6.5C5.67157 22 5 21.3284 5 20.5V8H4C3.44772 8 3 7.55228 3 7C3 6.44772 3.44772 6 4 6H5H7H8V3.5C8 2.67157 8.67157 2 9.5 2H14.5C15.3284 2 16 2.67157 16 3.5V6ZM7 8V20H17V8H7ZM10.5858 14L9.2929 15.2929C8.90237 15.6834 8.90237 16.3166 9.2929 16.7071C9.68342 17.0976 10.3166 17.0976 10.7071 16.7071L12 15.4142L13.2929 16.7071C13.6834 17.0976 14.3166 17.0976 14.7071 16.7071C15.0976 16.3166 15.0976 15.6834 14.7071 15.2929L13.4142 14L14.7071 12.7071C15.0976 12.3166 15.0976 11.6834 14.7071 11.2929C14.3166 10.9024 13.6834 10.9024 13.2929 11.2929L12 12.5858L10.7071 11.2929C10.3166 10.9024 9.68342 10.9024 9.29289 11.2929C8.90237 11.6834 8.90237 12.3166 9.29289 12.7071L10.5858 14Z",
                fill: "currentColor"
            }));
        }
        // EXTERNAL MODULE: external "styled-components"
                var H = t(232);
        var B =  t.n(H);
        // CONCATENATED MODULE: external "@splunk/react-ui/Box"
        const V = require("@splunk/react-ui/Box");
        var z =  t.n(V);
        // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
        const G = require("@splunk/react-ui/Clickable");
        var Z =  t.n(G);
        // CONCATENATED MODULE: external "@splunk/themes/mixins"
        const W = require("@splunk/themes/mixins");
        // CONCATENATED MODULE: ./src/File/ItemStyles.ts
        var U = B()(z()).withConfig({
            displayName: "ItemStyles__StyledBox",
            componentId: "sc-1fsk3kr-0"
        })([ "position:relative;width:100%;max-width:400px;margin:", ";background-color:", ";border-radius:", ";min-height:32px;color:", ";", " ", " ", " ", ";" ], (0, 
        b.pick)({
            enterprise: "5px auto 0",
            prisma: {
                comfortable: "8px auto 0",
                compact: "4px auto 0"
            }
        }), (0, b.pick)({
            enterprise: {
                light: b.variables.gray92,
                dark: b.variables.gray45
            },
            prisma: b.variables.interactiveColorBackground
        }), (0, b.pick)({
            enterprise: "2px",
            prisma: b.variables.borderRadius
        }), (0, b.pick)({
            enterprise: b.variables.textGray,
            prisma: b.variables.contentColorDefault
        }), (0, b.pick)({
            prisma: (0, H.css)([ "border:1px solid ", ";padding-left:12px;" ], b.variables.neutral200),
            enterprise: (0, H.css)([ "line-height:24px;" ])
        }), (0, b.pick)({
            enterprise: {
                compact: (0, H.css)([ "min-height:24px;" ])
            }
        }), (function(e) {
            var r = e.$error;
            return r && (0, b.pick)({
                enterprise: (0, H.css)([ "box-shadow:inset 0 0 0 1px ", ";" ], b.variables.errorColor),
                prisma: (0, H.css)([ "box-shadow:inset 0 0 0 1px ", ";background-color:", ";" ], b.variables.accentColorNegative, b.variables.interactiveColorBackground)
            });
        }), (function(e) {
            var r = e.$disabled;
            return r && (0, H.css)([ "cursor:not-allowed;background-color:", ";" ], (0, b.pick)({
                enterprise: {
                    dark: b.variables.gray60,
                    light: b.variables.gray96
                },
                prisma: (0, W.overlayColors)(b.variables.interactiveColorBackground, b.variables.interactiveColorBackgroundDisabled)
            }));
        }));
        var X = B().div.withConfig({
            displayName: "ItemStyles__StyledErrorText",
            componentId: "sc-1fsk3kr-1"
        })([ "font-size:", ";color:", ";line-height:16px;margin-top:4px;text-align:left;" ], b.variables.fontSizeSmall, b.variables.accentColorNegative);
        var J = B()(Z()).withConfig({
            displayName: "ItemStyles__StyledIconClickable",
            componentId: "sc-1fsk3kr-2"
        })([ "color:", ";border-radius:", ";padding:", ";text-align:center;", " ", " &:focus{color:", ";", "}&:hover{background-color:", ";color:", ";}" ], (0, 
        b.pick)({
            enterprise: "inherit",
            prisma: b.variables.contentColorDefault
        }), (0, b.pick)({
            enterprise: "0 2px 2px 0",
            prisma: "50%"
        }), (0, b.pick)({
            enterprise: "3px 0",
            prisma: {
                comfortable: "0 8px",
                compact: "0 6px"
            }
        }), (0, b.pick)({
            enterprise: (0, H.css)([ "flex-basis:", ";" ], b.variables.inputHeight)
        }), (function(e) {
            var r = e.$error;
            return r && (0, H.css)([ "", "" ], (0, b.pick)({
                enterprise: (0, H.css)([ "border-top:1px solid ", ";border-bottom:1px solid ", ";&:last-child{border-right:1px solid ", ";}svg{vertical-align:baseline;}" ], b.variables.errorColor, b.variables.errorColor, b.variables.errorColor)
            }));
        }), (0, b.pick)({
            enterprise: b.variables.linkColor,
            prisma: b.variables.contentColorActive
        }), (0, b.pick)({
            enterprise: (0, H.css)([ "box-shadow:", ";" ], b.variables.focusShadow),
            prisma: (0, H.css)([ "background-color:", ";box-shadow:0 0 0 3px ", ";" ], b.variables.interactiveColorOverlayHover, b.variables.focusColor)
        }), (0, b.pick)({
            enterprise: {
                light: b.variables.gray96,
                dark: b.variables.gray30
            },
            prisma: b.variables.interactiveColorOverlayHover
        }), (0, b.pick)({
            enterprise: b.variables.linkColor,
            prisma: b.variables.contentColorActive
        }));
        // for Prisma theme only
                var K = B()(Z()).withConfig({
            displayName: "ItemStyles__StyledTrashIcon",
            componentId: "sc-1fsk3kr-3"
        })([ "color:", ";border-radius:50%;padding:", ";margin-right:", ";text-align:center;&:focus{color:", ";background-color:", ";box-shadow:0 0 0 3px ", ";}&:hover{background-color:", ";color:", ";}" ], b.variables.contentColorDefault, (0, 
        b.pick)({
            comfortable: "0 8px",
            compact: "0 6px"
        }), (0, b.pick)({
            comfortable: "0 4px",
            compact: "0 6px"
        }), b.variables.contentColorActive, b.variables.interactiveColorOverlayHover, b.variables.focusColor, b.variables.interactiveColorOverlayHover, b.variables.contentColorActive);
        var Q = B().div.withConfig({
            displayName: "ItemStyles__StyledLabel",
            componentId: "sc-1fsk3kr-4"
        })([ "color:", ";overflow:hidden;white-space:nowrap;text-overflow:ellipsis;flex:1 0 0px;padding:", ";border-radius:", ";", "" ], b.variables.contentColorActive, (0, 
        b.pick)({
            enterprise: "3px 0 3px 10px",
            prisma: {
                comfortable: "10px 0 10px 0",
                compact: "6px 0 6px 0"
            }
        }), (0, b.pick)({
            enterprise: "2px",
            prisma: b.variables.borderRadius
        }), (function(e) {
            var r = e.$disabled;
            return r && (0, b.pick)({
                enterprise: {
                    light: (0, H.css)([ "color:", ";" ], b.variables.textGray),
                    dark: (0, H.css)([ "color:", ";" ], b.variables.textGray)
                },
                prisma: (0, H.css)([ "color:", ";" ], b.variables.contentColorDisabled)
            });
        }));
        // for Prisma theme only
                var Y = B().div.withConfig({
            displayName: "ItemStyles__StyledPaperClipIcon",
            componentId: "sc-1fsk3kr-5"
        })([ "color:", ";padding:", ";", "" ], b.variables.neutral400, (0, b.pick)({
            comfortable: "8px 4px 8px 0",
            compact: "4px 4px 4px 0"
        }), (function(e) {
            var r = e.$disabled;
            return r && (0, H.css)([ "color:", ";cursor:not-allowed;" ], b.variables.contentColorDisabled);
        }));
        // for Prisma theme only
                var ee = B().span.withConfig({
            displayName: "ItemStyles__StyledPercentage",
            componentId: "sc-1fsk3kr-6"
        })([ "color:", ";" ], b.variables.contentColorMuted);
        // for Prisma theme only
                var re = B().div.withConfig({
            displayName: "ItemStyles__StyledProgress",
            componentId: "sc-1fsk3kr-7"
        })([ "position:absolute;left:0;top:0;width:", "%;height:100%;background:", ";transition:width 300ms;border-radius:", ";" ], (function(e) {
            var r = e.$uploadPercentage;
            return r;
        }), b.variables.interactiveColorOverlaySelected, b.variables.borderRadius);
        // EXTERNAL MODULE: ./src/ScreenReaderContent/index.ts + 2 modules
                var te = t(6165);
        // CONCATENATED MODULE: ./src/File/Item.tsx
        function ne() {
            return ne = Object.assign ? Object.assign.bind() : function(e) {
                for (var r = 1; r < arguments.length; r++) {
                    var t = arguments[r];
                    for (var n in t) {
                        ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                    }
                }
                return e;
            }, ne.apply(null, arguments);
        }
        function ae(e, r) {
            if (null == e) return {};
            var t, n, a = ie(e, r);
            if (Object.getOwnPropertySymbols) {
                var i = Object.getOwnPropertySymbols(e);
                for (n = 0; n < i.length; n++) {
                    t = i[n], r.includes(t) || {}.propertyIsEnumerable.call(e, t) && (a[t] = e[t]);
                }
            }
            return a;
        }
        function ie(e, r) {
            if (null == e) return {};
            var t = {};
            for (var n in e) {
                if ({}.hasOwnProperty.call(e, n)) {
                    if (r.includes(n)) continue;
                    t[n] = e[n];
                }
            }
            return t;
        }
        var oe = {
            disabled: i().bool,
            elementRef: i().oneOfType([ i().func, i().object ]),
            error: i().bool,
            /** @private */
            index: i().number,
            itemId: i().any,
            name: i().string.isRequired,
            /** If the uploadPercentage is 0, the item is assumed to be queued. If the upload is complete or
   * not applicable, uploadPercentage must be undefined. */
            uploadPercentage: i().number
        };
        function le(t) {
            var n = t.disabled, a = t.error, i = a === void 0 ? false : a, o = t.itemId, l = t.name, s = t.index, c = s === void 0 ? 0 : s, p = t.uploadPercentage, u = ae(t, [ "disabled", "error", "itemId", "name", "index", "uploadPercentage" ]);
            // @docs-props-type ItemPropsBase
                        var d = (0, b.useSplunkTheme)(), v = d.isPrisma, f = d.isEnterprise, m = d.isCompact;
            var y = m ? "20px" : "24px";
            var g = (0, e.useContext)(_), C = g.onRequestRemove;
            var k = (0, e.useCallback)((function(e) {
                e.preventDefault();
                C === null || C === void 0 ? void 0 : C({
                    itemId: o,
                    name: l,
                    index: c
                });
            }), [ C, o, l, c ]);
            var O = (0, e.useContext)(_), L = O.onRequestRetry;
            var R = (0, e.useCallback)((function(e) {
                e.preventDefault();
                L === null || L === void 0 ? void 0 : L({
                    itemId: o,
                    name: l,
                    index: c
                });
            }), [ L, o, l, c ]);
            var P = (0, e.useContext)(_), q = P.disabled;
            var D = (0, I.sprintf)((0, h._)('Remove "%(filename)s"'), {
                filename: l
            });
            var j = (0, I.sprintf)((0, h._)('Retry "%(filename)s"'), {
                filename: l
            });
            var N = (0, I.sprintf)((0, h._)('Uploading "%(filename)s"'), {
                filename: l
            });
            var $ = (0, h._)("Remove");
            var A =  r().createElement("span", null,  r().createElement("span", {
                style: {
                    paddingRight: "4px"
                }
            }, (0, h._)("Uploading ")),  r().createElement(ee, null, p, "%"));
            var H = v && !x()(p) && p > 0 ? A : l;
            var B = n !== undefined ? n : q;
            
            return r().createElement(r().Fragment, null,  r().createElement(U, ne({
                "data-test": "item",
                "data-test-disabled": B,
                "data-test-error": i
            }, u, {
                flex: true,
                $error: i,
                $disabled: B
            }), v && x()(p) &&  r().createElement(Y, {
                "data-test": "paper-clip",
                $disabled: B
            },  r().createElement(T, {
                "aria-hidden": "true"
            })),  r().createElement(Q, {
                "data-test": "label",
                $error: i,
                $disabled: B
            }, H), L && i &&  r().createElement(J, {
                "data-test": "retry",
                $error: i,
                onClick: R,
                "aria-label": j
            },  r().createElement(M, null),  r().createElement(te["default"], null, (0, h._)("Retry"))), !B && (v && x()(p) && !i ?  r().createElement(K, {
                "data-test": "remove",
                onClick: k,
                "aria-label": D
            },  r().createElement(F, {
                screenReaderText: (0, h._)("Remove")
            })) :  r().createElement(J, {
                "data-test": "remove",
                $error: i,
                onClick: k,
                "aria-label": D
            }, f ?  r().createElement(S(), {
                hideDefaultTooltip: true,
                screenReaderText: $
            }) :  r().createElement(w(), {
                "aria-label": $,
                height: y,
                width: y
            }))), !x()(p) && p > 0 && (v ?  r().createElement(re, {
                $uploadPercentage: p,
                "aria-label": N,
                role: "progressbar",
                "aria-valuenow": p,
                "aria-valuemin": 0,
                "aria-valuemax": 100
            }) :  r().createElement(E(), {
                style: {
                    position: "absolute",
                    left: 0,
                    top: 0,
                    right: 0,
                    zIndex: 1
                },
                percentage: p,
                "aria-label": N
            }))), v && i &&  r().createElement(X, null, (0, h._)("Something went wrong.")));
        }
        le.propTypes = oe;
        /* harmony default export */        const se = le;
        // CONCATENATED MODULE: ./src/File/Icon.tsx
        function ce() {
            return ce = Object.assign ? Object.assign.bind() : function(e) {
                for (var r = 1; r < arguments.length; r++) {
                    var t = arguments[r];
                    for (var n in t) {
                        ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                    }
                }
                return e;
            }, ce.apply(null, arguments);
        }
        function pe(e) {
            
            return r().createElement(P(), ce({
                hideDefaultTooltip: true,
                screenReaderText: (0, h._)("File upload"),
                viewBox: "0 0 72 88"
            }, e),  r().createElement("path", {
                d: "M50,27 L68.0005854,27 C70.2022516,27 72,28.7919267 72,31.0023804 L72,83.9976196 C72,86.2074215 70.2094011,88 68.0005854,88 L3.99941455,88 C1.79774843,88 0,86.2080733 0,83.9976196 L0,31.0023804 C0,28.7925785 1.79059889,27 3.99941455,27 L21,27 L21,32 L5.99898406,32 C5.4472604,32 5,32.4408979 5,32.9958767 L5,82.0041233 C5,82.5541308 5.44605521,83 5.99898406,83 L66.0010159,83 C66.5527396,83 67,82.5591021 67,82.0041233 L67,32.9958767 C67,32.4458692 66.5539448,32 66.0010159,32 L50,32 L50,27 Z"
            }),  r().createElement("path", {
                d: "M41.9634682,10 L41.9634682,28 L46.9634682,28 L46.9634682,5 L44.4634682,5 L23.9634682,5 L23.9634682,10 L41.9634682,10 Z",
                transform: "translate(35.463468, 16.500000) rotate(-45.000000) translate(-35.463468, -16.500000) "
            }),  r().createElement("rect", {
                x: "33",
                y: "3",
                width: "5",
                height: "51"
            }));
        }
        // CONCATENATED MODULE: ./src/File/FileStyles.ts
        var ue = B().input.withConfig({
            displayName: "FileStyles__StyledInput",
            componentId: "sc-1wyh4cf-0"
        })([ "&[type='file']{width:0.1px;height:0.1px;opacity:0;overflow:hidden;position:absolute;z-index:-1;}" ]);
        var de = (0, H.css)([ "fill:", ";" ], (0, b.pick)({
            enterprise: {
                light: b.variables.gray60,
                dark: b.variables.gray80
            }
        }));
        var ve = B()(pe).withConfig({
            displayName: "FileStyles__StyledNormalIcon",
            componentId: "sc-1wyh4cf-1"
        })([ "", ";height:1.4em;width:1.4em;display:inline-block;vertical-align:middle;padding-bottom:3px;" ], de);
        var fe = B()(pe).withConfig({
            displayName: "FileStyles__StyledWindowIcon",
            componentId: "sc-1wyh4cf-2"
        })([ "", ";height:48px;width:48px;position:absolute;top:30px;left:50%;transform:translateX(-50%);" ], de);
        var me = B().label.withConfig({
            displayName: "FileStyles__StyledLink",
            componentId: "sc-1wyh4cf-3"
        })([ "", ";color:", ";cursor:pointer;font-size:inherit;font-weight:inherit;&:hover,&[data-focused]{text-decoration:underline;}&[data-focused]{box-shadow:", ";}" ], b.mixins.reset("inline"), b.variables.linkColor, (0, 
        b.pick)({
            enterprise: b.variables.focusShadowInset,
            prisma: b.variables.focusShadow
        }));
        var be = B().div.withConfig({
            displayName: "FileStyles__StyledNormalText",
            componentId: "sc-1wyh4cf-4"
        })([ "", " ", " ", " ", "" ], (0, b.pick)({
            enterprise: (0, H.css)([ "display:inline-block;" ]),
            prisma: (0, H.css)([ "display:inline-flex;flex-direction:column;justify-content:center;text-align:center;color:", ";border:1px dashed ", ";padding:8px;border-radius:", ";min-height:128px;&:not(:last-child){margin-bottom:4px;}" ], b.variables.contentColorActive, b.variables.contentColorMuted, b.variables.borderRadius)
        }), (function(e) {
            var r = e.$dragOver;
            return r && (0, H.css)([ "", "" ], (0, b.pick)({
                prisma: (0, H.css)([ "border:1px dashed ", ";background-color:", ";" ], b.variables.interactiveColorPrimary, b.mixins.colorWithAlpha(b.variables.interactiveColorPrimary, .1))
            }));
        }), (function(e) {
            var r = e.$error, t = e.$dragOver;
            return r && !t && (0, b.pick)({
                prisma: (0, H.css)([ "border:1px dashed ", ";color:", ";background-color:", ";" ], b.variables.accentColorNegative, b.variables.accentColorNegative, b.mixins.colorWithAlpha(b.variables.accentColorNegative, .1))
            });
        }), (function(e) {
            var r = e.$disabled;
            return r && (0, H.css)([ "border:none;min-height:0;color:", ";cursor:not-allowed;" ], (0, 
            b.pick)({
                prisma: b.variables.contentColorDisabled
            }));
        }));
        var he = B().div.withConfig({
            displayName: "FileStyles__StyledWindowText",
            componentId: "sc-1wyh4cf-5"
        })([ "margin-top:calc(", " * 4);margin-bottom:", ";font-size:", ";", "" ], b.variables.spacingLarge, (0, 
        b.pick)({
            enterprise: b.variables.spacingSmall,
            prisma: b.variables.spacingMedium
        }), b.variables.fontSizeXLarge, (0, b.pick)({
            prisma: (0, H.css)([ "color:", ";" ], b.variables.contentColorActive)
        }));
        var ye = B().div.withConfig({
            displayName: "FileStyles__StyledSubtitle",
            componentId: "sc-1wyh4cf-6"
        })([ "color:", ";padding:4px 8px;font-size:12px;line-height:16px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;" ], b.variables.contentColorMuted);
        var ge = B().div.withConfig({
            displayName: "FileStyles__StyledHelp",
            componentId: "sc-1wyh4cf-7"
        })([ "margin-bottom:", ";" ], b.variables.spacingXLarge);
        var xe = B().div.withConfig({
            displayName: "FileStyles__StyledWindowDrop",
            componentId: "sc-1wyh4cf-8"
        })([ "position:fixed;top:0;left:0;right:0;bottom:0;border:", ";z-index:", " + 10;" ], (0, 
        b.pick)({
            enterprise: "5px solid ".concat(b.variables.accentColorL10),
            prisma: "5px solid ".concat(b.variables.interactiveColorPrimary)
        }), b.variables.zindexModal);
        var Ce = B()(z()).withConfig({
            displayName: "FileStyles__StyledDropNormalTargetBox",
            componentId: "sc-1wyh4cf-9"
        })([ "", ";flex-direction:column;justify-content:center;text-align:center;border-radius:", ";", " ", " ", " ", " ", " ", "" ], b.mixins.reset("flex"), b.variables.borderRadius, (0, 
        b.pick)({
            enterprise: (0, H.css)([ "line-height:calc(", " - 2px);" ], b.variables.inputHeight)
        }), (0, b.pick)({
            enterprise: {
                comfortable: (0, H.css)([ "min-height:73px;padding:", ";" ], b.variables.spacingXSmall),
                compact: (0, H.css)([ "min-height:63px;padding:3px;" ])
            }
        }), (0, b.pick)({
            enterprise: {
                light: (0, H.css)([ "border:1px dashed ", ";" ], b.variables.borderColor),
                dark: (0, H.css)([ "border:1px dashed ", ";" ], b.variables.textGray)
            }
        }), (function(e) {
            var r = e.$dragOver;
            return r && (0, b.pick)({
                enterprise: (0, H.css)([ "border:1px solid ", ";" ], b.variables.accentColorL10)
            });
        }), (function(e) {
            var r = e.$error, t = e.$dragOver;
            return r && !t && (0, b.pick)({
                enterprise: (0, H.css)([ "border:1px solid ", ";" ], b.variables.errorColor)
            });
        }), (function(e) {
            var r = e.$disabled, t = e.$fileCount;
            return r && (0, H.css)([ "border:", ";color:", ";cursor:not-allowed;", " ", "" ], (0, 
            b.pick)({
                enterprise: "none",
                prisma: "1px solid ".concat(b.variables.contentColorDisabled)
            }), (0, b.pick)({
                enterprise: b.variables.textGray,
                prisma: b.variables.contentColorDisabled
            }), t === 0 && (0, b.pick)({
                enterprise: {
                    light: (0, H.css)([ "background-color:", ";" ], b.variables.gray96),
                    dark: (0, H.css)([ "background-color:", ";" ], b.variables.gray60)
                },
                prisma: (0, H.css)([ "border:1px dashed ", ";min-height:128px;" ], b.variables.contentColorDisabled)
            }), t !== 0 && (0, b.pick)({
                prisma: (0, H.css)([ "", "{padding:0;}" ], /* sc-sel */ be)
            }));
        }));
        var we = B()(z()).withConfig({
            displayName: "FileStyles__StyledWindowDropTargetBox",
            componentId: "sc-1wyh4cf-10"
        })([ "position:relative;text-align:center;min-height:250px;padding:", ";", "" ], b.variables.spacingLarge, (function(e) {
            var r = e.$disabled;
            return r && (0, H.css)([ "color:", ";" ], (0, b.pick)({
                enterprise: b.variables.textGray,
                prisma: b.variables.contentColorMuted
            }));
        }));
        // CONCATENATED MODULE: ./src/utils/ssrDocument.ts
        /* eslint-disable @typescript-eslint/no-empty-function */
        var ke = {
            body: {
                appendChild: function e() {
                    return [];
                }
            },
            addEventListener: function e() {},
            removeEventListener: function e() {},
            activeElement: {
                blur: function e() {},
                nodeName: ""
            },
            querySelector: function e() {
                return null;
            },
            querySelectorAll: function e() {
                return [];
            },
            getElementById: function e() {
                return null;
            },
            createEvent: function e() {
                return {
                    initEvent: function e() {}
                };
            },
            createElement: function e() {
                return {
                    children: [],
                    childNodes: [],
                    style: {},
                    setAttribute: function e() {},
                    getElementsByTagName: function e() {
                        return [];
                    }
                };
            },
            createElementNS: function e() {
                return {};
            },
            importNode: function e() {
                return null;
            },
            location: {
                hash: "",
                host: "",
                hostname: "",
                href: "",
                origin: "",
                pathname: "",
                protocol: "",
                search: ""
            }
        };
        function Se() {
            var e = typeof document !== "undefined" ? document : ke;
            return e;
        }
        var Oe = Se();
        /* harmony default export */        const Ee = /* unused pure expression or super */ null && Oe;
        // CONCATENATED MODULE: ./src/utils/ssrWindow.ts
        /* eslint-disable @typescript-eslint/no-empty-function */
        var Ie = {
            document: ke,
            navigator: {
                userAgent: ""
            },
            location: {
                hash: "",
                host: "",
                hostname: "",
                href: "",
                origin: "",
                pathname: "",
                protocol: "",
                search: ""
            },
            history: {
                replaceState: function e() {},
                pushState: function e() {},
                go: function e() {},
                back: function e() {}
            },
            CustomEvent: function e() {
                return this;
            },
            addEventListener: function e() {},
            removeEventListener: function e() {},
            getComputedStyle: function e() {
                return {
                    getPropertyValue: function e() {
                        return "";
                    }
                };
            },
            Image: function e() {},
            Date: function e() {},
            screen: {},
            setTimeout: function e() {},
            clearTimeout: function e() {},
            matchMedia: function e() {
                return {};
            },
            requestAnimationFrame: function e(r) {
                if (typeof setTimeout === "undefined") {
                    r();
                    return null;
                }
                return setTimeout(r, 0);
            },
            cancelAnimationFrame: function e(r) {
                if (typeof setTimeout === "undefined") {
                    return;
                }
                clearTimeout(r);
            }
        };
        function _e() {
            var e = typeof window !== "undefined" ? window : Ie;
            return e;
        }
        var Le = _e();
        /* harmony default export */        const Re = Le;
        // CONCATENATED MODULE: ./src/File/File.tsx
        function Pe(e) {
            "@babel/helpers - typeof";
            return Pe = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
                return typeof e;
            } : function(e) {
                return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
            }, Pe(e);
        }
        function qe() {
            return qe = Object.assign ? Object.assign.bind() : function(e) {
                for (var r = 1; r < arguments.length; r++) {
                    var t = arguments[r];
                    for (var n in t) {
                        ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                    }
                }
                return e;
            }, qe.apply(null, arguments);
        }
        function Te(e, r) {
            if (null == e) return {};
            var t, n, a = De(e, r);
            if (Object.getOwnPropertySymbols) {
                var i = Object.getOwnPropertySymbols(e);
                for (n = 0; n < i.length; n++) {
                    t = i[n], r.includes(t) || {}.propertyIsEnumerable.call(e, t) && (a[t] = e[t]);
                }
            }
            return a;
        }
        function De(e, r) {
            if (null == e) return {};
            var t = {};
            for (var n in e) {
                if ({}.hasOwnProperty.call(e, n)) {
                    if (r.includes(n)) continue;
                    t[n] = e[n];
                }
            }
            return t;
        }
        function je(e, r) {
            if (!(e instanceof r)) throw new TypeError("Cannot call a class as a function");
        }
        function Ne(e, r) {
            for (var t = 0; t < r.length; t++) {
                var n = r[t];
                n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), 
                Object.defineProperty(e, Ze(n.key), n);
            }
        }
        function $e(e, r, t) {
            return r && Ne(e.prototype, r), t && Ne(e, t), Object.defineProperty(e, "prototype", {
                writable: !1
            }), e;
        }
        function Me(e, r) {
            if ("function" != typeof r && null !== r) throw new TypeError("Super expression must either be null or a function");
            e.prototype = Object.create(r && r.prototype, {
                constructor: {
                    value: e,
                    writable: !0,
                    configurable: !0
                }
            }), Object.defineProperty(e, "prototype", {
                writable: !1
            }), r && Ae(e, r);
        }
        function Ae(e, r) {
            return Ae = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, r) {
                return e.__proto__ = r, e;
            }, Ae(e, r);
        }
        function Fe(e) {
            var r = Ve();
            return function() {
                var t, n = ze(e);
                if (r) {
                    var a = ze(this).constructor;
                    t = Reflect.construct(n, arguments, a);
                } else t = n.apply(this, arguments);
                return He(this, t);
            };
        }
        function He(e, r) {
            if (r && ("object" == Pe(r) || "function" == typeof r)) return r;
            if (void 0 !== r) throw new TypeError("Derived constructors may only return object or undefined");
            return Be(e);
        }
        function Be(e) {
            if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
            return e;
        }
        function Ve() {
            try {
                var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
            } catch (e) {}
            return (Ve = function r() {
                return !!e;
            })();
        }
        function ze(e) {
            return ze = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
                return e.__proto__ || Object.getPrototypeOf(e);
            }, ze(e);
        }
        function Ge(e, r, t) {
            return (r = Ze(r)) in e ? Object.defineProperty(e, r, {
                value: t,
                enumerable: !0,
                configurable: !0,
                writable: !0
            }) : e[r] = t, e;
        }
        function Ze(e) {
            var r = We(e, "string");
            return "symbol" == Pe(r) ? r : r + "";
        }
        function We(e, r) {
            if ("object" != Pe(e) || !e) return e;
            var t = e[Symbol.toPrimitive];
            if (void 0 !== t) {
                var n = t.call(e, r || "default");
                if ("object" != Pe(n)) return n;
                throw new TypeError("@@toPrimitive must return a primitive value.");
            }
            return ("string" === r ? String : Number)(e);
        }
        /** @public */
        /** @public */        var Ue = {
            accept: i().string,
            allowMultiple: i().bool,
            children: i().node,
            disabled: i().bool,
            dropAnywhere: i().bool,
            elementRef: i().oneOfType([ i().func, i().object ]),
            fullscreen: i().bool,
            error: i().bool,
            help: i().node,
            name: i().string,
            onRequestAdd: i().func,
            onRequestRemove: i().func,
            onRequestRetry: i().func,
            /** @private. */
            required: i().bool,
            supportsMessage: i().node,
            /** @private */
            splunkTheme: i().object
        };
        var Xe = {
            allowMultiple: false,
            disabled: false,
            dropAnywhere: false,
            error: false
        };
        /**
 * File provides the ability to accept files and present uploaded files. It does not provide
 * file readers, only a reference to the file. This can be used to post binary content, or
 * upload using an array buffer.
 */        var Je =  function(t) {
            Me(a, t);
            var n = Fe(a);
            // @docs-props-type FilePropsBase
                        function a(e) {
                var r;
                je(this, a);
                r = n.call(this, e);
                Ge(Be(r), "inputId", void 0);
                Ge(Be(r), "inputCount", void 0);
                Ge(Be(r), "handleInputChange", (function(e) {
                    var t;
                    r.addFiles((t = e.currentTarget.files) !== null && t !== void 0 ? t : undefined);
                }));
                Ge(Be(r), "handleInputFocus", (function() {
                    r.setState({
                        focusedInput: true
                    });
                }));
                Ge(Be(r), "handleInputBlur", (function() {
                    r.setState({
                        focusedInput: false
                    });
                }));
                Ge(Be(r), "handleDragOver", (function(e) {
                    if (!r.state.dragOver) {
                        r.setState({
                            dragOver: true
                        });
                    }
                    e.preventDefault();
                }));
                Ge(Be(r), "handleDragLeave", (function() {
                    r.setState({
                        dragOver: false
                    });
                }));
                Ge(Be(r), "handleDrop", (function(e) {
                    var t;
                    e.preventDefault();
                    r.handleDragLeave();
                    r.addFiles((t = e.dataTransfer) === null || t === void 0 ? void 0 : t.files);
                }));
                r.state = {
                    dragOver: false,
                    focusedInput: false
                };
                r.inputId = (0, y.createDOMID)();
                /* Each time a file is uploaded this is incremented and used to generate the
     * file input's key. In this way we get a new input without a value. */                r.inputCount = 0;
                r.handleDragLeave = l()(r.handleDragLeave, 300);
                return r;
            }
            $e(a, [ {
                key: "addFiles",
                value: function e(r) {
                    var t = this.props.name;
                    if (r !== undefined && r.length > 0) {
                        var n, a;
                        var i = Array.from(r);
                        (n = (a = this.props).onRequestAdd) === null || n === void 0 ? void 0 : n.call(a, this.props.allowMultiple ? i : [ i[0] ], {
                            name: t
                        });
                    } else {
                        var o, l;
                        (o = (l = this.props).onRequestAdd) === null || o === void 0 ? void 0 : o.call(l, [], {
                            name: t
                        });
                    }
                    this.inputCount += 1;
                }
            }, {
                key: "render",
                value: function t() {
                    var n = this.props, a = n.accept, i = n.allowMultiple, o = n.children, l = n.disabled, s = n.dropAnywhere, p = n.error, d = n.fullscreen, f = n.help, b = n.name, y = n.onRequestRemove, g = n.onRequestRetry, x = n.required, C = n.splunkTheme, w = Te(n, [ "accept", "allowMultiple", "children", "disabled", "dropAnywhere", "error", "fullscreen", "help", "name", "onRequestRemove", "onRequestRetry", "required", "splunkTheme" ]);
                    var k = C.isPrisma;
                    var S = 0;
                    var O = e.Children.toArray(o).filter(e.isValidElement).map((function(r, t) {
                        S += 1;
                        
                        return (0, e.cloneElement)(r, {
                            index: t,
                            key: r.key || r.props.itemId || "item-".concat(t)
                        });
                    }));
                    var E = d || s;
                    var I = E || l;
                    var _ = E && !l;
                    var R = d ? we : Ce;
                    var P = d ? he : be;
                    var q = d ? fe : ve;
                    var T = I ? false : this.state.dragOver || false;
                    var D = v()(a) && u()(a.split(",").join(", "));
                    var j = function e(r) {
                        g === null || g === void 0 ? void 0 : g({
                            event: r,
                            filename: r.name,
                            index: r.index,
                            itemId: r.itemId,
                            name: b
                        });
                    };
                    var N = function e(r) {
                        y === null || y === void 0 ? void 0 : y({
                            event: r,
                            index: r.index,
                            filename: r.name,
                            itemId: r.itemId,
                            name: b
                        });
                    };
                    var $ = null;
                    if (this.props.supportsMessage) {
                        $ =  r().createElement(ye, {
                            "data-test": "file-supports"
                        }, this.props.supportsMessage);
                    } else if (k && !l) {
                        $ = a ?  r().createElement(ye, {
                            "data-test": "file-supports"
                        }, (0, h._)("Supports ".concat(D))) :  r().createElement(ye, {
                            "data-test": "file-supports"
                        }, (0, h._)("Supports all file types"));
                    }
                    
                    return r().createElement(R, qe({
                        onDragOver: I ? undefined : this.handleDragOver,
                        onDragLeave: I ? undefined : this.handleDragLeave,
                        onDrop: I ? undefined : this.handleDrop,
                        $disabled: l,
                        $dragOver: T,
                        $error: p,
                        $fileCount: S,
                        "data-test": "file",
                        "data-test-disabled": l,
                        "data-test-error": p
                    }, c()(w, "onRequestAdd", "onRequestRemove")),  r().createElement(P, {
                        $disabled: l,
                        $dragOver: T,
                        $error: p
                    }, !l && !k &&  r().createElement(q, {
                        $dragOver: T,
                        $error: p
                    }), " ",  r().createElement("span", {
                        "data-test": "file-label"
                    }, !E && !l && (0, h._)("Drop your file here or"), _ && (0, h._)("Drop your file anywhere or"), " ", S === 0 && l && (0, 
                    h._)("No files selected"),  r().createElement(me, {
                        htmlFor: this.inputId,
                        "data-test": "file-link",
                        "data-focused": this.state.focusedInput || null
                    },  r().createElement(ue, {
                        accept: a,
                        "data-test": "file-input",
                        disabled: l,
                        id: this.inputId,
                        key: "file-input-".concat(this.inputCount),
                        multiple: i || undefined,
                        onBlur: this.handleInputBlur,
                        onChange: this.handleInputChange,
                        onFocus: this.handleInputFocus,
                        required: x,
                        type: "file"
                    }), !l && (0, h._)("upload file"))), $), !l && f &&  r().createElement(ge, {
                        "data-test": "help"
                    }, f), _ && this.state.dragOver &&  r().createElement(xe, {
                        "data-test": "file-window-drop",
                        onDragLeave: this.handleDragLeave
                    }), _ &&  r().createElement(r().Fragment, null,  r().createElement(m(), {
                        target: Re,
                        eventType: "dragover",
                        listener: this.handleDragOver
                    }),  r().createElement(m(), {
                        target: Re,
                        eventType: "drop",
                        listener: this.handleDrop
                    })),  r().createElement(L.Provider, {
                        // eslint-disable-next-line react/jsx-no-constructed-context-values
                        value: {
                            disabled: l,
                            onRequestRemove: y ? N : undefined,
                            onRequestRetry: g ? j : undefined
                        }
                    }, O));
                }
            } ]);
            return a;
        }(e.Component);
        Ge(Je, "propTypes", Ue);
        Ge(Je, "defaultProps", Xe);
        Ge(Je, "Item", se);
        var Ke = (0, b.withSplunkTheme)(Je);
        var Qe = Ke;
        Qe.propTypes = Je.propTypes;
        Qe.Item = se;
        /* harmony default export */        const Ye = Qe;
    }) // CONCATENATED MODULE: ./src/File/index.ts
    ();
    module.exports = n;
    /******/})();
