// @splunk/react-ui/FormRows.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        Row: () => /* reexport */ Oe,
        default: () => /* reexport */ Ue
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const a = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/ui-utils/keyboard"
    const l = require("@splunk/ui-utils/keyboard");
    // CONCATENATED MODULE: ./src/FormRows/FormRowsContext.tsx
    var u =  (0, r.createContext)({
        disabledDeleteButton: false,
        sortable: true
    });
    u.displayName = "FormRows";
    /* harmony default export */    const c = u;
    // CONCATENATED MODULE: external "@splunk/themes"
    const s = require("@splunk/themes");
    // CONCATENATED MODULE: external "@splunk/react-icons/Plus"
    const p = require("@splunk/react-icons/Plus");
    var f =  e.n(p);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/Plus"
    const d = require("@splunk/react-icons/enterprise/Plus");
    var v =  e.n(d);
    // CONCATENATED MODULE: ./src/FormRows/icons/FormRowsPlusIcon.tsx
    var b = function e() {
        var t = (0, s.useSplunkTheme)(), r = t.isEnterprise, o = t.isCompact;
        var i = o ? "20px" : "24px";
        return r ?  n().createElement(v(), {
            screenReaderText: null,
            hideDefaultTooltip: true
        }) :  n().createElement(f(), {
            width: i,
            height: i
        });
    };
    /* harmony default export */    const y = b;
    // CONCATENATED MODULE: external "lodash/omit"
    const m = require("lodash/omit");
    var h =  e.n(m);
    // CONCATENATED MODULE: external "@splunk/react-icons/Cross"
    const g = require("@splunk/react-icons/Cross");
    var w =  e.n(g);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/Close"
    const x = require("@splunk/react-icons/enterprise/Close");
    var O =  e.n(x);
    // CONCATENATED MODULE: external "styled-components"
    const S = require("styled-components");
    var j =  e.n(S);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const R = require("@splunk/react-ui/Box");
    var P =  e.n(R);
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const E = require("@splunk/react-ui/Button");
    var k =  e.n(E);
    // CONCATENATED MODULE: external "@splunk/react-ui/ButtonSimple"
    const _ = require("@splunk/react-ui/ButtonSimple");
    var q =  e.n(_);
    // CONCATENATED MODULE: ./src/FormRows/FormRowsStyles.ts
    var C = j()(P()).withConfig({
        displayName: "FormRowsStyles__StyledBox",
        componentId: "sc-1j4j7h3-0"
    })([ "", ";" ], s.mixins.reset("block"));
    var T = j().div.withConfig({
        displayName: "FormRowsStyles__StyledHeader",
        componentId: "sc-1j4j7h3-1"
    })([ "padding-left:", ";", "" ], (0, s.pick)({
        enterprise: s.variables.spacingSmall,
        prisma: "16px"
    }), (0, s.pick)({
        prisma: {
            comfortable: (0, S.css)([ "padding-bottom:8px;" ]),
            compact: (0, S.css)([ "padding-bottom:6px;" ])
        }
    }));
    var D = j()(k()).withConfig({
        displayName: "FormRowsStyles__StyledAddRowButton",
        componentId: "sc-1j4j7h3-2"
    })([ "", "" ], (function(e) {
        var t = e.$sortable;
        return t && (0, s.pick)({
            prisma: (0, S.css)([ "margin-left:16px;" ])
        });
    }));
    var I = j().div.withConfig({
        displayName: "FormRowsStyles__StyledRows",
        componentId: "sc-1j4j7h3-3"
    })([ "position:relative;", "" ], (0, s.pick)({
        prisma: {
            comfortable: (0, S.css)([ "padding-bottom:12px;" ]),
            compact: (0, S.css)([ "padding-bottom:10px;" ])
        }
    }));
    var B = j().div.withConfig({
        displayName: "FormRowsStyles__Styled",
        componentId: "sc-1j4j7h3-4"
    })([ "", ";position:relative;&.sorting{z-index:calc(", " + 1);}", " &:focus{outline:0;box-shadow:", ";}" ], (0, 
    s.pick)({
        enterprise: (0, S.css)([ "padding:3px calc(", " + 3px) 3px 0;" ], s.variables.inputHeight),
        prisma: {
            comfortable: (0, S.css)([ "padding:8px calc(", " + 8px) 8px 0;" ], s.variables.inputHeight),
            compact: (0, S.css)([ "padding:6px calc(", " + 8px) 6px 0;" ], s.variables.inputHeight)
        }
    }), s.variables.zindexPopover, (function(e) {
        var t = e.$sortable;
        return t && (0, S.css)([ "padding-left:", ";" ], (0, s.pick)({
            enterprise: s.variables.spacingSmall,
            prisma: "16px"
        }));
    }), s.variables.focusShadow);
    var N = j().div.withConfig({
        displayName: "FormRowsStyles__StyledButton",
        componentId: "sc-1j4j7h3-5"
    })([ "position:absolute;right:0;top:", ";float:right;" ], (0, s.pick)({
        enterprise: "3px",
        prisma: {
            comfortable: "8px",
            compact: "6px"
        }
    }));
    var A = j()(q()).withConfig({
        displayName: "FormRowsStyles__StyledButtonSimple",
        componentId: "sc-1j4j7h3-6"
    })([ "float:right;width:", ";height:", ";text-align:center;" ], s.variables.inputHeight, s.variables.inputHeight);
    var H = j().div.withConfig({
        displayName: "FormRowsStyles__StyledDrag",
        componentId: "sc-1j4j7h3-7"
    })([ "position:absolute;left:0;right:0;content:'';cursor:move;background:", ";width:", ";top:", ";", "" ], s.variables.draggableBackground, (0, 
    s.pick)({
        enterprise: "7px",
        prisma: "6px"
    }), (0, s.pick)({
        enterprise: "4px",
        prisma: {
            comfortable: "18px",
            compact: "12px"
        }
    }), (0, s.pick)({
        enterprise: (0, S.css)([ "bottom:3px;opacity:0.5;" ]),
        prisma: (0, S.css)([ "height:20px;" ])
    }));
    // CONCATENATED MODULE: ./src/FormRows/RowInternal.tsx
    function M() {
        return M = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, M.apply(null, arguments);
    }
    function F(e, t) {
        if (null == e) return {};
        var r, n, o = K(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                r = i[n], t.includes(r) || {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function K(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (t.includes(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    /** @public */    var L = {
        children: i().node,
        index: i().number,
        onRequestRemove: i().func,
        renderDragHandle: i().func,
        style: i().object,
        value: i().node
    };
    var $ =  n().forwardRef((function(e, t) {
        var o = e.children, i = e.index, l = e.onRequestRemove, u = e.renderDragHandle, p = F(e, [ "children", "index", "onRequestRemove", "renderDragHandle" ]);
        var f = (0, s.useSplunkTheme)(), d = f.isCompact, v = f.isEnterprise;
        var b = d ? "20px" : "24px";
        var y = (0, r.useContext)(c), m = y.disabledDeleteButton, g = y.hasOnRequestAdd, x = y.onKeyDown, S = y.sortable;
        var j = (0, r.useCallback)((function(e) {
            // TODO: this callback doesn't work correctly if index isn't provided SUI-5639
            x === null || x === void 0 ? void 0 : x(e, {
                index: i !== null && i !== void 0 ? i : 0
            });
        }), [ x, i ]);
        var R = (0, r.useCallback)((function(e) {
            // TODO: this callback doesn't work correctly if index isn't provided SUI-5639
            l === null || l === void 0 ? void 0 : l(e, {
                index: i !== null && i !== void 0 ? i : 0
            });
        }), [ l, i ]);
        var P = l !== undefined;
        if (false) {}
        if (false) {}
        if (false) {}
        
        return n().createElement(B, M({
            $sortable: S,
            "data-test": "row",
            ref: t,
            onKeyDown: j
        }, h()(p, [ "onKeyDown", "onRequestRemove" ])), u && u(), o, P &&  n().createElement(N, null,  n().createElement(A, {
            appearance: "subtle",
            "aria-label": (0, a._)("Remove Row"),
            "data-test": "remove",
            disabled: m,
            onClick: R
        }, v ?  n().createElement(O(), {
            "data-test": "cross",
            inline: false,
            screenReaderText: null,
            size: "13px"
        }) :  n().createElement(w(), {
            "data-test": "cross",
            height: b,
            width: b
        }))));
    }));
    $.propTypes = L;
    /* harmony default export */    const z = $;
    // CONCATENATED MODULE: external "@dnd-kit/sortable"
    const V = require("@dnd-kit/sortable");
    // CONCATENATED MODULE: external "@dnd-kit/utilities"
    const G = require("@dnd-kit/utilities");
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const J = require("@splunk/react-ui/ScreenReaderContent");
    var Q =  e.n(J);
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function U(e, t) {
        if (e) {
            if (typeof e === "function") {
                e(t);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = t;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/FormRows/SortableRow.tsx
    function W(e, t) {
        if (null == e) return {};
        var r, n, o = X(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                r = i[n], t.includes(r) || {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function X(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (t.includes(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    function Y() {
        return Y = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, Y.apply(null, arguments)
        /** @public */;
    }
    var Z = {
        children: i().node,
        index: i().number,
        onRequestRemove: i().func,
        value: i().node
    };
    var ee =  n().forwardRef((function(e, t) {
        
        return n().createElement(H, Y({
            "data-test": "drag-handle",
            ref: t
        }, e),  n().createElement(Q(), null, (0, a._)("Press arrow up or arrow down to re-order items")));
    }));
    var te =  n().forwardRef((function(e, t) {
        var r = e.children, o = e.index, i = W(e, [ "children", "index" ]);
        // @docs-props-type RowPropsBase
                var a = (0, V.useSortable)({
            // @dnd-kit requires "id" to be greater than 0 https://github.com/clauderic/dnd-kit/issues/1150
            // TODO: FormRows drag functionality doesn't work without an index, but it's not a required prop SUI-5639
            id: (o || 0) + 1
        }), l = a.attributes, u = a.listeners, c = a.setActivatorNodeRef, s = a.setNodeRef, p = a.transform, f = a.transition;
        var d = {
            transform: G.CSS.Transform.toString(p),
            transition: f
        };
        var v = function e(r) {
            s(r);
            U(t, r);
        };
        
        return n().createElement(z, Y({
            index: o,
            ref: v,
            renderDragHandle: function e() {
                
                return n().createElement(ee, Y({
                    ref: c
                }, u));
            },
            style: d
        }, i, l), r);
    }));
    te.propTypes = Z;
    /* harmony default export */    const re = te;
    // CONCATENATED MODULE: ./src/FormRows/Row.tsx
    function ne(e) {
        "@babel/helpers - typeof";
        return ne = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, ne(e);
    }
    function oe() {
        return oe = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, oe.apply(null, arguments);
    }
    function ie(e, t) {
        if (null == e) return {};
        var r, n, o = ae(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                r = i[n], t.includes(r) || {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function ae(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (t.includes(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    function le(e, t) {
        if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function");
    }
    function ue(e, t) {
        for (var r = 0; r < t.length; r++) {
            var n = t[r];
            n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), 
            Object.defineProperty(e, he(n.key), n);
        }
    }
    function ce(e, t, r) {
        return t && ue(e.prototype, t), r && ue(e, r), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function se(e, t) {
        if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), t && pe(e, t);
    }
    function pe(e, t) {
        return pe = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, t) {
            return e.__proto__ = t, e;
        }, pe(e, t);
    }
    function fe(e) {
        var t = be();
        return function() {
            var r, n = ye(e);
            if (t) {
                var o = ye(this).constructor;
                r = Reflect.construct(n, arguments, o);
            } else r = n.apply(this, arguments);
            return de(this, r);
        };
    }
    function de(e, t) {
        if (t && ("object" == ne(t) || "function" == typeof t)) return t;
        if (void 0 !== t) throw new TypeError("Derived constructors may only return object or undefined");
        return ve(e);
    }
    function ve(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function be() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return (be = function t() {
            return !!e;
        })();
    }
    function ye(e) {
        return ye = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, ye(e);
    }
    function me(e, t, r) {
        return (t = he(t)) in e ? Object.defineProperty(e, t, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = r, e;
    }
    function he(e) {
        var t = ge(e, "string");
        return "symbol" == ne(t) ? t : t + "";
    }
    function ge(e, t) {
        if ("object" != ne(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var n = r.call(e, t || "default");
            if ("object" != ne(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    /** @public */    var we = {
        children: i().node,
        elementRef: i().oneOfType([ i().func, i().object ]),
        index: i().number,
        onRequestRemove: i().func,
        value: i().node
    };
    var xe =  function(e) {
        se(r, e);
        var t = fe(r);
        function r() {
            le(this, r);
            return t.apply(this, arguments);
        }
        ce(r, [ {
            key: "render",
            // @docs-props-type RowPropsBase
            value: function e() {
                var t = this.props, r = t.children, o = t.elementRef, i = ie(t, [ "children", "elementRef" ]);
                var a = this.context.sortable;
                var l = h()(i, [ "onKeyDown" ]);
                var u = a ? re : z;
                
                return n().createElement(u, oe({
                    ref: o
                }, l), r);
            }
        } ]);
        return r;
    }(r.Component);
    me(xe, "propTypes", we);
    me(xe, "contextType", c);
    /* harmony default export */    const Oe = xe;
    // CONCATENATED MODULE: external "@dnd-kit/core"
    const Se = require("@dnd-kit/core");
    // CONCATENATED MODULE: ./src/FormRows/SortableList.tsx
    var je = function e(t) {
        var o = t.children, i = t.onSortEnd;
        var a = (0, Se.useSensors)((0, Se.useSensor)(Se.PointerSensor), (0, Se.useSensor)(Se.MouseSensor));
        var l = r.Children.toArray(o).filter(r.isValidElement);
        var u = l.map((function(e) {
            return (e.props.index || 0) + 1;
        }));
        var c = (0, r.useCallback)((function(e) {
            var t = e.active, r = e.over;
            if (r && t.id !== r.id) {
                var n = u.indexOf(t.id);
                var o = u.indexOf(r.id);
                i({
                    oldIndex: n,
                    newIndex: o
                });
            }
        }), [ u, i ]);
        
        return n().createElement(Se.DndContext, {
            sensors: a,
            collisionDetection: Se.closestCenter,
            onDragEnd: c
        },  n().createElement(V.SortableContext, {
            items: u,
            strategy: V.verticalListSortingStrategy
        }, l));
    };
    /* harmony default export */    const Re = je;
    // CONCATENATED MODULE: ./src/FormRows/FormRows.tsx
    function Pe(e) {
        "@babel/helpers - typeof";
        return Pe = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, Pe(e);
    }
    function Ee() {
        return Ee = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, Ee.apply(null, arguments);
    }
    function ke(e, t) {
        if (null == e) return {};
        var r, n, o = _e(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                r = i[n], t.includes(r) || {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function _e(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (t.includes(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    function qe(e, t) {
        if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function");
    }
    function Ce(e, t) {
        for (var r = 0; r < t.length; r++) {
            var n = t[r];
            n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), 
            Object.defineProperty(e, Ke(n.key), n);
        }
    }
    function Te(e, t, r) {
        return t && Ce(e.prototype, t), r && Ce(e, r), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function De(e, t) {
        if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), t && Ie(e, t);
    }
    function Ie(e, t) {
        return Ie = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, t) {
            return e.__proto__ = t, e;
        }, Ie(e, t);
    }
    function Be(e) {
        var t = He();
        return function() {
            var r, n = Me(e);
            if (t) {
                var o = Me(this).constructor;
                r = Reflect.construct(n, arguments, o);
            } else r = n.apply(this, arguments);
            return Ne(this, r);
        };
    }
    function Ne(e, t) {
        if (t && ("object" == Pe(t) || "function" == typeof t)) return t;
        if (void 0 !== t) throw new TypeError("Derived constructors may only return object or undefined");
        return Ae(e);
    }
    function Ae(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function He() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return (He = function t() {
            return !!e;
        })();
    }
    function Me(e) {
        return Me = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, Me(e);
    }
    function Fe(e, t, r) {
        return (t = Ke(t)) in e ? Object.defineProperty(e, t, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = r, e;
    }
    function Ke(e) {
        var t = Le(e, "string");
        return "symbol" == Pe(t) ? t : t + "";
    }
    function Le(e, t) {
        if ("object" != Pe(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var n = r.call(e, t || "default");
            if ("object" != Pe(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    /** @public */    var $e = {
        addLabel: i().string,
        children: i().node,
        disabled: i().bool,
        elementRef: i().oneOfType([ i().func, i().object ]),
        header: i().node,
        menu: i().node,
        onRequestAdd: i().func,
        onRequestMove: i().func
    };
    var ze = {
        addLabel: (0, a._)("Add Row"),
        disabled: false,
        header: null
    };
    /**
 * Static function for adding a row.
 *
 * Example Use:
 *
 *      onRequestAdd() => {
 *          this.setState(state => ({
 *              items: FormRows.addRow(
 *                  <FormRows.Row
 *                      index={state.items.length}
 *                      key={createDOMID()}
 *                      onRequestRemove={this.handleRequestRemove}
 *                  >
 *                      <Text />
 *                  </FormRows.Row>,
 *                  state.items
 *              ),
 *          }));
 *      };
 *
 * @param element Row element to add.
 * @param items Array of current FormRows to add to.
 * @return New array of FormRows with new Row added.
 */    var Ve = function e(t, r) {
        return r.concat(t);
    };
    /**
 * Static function for moving a row.
 *
 * Example Use:
 *
 *      onRequestMove({ fromIndex, toIndex }) => {
 *          this.setState(state => ({
 *              items: FormRows.moveRow(fromIndex, toIndex, state.items),
 *          }));
 *      };
 *
 * @param fromIndex Current index of row to move.
 * @param toIndex New index to move row to.
 * @param items Array of current FormRows.
 * @return New array of FormRows arranged in new order.
 */    var Ge = function e(t, n, o) {
        var i = o.filter((function(e, r) {
            return r !== t;
        }));
        i.splice(n, 0, o[t]);
        return i.map((function(e, t) {
            
            return (0, r.cloneElement)(e, {
                index: t
            });
        }));
    };
    /**
 * Static function for removing a row.
 *
 * Example Use:
 *
 *      onRequestRemove(e, { index }) => {
 *          this.setState(state => ({
 *              items: FormRows.removeRow(index, state.items),
 *          }));
 *      };
 *
 * @param index Index of Row to delete.
 * @param items Array of current FormRows.
 * @return New array of FormRows with Row of given index deleted.
 */    var Je = function e(t, n) {
        return n.filter((function(e, r) {
            return r !== t;
        })).map((function(e, t) {
            
            return (0, r.cloneElement)(e, {
                index: t
            });
        }));
    };
    var Qe =  function(e) {
        De(r, e);
        var t = Be(r);
        function r() {
            var e;
            qe(this, r);
            for (var o = arguments.length, i = new Array(o), a = 0; a < o; a++) {
                i[a] = arguments[a];
            }
            e = t.call.apply(t, [ this ].concat(i));
            Fe(Ae(e), "handleKeyDown", (function(t, r) {
                var o = r.index;
                if (t.currentTarget !== t.target) {
                    return;
                }
                if ((0, l.keycode)(t.nativeEvent) === "up") {
                    t.preventDefault();
                    if (o > 0) {
                        var i, a;
                        (i = (a = e.props).onRequestMove) === null || i === void 0 ? void 0 : i.call(a, {
                            fromIndex: o,
                            toIndex: o - 1
                        });
                    }
                } else if ((0, l.keycode)(t.nativeEvent) === "down") {
                    t.preventDefault();
                    if (o < n().Children.toArray(e.props.children).length - 1) {
                        var u, c;
                        (u = (c = e.props).onRequestMove) === null || u === void 0 ? void 0 : u.call(c, {
                            fromIndex: o,
                            toIndex: o + 1
                        });
                    }
                }
            }));
            Fe(Ae(e), "handleSortEnd", (function(t) {
                var r, n;
                var o = t.oldIndex, i = t.newIndex;
                (r = (n = e.props).onRequestMove) === null || r === void 0 ? void 0 : r.call(n, {
                    fromIndex: o,
                    toIndex: i
                });
            }));
            return e;
        }
 // @docs-props-type FormRowsPropsBase
                Te(r, [ {
            key: "render",
            value: function e() {
                var t = this.props, r = t.addLabel, o = t.children, i = t.disabled, a = t.header, l = t.menu, u = t.onRequestAdd, s = t.onRequestMove, p = ke(t, [ "addLabel", "children", "disabled", "header", "menu", "onRequestAdd", "onRequestMove" ]);
                var f = s !== undefined && !i;
                var d = f ? T : "div";
                var v = u !== undefined;
                
                return n().createElement(C, Ee({
                    "data-test": "form-rows"
                }, p),  n().createElement(c.Provider, {
                    // eslint-disable-next-line react/jsx-no-constructed-context-values
                    value: {
                        disabledDeleteButton: i,
                        onKeyDown: this.handleKeyDown,
                        hasOnRequestAdd: v,
                        sortable: f
                    }
                }, a &&  n().createElement(d, null, a),  n().createElement(I, null,  n().createElement(Re, {
                    onSortEnd: this.handleSortEnd
                }, o)), l || v &&  n().createElement(D, {
                    disabled: i,
                    appearance: "subtle",
                    "data-test": "add-row",
                    icon:  n().createElement(y, null),
                    label: r,
                    onClick: u,
                    $sortable: f
                })));
            }
        } ]);
        return r;
    }(r.Component);
    Fe(Qe, "propTypes", $e);
    Fe(Qe, "defaultProps", ze);
    Fe(Qe, "Row", Oe);
    Fe(Qe, "addRow", Ve);
    Fe(Qe, "moveRow", Ge);
    Fe(Qe, "removeRow", Je);
    /* harmony default export */    const Ue = Qe;
    // CONCATENATED MODULE: ./src/FormRows/index.ts
    module.exports = t;
    /******/})();
