// @splunk/react-ui/Heading.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var i in r) {
                /******/ if (e.o(r, i) && !e.o(t, i)) {
                    /******/ Object.defineProperty(t, i, {
                        enumerable: true,
                        get: r[i]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        default: () => /* reexport */ d
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var i =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const n = require("prop-types");
    var a =  e.n(n);
    // CONCATENATED MODULE: external "styled-components"
    const l = require("styled-components");
    var o =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/themes"
    const s = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Heading/HeadingStyles.ts
    var c = o().h2.withConfig({
        displayName: "HeadingStyles__StyledHeading",
        componentId: "sc-15s3yx3-0"
    })([ "", ";color:", ";margin:1.414em 0 0.4em;text-rendering:optimizelegibility;", " ", "" ], (function(e) {
        var t = e.$variant;
        if (t === "title4") {
            return (0, s.pickVariant)("$isSection", {
                true: {
                    enterprise: {
                        compact: s.mixins.typography("title4", {
                            weight: "semiBold"
                        }),
                        comfortable: s.mixins.typography("title4", {
                            size: 14,
                            weight: "semiBold"
                        })
                    },
                    prisma: s.mixins.typography("title4", {
                        color: "default",
                        size: 14
                    })
                },
                false: {
                    enterprise: {
                        compact: s.mixins.typography("title4"),
                        comfortable: s.mixins.typography("title4", {
                            size: 14
                        })
                    },
                    prisma: s.mixins.typography("title4")
                }
            });
        }
        return s.mixins.typography(t);
    }), (0, s.pick)({
        prisma: (0, s.pickVariant)("$variant", {
            title5: s.mixins.typography("title5", {
                color: "default"
            })
        })
    }), (0, s.pickVariant)("$isSection", {
        true: {
            enterprise: {
                dark: (0, l.css)([ "color:", ";" ], s.variables.gray80),
                light: (0, l.css)([ "color:", ";" ], s.variables.gray45)
            }
        }
    }), (function(e) {
        var t = e.$isSection;
        return t && (0, l.css)([ "margin:0.707em 0 0.2em;" ]);
    }));
    // CONCATENATED MODULE: ./src/Heading/Heading.tsx
    function p() {
        return p = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var i in r) {
                    ({}).hasOwnProperty.call(r, i) && (e[i] = r[i]);
                }
            }
            return e;
        }, p.apply(null, arguments);
    }
    function u(e, t) {
        if (null == e) return {};
        var r, i, n = f(e, t);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (i = 0; i < a.length; i++) {
                r = a[i], t.includes(r) || {}.propertyIsEnumerable.call(e, r) && (n[r] = e[r]);
            }
        }
        return n;
    }
    function f(e, t) {
        if (null == e) return {};
        var r = {};
        for (var i in e) {
            if ({}.hasOwnProperty.call(e, i)) {
                if (t.includes(i)) continue;
                r[i] = e[i];
            }
        }
        return r;
    }
    var y = {
        children: a().node,
        elementRef: a().oneOfType([ a().func, a().object ]),
        level: a().oneOf([ 1, 2, 3, 4, "s", "ss", 5, 6 ]),
        variant: a().oneOf([ "title1", "title2", "title3", "title4", "title5", "title6", "title7" ])
    };
    // h1-h6 use the same intrinsic element type
        var m = {
        1: "h1",
        2: "h2",
        3: "h3",
        4: "h4",
        s: "h4",
        ss: "h5",
        5: "h5",
        6: "h6"
    };
    var h = {
        h1: "title1",
        h2: "title2",
        h3: "title3",
        h4: "title4",
        h5: "title5",
        h6: "title6"
    };
    function v(e) {
        var t = e.children, r = e.elementRef, n = e.level, a = n === void 0 ? 2 : n, l = e.variant, o = u(e, [ "children", "elementRef", "level", "variant" ]);
        // @docs-props-type HeadingPropsBase
                var s = function e(t) {
            return t.toString().charAt(0) === "s";
        };
        var f = m[a];
        var y = l || h[f];
        if (false) {}
        
        return i().createElement(c, p({
            $isSection: s(a),
            "data-test": "heading",
            ref: r,
            as: f,
            $variant: y
        }, o), t);
    }
    v.propTypes = y;
    /* harmony default export */    const d = v;
    // CONCATENATED MODULE: ./src/Heading/index.ts
    module.exports = t;
    /******/})();
