// @splunk/react-ui/Image.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var a in t) {
                /******/ if (e.o(t, a) && !e.o(r, a)) {
                    /******/ Object.defineProperty(r, a, {
                        enumerable: true,
                        get: t[a]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ Z
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var a =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var n =  e.n(o);
    // CONCATENATED MODULE: external "lodash/endsWith"
    const i = require("lodash/endsWith");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "lodash/omit"
    const s = require("lodash/omit");
    var c =  e.n(s);
    // CONCATENATED MODULE: external "lodash/toLower"
    const u = require("lodash/toLower");
    var p =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/react-ui/File"
    const d = require("@splunk/react-ui/File");
    var f =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/ui-utils/format"
    const v = require("@splunk/ui-utils/format");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const m = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/themes"
    const b = require("@splunk/themes");
    // CONCATENATED MODULE: external "@splunk/react-icons/Cross"
    const y = require("@splunk/react-icons/Cross");
    var g =  e.n(y);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/Close"
    const h = require("@splunk/react-icons/enterprise/Close");
    var w =  e.n(h);
    // CONCATENATED MODULE: ./src/Image/icons/Cross.tsx
    var x = function e() {
        var r = (0, b.useSplunkTheme)(), t = r.isEnterprise, o = r.isCompact;
        var n = o ? "20px" : "24px";
        var i = (0, m._)("Remove");
        var l = t ?  a().createElement(w(), {
            hideDefaultTooltip: true,
            screenReaderText: i
        }) :  a().createElement(g(), {
            "aria-label": i,
            height: n,
            width: n
        });
        return l;
    };
    /* harmony default export */    const k = x;
    // CONCATENATED MODULE: external "styled-components"
    const C = require("styled-components");
    var S =  e.n(C);
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const R = require("@splunk/react-ui/Clickable");
    var O =  e.n(R);
    // CONCATENATED MODULE: external "@splunk/react-ui/Message"
    const I = require("@splunk/react-ui/Message");
    var _ =  e.n(I);
    // CONCATENATED MODULE: ./src/Image/ImageStyles.ts
    var j = S().div.withConfig({
        displayName: "ImageStyles__StyledImagePreview",
        componentId: "sc-16ooypp-0"
    })([ "position:relative;min-height:", ";", "" ], (0, b.pick)({
        enterprise: "30px",
        prisma: b.variables.inputHeight
    }), (0, b.pick)({
        enterprise: (0, C.css)([ "border:", ";" ], b.variables.border),
        prisma: (0, C.css)([ "border-radius:", ";overflow:hidden;margin-top:12px;background-color:", ";" ], b.variables.borderRadius, b.variables.interactiveColorBackground)
    }));
    var E = S().div.withConfig({
        displayName: "ImageStyles__StyledLabelContainer",
        componentId: "sc-16ooypp-1"
    })([ "", ";flex-direction:row;justify-content:space-between;background-color:", ";color:", ";opacity:0.9;position:absolute;width:100%;" ], b.mixins.reset("flex"), (0, 
    b.pick)({
        enterprise: {
            light: b.variables.gray92,
            dark: b.variables.gray30
        },
        prisma: b.mixins.overlayColors(b.variables.interactiveColorBackground, b.variables.interactiveColorOverlaySelected)
    }), b.variables.contentColorMuted);
    var P = S()(O()).withConfig({
        displayName: "ImageStyles__StyledRemoveClickable",
        componentId: "sc-16ooypp-2"
    })([ "cursor:pointer;", " padding:", ";text-align:center;height:inherit;&:focus{", "}&:hover{background-color:", ";color:", ";}" ], (0, 
    b.pick)({
        enterprise: (0, C.css)([ "flex:0 0 26px;" ]),
        prisma: (0, C.css)([ "color:", ";border-radius:50%;" ], b.variables.contentColorMuted)
    }), (0, b.pick)({
        enterprise: "3px 0",
        prisma: {
            comfortable: "0 8px",
            compact: "0 6px"
        }
    }), (0, b.pick)({
        enterprise: (0, C.css)([ "box-shadow:", ";color:", ";" ], b.variables.focusShadow, b.variables.linkColor),
        prisma: (0, C.css)([ "color:", ";background-color:", ";box-shadow:0 0 0 3px ", ";" ], b.variables.contentColorActive, b.variables.interactiveColorOverlayHover, b.variables.focusColor)
    }), (0, b.pick)({
        enterprise: {
            light: b.variables.gray96,
            dark: b.variables.gray30
        },
        prisma: b.variables.interactiveColorOverlayHover
    }), (0, b.pick)({
        enterprise: b.variables.linkColor,
        prisma: b.variables.contentColorActive
    }));
    var q = S().div.withConfig({
        displayName: "ImageStyles__StyledLabel",
        componentId: "sc-16ooypp-3"
    })([ "color:", ";overflow:hidden;white-space:nowrap;text-overflow:ellipsis;flex:1 0 0px;padding:", ";", "" ], b.variables.contentColorActive, (0, 
    b.pick)({
        enterprise: "3px 0 3px 10px",
        prisma: {
            comfortable: "10px 10px",
            compact: "6px 10px"
        }
    }), (function(e) {
        var r = e.$disabled;
        return r && (0, C.css)([ "background-color:", ";color:", ";cursor:not-allowed;" ], (0, 
        b.pick)({
            enterprise: b.variables.gray96,
            prisma: b.variables.interactiveColorBackgroundDisabled
        }), b.variables.contentColorDisabled);
    }));
    var D = S().img.withConfig({
        displayName: "ImageStyles__StyledImageThumbnail",
        componentId: "sc-16ooypp-4"
    })([ "", ";margin:0 auto;max-height:", "px;max-width:100%;" ], b.mixins.reset("block"), (function(e) {
        return e.maxHeight;
    }));
    var T = S()(_()).withConfig({
        displayName: "ImageStyles__StyledErrorMessage",
        componentId: "sc-16ooypp-5"
    })([ "margin-top:", ";" ], (0, b.pick)({
        enterprise: b.variables.spacingXSmall,
        prisma: "12px"
    }));
    // CONCATENATED MODULE: ./src/Image/Image.tsx
    function A(e) {
        "@babel/helpers - typeof";
        return A = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, A(e);
    }
    function F() {
        return F = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var a in t) {
                    ({}).hasOwnProperty.call(t, a) && (e[a] = t[a]);
                }
            }
            return e;
        }, F.apply(null, arguments);
    }
    function H(e, r) {
        if (!(e instanceof r)) throw new TypeError("Cannot call a class as a function");
    }
    function U(e, r) {
        for (var t = 0; t < r.length; t++) {
            var a = r[t];
            a.enumerable = a.enumerable || !1, a.configurable = !0, "value" in a && (a.writable = !0), 
            Object.defineProperty(e, J(a.key), a);
        }
    }
    function M(e, r, t) {
        return r && U(e.prototype, r), t && U(e, t), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function N(e, r) {
        if ("function" != typeof r && null !== r) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(r && r.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), r && B(e, r);
    }
    function B(e, r) {
        return B = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, r) {
            return e.__proto__ = r, e;
        }, B(e, r);
    }
    function L(e) {
        var r = X();
        return function() {
            var t, a = z(e);
            if (r) {
                var o = z(this).constructor;
                t = Reflect.construct(a, arguments, o);
            } else t = a.apply(this, arguments);
            return $(this, t);
        };
    }
    function $(e, r) {
        if (r && ("object" == A(r) || "function" == typeof r)) return r;
        if (void 0 !== r) throw new TypeError("Derived constructors may only return object or undefined");
        return W(e);
    }
    function W(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function X() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return (X = function r() {
            return !!e;
        })();
    }
    function z(e) {
        return z = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, z(e);
    }
    function G(e, r, t) {
        return (r = J(r)) in e ? Object.defineProperty(e, r, {
            value: t,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = t, e;
    }
    function J(e) {
        var r = K(e, "string");
        return "symbol" == A(r) ? r : r + "";
    }
    function K(e, r) {
        if ("object" != A(e) || !e) return e;
        var t = e[Symbol.toPrimitive];
        if (void 0 !== t) {
            var a = t.call(e, r || "default");
            if ("object" != A(a)) return a;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    /** @public */    var Q = {
        allowExtensions: n().arrayOf(n().string),
        defaultFilename: n().string,
        defaultImageDataURI: n().string,
        disabled: n().bool,
        dropAnywhere: n().bool,
        elementRef: n().oneOfType([ n().func, n().object ]),
        error: n().bool,
        maxHeight: n().number,
        onImageChange: n().func
    };
    var V = {
        allowExtensions: [ "gif", "jpeg", "jpg", "png" ],
        disabled: false,
        dropAnywhere: false,
        error: false,
        maxHeight: 180
    };
    /**
 * Image provides the ability to accept image files and present a preview of the image.
 */    var Y =  function(e) {
        N(t, e);
        var r = L(t);
        M(t, null, [ {
            key: "isAllowedFilename",
            // @docs-props-type ImagePropsBase
            value: function e(r, t) {
                return !!r && t.some((function(e) {
                    return l()(p()(r), ".".concat(p()(e)));
                }));
            }
        } ]);
        function t(e) {
            var a;
            H(this, t);
            a = r.call(this, e);
            G(W(a), "fileReader", void 0);
            G(W(a), "handleAddFiles", (function(e) {
                if (e.length > 0) {
                    var r = e[0];
                    if (a.fileReader.readyState === 1) {
                        a.fileReader.abort();
                    }
                    a.fileReader.onload = function() {
                        var e = {
                            filename: r.name,
                            imageDataURI: a.fileReader.result
                        };
                        a.setState(e);
                        if (t.isAllowedFilename(r.name, a.props.allowExtensions)) {
                            var o, n;
                            (o = (n = a.props).onImageChange) === null || o === void 0 ? void 0 : o.call(n, e);
                        }
                    };
                    a.fileReader.readAsDataURL(r);
                }
            }));
            G(W(a), "handleRemoveFile", (function() {
                var e, r;
                if (a.fileReader.readyState === 1) {
                    a.fileReader.abort();
                }
                var t = {
                    filename: null,
                    imageDataURI: null
                };
                a.setState(t);
                (e = (r = a.props).onImageChange) === null || e === void 0 ? void 0 : e.call(r, t);
            }));
            a.state = {
                filename: e.defaultFilename,
                imageDataURI: e.defaultImageDataURI
            };
            a.fileReader = new FileReader;
            return a;
        }
        M(t, [ {
            key: "componentDidUpdate",
            value: function e(r) {
                if (false) {}
            }
        }, {
            key: "render",
            value: function e() {
                var r = this.props, o = r.allowExtensions, n = r.maxHeight, i = r.disabled, l = r.error;
                var s = this.state, u = s.filename, p = s.imageDataURI;
                var d = o.map((function(e) {
                    return ".".concat(e);
                })).join(", ");
                // TODO: Error text should be customizable, error should support a string prop for customization SUI-5414
                                var b = u && !t.isAllowedFilename(u, o) || l;
                var y = (0, m._)("This file extension is unsupported.");
                var g = (0, v.sprintf)((0, m._)('Remove "%(filename)s"'), {
                    filename: u
                });
                
                return a().createElement("div", {
                    "data-test": "image"
                },  a().createElement(f(), F({}, c()(this.props, [ "allowExtensions", "defaultFilename", "defaultImageDataURI", "maxHeight", "onImageChange" ]), {
                    accept: d,
                    allowMultiple: false,
                    error: b,
                    onRequestAdd: this.handleAddFiles,
                    "data-test": "file"
                }), u && p && !b &&  a().createElement(j, null,  a().createElement(E, null,  a().createElement(q, {
                    "data-test": "label",
                    $disabled: i
                }, u), !i &&  a().createElement(P, {
                    onClick: this.handleRemoveFile,
                    "data-test": "remove",
                    "aria-label": g
                },  a().createElement(k, null))),  a().createElement(D, {
                    alt: u,
                    "data-test": "preview",
                    maxHeight: n,
                    src: p
                }))), b &&  a().createElement(T, {
                    appearance: "fill",
                    type: "error"
                }, y));
            }
        } ]);
        return t;
    }(t.Component);
    G(Y, "propTypes", Q);
    G(Y, "defaultProps", V);
    /* harmony default export */    const Z = Y;
    // CONCATENATED MODULE: ./src/Image/index.ts
    module.exports = r;
    /******/})();
