// @splunk/react-ui/Layer.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var n = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, n) => {
            /******/ for (var r in n) {
                /******/ if (e.o(n, r) && !e.o(t, r)) {
                    /******/ Object.defineProperty(t, r, {
                        enumerable: true,
                        get: n[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/global */
    /******/    (() => {
        /******/ e.g = function() {
            /******/ if (typeof globalThis === "object") return globalThis;
            /******/            try {
                /******/ return this || new Function("return this")();
                /******/            } catch (e) {
                /******/ if (typeof window === "object") return window;
                /******/            }
            /******/        }();
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        LayerContext: () => /* reexport */ h,
        LayerStackContext: () => /* reexport */ v,
        LayerStackGlobalProvider: () => /* reexport */ b,
        default: () => /* reexport */ J
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var r =  e.n(n);
    // CONCATENATED MODULE: external "react-dom"
    const o = require("react-dom");
    // CONCATENATED MODULE: external "prop-types"
    const i = require("prop-types");
    var a =  e.n(i);
    // CONCATENATED MODULE: external "lodash/includes"
    const u = require("lodash/includes");
    var c =  e.n(u);
    // CONCATENATED MODULE: external "lodash/last"
    const l = require("lodash/last");
    var s =  e.n(l);
    // CONCATENATED MODULE: external "lodash/pull"
    const f = require("lodash/pull");
    var p =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/react-ui/EventListener"
    const y = require("@splunk/react-ui/EventListener");
    var d =  e.n(y);
    // CONCATENATED MODULE: ./src/Layer/LayerStack.tsx
    /**
 * `LayerStackContext` is used to manage the array that `Layer` (and components that
 * depend on it, such as `Modal`) uses to determine the order of open layers.
 * @public
 */
    var v =  r().createContext([]);
    var h =  r().createContext({});
    var m = {
        children: a().node,
        name: a().string,
        scope: a().object,
        separateStackingContexts: a().bool
    };
    /* global global */
    /**
 * A `LayerStackContext` provider that stores a shared layer stack using a global variable.
 * Applications should only use this provider if there's a known need to support multiple
 * instances of this library on the same page.
 */    function b(t) {
        var o = t.children, i = t.name, a = i === void 0 ? "__splunkui_layer_instances__" : i, u = t.scope, c = u === void 0 ? typeof window !== "undefined" ? window : e.g : u, l = t.separateStackingContexts, s = l === void 0 ? false : l;
        if (!c[a]) {
            c[a] = [];
 // eslint-disable-line no-param-reassign
                }
        var f = (0, n.useMemo)((function() {
            return {
                separateStackingContexts: s
            };
        }), [ s ]);
        
        return r().createElement(v.Provider, {
            value: c[a]
        },  r().createElement(h.Provider, {
            value: f
        }, o));
    }
    b.propTypes = m;
    // CONCATENATED MODULE: external "styled-components"
    const g = require("styled-components");
    var w =  e.n(g);
    // CONCATENATED MODULE: external "@splunk/themes"
    const C = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Layer/LayerStyles.ts
    var k = w().div.withConfig({
        displayName: "LayerStyles__StyledLayer",
        componentId: "ii6psl-0"
    })([ "", "" ], (function(e) {
        var t = e.$separateStackingContexts;
        return t && (0, g.css)([ "isolation:isolate;position:relative;z-index:", ";" ], C.variables.zindexLayer);
    }));
    // CONCATENATED MODULE: ./src/utils/ssrDocument.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var S = {
        body: {
            appendChild: function e() {
                return [];
            }
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        activeElement: {
            blur: function e() {},
            nodeName: ""
        },
        querySelector: function e() {
            return null;
        },
        querySelectorAll: function e() {
            return [];
        },
        getElementById: function e() {
            return null;
        },
        createEvent: function e() {
            return {
                initEvent: function e() {}
            };
        },
        createElement: function e() {
            return {
                children: [],
                childNodes: [],
                style: {},
                setAttribute: function e() {},
                getElementsByTagName: function e() {
                    return [];
                }
            };
        },
        createElementNS: function e() {
            return {};
        },
        importNode: function e() {
            return null;
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        }
    };
    function E() {
        var e = typeof document !== "undefined" ? document : S;
        return e;
    }
    var O = E();
    /* harmony default export */    const T = /* unused pure expression or super */ null && O;
    // CONCATENATED MODULE: ./src/utils/ssrWindow.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var L = {
        document: S,
        navigator: {
            userAgent: ""
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        },
        history: {
            replaceState: function e() {},
            pushState: function e() {},
            go: function e() {},
            back: function e() {}
        },
        CustomEvent: function e() {
            return this;
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        getComputedStyle: function e() {
            return {
                getPropertyValue: function e() {
                    return "";
                }
            };
        },
        Image: function e() {},
        Date: function e() {},
        screen: {},
        setTimeout: function e() {},
        clearTimeout: function e() {},
        matchMedia: function e() {
            return {};
        },
        requestAnimationFrame: function e(t) {
            if (typeof setTimeout === "undefined") {
                t();
                return null;
            }
            return setTimeout(t, 0);
        },
        cancelAnimationFrame: function e(t) {
            if (typeof setTimeout === "undefined") {
                return;
            }
            clearTimeout(t);
        }
    };
    function j() {
        var e = typeof window !== "undefined" ? window : L;
        return e;
    }
    var P = j();
    /* harmony default export */    const _ = P;
    // CONCATENATED MODULE: ./src/Layer/Layer.tsx
    function x(e) {
        "@babel/helpers - typeof";
        return x = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, x(e);
    }
    function q(e, t) {
        if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function");
    }
    function R(e, t) {
        for (var n = 0; n < t.length; n++) {
            var r = t[n];
            r.enumerable = r.enumerable || !1, r.configurable = !0, "value" in r && (r.writable = !0), 
            Object.defineProperty(e, z(r.key), r);
        }
    }
    function A(e, t, n) {
        return t && R(e.prototype, t), n && R(e, n), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function D(e, t) {
        if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), t && M(e, t);
    }
    function M(e, t) {
        return M = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, t) {
            return e.__proto__ = t, e;
        }, M(e, t);
    }
    function F(e) {
        var t = W();
        return function() {
            var n, r = B(e);
            if (t) {
                var o = B(this).constructor;
                n = Reflect.construct(r, arguments, o);
            } else n = r.apply(this, arguments);
            return N(this, n);
        };
    }
    function N(e, t) {
        if (t && ("object" == x(t) || "function" == typeof t)) return t;
        if (void 0 !== t) throw new TypeError("Derived constructors may only return object or undefined");
        return K(e);
    }
    function K(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function W() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return (W = function t() {
            return !!e;
        })();
    }
    function B(e) {
        return B = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, B(e);
    }
    function I(e, t, n) {
        return (t = z(t)) in e ? Object.defineProperty(e, t, {
            value: n,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = n, e;
    }
    function z(e) {
        var t = U(e, "string");
        return "symbol" == x(t) ? t : t + "";
    }
    function U(e, t) {
        if ("object" != x(e) || !e) return e;
        var n = e[Symbol.toPrimitive];
        if (void 0 !== n) {
            var r = n.call(e, t || "default");
            if ("object" != x(r)) return r;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    /** @public */
    /** @public */    var $ = [ "clickAway", "escapeKey" ];
    var G = {
        children: a().node,
        closeReasons: a().arrayOf(a().oneOf($)),
        onRequestClose: a().func,
        open: a().bool
    };
    var V = {
        closeReasons: $,
        open: false
    };
    var H =  function(e) {
        D(n, e);
        var t = F(n);
        // @docs-props-type LayerPropsBase
        // eslint-disable-next-line react/sort-comp
        // LayerStackContext (an array) keeps track of the current instances of Layer. This is
        // used by Layer#handleKeyDown to determine if the escapeKey event should be handled by
        // the current instance. Only the topmost Layer instance should honor the escapeKey.
                function n(e) {
            var r;
            q(this, n);
            r = t.call(this, e);
            I(K(r), "layerClickEvent", null);
            I(K(r), "handleClickOnLayer", (function(e) {
                var t = e.nativeEvent;
                r.layerClickEvent = t;
            }));
            I(K(r), "handleClickOnWindow", (function(e) {
                var t, n;
                // clicks inside the layer should not be considered clickAways
                                if (!r.props.open || !c()(r.props.closeReasons, "clickAway") || r.layerClickEvent === e) {
                    return;
                }
                (t = (n = r.props).onRequestClose) === null || t === void 0 ? void 0 : t.call(n, {
                    event: e,
                    reason: "clickAway"
                });
            }));
            I(K(r), "handleKeyDownOnWindow", (function(e) {
                if (r.props.open && e.key === "Escape" && s()(r.getLayerStack()) === K(r) && c()(r.props.closeReasons, "escapeKey")) {
                    var t, n;
                    (t = (n = r.props).onRequestClose) === null || t === void 0 ? void 0 : t.call(n, {
                        event: e,
                        reason: "escapeKey"
                    });
                }
            }));
            var o = E();
            if (!n.layerContainer) {
                n.layerContainer = o.createElement("div");
                n.layerContainer.setAttribute("data-test", "layer-container");
                o.body.appendChild(n.layerContainer);
            }
            return r;
        }
        A(n, [ {
            key: "componentDidMount",
            value: function e() {
                if (this.props.open) {
                    this.getLayerStack().push(this);
                }
            }
        }, {
            key: "componentDidUpdate",
            value: function e(t) {
                if (!t.open && this.props.open) {
                    this.getLayerStack().push(this);
                } else if (t.open && !this.props.open) {
                    p()(this.getLayerStack(), this);
                }
            }
        }, {
            key: "componentWillUnmount",
            value: function e() {
                p()(this.getLayerStack(), this);
            }
        }, {
            key: "getLayerStack",
            value: function e() {
                return this.context;
            }
 // the fullscreen API does not render elements outside the fullscreen element
            // Layer renders in document.body, which means that anything using Layer will not render in fullscreen
            // this code moves the layer container into the fullscreen element, ensuring that it will render
                }, {
            key: "render",
            value: function e() {
                var t = this;
                var i = this.props, a = i.children, u = i.open;
                var c =  r().createElement(d(), {
                    target: _,
                    eventType: "fullscreenchange",
                    listener: n.handleFullScreenChange,
                    key: "fullScreenChange"
                });
                if (u) {
                    var l =  (0, o.createPortal)( r().createElement(h.Consumer, null, (function(e) {
                        var n = e.separateStackingContexts, o = n === void 0 ? false : n;
                        
                        return r().createElement(k, {
                            $separateStackingContexts: o,
                            "data-test": "layer",
                            onMouseDown: t.handleClickOnLayer,
                            onTouchStart: t.handleClickOnLayer
                        }, a);
                    })), n.layerContainer);
                    
                    return r().createElement(r().Fragment, null,  r().createElement(d(), {
                        target: _,
                        eventType: "keydown",
                        listener: this.handleKeyDownOnWindow,
                        key: "eventListenerKeydown"
                    }),  r().createElement(d(), {
                        target: _,
                        eventType: "mousedown",
                        listener: this.handleClickOnWindow,
                        key: "eventListenerMouseDown"
                    }),  r().createElement(d(), {
                        target: _,
                        eventType: "touchstart",
                        listener: this.handleClickOnWindow,
                        key: "eventListenerTouchStart",
                        options: {
                            passive: true
                        }
                    }), c, l);
                }
                // the fragment has to be present to avoid type issues with React 18
                // eslint-disable-next-line react/jsx-no-useless-fragment
                                
                return r().createElement(r().Fragment, null, c);
            }
        } ]);
        return n;
    }(n.Component);
    I(H, "layerContainer", null);
    I(H, "possibleCloseReasons", $);
    I(H, "propTypes", G);
    I(H, "defaultProps", V);
    I(H, "contextType", v);
    I(H, "handleFullScreenChange", (function() {
        if (H.layerContainer) {
            var e = E();
            if (e.fullscreenElement != null) {
                e.fullscreenElement.appendChild(H.layerContainer);
            } else {
                e.body.appendChild(H.layerContainer);
            }
        }
    }));
    /* harmony default export */    const J = H;
    // CONCATENATED MODULE: ./src/Layer/index.ts
    module.exports = t;
    /******/})();
