// @splunk/react-ui/Link.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        default: () => /* reexport */ Q
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const a = require("@splunk/react-ui/ScreenReaderContent");
    var l =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/react-ui/Dropdown"
    const u = require("@splunk/react-ui/Dropdown");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const c = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "styled-components"
    const s = require("styled-components");
    var p =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/react-icons/ArrowSquareTopRight"
    const f = require("@splunk/react-icons/ArrowSquareTopRight");
    var d =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/External"
    const b = require("@splunk/react-icons/enterprise/External");
    var y =  e.n(b);
    // CONCATENATED MODULE: external "@splunk/themes"
    const v = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Link/icons/External.tsx
    var m = (0, s.css)([ "width:0.8em;height:0.8em;vertical-align:baseline;" ]);
    var h = p()(y()).withConfig({
        displayName: "External__StyledEnterprise",
        componentId: "sc-13bdx45-0"
    })([ "", " margin:0 0 0 3px;" ], m);
    var w = p()(d()).withConfig({
        displayName: "External__StyledPrisma",
        componentId: "sc-13bdx45-1"
    })([ "", " margin:0 0 0 4px;" ], m);
    function g() {
        var e = (0, v.useSplunkTheme)(), t = e.isEnterprise;
        return t ?  n().createElement(h, {
            screenReaderText: null
        }) :  n().createElement(w, null);
    }
    /* harmony default export */    const O = g;
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const j = require("@splunk/react-ui/Clickable");
    var x =  e.n(j);
    // CONCATENATED MODULE: ./src/Link/LinkStyles.ts
    var S = p()(x()).withConfig({
        displayName: "LinkStyles__StyledClickable",
        componentId: "sc-1hhltcf-0"
    })([ "", ";color:", ";text-decoration:none;font-size:inherit;font-weight:inherit;line-height:inherit;border-radius:", ";&:not([disabled],[aria-disabled='true']){cursor:pointer;", ";&:focus{box-shadow:", ";outline:0;&:active{box-shadow:none;}}}&[disabled],&[aria-disabled='true']{color:", ";}" ], v.mixins.reset("inline"), (0, 
    v.pick)({
        enterprise: v.variables.linkColor,
        prisma: v.variables.interactiveColorPrimary
    }), (0, v.pick)({
        enterprise: 0,
        prisma: "1px"
    }), (0, v.pick)({
        enterprise: (0, s.css)([ "&:hover{text-decoration:underline;}" ]),
        prisma: (0, s.css)([ "&:hover:not(:focus),&:active{text-decoration:underline;text-underline-position:under;}" ])
    }), v.variables.focusShadow, v.variables.contentColorDisabled);
    // CONCATENATED MODULE: ./src/Link/Link.tsx
    function P(e) {
        "@babel/helpers - typeof";
        return P = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, P(e);
    }
    function k() {
        return k = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, k.apply(null, arguments);
    }
    function _(e, t) {
        if (null == e) return {};
        var r, n, o = E(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                r = i[n], t.includes(r) || {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function E(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (t.includes(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    function C(e, t) {
        if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function");
    }
    function T(e, t) {
        for (var r = 0; r < t.length; r++) {
            var n = t[r];
            n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), 
            Object.defineProperty(e, z(n.key), n);
        }
    }
    function q(e, t, r) {
        return t && T(e.prototype, t), r && T(e, r), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function I(e, t) {
        if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), t && N(e, t);
    }
    function N(e, t) {
        return N = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, t) {
            return e.__proto__ = t, e;
        }, N(e, t);
    }
    function R(e) {
        var t = A();
        return function() {
            var r, n = B(e);
            if (t) {
                var o = B(this).constructor;
                r = Reflect.construct(n, arguments, o);
            } else r = n.apply(this, arguments);
            return M(this, r);
        };
    }
    function M(e, t) {
        if (t && ("object" == P(t) || "function" == typeof t)) return t;
        if (void 0 !== t) throw new TypeError("Derived constructors may only return object or undefined");
        return D(e);
    }
    function D(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function A() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return (A = function t() {
            return !!e;
        })();
    }
    function B(e) {
        return B = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, B(e);
    }
    function L(e, t, r) {
        return (t = z(t)) in e ? Object.defineProperty(e, t, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = r, e;
    }
    function z(e) {
        var t = F(e, "string");
        return "symbol" == P(t) ? t : t + "";
    }
    function F(e, t) {
        if ("object" != P(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var n = r.call(e, t || "default");
            if ("object" != P(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    var G = {
        disabled: false,
        openInNewContext: false
    };
    var H = {
        /** @private */
        allowDisabledLink: i().bool,
        children: i().node,
        disabled: i().bool,
        elementRef: i().oneOfType([ i().func, i().object ]),
        openInNewContext: i().oneOfType([ i().bool, i().string ]),
        to: i().string
    };
    var J = (0, c._)("(Opens new window)");
    /**
 * `Link` is a simple method for configuring `Button` for inline links. For more complex behaviors,
 * see the `Button` documentation.
 */    var K =  function(e) {
        I(r, e);
        var t = R(r);
        function r() {
            var e;
            C(this, r);
            for (var n = arguments.length, o = new Array(n), i = 0; i < n; i++) {
                o[i] = arguments[i];
            }
            e = t.call.apply(t, [ this ].concat(o));
            L(D(e), "component", null);
            L(D(e), "handleMount", (function(t) {
                e.component = t;
            }));
            return e;
        }
 // @docs-props-type LinkPropsBase
                q(r, [ {
            key: "focus",
            /**
     * Place focus on the link.
     */
            // eslint-disable-next-line react/no-unused-class-component-methods
            value: function e() {
                var t;
                (t = this.component) === null || t === void 0 ? void 0 : t.focus();
            }
        }, {
            key: "render",
            value: function e() {
                var t = this.props, r = t.children, o = t.openInNewContext, i = _(t, [ "children", "openInNewContext" ]);
                var a = false;
                var u;
                if (o) {
                    a = true;
                    u = typeof o === "string" ? o : J;
                }
                
                return n().createElement(S, k({
                    "data-test": "link",
                    ref: this.handleMount,
                    openInNewContext: a
                }, i), r, o &&  n().createElement(n().Fragment, null,  n().createElement(O, null),  n().createElement(l(), null, u)));
            }
        } ]);
        return r;
    }(r.Component);
    L(K, "propTypes", H);
    L(K, "defaultProps", G);
    L(K, u.legacyRefMode, true);
    /* harmony default export */    const Q = K;
    // CONCATENATED MODULE: ./src/Link/index.ts
    module.exports = t;
    /******/})();
