// @splunk/react-ui/List.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        Item: () => /* reexport */ v,
        default: () => /* reexport */ j
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const l = require("prop-types");
    var o =  e.n(l);
    // CONCATENATED MODULE: external "styled-components"
    const i = require("styled-components");
    var a =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/themes"
    const s = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/List/ListStyles.ts
    var c = a().ul.withConfig({
        displayName: "ListStyles__StyledUl",
        componentId: "sc-1yyh8l9-0"
    })([ "", ";", ";counter-reset:list-item;list-style-type:", ";padding-left:40px;margin-bottom:1.3em;margin-top:1em;" ], s.mixins.reset("block"), s.mixins.typography("body"), (function(e) {
        var t = e.$listStyleType;
        return t;
    }));
    var u = a().li.withConfig({
        displayName: "ListStyles__StyledItem",
        componentId: "sc-1yyh8l9-1"
    })([ "", ";list-style-type:inherit;" ], s.mixins.reset("list-item"));
    // CONCATENATED MODULE: ./src/List/Item.tsx
    function f() {
        return f = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, f.apply(null, arguments);
    }
    function p(e, t) {
        if (null == e) return {};
        var r, n, l = y(e, t);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (n = 0; n < o.length; n++) {
                r = o[n], t.includes(r) || {}.propertyIsEnumerable.call(e, r) && (l[r] = e[r]);
            }
        }
        return l;
    }
    function y(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (t.includes(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var d = {
        children: o().node,
        elementRef: o().oneOfType([ o().func, o().object ])
    };
    /**
 * A container for items of a `List`.
 */    function m(e) {
        var t = e.children, r = e.elementRef, l = p(e, [ "children", "elementRef" ]);
        // @docs-props-type ItemPropsBase
                
        return n().createElement(u, f({
            "data-test": "item",
            ref: r
        }, l), t);
    }
    m.propTypes = d;
    /* harmony default export */    const v = m;
    // CONCATENATED MODULE: ./src/List/List.tsx
    function b() {
        return b = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, b.apply(null, arguments);
    }
    function h(e, t) {
        if (null == e) return {};
        var r, n, l = O(e, t);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (n = 0; n < o.length; n++) {
                r = o[n], t.includes(r) || {}.propertyIsEnumerable.call(e, r) && (l[r] = e[r]);
            }
        }
        return l;
    }
    function O(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (t.includes(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var g = {
        children: o().node,
        elementRef: o().oneOfType([ o().func, o().object ]),
        ordered: o().bool,
        type: o().oneOf([ "disc", "decimal", "lower-alpha", "upper-alpha" ])
    };
    function S(e) {
        var t = e.children, r = e.elementRef, l = e.type, o = l === void 0 ? "disc" : l, i = e.ordered, a = h(e, [ "children", "elementRef", "type", "ordered" ]);
        // @docs-props-type ListPropsBase
        // Default prop value is "disc", we cannot determine if user or defaultProps is setting the value
                if (false) {}
        
        return n().createElement(c, b({
            as: i || o !== "disc" ? "ol" : "ul",
            "data-test": "list",
            ref: r,
            $listStyleType: i ? "decimal" : o
        }, a), t);
    }
    S.propTypes = g;
    S.Item = v;
    /* harmony default export */    const j = S;
    // CONCATENATED MODULE: ./src/List/index.ts
    module.exports = t;
    /******/})();
