// @splunk/react-ui/Markdown.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var n = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, n) => {
            /******/ for (var t in n) {
                /******/ if (e.o(n, t) && !e.o(r, t)) {
                    /******/ Object.defineProperty(r, t, {
                        enumerable: true,
                        get: n[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        MarkdownAnchorHeading: () => /* reexport */ ee,
        MarkdownBlockquote: () => /* reexport */ k,
        MarkdownCode: () => /* reexport */ E,
        MarkdownCodeBlock: () => /* reexport */ B,
        MarkdownHeading: () => /* reexport */ G,
        MarkdownImage: () => /* reexport */ le,
        MarkdownItem: () => /* reexport */ Ee,
        MarkdownLink: () => /* reexport */ ye,
        MarkdownList: () => /* reexport */ ke,
        MarkdownParagraph: () => /* reexport */ _e,
        default: () => /* reexport */ Ue
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var t =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "commonmark-react-renderer"
    const l = require("commonmark-react-renderer");
    var a =  e.n(l);
    // CONCATENATED MODULE: external "commonmark"
    const c = require("commonmark");
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const u = require("@splunk/react-ui/Box");
    var s =  e.n(u);
    // CONCATENATED MODULE: external "styled-components"
    const f = require("styled-components");
    var p =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/themes"
    const d = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Markdown/MarkdownStyles.ts
    var y = p().div.withConfig({
        displayName: "MarkdownStyles__StyledCodeBlock",
        componentId: "sc-1xadih-0"
    })([ "", ";margin:", " 0;background-color:", ";" ], (0, d.pick)({
        enterprise: (0, f.css)([ "padding:", ";" ], d.variables.spacingSmall),
        prisma: (0, f.css)([ "padding:10px ", ";" ], d.variables.spacingLarge)
    }), (0, d.pick)({
        enterprise: d.variables.spacingSmall,
        prisma: d.variables.spacingLarge
    }), (0, d.pick)({
        enterprise: {
            light: d.variables.gray96,
            dark: d.variables.gray22
        },
        prisma: d.variables.backgroundColorSection
    }));
    var b = p().code.withConfig({
        displayName: "MarkdownStyles__StyledCodeInline",
        componentId: "sc-1xadih-1"
    })([ "", ";font-family:", ";background-color:", ";" ], d.mixins.reset("inline"), d.variables.monoFontFamily, (0, 
    d.pick)({
        enterprise: {
            light: d.variables.gray92,
            dark: d.variables.gray22
        },
        prisma: d.variables.neutral200
    }));
    var v = p().blockquote.withConfig({
        displayName: "MarkdownStyles__StyledHint",
        componentId: "sc-1xadih-2"
    })([ "", " ", " background-color:", ";margin-bottom:15px;padding:", ";border-left:10px solid ", ";>:last-child{margin-bottom:0;}" ], d.mixins.reset("block"), d.mixins.clearfix(), (0, 
    d.pick)({
        enterprise: {
            light: d.variables.infoColorL50,
            dark: d.variables.infoColorD50
        },
        prisma: d.variables.neutral100
    }), d.variables.spacingSmall, d.variables.infoColor);
    var m = p().img.withConfig({
        displayName: "MarkdownStyles__StyledImage",
        componentId: "sc-1xadih-3"
    })([ "max-width:100%;" ]);
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownBlockquote.tsx
    function g(e, r) {
        if (null == e) return {};
        var n, t, o = O(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (t = 0; t < i.length; t++) {
                n = i[t], r.includes(n) || {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function O(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (r.includes(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    var h = {
        children: i().node
    };
    function w(e) {
        var r = e.children, n = g(e, [ "children" ]);
        
        return t().createElement(v, n, r);
    }
    w.propTypes = h;
    /* harmony default export */    const k = w;
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownCode.tsx
    function P(e, r) {
        if (null == e) return {};
        var n, t, o = j(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (t = 0; t < i.length; t++) {
                n = i[t], r.includes(n) || {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function j(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (r.includes(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    var S = {
        literal: i().string
    };
    function R(e) {
        var r = e.literal, n = P(e, [ "literal" ]);
        
        return t().createElement(b, n, r);
    }
    R.propTypes = S;
    /* harmony default export */    const E = R;
    // CONCATENATED MODULE: external "@splunk/react-ui/Code"
    const I = require("@splunk/react-ui/Code");
    var q =  e.n(I);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const M = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownCodeBlock.tsx
    function C(e, r) {
        if (null == e) return {};
        var n, t, o = x(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (t = 0; t < i.length; t++) {
                n = i[t], r.includes(n) || {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function x(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (r.includes(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    var T = {
        literal: i().string,
        language: i().string
    };
    function _(e) {
        var r = e.literal, n = e.language, o = C(e, [ "literal", "language" ]);
        
        return t().createElement(y, o,  t().createElement(q(), {
            "aria-label": (0, M._)("Code Block"),
            language: n,
            value: r,
            style: {
                margin: 0
            }
        }));
    }
    _.propTypes = T;
    /* harmony default export */    const B = _;
    // CONCATENATED MODULE: external "@splunk/react-ui/Heading"
    const L = require("@splunk/react-ui/Heading");
    var H =  e.n(L);
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownHeading.tsx
    function D() {
        return D = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, D.apply(null, arguments);
    }
    function N(e, r) {
        if (null == e) return {};
        var n, t, o = A(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (t = 0; t < i.length; t++) {
                n = i[t], r.includes(n) || {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function A(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (r.includes(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    var F = {
        children: i().node
    };
    function z(e) {
        var r = e.level, n = e.children, o = N(e, [ "level", "children" ]);
        
        return t().createElement(H(), D({
            level: Math.min(r, 4)
        }, o), n);
    }
    z.propTypes = F;
    /* harmony default export */    const G = z;
    // CONCATENATED MODULE: external "@splunk/react-ui/Anchor"
    const J = require("@splunk/react-ui/Anchor");
    var K =  e.n(J);
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownAnchorHeading.tsx
    function Q() {
        return Q = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, Q.apply(null, arguments);
    }
    function U(e, r) {
        if (null == e) return {};
        var n, t, o = V(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (t = 0; t < i.length; t++) {
                n = i[t], r.includes(n) || {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function V(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (r.includes(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    function W(e) {
        "@babel/helpers - typeof";
        return W = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, W(e);
    }
    var X = {
        children: i().node
    };
    // When this is used in MDX, the node isn't guaranteed to be text,
    // so this method extracts the actual text from the node.
        function Y(e) {
        var r = W(e);
        var n = "";
        if (e != null) {
            if (r === "string" || r === "number") {
                n = e.toString();
            } else if (r === "boolean") {
                n = e ? "true" : "false";
            } else if (W(e) === "object") {
                if (e instanceof Array) {
                    n = e.map(Y).join(" ");
                } else if ("props" in e) {
                    n = Y(e.props.children);
                }
            }
        }
        return n;
    }
    function Z(e) {
        return n.Children.toArray(e).map(Y).join(" ").toLowerCase().replace(/[^a-z0-9-. ]/g, "").replace(/[-. ]+/g, "-");
    }
    function $(e) {
        var r = e.level, n = e.children, o = U(e, [ "level", "children" ]);
        
        return t().createElement(G, Q({
            level: Math.min(r, 4)
        }, o),  t().createElement(K(), {
            name: Z(n)
        }, n));
    }
    $.propTypes = X;
    /* harmony default export */    const ee = $;
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownImage.tsx
    function re() {
        return re = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, re.apply(null, arguments);
    }
    function ne(e, r) {
        if (null == e) return {};
        var n, t, o = te(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (t = 0; t < i.length; t++) {
                n = i[t], r.includes(n) || {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function te(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (r.includes(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    var oe = {
        alt: i().string,
        src: i().string,
        title: i().string
    };
    function ie(e) {
        var r = e.src, n = e.title, o = e.alt, i = ne(e, [ "src", "title", "alt" ]);
        
        return t().createElement(m, re({
            src: r,
            title: n,
            alt: o
        }, i));
    }
    ie.propTypes = oe;
    /* harmony default export */    const le = ie;
    // CONCATENATED MODULE: external "@splunk/react-ui/Link"
    const ae = require("@splunk/react-ui/Link");
    var ce =  e.n(ae);
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownLink.tsx
    function ue() {
        return ue = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, ue.apply(null, arguments);
    }
    function se(e, r) {
        if (null == e) return {};
        var n, t, o = fe(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (t = 0; t < i.length; t++) {
                n = i[t], r.includes(n) || {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function fe(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (r.includes(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    var pe = {
        children: i().node,
        href: i().string,
        title: i().string
    };
    function de(e) {
        var r = e.children, n = e.href, o = se(e, [ "children", "href" ]);
        
        return t().createElement(ce(), ue({
            to: n
        }, o), r);
    }
    de.propTypes = pe;
    /* harmony default export */    const ye = de;
    // CONCATENATED MODULE: external "@splunk/react-ui/List"
    const be = require("@splunk/react-ui/List");
    var ve =  e.n(be);
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownList.tsx
    function me() {
        return me = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, me.apply(null, arguments);
    }
    function ge(e, r) {
        if (null == e) return {};
        var n, t, o = Oe(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (t = 0; t < i.length; t++) {
                n = i[t], r.includes(n) || {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function Oe(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (r.includes(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    var he = {
        children: i().node,
        type: i().string
    };
    function we(e) {
        var r = e.children, n = e.type, o = ge(e, [ "children", "type" ]);
        if (n === "ordered") {
            
            return t().createElement(ve(), me({}, o, {
                ordered: true
            }), r);
        }
        
        return t().createElement(ve(), o, r);
    }
    we.propTypes = he;
    /* harmony default export */    const ke = we;
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownItem.tsx
    function Pe(e, r) {
        if (null == e) return {};
        var n, t, o = je(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (t = 0; t < i.length; t++) {
                n = i[t], r.includes(n) || {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function je(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (r.includes(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    var Se = {
        children: i().node
    };
    function Re(e) {
        var r = e.children, n = Pe(e, [ "children" ]);
        
        return t().createElement(be.Item, n, r);
    }
    Re.propTypes = Se;
    /* harmony default export */    const Ee = Re;
    // CONCATENATED MODULE: external "@splunk/react-ui/Paragraph"
    const Ie = require("@splunk/react-ui/Paragraph");
    var qe =  e.n(Ie);
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownParagraph.tsx
    function Me(e, r) {
        if (null == e) return {};
        var n, t, o = Ce(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (t = 0; t < i.length; t++) {
                n = i[t], r.includes(n) || {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function Ce(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (r.includes(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    var xe = {
        children: i().node
    };
    function Te(e) {
        var r = e.children, n = Me(e, [ "children" ]);
        
        return t().createElement(qe(), n, r);
    }
    Te.propTypes = xe;
    /* harmony default export */    const _e = Te;
    // CONCATENATED MODULE: ./src/Markdown/Markdown.tsx
    function Be(e) {
        "@babel/helpers - typeof";
        return Be = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, Be(e);
    }
    function Le() {
        return Le = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, Le.apply(null, arguments);
    }
    function He(e, r) {
        var n = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var t = Object.getOwnPropertySymbols(e);
            r && (t = t.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }))), n.push.apply(n, t);
        }
        return n;
    }
    function De(e) {
        for (var r = 1; r < arguments.length; r++) {
            var n = null != arguments[r] ? arguments[r] : {};
            r % 2 ? He(Object(n), !0).forEach((function(r) {
                Ne(e, r, n[r]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(n)) : He(Object(n)).forEach((function(r) {
                Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(n, r));
            }));
        }
        return e;
    }
    function Ne(e, r, n) {
        return (r = Ae(r)) in e ? Object.defineProperty(e, r, {
            value: n,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = n, e;
    }
    function Ae(e) {
        var r = Fe(e, "string");
        return "symbol" == Be(r) ? r : r + "";
    }
    function Fe(e, r) {
        if ("object" != Be(e) || !e) return e;
        var n = e[Symbol.toPrimitive];
        if (void 0 !== n) {
            var t = n.call(e, r || "default");
            if ("object" != Be(t)) return t;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    function ze(e, r) {
        if (null == e) return {};
        var n, t, o = Ge(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (t = 0; t < i.length; t++) {
                n = i[t], r.includes(n) || {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function Ge(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (r.includes(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    /** @public */
    /** @public */
    /** @public */
    /** @public */
    /** @public */
    /** @public */
    /** @public */
    /** @public */
    /** @public */
    /** @public */    var Je = {
        blockquoteRenderer: i().func,
        codeBlockRenderer: i().func,
        codeRenderer: i().func,
        elementRef: i().oneOfType([ i().func, i().object ]),
        headingRenderer: i().func,
        imageRenderer: i().func,
        itemRenderer: i().func,
        linkRenderer: i().func,
        listRenderer: i().func,
        paragraphRenderer: i().func,
        text: i().string.isRequired,
        rendererOptions: i().object
    };
    var Ke = new c.Parser;
    /**
 * The `Markdown` component renders the given Markdown text as a React component.
 * The component prefers @splunk/react-ui components over plain HTML components. For example
 * links are rendered as the `@splunk/react-ui/Link` component instead of plain `<a>` tag.
 */    function Qe(e) {
        var r = e.blockquoteRenderer, n = e.codeBlockRenderer, o = e.codeRenderer, i = e.elementRef, l = e.headingRenderer, c = e.imageRenderer, u = e.itemRenderer, f = e.linkRenderer, p = e.listRenderer, d = e.paragraphRenderer, y = e.text, b = e.rendererOptions, v = ze(e, [ "blockquoteRenderer", "codeBlockRenderer", "codeRenderer", "elementRef", "headingRenderer", "imageRenderer", "itemRenderer", "linkRenderer", "listRenderer", "paragraphRenderer", "text", "rendererOptions" ]);
        // @docs-props-type MarkdownPropsBase
        // keep this option object inside the constructor, otherwise the renderers
        // below will confuse the doc-gen
                var m = {
            escapeHtml: true,
            renderers: {
                CodeBlock: n || B,
                Code: o || E,
                Heading: l || G,
                Link: f || ye,
                List: p || ke,
                Item: u || Ee,
                Paragraph: d || _e
            }
        };
        // not setting MarkdownBlockquote as the default
                if (r) {
            m.renderers.Blockquote = r;
        }
        // not setting MarkdownImage as the default
                if (c) {
            m.renderers.Image = c;
        }
        var g = new (a())(De(De({}, m), b));
        var O = Ke.parse(y);
        var h = g.render(O);
        
        return t().createElement(s(), Le({
            "data-test": "markdown",
            elementRef: i
        }, v), h);
    }
    Qe.propTypes = Je;
    /* harmony default export */    const Ue = Qe;
    // CONCATENATED MODULE: ./src/Markdown/index.ts
    module.exports = r;
    /******/})();
