// @splunk/react-ui/Menu.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        Divider: () => /* reexport */ w,
        Heading: () => /* reexport */ M,
        Item: () => /* reexport */ Ye,
        MenuContext: () => /* reexport */ h,
        default: () => /* reexport */ pt
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const i = require("prop-types");
    var o =  e.n(i);
    // CONCATENATED MODULE: external "lodash/omit"
    const a = require("lodash/omit");
    var l =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/react-ui/Popover"
    const s = require("@splunk/react-ui/Popover");
    // CONCATENATED MODULE: external "@splunk/ui-utils/keyboard"
    const c = require("@splunk/ui-utils/keyboard");
    // CONCATENATED MODULE: external "@splunk/ui-utils/focus"
    const p = require("@splunk/ui-utils/focus");
    // CONCATENATED MODULE: external "styled-components"
    const u = require("styled-components");
    var d =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/react-ui/Divider"
    const f = require("@splunk/react-ui/Divider");
    var v =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/themes"
    const m = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Menu/DividerStyles.ts
    var b = d()(v()).withConfig({
        displayName: "DividerStyles__Styled",
        componentId: "l3zfh3-0"
    })([ "", "" ], (0, m.pick)({
        prisma: {
            comfortable: (0, u.css)([ "margin:8px 0;" ]),
            compact: (0, u.css)([ "margin:6px 0;" ])
        }
    }));
    // CONCATENATED MODULE: ./src/Menu/MenuContext.tsx
    var y =  (0, r.createContext)({
        role: "menu",
        preventFocus: false
    });
    /* harmony default export */    const h = y;
    // CONCATENATED MODULE: ./src/Menu/Divider.tsx
    function g() {
        return g = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, g.apply(null, arguments);
    }
    var k = {};
    /**
 * A non-interactive menu item used to visually separate groups of items in the menu.
 */    function x(e) {
        var t = g({}, e);
        // @docs-props-type DividerPropsBase
                var i = (0, r.useContext)(h), o = i.role;
        
        return n().createElement(b, g({
            "aria-hidden": o === "listbox",
            "data-test": "divider",
            role: "separator"
        }, t));
    }
    x.propTypes = k;
    /* Remove the item if it is the first item after filtering */    x.filterFirst = true;
    /* Remove consecutive items with filterConsecutive = true (Dividers and Headings) */    x.filterConsecutive = true;
    /* Remove the item if it is the last item after filtering. */    x.filterLast = true;
    /* Tag the item as Divider */    x.as = "Divider";
    /* harmony default export */    const w = x;
    // CONCATENATED MODULE: external "@splunk/react-ui/Heading"
    const S = require("@splunk/react-ui/Heading");
    var O =  e.n(S);
    // CONCATENATED MODULE: external "@splunk/themes/useSplunkTheme"
    const C = require("@splunk/themes/useSplunkTheme");
    var P =  e.n(C);
    // CONCATENATED MODULE: ./src/Menu/HeadingStyles.ts
    var E = d().div.withConfig({
        displayName: "HeadingStyles__Styled",
        componentId: "mcd2ws-0"
    })([ "", " border-top:1px solid transparent;", ";&:not(:first-child){", ";}" ], m.mixins.reset("block"), (0, 
    m.pick)({
        enterprise: (0, u.css)([ "padding:", " ", " 6px;" ], m.variables.spacingXSmall, m.variables.spacingSmall),
        prisma: {
            comfortable: "padding: 13px 16px 10px 16px",
            compact: "padding: 9px 16px 6px 16px"
        }
    }), (0, m.pick)({
        enterprise: (0, u.css)([ "border-top:", ";" ], m.variables.border),
        prisma: {
            comfortable: (0, u.css)([ "border-top:1px solid ", ";margin-top:6px;padding-top:19px;" ], m.variables.neutral200),
            compact: (0, u.css)([ "border-top:1px solid ", ";margin-top:6px;padding-top:15px;" ], m.variables.neutral200)
        }
    }));
    var I = d().h5.withConfig({
        displayName: "HeadingStyles__PrismaSectionTitle",
        componentId: "mcd2ws-1"
    })([ "", " font-size:", ";line-height:", ";font-weight:", ";letter-spacing:0.02em;" ], m.mixins.reset("block"), m.variables.fontSizeSmall, m.variables.lineHeight, m.variables.fontWeightBold);
    // CONCATENATED MODULE: ./src/Menu/Heading.tsx
    function j(e) {
        "@babel/helpers - typeof";
        return j = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, j(e);
    }
    function _() {
        return _ = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, _.apply(null, arguments);
    }
    function T(e, t) {
        var r = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            t && (n = n.filter((function(t) {
                return Object.getOwnPropertyDescriptor(e, t).enumerable;
            }))), r.push.apply(r, n);
        }
        return r;
    }
    function D(e) {
        for (var t = 1; t < arguments.length; t++) {
            var r = null != arguments[t] ? arguments[t] : {};
            t % 2 ? T(Object(r), !0).forEach((function(t) {
                q(e, t, r[t]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : T(Object(r)).forEach((function(t) {
                Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
            }));
        }
        return e;
    }
    function q(e, t, r) {
        return (t = N(t)) in e ? Object.defineProperty(e, t, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = r, e;
    }
    function N(e) {
        var t = A(e, "string");
        return "symbol" == j(t) ? t : t + "";
    }
    function A(e, t) {
        if ("object" != j(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var n = r.call(e, t || "default");
            if ("object" != j(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    function R(e, t) {
        if (null == e) return {};
        var r, n, i = F(e, t);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (n = 0; n < o.length; n++) {
                r = o[n], t.includes(r) || {}.propertyIsEnumerable.call(e, r) && (i[r] = e[r]);
            }
        }
        return i;
    }
    function F(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (t.includes(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var H = {
        children: o().node,
        title: o().bool,
        outerStyle: o().object
    };
    /**
 * A non-interactive `Menu` item used to separate and label groups of `Menu` items.
 */    function $(e) {
        var t = e.children, i = e.outerStyle, o = e.title, a = R(e, [ "children", "outerStyle", "title" ]);
        var l = (0, r.useContext)(h), s = l.role;
        var c = P()(), p = c.family;
        var u = (0, r.useMemo)((function() {
            return D(D({}, i), {}, {
                margin: 0
            });
        }), [ i ]);
        var d = p === "prisma" ?  n().createElement(I, _({
            as: o ? "h4" : "h5",
            "data-test": "heading",
            style: i
        }, a), t) :  n().createElement(O(), _({
            "data-test": "heading",
            level: o ? 4 : "ss",
            style: u
        }, a), t);
        // headings are not supported in role="listbox"
        // this causes Firefox to incorrectly read the total number
        // of items in the list if it contains a heading or divider
        // so we set them to aria-hidden to prevent that
                
        return n().createElement(E, {
            "aria-hidden": s === "listbox" ? true : undefined
        }, d);
    }
    $.propTypes = H;
    /* Remove consecutive items with filterConsecutive = true (Dividers and Headings) */    $.filterConsecutive = true;
    /* Remove the item if it is the last item after filtering. */    $.filterLast = true;
    /* Tag the item as a Heading */    $.as = "Heading";
    /* harmony default export */    const M = $;
    // CONCATENATED MODULE: external "lodash/isString"
    const L = require("lodash/isString");
    var z =  e.n(L);
    // CONCATENATED MODULE: external "@splunk/react-icons/ChevronRight"
    const K = require("@splunk/react-icons/ChevronRight");
    var B =  e.n(K);
    // CONCATENATED MODULE: external "@splunk/react-icons/Checkmark"
    const W = require("@splunk/react-icons/Checkmark");
    var G =  e.n(W);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/Check"
    const V = require("@splunk/react-icons/enterprise/Check");
    var U =  e.n(V);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/ChevronRight"
    const X = require("@splunk/react-icons/enterprise/ChevronRight");
    var Y =  e.n(X);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const J = require("@splunk/react-ui/ScreenReaderContent");
    var Q =  e.n(J);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const Z = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const ee = require("@splunk/react-ui/Clickable");
    var te =  e.n(ee);
    // CONCATENATED MODULE: external "@splunk/react-ui/Switch"
    const re = require("@splunk/react-ui/Switch");
    var ne =  e.n(re);
    // CONCATENATED MODULE: ./src/Menu/ItemStyles.ts
    var ie = (0, u.css)([ "color:", ";font-size:", ";line-height:", ";overflow:inherit;white-space:inherit;text-overflow:inherit;" ], (0, 
    m.pick)({
        prisma: m.variables.contentColorDefault,
        enterprise: m.variables.textGray
    }), m.variables.fontSizeSmall, (0, m.pick)({
        prisma: "16px"
    }));
    var oe = d().span.withConfig({
        displayName: "ItemStyles__StyledItemDescriptionBottom",
        componentId: "sc-4kc053-0"
    })([ "", ";display:block;" ], ie);
    var ae = d().span.withConfig({
        displayName: "ItemStyles__StyledItemDescriptionRight",
        componentId: "sc-4kc053-1"
    })([ "", ";display:flex;align-items:center;float:right;", " max-width:50%;min-height:20px;text-align:right;box-sizing:border-box;" ], ie, (0, 
    m.pick)({
        enterprise: (0, u.css)([ "padding-left:", ";" ], m.variables.spacingLarge)
    }));
    var le = d().div.withConfig({
        displayName: "ItemStyles__StyledItemSelectedAdornment",
        componentId: "sc-4kc053-2"
    })([ "position:absolute;", " color:", ";" ], (0, m.pick)({
        prisma: (0, u.css)([ "right:16px;" ]),
        enterprise: (0, u.css)([ "right:3px;left:8px;" ])
    }), (0, m.pick)({
        prisma: m.variables.contentColorActive,
        enterprise: m.variables.accentColorL10
    }));
    var se = d().span.withConfig({
        displayName: "ItemStyles__StyledItemAdornment",
        componentId: "sc-4kc053-3"
    })([ "flex:0 0 auto;padding-right:", ";", " min-width:10px;display:inline-block;text-align:center;vertical-align:", ";", "" ], (0, 
    m.pick)({
        prisma: "8px",
        enterprise: "3px"
    }), (function(e) {
        var t = e.endAdornment;
        return t && (0, u.css)([ "padding-left:", ";" ], (0, m.pick)({
            prisma: "8px",
            enterprise: "3px"
        }));
    }), (0, m.pick)({
        prisma: "initial",
        enterprise: "middle"
    }), (0, m.pick)({
        prisma: (0, u.css)([ "color:", ";display:inline-flex;align-items:center;min-width:20px;min-height:20px;& > svg{width:20px;height:20px;}" ], m.variables.contentColorMuted),
        enterprise: (0, u.css)([ "transform:translateY(-1px);" ])
    }));
    var ce = d()(te()).withConfig({
        displayName: "ItemStyles__StyledClickable",
        componentId: "sc-4kc053-4"
    })([ "display:block;position:relative;cursor:pointer;color:", ";word-wrap:break-word;max-width:100%;width:100%;padding:", ";", " ", " ", " &:not([disabled],[aria-disabled='true']){&:hover{background:", ";", "}", " ", "}&:focus{outline:0;box-shadow:", ";", "}&[disabled],&[aria-disabled='true']{color:", ";", ",", "{color:inherit;}", "{color:", ";}", "}", "" ], (0, 
    m.pick)({
        prisma: m.variables.contentColorActive,
        enterprise: {
            dark: m.variables.gray96,
            light: m.variables.gray22
        }
    }), (0, m.pick)({
        prisma: {
            comfortable: "10px 16px",
            compact: "6px 16px"
        },
        enterprise: "6px 10px"
    }), (function(e) {
        var t = e.$isSelectable, r = e.$selectableAppearance;
        return t && (0, m.pick)({
            prisma: (0, u.css)([ "padding-right:44px;", "" ], (function() {
                return r === "checkbox" && "padding-left: 42px;";
            })),
            enterprise: (0, u.css)([ "padding-right:10px;padding-left:", ";" ], (function() {
                return r === "checkbox" ? "32px" : "28px";
            }))
        });
    }), (function(e) {
        var t = e.$active, r = e.$preventFocus;
        return t && !r && (0, u.css)([ "box-shadow:", ";" ], m.variables.focusShadowInset);
    }), (function(e) {
        var t = e.$active;
        return t && (0, u.css)([ "background:", ";font-weight:", ";" ], (0, m.pick)({
            enterprise: {
                light: m.variables.gray92,
                dark: m.variables.gray22
            },
            prisma: m.variables.neutral100
        }), m.variables.fontWeightBold);
    }), (0, m.pick)({
        prisma: m.variables.interactiveColorOverlayHover,
        enterprise: m.variables.backgroundColorHover
    }), (function(e) {
        var t = e.$selected;
        return t && (0, m.pick)({
            prisma: (0, u.css)([ "background:", ";" ], m.mixins.overlayColors(m.variables.interactiveColorOverlaySelected, m.variables.interactiveColorOverlayHover))
        });
    }), (0, m.pick)({
        prisma: (0, u.css)([ "&:active{background:", ";}" ], m.variables.interactiveColorOverlayActive)
    }), (function(e) {
        var t = e.$selected;
        return t && (0, m.pick)({
            prisma: (0, u.css)([ "background:", ";&:active{background:", ";}" ], m.variables.interactiveColorOverlaySelected, m.mixins.overlayColors(m.variables.interactiveColorOverlaySelected, m.variables.interactiveColorOverlayActive))
        });
    }), m.variables.focusShadowInset, (0, m.pick)({
        prisma: (0, u.css)([ "background:", ";" ], m.variables.interactiveColorOverlayHover)
    }), (0, m.pick)({
        enterprise: {
            light: m.variables.gray45,
            dark: m.variables.contentColorDisabled
        },
        prisma: m.variables.contentColorDisabled
    }), /* sc-sel */ oe, /* sc-sel */ ae, /* sc-sel */ le, (0, m.pick)({
        prisma: m.variables.contentColorDisabled,
        enterprise: m.variables.gray80
    }), (0, m.pick)({
        prisma: (0, u.css)([ "& > * > ", "{color:", ";}" ], /* sc-sel */ se, m.variables.contentColorDisabled)
    }), (function(e) {
        var t = e.$selected;
        return t && (0, m.pick)({
            prisma: (0, u.css)([ "", "{color:", ";}" ], /* sc-sel */ se, m.variables.contentColorActive)
        });
    }));
    var pe = d()(ne()).withConfig({
        displayName: "ItemStyles__StyledSwitch",
        componentId: "sc-4kc053-5"
    })([ "position:absolute;left:", ";top:", ";" ], (0, m.pick)({
        prisma: "16px",
        enterprise: "8px"
    }), (0, m.pick)({
        prisma: "0px",
        enterprise: {
            comfortable: "-1px",
            compact: "2px"
        }
    }));
    var ue = d().span.withConfig({
        displayName: "ItemStyles__StyledInnerWrapper",
        componentId: "sc-4kc053-6"
    })([ "display:flex;align-items:flex-start;" ]);
    var de = d().span.withConfig({
        displayName: "ItemStyles__StyledTitleAndDescriptionWrapper",
        componentId: "sc-4kc053-7"
    })([ "display:flex;flex-wrap:wrap;flex-direction:column;max-width:100%;width:100%;align-self:center;word-break:break-word;white-space:normal;", "" ], (function(e) {
        var t = e.$truncate;
        return t && (0, u.css)([ "white-space:nowrap;overflow:hidden;text-overflow:ellipsis;" ]);
    }));
    var fe = d().div.withConfig({
        displayName: "ItemStyles__StyledLabel",
        componentId: "sc-4kc053-8"
    })([ "overflow:inherit;white-space:inherit;text-overflow:inherit;max-width:100%;min-height:20px;", "" ], (function(e) {
        var t = e.$truncate;
        return t && (0, u.css)([ "display:block;clear:both;" ]);
    }));
    var ve = d().span.withConfig({
        displayName: "ItemStyles__StyledMatch",
        componentId: "sc-4kc053-9"
    })([ "border-bottom:1px solid ", ";" ], (0, m.pick)({
        prisma: m.variables.contentColorActive,
        enterprise: {
            dark: m.variables.gray96,
            light: m.variables.gray45
        }
    }));
    var me = d().span.withConfig({
        displayName: "ItemStyles__StyledSubmenu",
        componentId: "sc-4kc053-10"
    })([ "float:right;padding-left:", ";color:", ";", "" ], m.variables.spacingSmall, (0, 
    m.pick)({
        prisma: "inherit",
        enterprise: m.variables.textGray
    }), (0, m.pick)({
        prisma: (0, u.css)([ "& > svg{width:20px;height:20px;}" ])
    }));
    // CONCATENATED MODULE: external "@splunk/react-icons/ArrowSquareTopRight"
    const be = require("@splunk/react-icons/ArrowSquareTopRight");
    var ye =  e.n(be);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/External"
    const he = require("@splunk/react-icons/enterprise/External");
    var ge =  e.n(he);
    // CONCATENATED MODULE: ./src/Menu/icons/External.tsx
    var ke = (0, u.css)([ "width:0.8em;height:0.8em;vertical-align:baseline;" ]);
    var xe = d()(ge()).withConfig({
        displayName: "External__StyledEnterprise",
        componentId: "sc-1turs2s-0"
    })([ "", " margin:0 0 0 3px;" ], ke);
    var we = d()(ye()).withConfig({
        displayName: "External__StyledPrisma",
        componentId: "sc-1turs2s-1"
    })([ "", " margin:0 0 0 4px;" ], ke);
    function Se() {
        var e = (0, m.useSplunkTheme)(), t = e.isEnterprise, r = e.isCompact;
        var i = r ? "20px" : "24px";
        return t ?  n().createElement(xe, {
            screenReaderText: null
        }) :  n().createElement(we, {
            height: i,
            width: i
        });
    }
    /* harmony default export */    const Oe = Se;
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function Ce(e, t) {
        if (e) {
            if (typeof e === "function") {
                e(t);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = t;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/utils/scrollIntoViewIfNeeded.ts
    // A utility that attempts to move an element into view by scrolling it's derived parent.
    var Pe = function e(t) {
        if (!t) {
            return;
        }
        var r = t.offsetParent;
        if (!r) {
            return;
        }
        // Below the bottom of the container.
                if (r.scrollTop + r.clientHeight < t.offsetTop + t.clientHeight) {
            r.scrollTop = t.offsetTop + t.clientHeight - r.clientHeight;
            // Above the top of the container.
                } else if (r.scrollTop > t.offsetTop) {
            r.scrollTop = t.offsetTop;
        }
    };
    // CONCATENATED MODULE: ./src/Menu/Item.tsx
    function Ee(e) {
        "@babel/helpers - typeof";
        return Ee = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, Ee(e);
    }
    function Ie() {
        return Ie = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, Ie.apply(null, arguments);
    }
    function je(e, t) {
        if (null == e) return {};
        var r, n, i = _e(e, t);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (n = 0; n < o.length; n++) {
                r = o[n], t.includes(r) || {}.propertyIsEnumerable.call(e, r) && (i[r] = e[r]);
            }
        }
        return i;
    }
    function _e(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (t.includes(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    function Te(e, t) {
        if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function");
    }
    function De(e, t) {
        for (var r = 0; r < t.length; r++) {
            var n = t[r];
            n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), 
            Object.defineProperty(e, ze(n.key), n);
        }
    }
    function qe(e, t, r) {
        return t && De(e.prototype, t), r && De(e, r), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function Ne(e, t) {
        if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), t && Ae(e, t);
    }
    function Ae(e, t) {
        return Ae = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, t) {
            return e.__proto__ = t, e;
        }, Ae(e, t);
    }
    function Re(e) {
        var t = $e();
        return function() {
            var r, n = Me(e);
            if (t) {
                var i = Me(this).constructor;
                r = Reflect.construct(n, arguments, i);
            } else r = n.apply(this, arguments);
            return Fe(this, r);
        };
    }
    function Fe(e, t) {
        if (t && ("object" == Ee(t) || "function" == typeof t)) return t;
        if (void 0 !== t) throw new TypeError("Derived constructors may only return object or undefined");
        return He(e);
    }
    function He(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function $e() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return ($e = function t() {
            return !!e;
        })();
    }
    function Me(e) {
        return Me = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, Me(e);
    }
    function Le(e, t, r) {
        return (t = ze(t)) in e ? Object.defineProperty(e, t, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = r, e;
    }
    function ze(e) {
        var t = Ke(e, "string");
        return "symbol" == Ee(t) ? t : t + "";
    }
    function Ke(e, t) {
        if ("object" != Ee(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var n = r.call(e, t || "default");
            if ("object" != Ee(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    var Be = {
        active: o().bool,
        children: o().node,
        description: o().string,
        descriptionPosition: o().oneOf([ "right", "bottom" ]),
        disabled: o().oneOfType([ o().bool, o().oneOf([ "dimmed" ]) ]),
        elementRef: o().oneOfType([ o().func, o().object ]),
        endAdornment: o().node,
        hasSubmenu: o().bool,
        icon: o().node,
        /** @private. The internal key of the tab */
        itemKey: o().number,
        matchRanges: o().arrayOf(o().shape({
            start: o().number.isRequired,
            end: o().number.isRequired
        })),
        onClick: o().func,
        /** @private */
        onFocus: o().func,
        openInNewContext: o().oneOfType([ o().bool, o().string ]),
        /** @private */
        preventFocus: o().bool,
        role: o().oneOf([ "menuitem", "menuitemradio", "menuitemcheckbox", "listboxitem", "option" ]),
        selectable: o().bool,
        selectableAppearance: o().oneOf([ "checkmark", "checkbox" ]),
        selected: o().oneOfType([ o().bool, o().oneOf([ "some" ]) ]),
        /** @private */
        splunkTheme: o().object,
        startAdornment: o().node,
        to: o().string,
        truncate: o().bool
    };
    var We = {
        active: false,
        descriptionPosition: "bottom",
        hasSubmenu: false,
        openInNewContext: false,
        preventFocus: false,
        selectable: false,
        selectableAppearance: "checkmark",
        selected: false,
        truncate: false
    };
    var Ge = (0, Z._)("(Opens new window)");
    var Ve =  function(e) {
        Ne(r, e);
        var t = Re(r);
        qe(r, null, [ {
            key: "validateProps",
            // @docs-props-type ItemPropsBase
            value: function e(t) {
                if (false) {}
            }
        } ]);
        function r(e) {
            var n;
            Te(this, r);
            n = t.call(this, e);
            Le(He(n), "el", null);
            Le(He(n), "handleFocus", (function(e) {
                var t = n.props, r = t.onFocus, i = t.itemKey;
                r === null || r === void 0 ? void 0 : r(e, {
                    itemKey: i
                });
            }));
            Le(He(n), "handleMount", (function(e) {
                n.el = e;
                Ce(n.props.elementRef, e);
            }));
            r.validateProps(e);
            return n;
        }
        qe(r, [ {
            key: "componentDidUpdate",
            value: function e() {
                r.validateProps(this.props);
            }
        }, {
            key: "focus",
            // eslint-disable-next-line react/no-unused-class-component-methods
            value: function e() {
                var t;
                (t = this.el) === null || t === void 0 ? void 0 : t.focus();
            }
 // eslint-disable-next-line react/no-unused-class-component-methods
                }, {
            key: "scrollIntoViewIfNeeded",
            value: function e() {
                Pe(this.el);
            }
        }, {
            key: "renderLabel",
            value: function e() {
                var t = this.props, r = t.children, i = t.matchRanges;
                if (!i || !z()(r)) {
                    return r;
                }
                var o = [];
                // before first match. May be empty string.
                                o.push(r.substring(0, i[0].start));
                i.forEach((function(e, t) {
                    o.push(
                    
                    // eslint-disable-next-line react/no-array-index-key
                    n().createElement(ve, {
                        key: t,
                        "data-test": "match"
                    }, r.substring(e.start, e.end)));
                    if (t < i.length - 1) {
                        o.push(r.substring(e.end, i[t + 1].start));
                    } else {
                        o.push(r.substring(e.end, r.length));
                    }
                }));
                return o;
            }
        }, {
            key: "render",
            value: function e() {
                var t = this.props, r = t.active, i = t.children, o = t.hasSubmenu, a = t.selectable, s = t.selectableAppearance, c = t.selected, p = t.icon, u = t.description, d = t.disabled, f = t.onClick, v = t.role, m = t.splunkTheme, b = t.to, y = t.truncate, h = t.descriptionPosition, g = t.openInNewContext, k = t.endAdornment, x = t.startAdornment, w = je(t, [ "active", "children", "hasSubmenu", "selectable", "selectableAppearance", "selected", "icon", "description", "disabled", "onClick", "role", "splunkTheme", "to", "truncate", "descriptionPosition", "openInNewContext", "endAdornment", "startAdornment" ]);
                var S = m.isEnterprise, O = m.isPrisma;
                var C = this.context.preventFocus;
                var P = c === true || c === "some";
                var E = a || P;
                var I = v || {
                    nonselectable: "menuitem",
                    checkmark: "menuitemradio",
                    checkbox: "menuitemcheckbox"
                }[E ? s : "nonselectable"];
                var j = {
                    "aria-haspopup": o ? true : undefined
                };
                if (I === "menuitemradio" || I === "menuitemcheckbox") {
                    j["aria-checked"] = c === "some" ? "mixed" : P;
                } else if (I === "option") {
                    j["aria-selected"] = P;
                }
                var _ = O ? "right" : "left";
                var T = u && h === "right";
                var D = u && !T;
                var q = c && s === "checkmark" &&  n().createElement(le, null, O ?  n().createElement(G(), {
                    inline: true,
                    height: "16",
                    width: "16"
                }) :  n().createElement(U(), {
                    hideDefaultTooltip: true,
                    inline: true,
                    screenReaderText: null,
                    size: .85
                }));
                var N = false;
                var A;
                if (g) {
                    N = true;
                    A = typeof g === "string" ? g : Ge;
                }
                
                return n().createElement(ce, Ie({
                    $isSelectable: E,
                    $selectableAppearance: s,
                    $selected: P,
                    $active: r,
                    $preventFocus: C || false,
                    "data-test-selected": E ? P : null,
                    "data-test": "item",
                    "data-has-icon": !!p || !!x || !!k,
                    disabled: d,
                    onClick: f,
                    onFocus: this.handleFocus,
                    onMouseDown: function e(t) {
                        if (C) {
                            t.preventDefault();
                        }
                    },
                    role: I,
                    tabIndex: C ? -1 : undefined,
                    to: b,
                    title: y && z()(i) ? i : undefined,
                    openInNewContext: N
                }, j, l()(w, "onFocus"), {
                    elementRef: this.handleMount
                }), a && s === "checkbox" &&  n().createElement(pe, {
                    interactive: false,
                    value: "menu-item",
                    selectedLabel: "Selected",
                    selected: c,
                    disabled: Boolean(d)
                }), o &&  n().createElement(me, null, S ?  n().createElement(Y(), {
                    hideDefaultTooltip: true,
                    screenReaderText: null
                }) :  n().createElement(B(), null)), T &&  n().createElement(ae, {
                    "data-test": "description"
                }, u),  n().createElement(ue, null, x ?  n().createElement(se, null, x) : p &&  n().createElement(se, null, p),  n().createElement(de, {
                    $truncate: y
                }, _ === "left" && q,  n().createElement(fe, {
                    $truncate: y,
                    "data-test": "label"
                }, this.renderLabel(), g &&  n().createElement(n().Fragment, null,  n().createElement(Oe, null),  n().createElement(Q(), null, A))), D &&  n().createElement(oe, {
                    "data-test": "description"
                }, u), _ === "right" && q), k &&  n().createElement(se, {
                    endAdornment: k
                }, k)));
            }
        } ]);
        return r;
    }(r.Component);
    Le(Ve, "propTypes", Be);
    Le(Ve, "defaultProps", We);
    Le(Ve, "as", void 0);
    Le(Ve, "contextType", h);
    var Ue = (0, m.withSplunkTheme)(Ve);
    var Xe = Ue;
    Xe.propTypes = Ve.propTypes;
    Xe.as = "Item";
    /* harmony default export */    const Ye = Xe;
    // only for styled components using Item
        // CONCATENATED MODULE: external "@splunk/react-ui/Scroll"
    const Je = require("@splunk/react-ui/Scroll");
    var Qe =  e.n(Je);
    // CONCATENATED MODULE: ./src/Menu/MenuStyles.ts
    var Ze = (0, u.css)([ "border-top:1px solid ", ";" ], (0, m.pick)({
        enterprise: m.variables.gray60,
        prisma: m.variables.neutral200
    }));
    var et = (0, u.css)([ "", ";background-color:", ";border-radius:", ";min-width:60px;overflow:auto;position:relative;", " &:focus{outline:0;box-shadow:", ";", "}& + &{", "}" ], m.mixins.reset("block"), (0, 
    m.pick)({
        enterprise: m.variables.backgroundColor,
        prisma: m.variables.backgroundColorPopup
    }), m.variables.borderRadius, (0, m.pick)({
        prisma: (0, u.css)([ "padding:8px 0;" ])
    }), m.variables.focusShadowInset, (0, m.pick)({
        prisma: (0, u.css)([ "background:", ";" ], m.variables.interactiveColorOverlayHover)
    }), Ze);
    var tt = d().div.withConfig({
        displayName: "MenuStyles__Styled",
        componentId: "sc-1olffp9-0"
    })([ "", "" ], et);
    var rt = d()(Qe()).withConfig({
        displayName: "MenuStyles__StyledScroll",
        componentId: "sc-1olffp9-1"
    })([ "", " ", " + &,& + ", "{", "}" ], et, /* sc-sel */ tt, /* sc-sel */ tt, Ze);
    // CONCATENATED MODULE: ./src/Menu/Menu.tsx
    function nt() {
        return nt = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, nt.apply(null, arguments);
    }
    function it(e, t) {
        if (null == e) return {};
        var r, n, i = ot(e, t);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (n = 0; n < o.length; n++) {
                r = o[n], t.includes(r) || {}.propertyIsEnumerable.call(e, r) && (i[r] = e[r]);
            }
        }
        return i;
    }
    function ot(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (t.includes(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var at = {
        children: o().node,
        elementRef: o().oneOfType([ o().func, o().object ]),
        focusMode: o().oneOf([ "roving", "normal", "never" ]),
        retainFocus: o().bool,
        stopScrollPropagation: o().bool
    };
    // the default focus control for Menu: loop, verticalArrows(up/down keys), home/end keys
        var lt = function e(t) {
        return {
            enableLoop: true,
            orientation: "vertical",
            enableTab: t,
            enableHomeEnd: true
        };
    };
    var st = function e(t) {
        var r = (0, c.keycode)(t.nativeEvent);
        // Handle special case for tab navigation
                if (r === "tab" && (0, p.isTabKey)(t)) {
            return t.shiftKey ? "tabShift" : "tab";
        }
        return r;
    };
    function ct(e) {
        var t = e.children, i = e.elementRef, o = e.focusMode, a = e.retainFocus, c = e.stopScrollPropagation, u = it(e, [ "children", "elementRef", "focusMode", "retainFocus", "stopScrollPropagation" ]);
        // @docs-props-type MenuPropsBase
                if (false) {}
        var d = (0, r.useContext)(h), f = d.role, v = f === void 0 ? "menu" : f;
        var m = (0, r.useContext)(s.PopoverMenuContext), b = m.retainFocus;
        var y = (0, r.useRef)(null);
        var g = true;
        if (a != null) {
            g = a;
        } else if (o === "roving") {
            g = false;
        } else if (b != null) {
            g = b;
        }
        var k = r.Children.toArray(t).filter(r.isValidElement).reduce((function(e, t, r, n) {
            /* Filter out initial Dividers
     * Requires reduce() over filter() because a Heading may have been
     * before the Divider.
     */
            if (t.type.as === "Divider" && e.length === 0) {
                return e;
            }
            // Filter out consecutive Dividers and Headings
            // If Divider is next to Heading then it will remove both Heading and Divider unless Heading has title prop set to it
                        if ((t.type.as === "Heading" || t.type.as === "Divider") && !t.props.title && t.type.filterConsecutive && n.length > r + 1) {
                var i = n[r + 1];
                if (i.type.as === "Heading" || i.type.as === "Divider") {
                    return e;
                }
            }
            // Filter out last Dividers and Headings
                        if ((t.type.as === "Divider" || t.type.as === "Heading") && r === n.length - 1) {
                return e;
            }
            e.push(t);
            return e;
        }), []);
        var x = (0, r.useCallback)((function(e) {
            y.current = e;
            Ce(i, e);
        }), [ i ]);
        var w = lt(g);
        var S = (0, r.useCallback)((function(e) {
            // if command key on Mac was pressed ignore
            if (e.nativeEvent.metaKey) {
                return;
            }
            var t = st(e);
            var r = t === "tab" || t === "tabShift";
            var n = t === "enter";
            // Allow default behavior for the enter key.
            // If tab navigation is enabled, prevent default behavior for all other keys.
            // If tab navigation is disabled, allow default behavior for tab and tab+shift key
                        if (!n && (w.enableTab || !r)) {
                e.preventDefault();
            }
            var i = y.current;
            var o = [];
            if (i) {
                o = (0, p.getSortedTabbableElements)(i, {
                    ignoreTabIndex: true
                });
            }
            var a = (e === null || e === void 0 ? void 0 : e.target) instanceof Element ? e.target : null;
            var l = i === null || i === void 0 ? void 0 : i.querySelector(":focus");
            var s = a || l;
            var c = s ? o.indexOf(s) : -1;
            (0, p.handleFocus)(t, o, c, w);
        }), [ w ]);
        var O = function e() {
            return {
                role: "menu",
                preventFocus: o === "never"
            };
        };
        var C = o === "normal" || o === "never" ? undefined : S;
        var P = {
            "data-test": "menu",
            onKeyDown: C,
            role: v
        };
        return c ?  n().createElement(rt, nt({}, P, {
            elementRef: x,
            stopScrollPropagation: true
        }, l()(u, "tagName")), k) :  n().createElement(tt, nt({}, P, {
            ref: x
        }, u),  n().createElement(h.Provider, {
            value: O()
        }, k));
    }
    ct.propTypes = at;
    ct.Item = Ye;
    ct.Divider = w;
    ct.Heading = M;
    /* harmony default export */    const pt = ct;
    // CONCATENATED MODULE: ./src/Menu/index.ts
    module.exports = t;
    /******/})();
