// @splunk/react-ui/MessageBar.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var a = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(a, {
                a
            });
            /******/            return a;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, a) => {
            /******/ for (var s in a) {
                /******/ if (e.o(a, s) && !e.o(r, s)) {
                    /******/ Object.defineProperty(r, s, {
                        enumerable: true,
                        get: a[s]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ se
    });
    // CONCATENATED MODULE: external "react"
    const a = require("react");
    var s =  e.n(a);
    // CONCATENATED MODULE: external "prop-types"
    const i = require("prop-types");
    var n =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/react-icons/Cross"
    const o = require("@splunk/react-icons/Cross");
    var t =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/Close"
    const l = require("@splunk/react-icons/enterprise/Close");
    var c =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const p = require("@splunk/react-ui/ScreenReaderContent");
    var d =  e.n(p);
    // CONCATENATED MODULE: external "@splunk/themes"
    const u = require("@splunk/themes");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const v = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "styled-components"
    const b = require("styled-components");
    var g =  e.n(b);
    // CONCATENATED MODULE: external "@splunk/react-icons/CheckCircle"
    const h = require("@splunk/react-icons/CheckCircle");
    var y =  e.n(h);
    // CONCATENATED MODULE: external "@splunk/react-icons/ExclamationCircle"
    const m = require("@splunk/react-icons/ExclamationCircle");
    var k =  e.n(m);
    // CONCATENATED MODULE: external "@splunk/react-icons/InformationCircle"
    const f = require("@splunk/react-icons/InformationCircle");
    var x =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/react-icons/ExclamationTriangle"
    const C = require("@splunk/react-icons/ExclamationTriangle");
    var w =  e.n(C);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/Error"
    const S = require("@splunk/react-icons/enterprise/Error");
    var _ =  e.n(S);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/InfoCircle"
    const I = require("@splunk/react-icons/enterprise/InfoCircle");
    var z =  e.n(I);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/Success"
    const q = require("@splunk/react-icons/enterprise/Success");
    var E =  e.n(q);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/Warning"
    const O = require("@splunk/react-icons/enterprise/Warning");
    var B =  e.n(O);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const M = require("@splunk/react-ui/Box");
    var j =  e.n(M);
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const N = require("@splunk/react-ui/Clickable");
    var R =  e.n(N);
    // CONCATENATED MODULE: ./src/MessageBar/MessageBarStyles.ts
    var T = g().div.withConfig({
        displayName: "MessageBarStyles__MessageBarContent",
        componentId: "sc-1uyhzza-0"
    })([ "", ";color:inherit;padding:0 8px;text-align:center;margin-right:auto;& a,& button[data-test='link']{display:inline-block;color:inherit;text-decoration:underline;text-underline-position:under;padding:0 3px 2px 3px;&:not([disabled]):focus{border-radius:", ";outline:0;", ";}}" ], u.mixins.reset("inline"), u.variables.borderRadius, (0, 
    u.pick)({
        enterprise: {
            dark: (0, b.css)([ "box-shadow:0 0 0 2px ", ";" ], u.variables.gray80),
            light: (0, b.css)([ "box-shadow:0 0 0 2px ", ";" ], u.variables.gray25)
        },
        prisma: {
            light: (0, u.pickVariant)("$type", {
                info: (0, b.css)([ "box-shadow:0 0 0 3px ", ";" ], u.variables.black),
                warning: (0, b.css)([ "box-shadow:0 0 0 3px ", ";" ], u.variables.black),
                error: (0, b.css)([ "box-shadow:0 0 0 3px ", ";" ], u.variables.white),
                success: (0, b.css)([ "box-shadow:0 0 0 3px ", ";" ], u.variables.white)
            }),
            dark: (0, b.css)([ "box-shadow:0 0 0 3px ", ";" ], u.variables.black)
        }
    }));
    var W = g()(R()).withConfig({
        displayName: "MessageBarStyles__MessageBarCloseWrapper",
        componentId: "sc-1uyhzza-1"
    })([ "", ";border:1px solid transparent;border-radius:", ";color:inherit;cursor:pointer;padding:", ";", " &:hover,&:focus{background:", ";border:1px solid;border-color:", ";color:", ";}&:active{", ";}", "" ], u.mixins.reset("flex"), u.variables.borderRadius, (0, 
    u.pick)({
        enterprise: u.variables.spacingXSmall,
        prisma: "3px"
    }), (0, u.pick)({
        enterprise: (0, b.css)([ "margin-top:1px;" ])
    }), (0, u.pick)({
        prisma: u.variables.interactiveColorOverlayHover,
        enterprise: {
            dark: u.variables.gray30,
            light: u.variables.gray92
        }
    }), (0, u.pick)({
        prisma: "inherit",
        enterprise: u.variables.gray80
    }), (0, u.pick)({
        enterprise: {
            dark: u.variables.gray96,
            light: u.variables.linkColor
        }
    }), (0, u.pick)({
        prisma: (0, b.css)([ "background:", ";" ], u.variables.interactiveColorOverlayActive),
        enterprise: (0, b.css)([ "box-shadow:", ";" ], u.variables.focusShadow)
    }), (0, u.pick)({
        prisma: (0, b.css)([ "&:focus{border-radius:", ";", ";}" ], u.variables.borderRadius, (0, 
        u.pick)({
            light: (0, u.pickVariant)("$type", {
                info: (0, b.css)([ "box-shadow:0 0 0 3px ", ";" ], u.variables.black),
                warning: (0, b.css)([ "box-shadow:0 0 0 3px ", ";" ], u.variables.black),
                error: (0, b.css)([ "box-shadow:0 0 0 3px ", ";" ], u.variables.white),
                success: (0, b.css)([ "box-shadow:0 0 0 3px ", ";" ], u.variables.white)
            }),
            dark: (0, b.css)([ "box-shadow:0 0 0 3px ", ";" ], u.variables.black)
        }))
    }));
    var P = (0, b.css)([ "margin-left:auto;" ]);
    var $ = g()(y()).withConfig({
        displayName: "MessageBarStyles__StyledCheckCircleIcon",
        componentId: "sc-1uyhzza-2"
    })([ "", "" ], P);
    var A = g()(k()).withConfig({
        displayName: "MessageBarStyles__StyledExclamationCircleIcon",
        componentId: "sc-1uyhzza-3"
    })([ "", "" ], P);
    var L = g()(x()).withConfig({
        displayName: "MessageBarStyles__StyledInformationCircleIcon",
        componentId: "sc-1uyhzza-4"
    })([ "", "" ], P);
    var V = g()(w()).withConfig({
        displayName: "MessageBarStyles__StyledExclamationTriangleIcon",
        componentId: "sc-1uyhzza-5"
    })([ "", "" ], P);
    var D = g()(_()).withConfig({
        displayName: "MessageBarStyles__StyledEnterpriseErrorIcon",
        componentId: "sc-1uyhzza-6"
    })([ "", " color:", ";" ], P, u.variables.errorColor);
    var H = g()(z()).withConfig({
        displayName: "MessageBarStyles__StyledEnterpriseInfoCircleIcon",
        componentId: "sc-1uyhzza-7"
    })([ "", " color:", ";" ], P, u.variables.infoColor);
    var X = g()(E()).withConfig({
        displayName: "MessageBarStyles__StyledEnterpriseSuccessIcon",
        componentId: "sc-1uyhzza-8"
    })([ "", " color:", ";" ], P, u.variables.successColor);
    var F = g()(B()).withConfig({
        displayName: "MessageBarStyles__StyledEnterpriseWarningIcon",
        componentId: "sc-1uyhzza-9"
    })([ "", " color:", ";" ], P, u.variables.warningColor);
    var G = g()(j()).withConfig({
        displayName: "MessageBarStyles__MessageBarWrapper",
        componentId: "sc-1uyhzza-10"
    })([ "", ";display:flex;align-items:center;justify-content:center;min-height:40px;margin-bottom:", ";", ";word-wrap:break-word;color:", ";", ";" ], u.mixins.reset("block"), u.variables.spacingSmall, (function(e) {
        var r = e.$hasCloseButton;
        return r ? (0, b.css)([ "", "" ], (0, u.pick)({
            prisma: (0, b.css)([ "padding:0 8px 0 32px;" ]),
            enterprise: (0, b.css)([ "padding:0 8px 0 38px;" ])
        })) : (0, b.css)([ "", "" ], (0, u.pick)({
            prisma: (0, b.css)([ "padding:0 33px 0 32px;" ]),
            enterprise: (0, b.css)([ "padding:0 26px 0 32px;" ])
        }));
    }), (0, u.pick)({
        prisma: (0, u.pickVariant)("$type", {
            info: {
                light: u.variables.contentColorActive,
                dark: u.variables.contentColorInverted
            },
            success: u.variables.contentColorInverted,
            warning: u.variables.black,
            error: u.variables.contentColorInverted
        }),
        enterprise: {
            dark: u.variables.gray96,
            light: u.variables.gray25
        }
    }), (0, u.pickVariant)("$type", {
        info: (0, b.css)([ "background-color:", ";" ], (0, u.pick)({
            prisma: {
                dark: u.variables.white,
                light: u.variables.neutral200
            },
            enterprise: {
                dark: u.mixins.colorWithAlpha(u.variables.infoColor, .5),
                light: u.variables.infoColorL50
            }
        })),
        success: (0, b.css)([ "background-color:", ";" ], (0, u.pick)({
            prisma: u.variables.accentColorPositive,
            enterprise: {
                dark: u.mixins.colorWithAlpha(u.variables.successColor, .5),
                light: u.variables.successColorL50
            }
        })),
        warning: (0, b.css)([ "background-color:", ";" ], (0, u.pick)({
            prisma: u.variables.accentColorWarning,
            enterprise: {
                dark: u.mixins.colorWithAlpha(u.variables.warningColor, .5),
                light: u.variables.warningColorL50
            }
        })),
        error: (0, b.css)([ "background-color:", ";" ], (0, u.pick)({
            prisma: u.variables.accentColorNegative,
            enterprise: {
                dark: u.mixins.colorWithAlpha(u.variables.errorColor, .5),
                light: u.variables.errorColorL50
            }
        }))
    }));
    // CONCATENATED MODULE: ./src/MessageBar/MessageBar.tsx
    function J() {
        return J = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var a = arguments[r];
                for (var s in a) {
                    ({}).hasOwnProperty.call(a, s) && (e[s] = a[s]);
                }
            }
            return e;
        }, J.apply(null, arguments);
    }
    function K(e, r) {
        if (null == e) return {};
        var a, s, i = Q(e, r);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            for (s = 0; s < n.length; s++) {
                a = n[s], r.includes(a) || {}.propertyIsEnumerable.call(e, a) && (i[a] = e[a]);
            }
        }
        return i;
    }
    function Q(e, r) {
        if (null == e) return {};
        var a = {};
        for (var s in e) {
            if ({}.hasOwnProperty.call(e, s)) {
                if (r.includes(s)) continue;
                a[s] = e[s];
            }
        }
        return a;
    }
    var U = {
        children: n().node.isRequired,
        onRequestClose: n().func,
        type: n().oneOf([ "info", "success", "warning", "error" ]).isRequired
    };
    var Y = Object.freeze({
        info: (0, v._)("Info"),
        warning: (0, v._)("Warning"),
        error: (0, v._)("Alert"),
        success: (0, v._)("Success")
    });
    var Z = Object.freeze({
        info: H,
        warning: F,
        error: D,
        success: X
    });
    var ee = Object.freeze({
        info: L,
        warning: V,
        error: A,
        success: $
    });
    var re = "24px";
    function ae(e) {
        var r = e.children, a = e.type, i = e.onRequestClose, n = K(e, [ "children", "type", "onRequestClose" ]);
        // @docs-props-type MessageBarPropsBase
                var o = (0, u.useSplunkTheme)(), l = o.isEnterprise;
        var p = l && a === "error" ? "22px" : re;
        var b = l && a === "warning" ? "21px" : re;
        var g;
        if (l) {
            var h = Z[a];
            g =  s().createElement(h, {
                height: b,
                width: p,
                hideDefaultTooltip: true,
                screenReaderText: null,
                "aria-hidden": true,
                "data-test": "icon"
            });
        } else {
            var y = ee[a];
            g =  s().createElement(y, {
                height: re,
                width: re,
                variant: "filled",
                "aria-hidden": true,
                "data-test": "icon"
            });
        }
        
        return s().createElement(G, J({
            $type: a,
            "data-test": "message-bar",
            "data-test-type": a
        }, n, {
            role: "region",
            $hasCloseButton: Boolean(i)
        }), g,  s().createElement(d(), null, Y[a]),  s().createElement(T, {
            $type: a,
            "data-test": "content"
        }, r), i &&  s().createElement(W, {
            $type: a,
            onClick: i
        }, l ?  s().createElement(c(), {
            height: "12px",
            hideDefaultTooltip: true,
            screenReaderText: null,
            width: "12px"
        }) :  s().createElement(t(), {
            height: "19px",
            style: {
                margin: -.5
            },
            width: "19px"
        }),  s().createElement(d(), null, (0, v._)("Close"))));
    }
    ae.propTypes = U;
    /* harmony default export */    const se = ae;
    // CONCATENATED MODULE: ./src/MessageBar/index.ts
    module.exports = r;
    /******/})();
