// @splunk/react-ui/Message.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var a = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(a, {
                a
            });
            /******/            return a;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, a) => {
            /******/ for (var n in a) {
                /******/ if (e.o(a, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: a[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        Link: () => /* reexport */ z,
        default: () => /* reexport */ pe
    });
    // CONCATENATED MODULE: external "react"
    const a = require("react");
    var n =  e.n(a);
    // CONCATENATED MODULE: external "prop-types"
    const i = require("prop-types");
    var o =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/react-icons/CheckCircle"
    const t = require("@splunk/react-icons/CheckCircle");
    var s =  e.n(t);
    // CONCATENATED MODULE: external "@splunk/react-icons/Cross"
    const c = require("@splunk/react-icons/Cross");
    var l =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/Close"
    const p = require("@splunk/react-icons/enterprise/Close");
    var u =  e.n(p);
    // CONCATENATED MODULE: external "@splunk/react-icons/ExclamationCircle"
    const v = require("@splunk/react-icons/ExclamationCircle");
    var d =  e.n(v);
    // CONCATENATED MODULE: external "@splunk/react-icons/InformationCircle"
    const b = require("@splunk/react-icons/InformationCircle");
    var f =  e.n(b);
    // CONCATENATED MODULE: external "@splunk/react-icons/ExclamationTriangle"
    const g = require("@splunk/react-icons/ExclamationTriangle");
    var m =  e.n(g);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/Error"
    const y = require("@splunk/react-icons/enterprise/Error");
    var k =  e.n(y);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/InfoCircle"
    const h = require("@splunk/react-icons/enterprise/InfoCircle");
    var C =  e.n(h);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/Success"
    const x = require("@splunk/react-icons/enterprise/Success");
    var w =  e.n(x);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/Warning"
    const O = require("@splunk/react-icons/enterprise/Warning");
    var S =  e.n(O);
    // CONCATENATED MODULE: external "@splunk/react-ui/MessageBar"
    const j = require("@splunk/react-ui/MessageBar");
    var q =  e.n(j);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const _ = require("@splunk/react-ui/ScreenReaderContent");
    var E =  e.n(_);
    // CONCATENATED MODULE: external "@splunk/themes"
    const I = require("@splunk/themes");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const P = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "styled-components"
    const R = require("styled-components");
    var T =  e.n(R);
    // CONCATENATED MODULE: external "@splunk/react-ui/Link"
    const $ = require("@splunk/react-ui/Link");
    var L =  e.n($);
    // CONCATENATED MODULE: ./src/Message/LinkStyles.ts
    var M = T()(L()).withConfig({
        displayName: "LinkStyles__StyledLink",
        componentId: "w06zjv-0"
    })([ "" ]);
    // CONCATENATED MODULE: ./src/Message/Link.tsx
    function A(e, r) {
        if (e == null) return {};
        var a = N(e, r);
        var n, i;
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (i = 0; i < o.length; i++) {
                n = o[i];
                if (r.indexOf(n) >= 0) continue;
                if (!Object.prototype.propertyIsEnumerable.call(e, n)) continue;
                a[n] = e[n];
            }
        }
        return a;
    }
    function N(e, r) {
        if (e == null) return {};
        var a = {};
        var n = Object.keys(e);
        var i, o;
        for (o = 0; o < n.length; o++) {
            i = n[o];
            if (r.indexOf(i) >= 0) continue;
            a[i] = e[i];
        }
        return a;
    }
    var W = {
        children: o().node,
        elementRef: o().oneOfType([ o().func, o().object ]),
        openInNewContext: o().bool,
        to: o().string
    };
    /**
 * @deprecated `Message.Link` is deprecated and will be removed in a future major version. Use `Link` instead.
 *
 * A `Link` styled to work with `Message`'s `banner` appearance.
 */    function X(e) {
        var r = e.children, a = A(e, [ "children" ]);
        // @docs-props-type LinkPropsBase
                if (false) {}
        
        return n().createElement(M, a, r);
    }
    X.propTypes = W;
    /* harmony default export */    const z = X;
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const V = require("@splunk/react-ui/Box");
    var B =  e.n(V);
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const D = require("@splunk/react-ui/Clickable");
    var H =  e.n(D);
    // CONCATENATED MODULE: ./src/Message/TitleStyles.ts
    var F = T().p.withConfig({
        displayName: "TitleStyles__StyledTitle",
        componentId: "sc-6gbjha-0"
    })([ "", " ", "" ], I.mixins.reset("block"), (0, I.pick)({
        prisma: (0, R.css)([ "font-size:14px;line-height:20px;" ])
    }));
    // CONCATENATED MODULE: ./src/Message/MessageStyles.ts
    var G = T().div.withConfig({
        displayName: "MessageStyles__StyledContent",
        componentId: "eg66af-0"
    })([ "display:flex;align-items:baseline;flex:1;", ";", " ", ";" ], (0, I.pick)({
        prisma: (0, R.css)([ "padding:", " 0;color:", ";" ], I.variables.spacingMedium, I.variables.contentColorActive),
        enterprise: (0, R.css)([ "padding:12px 0;" ])
    }), (0, I.pick)({
        prisma: (0, R.css)([ "padding-left:", ";" ], I.variables.spacingXSmall),
        enterprise: (0, R.css)([ "padding-left:15px;" ])
    }), (function(e) {
        var r = e.$fillStyle;
        return r && (0, I.pick)({
            prisma: (0, R.css)([ "padding-left:", ";" ], I.variables.spacingMedium)
        });
    }));
    var J = T()(H()).withConfig({
        displayName: "MessageStyles__StyledRemove",
        componentId: "eg66af-1"
    })([ "", ";border:1px solid transparent;border-radius:", ";color:", ";", ";cursor:pointer;position:absolute;", " padding:8px;&:hover,&:focus{background:", ";color:", ";}&:active{", ";}", "" ], I.mixins.reset("flex"), (0, 
    I.pick)({
        prisma: "50%",
        enterprise: I.variables.borderRadius
    }), (0, I.pick)({
        prisma: I.variables.contentColorMuted,
        enterprise: {
            dark: I.variables.gray96,
            light: I.variables.gray45
        }
    }), (function(e) {
        var r = e.$appearance;
        return r === "banner" && (0, I.pick)({
            prisma: (0, R.css)([ "color:", ";" ], I.variables.contentColorInverted)
        });
    }), (0, I.pick)({
        prisma: (0, R.css)([ "top:3px;right:2px;" ]),
        enterprise: (0, R.css)([ "top:8px;right:8px;" ])
    }), (0, I.pick)({
        prisma: I.variables.interactiveColorOverlayHover,
        enterprise: {
            dark: I.variables.gray30,
            light: I.variables.gray92
        }
    }), (0, I.pick)({
        enterprise: {
            dark: I.variables.gray96,
            light: I.variables.linkColor
        },
        prisma: I.variables.contentColorActive
    }), (0, I.pick)({
        prisma: (0, R.css)([ "background:", ";" ], I.variables.interactiveColorOverlayActive),
        enterprise: (0, R.css)([ "box-shadow:", ";" ], I.variables.focusShadow)
    }), (0, I.pick)({
        prisma: (0, R.css)([ "&:focus{box-shadow:0 0 0 3px ", ";}" ], I.variables.focusColor)
    }));
    var K = T().span.withConfig({
        displayName: "MessageStyles__StyledIconWrapper",
        componentId: "eg66af-2"
    })([ "position:relative;min-width:", ";text-align:center;color:", ";", " ", "" ], (0, 
    I.pick)({
        prisma: I.variables.spacingXLarge,
        enterprise: "25px"
    }), I.variables.white, (0, I.pick)({
        enterprise: (0, R.css)([ "top:-2px;" ]),
        prisma: (0, R.css)([ "top:-1.8px;" ])
    }), (0, I.pickVariant)("$type", {
        info: (0, R.css)([ "color:", ";" ], (0, I.pick)({
            prisma: I.variables.contentColorMuted,
            enterprise: I.variables.infoColor
        })),
        success: (0, R.css)([ "color:", ";" ], I.variables.accentColorPositive),
        warning: (0, R.css)([ "color:", ";" ], I.variables.accentColorWarning),
        error: (0, R.css)([ "color:", ";" ], I.variables.accentColorNegative)
    }));
    var Q = T()(B()).withConfig({
        displayName: "MessageStyles__StyledBox",
        componentId: "eg66af-3"
    })([ "", ";align-items:baseline;position:relative;border-radius:", ";margin-bottom:", ";", " word-wrap:break-word;", " ", "" ], I.mixins.reset("flex"), (0, 
    I.pick)({
        prisma: I.variables.borderRadius,
        enterprise: "5px"
    }), I.variables.spacingSmall, (0, I.pickVariant)("$hasRemoveIcon", {
        true: {
            prisma: (0, R.css)([ "padding-right:", ";" ], I.variables.spacingXXXLarge),
            enterprise: (0, R.css)([ "padding-right:", ";" ], I.variables.spacingXXLarge)
        },
        false: (0, R.css)([ "padding-right:", ";" ], I.variables.spacingSmall)
    }), (function(e) {
        var r = e.$fillStyle;
        return r && (0, I.pick)({
            enterprise: (0, R.css)([ "& > ", "{left:", ";}", "" ], 
            /* sc-sel */
            K, I.variables.spacingXSmall, (0, I.pickVariant)("$type", {
                info: (0, R.css)([ "", "" ], (0, I.pick)({
                    light: (0, R.css)([ "background-color:", ";border:1px solid ", ";" ], I.variables.infoColorL50, I.variables.infoColor),
                    dark: (0, R.css)([ "background-color:", ";" ], I.mixins.colorWithAlpha(I.variables.infoColor, .5))
                })),
                success: (0, R.css)([ "", "" ], (0, I.pick)({
                    light: (0, R.css)([ "background-color:", ";border:1px solid ", ";" ], I.variables.successColorL50, I.variables.successColor),
                    dark: (0, R.css)([ "background-color:", ";" ], I.mixins.colorWithAlpha(I.variables.successColor, .5))
                })),
                warning: (0, R.css)([ "", "" ], (0, I.pick)({
                    light: (0, R.css)([ "background-color:", ";border:1px solid ", ";" ], I.variables.warningColorL50, I.variables.warningColor),
                    dark: (0, R.css)([ "background-color:", ";" ], I.mixins.colorWithAlpha(I.variables.warningColor, .5))
                })),
                error: (0, R.css)([ "", "" ], (0, I.pick)({
                    light: (0, R.css)([ "background-color:", ";border:1px solid ", ";" ], I.variables.errorColorL50, I.variables.errorColor),
                    dark: (0, R.css)([ "background-color:", ";" ], I.mixins.colorWithAlpha(I.variables.errorColor, .5))
                }))
            })),
            prisma: (0, R.css)([ "", ";box-shadow:", ";& > ", "{color:", ";}" ], (function(e) {
                var r = e.$type;
                var a = {
                    info: I.variables.contentColorActive,
                    success: I.variables.accentColorPositive,
                    warning: I.variables.accentColorWarning,
                    error: I.variables.accentColorNegative
                };
                var n = a[r];
                return (0, R.css)([ "background:linear-gradient( to right,", " 0,", " ", ",", " ", ",", " 100% );" ], n, n, I.variables.spacingXLarge, I.variables.backgroundColorPopup, I.variables.spacingXLarge, I.variables.backgroundColorPopup);
            }), I.variables.embossShadow, 
            /* sc-sel */
            K, I.variables.backgroundColorPopup)
        });
    }), (0, I.pick)({
        prisma: (0, R.css)([ "& ", "{color:", ";}" ], 
        /* sc-sel */
        F, (0, I.pickVariant)("$type", {
            info: I.variables.contentColorActive,
            warning: I.variables.accentColorWarning,
            error: I.variables.accentColorNegative,
            success: I.variables.accentColorPositive
        }))
    }));
    // CONCATENATED MODULE: ./src/Message/Title.tsx
    function U(e, r) {
        if (e == null) return {};
        var a = Y(e, r);
        var n, i;
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (i = 0; i < o.length; i++) {
                n = o[i];
                if (r.indexOf(n) >= 0) continue;
                if (!Object.prototype.propertyIsEnumerable.call(e, n)) continue;
                a[n] = e[n];
            }
        }
        return a;
    }
    function Y(e, r) {
        if (e == null) return {};
        var a = {};
        var n = Object.keys(e);
        var i, o;
        for (o = 0; o < n.length; o++) {
            i = n[o];
            if (r.indexOf(i) >= 0) continue;
            a[i] = e[i];
        }
        return a;
    }
    var Z = {
        children: o().node
    };
    /**
 * A title component for use in `Message`.
 */    function ee(e) {
        var r = e.children, a = U(e, [ "children" ]);
        // @docs-props-type TitleProps
                
        return n().createElement(F, a, r);
    }
    ee.propTypes = Z;
    /* harmony default export */    const re = ee;
    // CONCATENATED MODULE: ./src/Message/Message.tsx
    function ae() {
        ae = Object.assign || function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var a = arguments[r];
                for (var n in a) {
                    if (Object.prototype.hasOwnProperty.call(a, n)) {
                        e[n] = a[n];
                    }
                }
            }
            return e;
        };
        return ae.apply(this, arguments);
    }
    function ne(e, r) {
        if (e == null) return {};
        var a = ie(e, r);
        var n, i;
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (i = 0; i < o.length; i++) {
                n = o[i];
                if (r.indexOf(n) >= 0) continue;
                if (!Object.prototype.propertyIsEnumerable.call(e, n)) continue;
                a[n] = e[n];
            }
        }
        return a;
    }
    function ie(e, r) {
        if (e == null) return {};
        var a = {};
        var n = Object.keys(e);
        var i, o;
        for (o = 0; o < n.length; o++) {
            i = n[o];
            if (r.indexOf(i) >= 0) continue;
            a[i] = e[i];
        }
        return a;
    }
    var oe = {
        appearance: o().oneOf([ "default", "fill", "banner" ]),
        children: o().node,
        elementRef: o().oneOfType([ o().func, o().object ]),
        onRequestRemove: o().func,
        type: o().oneOf([ "info", "success", "warning", "error" ])
    };
    var te = Object.freeze({
        info: (0, P._)("Info"),
        warning: (0, P._)("Warning"),
        error: (0, P._)("Alert"),
        success: (0, P._)("Success")
    });
    var se = Object.freeze({
        info: C(),
        warning: S(),
        error: k(),
        success: w()
    });
    var ce = Object.freeze({
        info: f(),
        warning: m(),
        error: d(),
        success: s()
    });
    function le(e) {
        var r = e.appearance, i = r === void 0 ? "default" : r, o = e.children, t = e.type, s = t === void 0 ? "warning" : t, c = e.onRequestRemove, p = ne(e, [ "appearance", "children", "type", "onRequestRemove" ]);
        // @docs-props-type MessagePropsBase
                var v = (0, I.useSplunkTheme)(), d = v.isEnterprise;
        if (false) {}
 // double check this can only be button
                var b = (0, a.useCallback)((function(e) {
            c === null || c === void 0 ? void 0 : c(e);
        }), [ c ]);
        if (i === "banner") {
            
            return n().createElement(q(), ae({
                type: s,
                "data-test": "banner",
                onRequestClose: b
            }, p), o);
        }
        var f = d ? se : ce;
        var g = f[s];
        var m = i === "fill";
        var y = i === "default" || i === "fill";
        
        return n().createElement(Q, ae({
            $appearance: i,
            $fillStyle: m,
            $hasRemoveIcon: !!c,
            $type: s,
            "data-test-type": s,
            "data-test": "message"
        }, p), y &&  n().createElement(K, {
            $fillStyle: m,
            $type: s
        }, d ?  n().createElement(g, {
            height: "24px",
            width: "24px",
            hideDefaultTooltip: true,
            screenReaderText: null
        }) :  n().createElement(g, {
            variant: "filled"
        }),  n().createElement(E(), null, te[s])),  n().createElement(G, {
            "data-test": "content",
            $fillStyle: m
        }, o), c &&  n().createElement(J, {
            "data-test": "remove",
            onClick: b,
            $appearance: i
        }, d ?  n().createElement(u(), {
            height: "12px",
            hideDefaultTooltip: true,
            screenReaderText: null,
            width: "12px"
        }) :  n().createElement(l(), {
            height: "20px",
            width: "20px"
        })));
    }
    le.propTypes = oe;
    le.Title = re;
    le.Link = z;
    /* harmony default export */    const pe = le;
    // CONCATENATED MODULE: ./src/Message/index.ts
    module.exports = r;
    /******/})();
