// @splunk/react-ui/ModalLayer.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        default: () => /* reexport */ T
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/react-ui/Animation"
    const a = require("@splunk/react-ui/Animation");
    // CONCATENATED MODULE: external "@splunk/react-ui/Layer"
    const c = require("@splunk/react-ui/Layer");
    var l =  e.n(c);
    // CONCATENATED MODULE: external "styled-components"
    const u = require("styled-components");
    var p =  e.n(u);
    // CONCATENATED MODULE: external "react-spring"
    const s = require("react-spring");
    // CONCATENATED MODULE: external "@splunk/themes"
    const f = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/ModalLayer/ModalLayerStyles.ts
    var y = (0, u.css)([ "", ";position:fixed;top:0;right:0;bottom:0;left:0;" ], f.mixins.reset("block"));
    var d = p()(s.animated.div).withConfig({
        displayName: "ModalLayerStyles__StyledClickAwayOverlay",
        componentId: "sc-1sa3n1j-0"
    })([ "", ";background-color:", ";z-index:", ";" ], y, f.variables.backgroundColorScrim, f.variables.zindexModalBackdrop);
    // CONCATENATED MODULE: ./src/ModalLayer/ModalLayer.tsx
    function v(e) {
        "@babel/helpers - typeof";
        return v = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, v(e);
    }
    function b(e, t) {
        var r = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            t && (n = n.filter((function(t) {
                return Object.getOwnPropertyDescriptor(e, t).enumerable;
            }))), r.push.apply(r, n);
        }
        return r;
    }
    function m(e) {
        for (var t = 1; t < arguments.length; t++) {
            var r = null != arguments[t] ? arguments[t] : {};
            t % 2 ? b(Object(r), !0).forEach((function(t) {
                E(e, t, r[t]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : b(Object(r)).forEach((function(t) {
                Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
            }));
        }
        return e;
    }
    function O(e, t) {
        if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function");
    }
    function h(e, t) {
        for (var r = 0; r < t.length; r++) {
            var n = t[r];
            n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), 
            Object.defineProperty(e, q(n.key), n);
        }
    }
    function g(e, t, r) {
        return t && h(e.prototype, t), r && h(e, r), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function w(e, t) {
        if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), t && j(e, t);
    }
    function j(e, t) {
        return j = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, t) {
            return e.__proto__ = t, e;
        }, j(e, t);
    }
    function P(e) {
        var t = A();
        return function() {
            var r, n = _(e);
            if (t) {
                var o = _(this).constructor;
                r = Reflect.construct(n, arguments, o);
            } else r = n.apply(this, arguments);
            return S(this, r);
        };
    }
    function S(e, t) {
        if (t && ("object" == v(t) || "function" == typeof t)) return t;
        if (void 0 !== t) throw new TypeError("Derived constructors may only return object or undefined");
        return k(e);
    }
    function k(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function A() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return (A = function t() {
            return !!e;
        })();
    }
    function _(e) {
        return _ = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, _(e);
    }
    function E(e, t, r) {
        return (t = q(t)) in e ? Object.defineProperty(e, t, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = r, e;
    }
    function q(e) {
        var t = M(e, "string");
        return "symbol" == v(t) ? t : t + "";
    }
    function M(e, t) {
        if ("object" != v(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var n = r.call(e, t || "default");
            if ("object" != v(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    var C = {
        childrenInAnimation: i().bool,
        getDefaultMotionStyle: i().func,
        getMotionStyle: i().func,
        onRequestClose: i().func,
        open: i().bool,
        renderModal: i().func.isRequired,
        scrim: i().oneOf([ "visible", "hidden", "interactive" ])
    };
    var R = {
        childrenInAnimation: false,
        open: false,
        scrim: "interactive"
    };
    var D =  function(e) {
        w(r, e);
        var t = P(r);
        g(r, null, [ {
            key: "getDerivedStateFromProps",
            value: function e(t, r) {
                if (t.open !== r.prevOpen) {
                    return {
                        animating: true,
                        prevOpen: t.open
                    };
                }
                return null;
            }
        } ]);
        function r(e) {
            var o;
            O(this, r);
            o = t.call(this, e);
            E(k(o), "allowAnimationUpdates", true);
            E(k(o), "handleClickAway", (function(e) {
                var t, r;
                (t = (r = o.props).onRequestClose) === null || t === void 0 ? void 0 : t.call(r, {
                    reason: "clickAway",
                    event: e
                });
            }));
            E(k(o), "handleAnimationEnd", (function() {
                if (o.allowAnimationUpdates) {
                    o.setState({
                        animating: false
                    });
                }
            }));
            E(k(o), "renderLayer", (function() {
                var e = o.props, t = e.getDefaultMotionStyle, r = e.getMotionStyle, i = e.renderModal, c = e.scrim;
                var l = o.props.open || o.state.animating;
                var u = l && c !== "hidden";
                
                return n().createElement(a.Spring, {
                    from: m(m({}, t === null || t === void 0 ? void 0 : t()), {}, {
                        overlayOpacity: 0,
                        opacity: 0
                    }),
                    to: m(m({}, r === null || r === void 0 ? void 0 : r()), {}, {
                        overlayOpacity: o.props.open ? 1 : 0,
                        opacity: o.props.open ? 1 : 0
                    }),
                    onRest: o.handleAnimationEnd
                }, (function(e) {
                    
                    return n().createElement("div", null, /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
                    i(e), u &&  n().createElement(d, {
                        "data-test": "modal-overlay",
                        onMouseDown: c === "interactive" ? o.handleClickAway : undefined,
                        key: "clickAway",
                        style: {
                            /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
                            opacity: e.overlayOpacity
                        }
                    }));
                }));
            }));
            o.state = {
                animating: false,
                prevOpen: e.open
            };
            return o;
        }
        g(r, [ {
            key: "componentWillUnmount",
            value: function e() {
                this.allowAnimationUpdates = false;
            }
 /* eslint-disable jsx-a11y/no-static-element-interactions */        }, {
            key: "render",
            /* eslint-enable jsx-a11y/no-static-element-interactions */ value: function e() {
                var t = this.props.open || this.state.animating || this.props.childrenInAnimation;
                
                return n().createElement(l(), {
                    open: t,
                    closeReasons: [ "escapeKey" ],
                    onRequestClose: this.props.onRequestClose
                }, t && this.renderLayer());
            }
        } ]);
        return r;
    }(r.Component);
    E(D, "propTypes", C);
    E(D, "defaultProps", R);
    /* harmony default export */    const T = D;
    // CONCATENATED MODULE: ./src/ModalLayer/index.ts
    module.exports = t;
    /******/})();
