// @splunk/react-ui/Monogram.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ N,
        getInitials: () => /* reexport */ $
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var n =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const i = require("prop-types");
    var o =  e.n(i);
    // CONCATENATED MODULE: external "lodash/omit"
    const a = require("lodash/omit");
    var c =  e.n(a);
    // CONCATENATED MODULE: external "styled-components"
    const s = require("styled-components");
    var l =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const u = require("@splunk/react-ui/Clickable");
    var f =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/themes"
    const p = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Monogram/MonogramStyles.ts
    // Thanks: https://gist.github.com/hyamamoto/fd435505d29ebfa3d9716fd2be8d42f0
    var d = function e(r) {
        var t = 0;
        for (var n = 0; n < r.length; ) {
            t = (t << 5) - t + r.charCodeAt(n++) | 0;
 // eslint-disable-line no-bitwise, no-plusplus
                }
        return t;
    };
    var b = [ "#006d9c", "#ec9960", "#af575a", "#62b3b2", "#4fa484", "#f8be34", "#5a4575", "#708794", "#294e70", "#b6c75a", "#5cc05c", "#007abd", "#dc4e41", "#f1813f", "#f8be34", "#53a051", "#006d9c", "#77d6d8", "#f589ad", "#6a2c5d" ];
    var m = (0, s.css)([ "", ";align-items:center;justify-content:center;border-radius:50%;font-weight:normal;", " background-color:", ";", ";& + &{margin-left:4px;}" ], p.mixins.reset("inline-flex"), (0, 
    p.pick)({
        prisma: (0, s.css)([ "color:", ";" ], p.variables.contentColorActive)
    }), (function(e) {
        var r = e.$backgroundColor, t = e.$initials;
        if (r === "theme") {
            return (0, p.pick)({
                enterprise: {
                    dark: p.variables.gray45,
                    light: p.variables.gray92
                },
                prisma: p.variables.neutral100
            });
        }
        if (r === "auto") {
            return b[d(t) % b.length];
        }
        return r;
    }), (function(e) {
        var r = e.$size;
        if (typeof r === "number") {
            return (0, s.css)([ "height:", "px;width:", "px;" ], r, r);
        }
        if (typeof r === "string") {
            return (0, p.pickVariant)("$size", {
                small: (0, s.css)([ "font-size:", ";", "" ], p.variables.fontSize, (0, p.pick)({
                    enterprise: (0, s.css)([ "height:32px;width:32px;" ]),
                    prisma: (0, s.css)([ "height:24px;width:24px;" ])
                })),
                medium: (0, s.css)([ "height:48px;width:48px;font-size:", ";" ], p.variables.fontSize),
                large: (0, s.css)([ "height:80px;width:80px;font-size:", ";" ], p.variables.fontSizeXXLarge),
                xlarge: (0, s.css)([ "width:144px;height:144px;font-size:36px;font-weight:700;" ])
            });
        }
        return (0, s.css)([ "height:48px;width:48px;font-size:", ";" ], p.variables.fontSize);
    }));
    var v = l().div.withConfig({
        displayName: "MonogramStyles__Styled",
        componentId: "r5ud6k-0"
    })([ "", "" ], m);
    var y = l()(f()).withConfig({
        displayName: "MonogramStyles__StyledClickableComponent",
        componentId: "r5ud6k-1"
    })([ "", "" ], m);
    var g = l()(y).withConfig({
        displayName: "MonogramStyles__StyledClickable",
        componentId: "r5ud6k-2"
    })([ "cursor:pointer;position:relative;:focus,:hover{", "}:focus{box-shadow:", ";z-index:1;}:active{", "}" ], (0, 
    p.pick)({
        enterprise: (0, s.css)([ "box-shadow:", ";" ], p.variables.focusShadowInset),
        prisma: (0, s.css)([ "border:1px solid transparent;background-color:", ";" ], p.variables.neutral200)
    }), (0, p.pick)({
        enterprise: p.variables.focusShadowInset,
        prisma: p.variables.focusShadow
    }), (0, p.pick)({
        enterprise: (0, s.css)([ "opacity:0.85;" ]),
        prisma: (0, s.css)([ "border:1px solid transparent;background-color:", ";opacity:1;" ], p.variables.neutral200)
    }));
    var h = l().span.withConfig({
        displayName: "MonogramStyles__Initials",
        componentId: "r5ud6k-3"
    })([ "opacity:0.8;text-transform:capitalize;" ]);
    // CONCATENATED MODULE: ./src/Monogram/Monogram.tsx
    function O(e, r) {
        var t = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            if (r) n = n.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }));
            t.push.apply(t, n);
        }
        return t;
    }
    function k(e) {
        for (var r = 1; r < arguments.length; r++) {
            var t = arguments[r] != null ? arguments[r] : {};
            if (r % 2) {
                O(Object(t), true).forEach((function(r) {
                    w(e, r, t[r]);
                }));
            } else if (Object.getOwnPropertyDescriptors) {
                Object.defineProperties(e, Object.getOwnPropertyDescriptors(t));
            } else {
                O(Object(t)).forEach((function(r) {
                    Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
                }));
            }
        }
        return e;
    }
    function w(e, r, t) {
        if (r in e) {
            Object.defineProperty(e, r, {
                value: t,
                enumerable: true,
                configurable: true,
                writable: true
            });
        } else {
            e[r] = t;
        }
        return e;
    }
    function j() {
        j = Object.assign || function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    if (Object.prototype.hasOwnProperty.call(t, n)) {
                        e[n] = t[n];
                    }
                }
            }
            return e;
        };
        return j.apply(this, arguments);
    }
    function x(e, r) {
        return I(e) || P(e, r) || C(e, r) || S();
    }
    function S() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function C(e, r) {
        if (!e) return;
        if (typeof e === "string") return z(e, r);
        var t = Object.prototype.toString.call(e).slice(8, -1);
        if (t === "Object" && e.constructor) t = e.constructor.name;
        if (t === "Map" || t === "Set") return Array.from(e);
        if (t === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t)) return z(e, r);
    }
    function z(e, r) {
        if (r == null || r > e.length) r = e.length;
        for (var t = 0, n = new Array(r); t < r; t++) {
            n[t] = e[t];
        }
        return n;
    }
    function P(e, r) {
        if (typeof Symbol === "undefined" || !(Symbol.iterator in Object(e))) return;
        var t = [];
        var n = true;
        var i = false;
        var o = undefined;
        try {
            for (var a = e[Symbol.iterator](), c; !(n = (c = a.next()).done); n = true) {
                t.push(c.value);
                if (r && t.length === r) break;
            }
        } catch (e) {
            i = true;
            o = e;
        } finally {
            try {
                if (!n && a["return"] != null) a["return"]();
            } finally {
                if (i) throw o;
            }
        }
        return t;
    }
    function I(e) {
        if (Array.isArray(e)) return e;
    }
    function _(e, r) {
        if (e == null) return {};
        var t = E(e, r);
        var n, i;
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (i = 0; i < o.length; i++) {
                n = o[i];
                if (r.indexOf(n) >= 0) continue;
                if (!Object.prototype.propertyIsEnumerable.call(e, n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    function E(e, r) {
        if (e == null) return {};
        var t = {};
        var n = Object.keys(e);
        var i, o;
        for (o = 0; o < n.length; o++) {
            i = n[o];
            if (r.indexOf(i) >= 0) continue;
            t[i] = e[i];
        }
        return t;
    }
    var A = {
        backgroundColor: o().string,
        elementRef: o().oneOfType([ o().func, o().object ]),
        initials: o().string.isRequired,
        name: o().string,
        onClick: o().func,
        size: o().oneOfType([ o().oneOf([ "small", "medium", "large", "xlarge" ]), o().number ])
    };
    function M(e) {
        return !!e.onClick;
    }
    /**
 * Returns a suitable set of initials for a name. Uses the first character of each
 * name segment and omits middle segments if the segment count is greater than three.
 * @param {string} name - The full name.
 * @returns {string} Limited to three characters. Empty if `name` is empty.
 * @public
 */    function $(e) {
        var r = (e.match(/[^\s-]+/g) || []).map((function(e) {
            return e[0];
        })).join("");
        return r.length > 3 ? "".concat(r[0]).concat(r[r.length - 1]) : r;
    }
    function q(e) {
        var r = e.backgroundColor, t = e.initials, n = e.size, i = _(e, [ "backgroundColor", "initials", "size" ]);
        var o = {
            "data-test": "monogram",
            $backgroundColor: r,
            $initials: t,
            $size: n
        };
        return [ o, i ];
    }
    function R(e) {
        var r = q(e), i = x(r, 2), o = i[0], a = i[1];
        var s = a.children, l = a.elementRef, u = a.name, f = a.onClick, p = _(a, [ "children", "elementRef", "name", "onClick" ]);
        var d = (0, t.useCallback)((function(e) {
            f(e, {
                name: u
            });
        }), [ f, u ]);
        
        return n().createElement(g, j({
            elementRef: l,
            onClick: d
        }, o, c()(p, Object.keys(A))), s);
    }
    function T(e) {
        var r = q(e), t = x(r, 2), i = t[0], o = t[1];
        var a = o.children, s = o.elementRef, l = _(o, [ "children", "elementRef" ]);
        
        return n().createElement(v, j({
            ref: s
        }, i, c()(l, Object.keys(A))), a);
    }
    function D(e) {
        var r = e.backgroundColor, t = r === void 0 ? "theme" : r, i = e.initials, o = e.size, a = o === void 0 ? "medium" : o, c = _(e, [ "backgroundColor", "initials", "size" ]);
        // @docs-props-type MonogramPropsBase
                if (false) {}
        var s =  n().createElement(h, {
            "data-test": "initials"
        }, i);
        var l = k({
            backgroundColor: t,
            size: a,
            initials: i
        }, c);
        return M(l) ?  n().createElement(R, l, s) :  n().createElement(T, l, s);
    }
    D.propTypes = A;
    /* harmony default export */    const N = D;
    // CONCATENATED MODULE: ./src/Monogram/index.ts
    module.exports = r;
    /******/})();
