// @splunk/react-ui/Number.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        default: () => /* reexport */ pe
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var a =  e.n(o);
    // CONCATENATED MODULE: external "lodash/has"
    const i = require("lodash/has");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "lodash/isFinite"
    const s = require("lodash/isFinite");
    var u =  e.n(s);
    // CONCATENATED MODULE: external "lodash/omit"
    const c = require("lodash/omit");
    var p =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/themes"
    const d = require("@splunk/themes");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const f = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/ui-utils/keyboard"
    const v = require("@splunk/ui-utils/keyboard");
    // CONCATENATED MODULE: external "@splunk/react-icons/SVGEnterprise"
    const m = require("@splunk/react-icons/SVGEnterprise");
    var b =  e.n(m);
    // CONCATENATED MODULE: ./src/Number/IncrementIcon.tsx
    function h() {
        return h = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, h.apply(null, arguments)
        /* eslint-disable max-len */;
    }
    function y(e) {
        
        return n().createElement(b(), h({
            viewBox: "0 0 933 600"
        }, e),  n().createElement("path", {
            fill: "currentColor",
            d: "M0 466.438L466.438 0l466.44 467.466-134.59 133.56-331.85-331.848-331.85 331.85L0 466.438z"
        }));
    }
    // CONCATENATED MODULE: external "styled-components"
    const g = require("styled-components");
    var x =  e.n(g);
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const k = require("@splunk/react-ui/Button");
    var w =  e.n(k);
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const O = require("@splunk/react-ui/Clickable");
    var S =  e.n(O);
    // CONCATENATED MODULE: external "@splunk/react-ui/Text"
    const j = require("@splunk/react-ui/Text");
    var C =  e.n(j);
    // CONCATENATED MODULE: ./src/Number/NumberStyles.ts
    var P = x()(C()).withConfig({
        displayName: "NumberStyles__StyledText",
        componentId: "sc-2ekj9s-0"
    })([ "&[data-inline]{", "}" ], (0, d.pick)({
        enterprise: {
            comfortable: (0, g.css)([ "width:100px;" ]),
            compact: (0, g.css)([ "width:80px;" ])
        },
        prisma: (0, g.css)([ "width:100px;" ])
    }));
    var T = x().div.withConfig({
        displayName: "NumberStyles__StyledControls",
        componentId: "sc-2ekj9s-1"
    })([ "width:", ";height:100%;", " ", "" ], (0, d.pick)({
        enterprise: {
            comfortable: "20px",
            compact: "16px"
        },
        prisma: {
            comfortable: "24px",
            compact: "24px"
        }
    }), (0, d.pick)({
        enterprise: (0, g.css)([ "transform:translateY(-1px);" ]),
        prisma: (0, g.css)([ "transform:translateX(1px);" ])
    }), (function(e) {
        var t = e.$isAppend;
        return t ? (0, g.css)([ "margin-right:8px;", "" ], (0, d.pick)({
            prisma: (0, g.css)([ "transform:translateX(-1px);" ])
        })) : (0, g.css)([ "margin-left:8px;" ]);
    }));
    var E = (0, g.css)([ "position:relative;padding:3px;font-size:inherit;min-width:0;min-height:0;height:", ";width:100%;" ], (0, 
    d.pick)({
        enterprise: "calc(50% + 1.5px)",
        prisma: "50%"
    }));
    var _ = x()(w()).withConfig({
        displayName: "NumberStyles__StyledPlusButton",
        componentId: "sc-2ekj9s-2"
    })([ "", ";top:0;border-bottom-left-radius:0;border-bottom-right-radius:0;", " &:not([disabled]){&,&:hover{", ";}}" ], E, (function(e) {
        var t = e.append;
        return t && (0, d.pick)({
            enterprise: {
                light: (0, g.css)([ "border-right:1px solid ", ";" ], d.variables.gray60),
                dark: (0, g.css)([ "border-right:", ";" ], d.variables.border)
            }
        });
    }), (0, d.pick)({
        enterprise: {
            dark: (0, g.css)([ "border-color:", ";" ], d.variables.gray20)
        }
    }));
    var V = x()(w()).withConfig({
        displayName: "NumberStyles__StyledMinusButton",
        componentId: "sc-2ekj9s-3"
    })([ "", ";bottom:1px;border-top-left-radius:0;border-top-right-radius:0;", " &:not([disabled]){&,&:hover{", ";}}" ], E, (function(e) {
        var t = e.append;
        return t && (0, d.pick)({
            enterprise: {
                light: (0, g.css)([ "border-right:1px solid ", ";" ], d.variables.gray60),
                dark: (0, g.css)([ "border-right:", ";" ], d.variables.border)
            }
        });
    }), (0, d.pick)({
        enterprise: {
            dark: (0, g.css)([ "border-color:", ";" ], d.variables.gray20)
        }
    }));
    var B = x()(S()).withConfig({
        displayName: "NumberStyles__prismaButton",
        componentId: "sc-2ekj9s-4"
    })([ "", ";display:flex;align-items:center;justify-content:center;", " &:hover:not([disabled]){", "}&:active:not([disabled]){", "}" ], E, (0, 
    d.pick)({
        prisma: (0, g.css)([ "color:", ";" ], d.variables.contentColorDefault)
    }), (0, d.pick)({
        prisma: (0, g.css)([ "background-color:", ";color:", ";" ], d.variables.interactiveColorOverlayHover, d.variables.contentColorActive)
    }), (0, d.pick)({
        prisma: (0, g.css)([ "background-color:", ";color:", ";" ], d.variables.interactiveColorOverlayActive, d.variables.contentColorActive)
    }));
    var D = x()(B).withConfig({
        displayName: "NumberStyles__StyledPrismaPlusButton",
        componentId: "sc-2ekj9s-5"
    })([ "border-top-right-radius:4px;" ]);
    var N = x()(B).withConfig({
        displayName: "NumberStyles__StyledPrismaMinusButton",
        componentId: "sc-2ekj9s-6"
    })([ "border-bottom-right-radius:4px;padding-bottom:5px;" ]);
    // CONCATENATED MODULE: external "lodash/isString"
    const I = require("lodash/isString");
    var q =  e.n(I);
    // CONCATENATED MODULE: ./src/Number/utils.ts
    var K = function e(t) {
        // A more robust and expandable solution would be to use formatToParts() once IE requirements are dropped.
        var r = new Intl.NumberFormat(t.replace("_", "-")).format(1.2);
        return r.indexOf(",") > -1 ? "," : ".";
    };
    var A = function e(t) {
        var r = t.val, n = t.min, o = t.max, a = t.roundTo;
        var i = r;
        if (!u()(r)) {
            return i;
        }
        if (n !== undefined && u()(n)) {
            i = Math.max(n, i);
        }
        if (o !== undefined && u()(o)) {
            i = Math.min(o, i);
        }
        var l = Math.pow(10, a);
        return Math.round(i * l) / l;
    };
    var M = function e(t) {
        var r = t.str, n = t.locale, o = t.min, a = t.max, i = t.roundTo;
        var l = r;
        if (!q()(l) || l === "") {
            return undefined;
        }
        if (K(n) === ".") {
            // remove leading non-numeric characters.
            l = l.replace(/^[^\d.-]/, "");
        } else {
            // remove leading non-numeric characters and replace comma with period
            l = l.replace(/^[^\d,-]/, "").replace(",", ".");
        }
        var s = A({
            val: parseFloat(l),
            min: o,
            max: a,
            roundTo: i
        });
        if (u()(s)) {
            return s;
        }
        return undefined;
    };
    var R = function e(t, r) {
        if (t === undefined || !u()(t)) {
            return "";
        }
        if (K(r) === ".") {
            return t.toString();
        }
        // comma separated case
                return t.toString().replace(".", ",");
    };
    var U = function e(t) {
        var r = t.value, n = t.max;
        return r !== undefined && u()(r) && n !== undefined && u()(n) && r >= n;
    };
    var F = function e(t) {
        var r = t.value, n = t.min;
        return r !== undefined && u()(r) && n !== undefined && u()(n) && r <= n;
    };
    var z = function e(t, r) {
        if (false) {}
    };
    // CONCATENATED MODULE: ./src/Number/Number.tsx
    function X(e) {
        "@babel/helpers - typeof";
        return X = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, X(e);
    }
    function $() {
        return $ = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, $.apply(null, arguments);
    }
    function L(e, t) {
        var r = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            t && (n = n.filter((function(t) {
                return Object.getOwnPropertyDescriptor(e, t).enumerable;
            }))), r.push.apply(r, n);
        }
        return r;
    }
    function G(e) {
        for (var t = 1; t < arguments.length; t++) {
            var r = null != arguments[t] ? arguments[t] : {};
            t % 2 ? L(Object(r), !0).forEach((function(t) {
                oe(e, t, r[t]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : L(Object(r)).forEach((function(t) {
                Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
            }));
        }
        return e;
    }
    function H(e, t) {
        if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function");
    }
    function Y(e, t) {
        for (var r = 0; r < t.length; r++) {
            var n = t[r];
            n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), 
            Object.defineProperty(e, ae(n.key), n);
        }
    }
    function J(e, t, r) {
        return t && Y(e.prototype, t), r && Y(e, r), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function Q(e, t) {
        if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), t && W(e, t);
    }
    function W(e, t) {
        return W = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, t) {
            return e.__proto__ = t, e;
        }, W(e, t);
    }
    function Z(e) {
        var t = re();
        return function() {
            var r, n = ne(e);
            if (t) {
                var o = ne(this).constructor;
                r = Reflect.construct(n, arguments, o);
            } else r = n.apply(this, arguments);
            return ee(this, r);
        };
    }
    function ee(e, t) {
        if (t && ("object" == X(t) || "function" == typeof t)) return t;
        if (void 0 !== t) throw new TypeError("Derived constructors may only return object or undefined");
        return te(e);
    }
    function te(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function re() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return (re = function t() {
            return !!e;
        })();
    }
    function ne(e) {
        return ne = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, ne(e);
    }
    function oe(e, t, r) {
        return (t = ae(t)) in e ? Object.defineProperty(e, t, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = r, e;
    }
    function ae(e) {
        var t = ie(e, "string");
        return "symbol" == X(t) ? t : t + "";
    }
    function ie(e, t) {
        if ("object" != X(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var n = r.call(e, t || "default");
            if ("object" != X(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    /** @public */
    /** @public */
    /** @public */    var le = {
        append: a().bool,
        children: a().node,
        defaultValue: a().number,
        describedBy: a().string,
        disabled: a().bool,
        elementRef: a().oneOfType([ a().func, a().object ]),
        inputRef: a().oneOfType([ a().func, a().object ]),
        error: a().bool,
        hideStepButtons: a().bool,
        inline: a().bool,
        inputId: a().string,
        labelledBy: a().string,
        locale: a().string,
        max: a().number,
        min: a().number,
        name: a().string,
        onBlur: a().func,
        onChange: a().func,
        onFocus: a().func,
        endAdornment: a().node,
        startAdornment: a().node,
        onKeyDown: a().func,
        onKeyUp: a().func,
        onSelect: a().func,
        placeholder: a().string,
        prepend: a().bool,
        roundTo: a().number,
        step: a().number,
        /** @private */
        splunkTheme: a().object,
        useSyntheticPlaceholder: a().bool,
        value: a().number
    };
    var se = {
        append: false,
        disabled: false,
        error: false,
        hideStepButtons: false,
        inline: false,
        locale: "en-US",
        placeholder: "",
        prepend: false,
        roundTo: 5,
        step: 1
    };
    var ue =  function(e) {
        Q(r, e);
        var t = Z(r);
        J(r, null, [ {
            key: "getDerivedStateFromProps",
            // @docs-props-type NumberPropsBase
            value: function e(t, r) {
                if (l()(t, "value") && t.value !== r.prevValueProp) {
                    var n = t.locale, o = t.min, a = t.max, i = t.roundTo;
                    var s = r.stringValue;
                    var u = M({
                        str: s,
                        locale: n,
                        min: o,
                        max: a,
                        roundTo: i
                    });
                    return t.value !== u ? {
                        prevValueProp: t.value,
                        stringValue: R(t.value, t.locale)
                    } : {
                        prevValueProp: t.value
                    };
                }
                return null;
            }
        } ]);
        function r(e) {
            var n;
            H(this, r);
            n = t.call(this, e);
            oe(te(n), "controlledExternally", void 0);
            oe(te(n), "text", null);
            oe(te(n), "handleInputBlur", (function(e, t) {
                var r, o;
                n.updateString();
                (r = (o = n.props).onBlur) === null || r === void 0 ? void 0 : r.call(o, e, t);
            }));
            oe(te(n), "handleInputChange", (function(e, t) {
                var r = t.value;
                var o = n.props, a = o.locale, i = o.min, l = o.max, s = o.roundTo, u = o.name, c = o.onChange;
                var p = M({
                    str: r,
                    locale: a,
                    min: i,
                    max: l,
                    roundTo: s
                });
                n.setState({
                    value: n.isControlled() ? undefined : p,
                    stringValue: r
                }, (function() {
                    if (n.getValue() !== p || !n.isControlled()) {
                        c === null || c === void 0 ? void 0 : c(e, {
                            value: p,
                            name: u,
                            reason: "input"
                        });
                    }
                }));
            }));
            oe(te(n), "handleKeyDown", (function(e) {
                var t, r;
                if (!e.metaKey && !e.ctrlKey && !e.altKey && (0, v.addsCharacter)(e.nativeEvent) !== false) {
                    // only allow numbers, negative and decimals
                    if (!(0, v.isNumeric)(e.nativeEvent, {
                        locale: n.props.locale
                    })) {
                        e.preventDefault();
                    }
                    // if minus is not allowed or there is already one, prevent input
                                        var o = n.props.min !== undefined && u()(n.props.min) && n.props.min >= 0;
                    if ((0, v.isMinus)(e.nativeEvent) && (o || n.state.stringValue.indexOf("-") >= 0)) {
                        e.preventDefault();
                    }
                    // if decimal is not allowed or there is already one, prevent input
                                        if ((0, v.isDecimal)(e.nativeEvent, {
                        locale: n.props.locale
                    }) && (n.props.roundTo <= 0 || n.state.stringValue.indexOf(K(n.props.locale)) >= 0)) {
                        e.preventDefault();
                    }
                }
                // Arrow up and down will increment
                                if ((0, v.keycode)(e.nativeEvent) === "up") {
                    n.increment(e, n.props.step);
                    e.preventDefault();
                } else if ((0, v.keycode)(e.nativeEvent) === "down") {
                    n.increment(e, -n.props.step);
                    e.preventDefault();
                }
                (t = (r = n.props).onKeyDown) === null || t === void 0 ? void 0 : t.call(r, e);
            }));
            oe(te(n), "handleKeyUp", (function(e) {
                var t, r;
                if ((0, v.keycode)(e.nativeEvent) === "enter") {
                    n.updateString();
                }
                (t = (r = n.props).onKeyUp) === null || t === void 0 ? void 0 : t.call(r, e);
            }));
            oe(te(n), "handleIncrement", (function(e) {
                n.increment(e, n.props.step);
            }));
            oe(te(n), "handleDecrement", (function(e) {
                n.increment(e, -n.props.step);
            }));
            n.controlledExternally = l()(e, "value");
            var o = n.isControlled() ? e.value : e.defaultValue;
            n.state = {
                // value stores internal state, only in uncontrolled mode
                value: n.isControlled() ? undefined : o,
                // prevValueProp is used to keep track of changes to the value prop, only in controlled mode
                prevValueProp: n.isControlled() ? o : undefined,
                // eslint-disable-line react/no-unused-state
                // stringValue stores the current contents of the input text box - it might be out of range, invalid, ...
                stringValue: R(o, e.locale)
            };
            if (false) {}
            if (l()(e, "useSyntheticPlaceholder")) {
                // eslint-disable-next-line no-console
                console.warn("The 'Number' prop 'useSyntheticPlaceholder' has been marked for deprecation.");
            }
            if (false) {}
            z(n.props.append, n.props.prepend);
            return n;
        }
        J(r, [ {
            key: "componentDidUpdate",
            value: function e(t) {
                if (false) {}
                if (false) {}
                z(this.props.append, this.props.prepend);
            }
        }, {
            key: "getValue",
            value: function e() {
                return this.isControlled() ? this.props.value : this.state.value;
            }
        }, {
            key: "setValue",
            value: function e(t, r) {
                var n = this;
                var o = this.props, a = o.locale, i = o.onChange, l = o.name;
                this.setState({
                    value: this.isControlled() ? undefined : r,
                    stringValue: R(r, a)
                }, (function() {
                    if (n.getValue() !== r || !n.isControlled()) {
                        i === null || i === void 0 ? void 0 : i(t, {
                            value: r,
                            name: l,
                            reason: "stepButton"
                        });
                    }
                }));
            }
        }, {
            key: "focus",
            /**
     * Place focus on the input.
     */
            // eslint-disable-next-line react/no-unused-class-component-methods
            value: function e(t) {
                if (this.text) {
                    this.text.focus(t);
                }
            }
        }, {
            key: "isControlled",
            value: function e() {
                return this.controlledExternally;
            }
        }, {
            key: "updateString",
            value: function e() {
                var t = this.getValue();
                this.setState({
                    stringValue: R(t, this.props.locale)
                });
            }
        }, {
            key: "increment",
            value: function e(t, r) {
                var n = this.props, o = n.locale, a = n.min, i = n.max, l = n.roundTo, s = n.defaultValue, c = n.step;
                var p = this.state.stringValue;
                var d = M({
                    str: p,
                    locale: o,
                    min: a,
                    max: i,
                    roundTo: l
                });
                var f;
                // increment appropriate number
                                if (d !== undefined && u()(d)) {
                    // increment the entered value
                    f = d + r;
                } else if (s !== undefined && u()(s)) {
                    // increment the defaultValue
                    f = s + r;
                } else if (r > 0) {
                    // increment up to max or step value
                    f = i || c;
                } else {
                    // increment down to min or 0
                    f = a || 0;
 // set to min or 0
                                }
                this.setValue(t, A({
                    val: f,
                    min: a,
                    max: i,
                    roundTo: l
                }));
            }
        }, {
            key: "render",
            value: function e() {
                var t = this;
                var r = this.props, o = r.append, a = r.describedBy, i = r.disabled, l = r.elementRef, s = r.error, u = r.hideStepButtons, c = r.inline, d = r.value, v = d === void 0 ? this.state.value : d, m = r.max, b = r.min, h = r.prepend, g = r.splunkTheme, x = r.endAdornment, k = r.startAdornment;
                var w = p()(this.props, [ "defaultValue", "hideStepButtons", "max", "min", "roundTo", "spellCheck", "step" ]);
                var O = g.isPrisma;
                var S = O ? {} : {
                    append: o,
                    prepend: !o,
                    error: s,
                    inline: false
                };
                var j = U({
                    value: v,
                    max: m
                });
                var C = F({
                    value: v,
                    min: b
                });
                var E =  n().createElement(y, {
                    width: "6px",
                    height: "4px",
                    screenReaderText: (0, f._)("Increment")
                });
                var B = G({
                    "data-test": "increment",
                    disabled: i || j,
                    onClick: this.handleIncrement,
                    tabIndex: -1
                }, S);
                var I = O ?  n().createElement(D, B, E) :  n().createElement(_, B, E);
                var q =  n().createElement(y, {
                    width: "6px",
                    height: "4px",
                    screenReaderText: (0, f._)("Decrement"),
                    style: {
                        transform: "rotateX(180deg)"
                    }
                });
                var K = G({
                    "data-test": "decrement",
                    disabled: i || C,
                    onClick: this.handleDecrement,
                    tabIndex: -1
                }, S);
                var A = O ?  n().createElement(N, K, q) :  n().createElement(V, K, q);
                var M = o && !h;
                var R = (!i || O) && M && !u &&  n().createElement(T, {
                    $isAppend: M
                }, I, A);
                var z = (!i || O) && !M && !u &&  n().createElement(T, {
                    $isAppend: M
                }, I, A);
                
                return n().createElement(P, $({
                    autoCapitalize: "off",
                    autoComplete: "off",
                    autoCorrect: "off",
                    spellCheck: false,
                    "data-test": "number",
                    elementRef: l,
                    ref: function e(r) {
                        t.text = r;
                    }
                }, w, {
                    inline: c,
                    onChange: this.handleInputChange,
                    onBlur: this.handleInputBlur,
                    onKeyDown: this.handleKeyDown,
                    onKeyUp: this.handleKeyUp,
                    value: this.state.stringValue,
                    "aria-describedby": a,
                    error: s,
                    "data-test-value": v,
                    startAdornment: k || R,
                    endAdornment: x || z,
                    role: "spinbutton",
                    "aria-valuenow": this.state.value,
                    "aria-valuemin": b,
                    "aria-valuemax": m,
                    "aria-invalid": s
                }));
            }
        } ]);
        return r;
    }(r.Component);
    oe(ue, "propTypes", le);
    oe(ue, "defaultProps", se);
    oe(ue, "displayName", "Number");
    var ce = (0, d.withSplunkTheme)(ue);
    ce.propTypes = ue.propTypes;
    /* harmony default export */    const pe = ce;
    // only for styled components using Number
        // CONCATENATED MODULE: ./src/Number/index.ts
    module.exports = t;
    /******/})();
