// @splunk/react-ui/Paginator.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var a = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(a, {
                a
            });
            /******/            return a;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, a) => {
            /******/ for (var r in a) {
                /******/ if (e.o(a, r) && !e.o(n, r)) {
                    /******/ Object.defineProperty(n, r, {
                        enumerable: true,
                        get: a[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        Compact: () => /* reexport */ W,
        PageControl: () => /* reexport */ oe,
        default: () => /* reexport */ ke
    });
    // CONCATENATED MODULE: external "react"
    const a = require("react");
    var r =  e.n(a);
    // CONCATENATED MODULE: external "prop-types"
    const t = require("prop-types");
    var l =  e.n(t);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const i = require("@splunk/react-ui/Box");
    var o =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/ui-utils/format"
    const s = require("@splunk/ui-utils/format");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const c = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/ChevronLeft"
    const u = require("@splunk/react-icons/enterprise/ChevronLeft");
    var p =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/react-icons/ChevronLeft"
    const d = require("@splunk/react-icons/ChevronLeft");
    var g =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/themes"
    const v = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Paginator/icons/ChevronLeft.tsx
    function f(e) {
        var n = e.className;
        var a = (0, v.useSplunkTheme)(), t = a.isCompact, l = a.isEnterprise;
        var i = t ? "20px" : "24px";
        return l ?  r().createElement(p(), {
            className: n,
            hideDefaultTooltip: true,
            screenReaderText: null
        }) :  r().createElement(g(), {
            className: n,
            height: i,
            width: i
        });
    }
    /* harmony default export */    const m = f;
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/ChevronRight"
    const b = require("@splunk/react-icons/enterprise/ChevronRight");
    var y =  e.n(b);
    // CONCATENATED MODULE: external "@splunk/react-icons/ChevronRight"
    const h = require("@splunk/react-icons/ChevronRight");
    var P =  e.n(h);
    // CONCATENATED MODULE: ./src/Paginator/icons/ChevronRight.tsx
    function C(e) {
        var n = e.className;
        var a = (0, v.useSplunkTheme)(), t = a.isCompact, l = a.isEnterprise;
        var i = t ? "20px" : "24px";
        return l ?  r().createElement(y(), {
            className: n,
            hideDefaultTooltip: true,
            screenReaderText: null
        }) :  r().createElement(P(), {
            className: n,
            height: i,
            width: i
        });
    }
    /* harmony default export */    const k = C;
    // CONCATENATED MODULE: external "styled-components"
    const S = require("styled-components");
    var O =  e.n(S);
    // CONCATENATED MODULE: external "@splunk/react-ui/ButtonSimple"
    const w = require("@splunk/react-ui/ButtonSimple");
    var x =  e.n(w);
    // CONCATENATED MODULE: ./src/Paginator/ButtonStyles.ts
    var E = O()(x()).withConfig({
        displayName: "ButtonStyles__StyledButtonSimple",
        componentId: "sc-1neztq7-0"
    })([ "flex-shrink:0;font-weight:", ";min-width:", ";min-height:", ";text-align:center;&:not([disabled],[aria-disabled='true']){", ";", ";}", "" ], v.variables.fontWeightNormal, v.variables.inputHeight, v.variables.inputHeight, (0, 
    v.pick)({
        enterprise: {
            // TODO: we can remove this pick once SUI-6160 fixes contentColorDefault in Enterprise
            // we're intentionally only overriding dark here, because light has enough contrast as-is
            // and changing only dark lets us avoid making an unnecessary change to Enterprise light in SUI 4
            dark: (0, S.css)([ "color:", ";" ], v.variables.textGray)
        },
        prisma: (0, S.css)([ "color:", ";" ], v.variables.contentColorDefault)
    }), (function(e) {
        var n = e.$current;
        return n && (0, S.css)([ "", ";", ";" ], v.mixins.typography({
            color: "active",
            weight: "bold"
        }), (0, v.pick)({
            // TODO: maintain this separation until SUI 5, where we unify on Prisma's approach
            enterprise: (0, S.css)([ "border-color:", ";" ], v.variables.accentColor),
            prisma: (0, S.css)([ "background-color:", ";" ], v.variables.interactiveColorOverlaySelected)
        }));
    }), (0, v.pick)({
        enterprise: {
            comfortable: (0, S.css)([ "padding:", ";" ], v.variables.spacingXSmall),
            compact: (0, S.css)([ "padding:3px;" ])
        },
        prisma: (0, S.css)([ "padding:", ";& + &{margin-left:8px;}" ], (0, v.pick)({
            comfortable: "5px 8px",
            compact: "4px"
        }))
    }));
    // CONCATENATED MODULE: ./src/Paginator/Button.tsx
    function _() {
        return _ = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var a = arguments[n];
                for (var r in a) {
                    ({}).hasOwnProperty.call(a, r) && (e[r] = a[r]);
                }
            }
            return e;
        }, _.apply(null, arguments);
    }
    function j(e, n) {
        if (null == e) return {};
        var a, r, t = N(e, n);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (r = 0; r < l.length; r++) {
                a = l[r], n.includes(a) || {}.propertyIsEnumerable.call(e, a) && (t[a] = e[a]);
            }
        }
        return t;
    }
    function N(e, n) {
        if (null == e) return {};
        var a = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (n.includes(r)) continue;
                a[r] = e[r];
            }
        }
        return a;
    }
    var I = {
        children: l().node,
        disabled: l().oneOfType([ l().bool, l().oneOf([ "dimmed" ]) ]),
        onClick: l().func,
        page: l().number,
        /** @private */
        selected: l().bool
    };
    function T(e) {
        var n = e.children, t = e.disabled, l = e.onClick, i = e.page, o = e.selected, s = j(e, [ "children", "disabled", "onClick", "page", "selected" ]);
        var c = (0, a.useCallback)((function(e) {
            if (i != null) {
                l === null || l === void 0 ? void 0 : l(e, {
                    page: i
                });
            } else {
                l === null || l === void 0 ? void 0 : l(e, {});
            }
        }), [ l, i ]);
        
        return r().createElement(E, _({
            $current: o || false,
            "data-test": "page",
            "data-test-page": i,
            appearance: "subtle",
            disabled: t,
            inline: false,
            onClick: c
        }, s), n);
    }
    T.propTypes = I;
    /* harmony default export */    const q = T;
    // CONCATENATED MODULE: ./src/Paginator/CompactStyles.ts
    var L = O()(o()).withConfig({
        displayName: "CompactStyles__StyledBox",
        componentId: "dwgzkj-0"
    })([ "flex:0 1 auto;flex-wrap:nowrap;margin-left:auto;" ]);
    var M = O().span.withConfig({
        displayName: "CompactStyles__StyledLabel",
        componentId: "dwgzkj-1"
    })([ "align-items:center;display:flex;flex-wrap:wrap;margin-right:", ";" ], v.variables.spacingSmall);
    // CONCATENATED MODULE: ./src/Paginator/Compact.tsx
    function R() {
        return R = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var a = arguments[n];
                for (var r in a) {
                    ({}).hasOwnProperty.call(a, r) && (e[r] = a[r]);
                }
            }
            return e;
        }, R.apply(null, arguments);
    }
    function D(e, n) {
        if (null == e) return {};
        var a, r, t = z(e, n);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (r = 0; r < l.length; r++) {
                a = l[r], n.includes(a) || {}.propertyIsEnumerable.call(e, a) && (t[a] = e[a]);
            }
        }
        return t;
    }
    function z(e, n) {
        if (null == e) return {};
        var a = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (n.includes(r)) continue;
                a[r] = e[r];
            }
        }
        return a;
    }
    /** @public */
    /** @public */    var G = {
        current: l().number,
        elementRef: l().oneOfType([ l().func, l().object ]),
        onChange: l().func,
        renderLabel: l().oneOfType([ l().bool, l().func ]),
        totalPages: l().number
    };
    var B = function e(n) {
        var a = n.current, r = n.totalPages;
        var t = null;
        if (a != null && r != null) {
            t = (0, s.sprintf)((0, c._)("%d of %d pages"), a, r);
        } else if (a != null) {
            t = (0, s.sprintf)((0, c._)("Page %d"), a);
        }
        return t;
    };
    function H(e) {
        var n = e.onChange, t = e.current, l = e.renderLabel, i = e.totalPages, o = D(e, [ "onChange", "current", "renderLabel", "totalPages" ]);
        // @docs-props-type PaginatorCompactPropsBase
                var s = (0, a.useCallback)((function(e, a) {
            var r = a.page;
            n === null || n === void 0 ? void 0 : n(e, {
                direction: "prev",
                page: r
            });
        }), [ n ]);
        var u = (0, a.useCallback)((function(e, a) {
            var r = a.page;
            n === null || n === void 0 ? void 0 : n(e, {
                direction: "next",
                page: r
            });
        }), [ n ]);
        var p = null;
        if (l) {
            var d = typeof l === "function" ? l : B;
            p = d({
                current: t,
                totalPages: i
            });
        }
        
        return r().createElement(L, R({
            "aria-label": (0, c._)("Pagination"),
            "data-test": "paginator",
            "data-test-current": t,
            flex: true,
            inline: true,
            tag: "nav"
        }, o), p &&  r().createElement(M, {
            "data-test": "label"
        }, p),  r().createElement(q, {
            "aria-label": (0, c._)("Go to previous page"),
            "data-test": "prev",
            disabled: t != null && t <= 1 ? "dimmed" : false,
            onClick: s,
            page: t != null ? t - 1 : undefined
        },  r().createElement(m, null)),  r().createElement(q, {
            "aria-label": (0, c._)("Go to next page"),
            "data-test": "next",
            disabled: t != null && i != null && t >= i ? "dimmed" : false,
            onClick: u,
            page: t != null ? t + 1 : undefined
        },  r().createElement(k, null)));
    }
    H.propTypes = G;
    /* harmony default export */    const W = H;
    // CONCATENATED MODULE: external "@splunk/react-ui/Select"
    const $ = require("@splunk/react-ui/Select");
    var A =  e.n($);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const V = require("@splunk/react-ui/ScreenReaderContent");
    var X =  e.n(V);
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const F = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: ./src/Paginator/PageSelectStyles.ts
    var J = O().span.withConfig({
        displayName: "PageSelectStyles__StyledWrapper",
        componentId: "haycz4-0"
    })([ "display:flex;" ]);
    var K = O().span.withConfig({
        displayName: "PageSelectStyles__StyledCount",
        componentId: "haycz4-1"
    })([ "align-items:center;display:flex;flex-wrap:wrap;margin:0 ", ";" ], v.variables.spacingSmall);
    var Q = O()(A()).withConfig({
        displayName: "PageSelectStyles__StyledSelect",
        componentId: "haycz4-2"
    })([ "flex:0 0 auto;" ]);
    // CONCATENATED MODULE: ./src/Paginator/PageSelect.tsx
    function U(e, n) {
        if (null == e) return {};
        var a, r, t = Y(e, n);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (r = 0; r < l.length; r++) {
                a = l[r], n.includes(a) || {}.propertyIsEnumerable.call(e, a) && (t[a] = e[a]);
            }
        }
        return t;
    }
    function Y(e, n) {
        if (null == e) return {};
        var a = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (n.includes(r)) continue;
                a[r] = e[r];
            }
        }
        return a;
    }
    /** @public */    var Z = {
        onChange: l().func,
        current: l().number,
        totalPages: l().number
    };
    function ee(e) {
        var n = e.onChange, t = e.current, l = t === void 0 ? 1 : t, i = e.totalPages, o = U(e, [ "onChange", "current", "totalPages" ]);
        // @docs-props-type PaginatorPageSelectPropsBase
                var u = (0, a.useRef)((0, F.createDOMID)("pages"));
        var p = (0, a.useRef)((0, F.createDOMID)("totalPages"));
        var d = function e(a, r) {
            var t = r.value;
            n === null || n === void 0 ? void 0 : n(a, {
                page: t
            });
        };
        var g = [];
        for (var v = 1; v <= i; v += 1) {
            g.push( r().createElement(A().Option, {
                label: "".concat(v),
                key: "".concat(v),
                value: v
            }));
        }
        var f = (0, s.sprintf)((0, c._)("of %d pages"), i);
        
        return r().createElement(J, o,  r().createElement(Q, {
            value: l,
            onChange: d,
            "aria-labelledby": "".concat(u.current, " ").concat(p.current)
        }, g),  r().createElement(X(), {
            id: u.current,
            "aria-hidden": true
        }, l),  r().createElement(K, {
            id: p.current,
            "aria-hidden": true
        }, " ", f));
    }
    ee.propTypes = Z;
    /* harmony default export */    const ne = ee;
    // CONCATENATED MODULE: ./src/Paginator/PageControl.tsx
    function ae() {
        return ae = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var a = arguments[n];
                for (var r in a) {
                    ({}).hasOwnProperty.call(a, r) && (e[r] = a[r]);
                }
            }
            return e;
        }, ae.apply(null, arguments);
    }
    function re(e, n) {
        if (null == e) return {};
        var a, r, t = te(e, n);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (r = 0; r < l.length; r++) {
                a = l[r], n.includes(a) || {}.propertyIsEnumerable.call(e, a) && (t[a] = e[a]);
            }
        }
        return t;
    }
    function te(e, n) {
        if (null == e) return {};
        var a = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (n.includes(r)) continue;
                a[r] = e[r];
            }
        }
        return a;
    }
    /** @public */    var le = {
        elementRef: l().oneOfType([ l().func, l().object ]),
        onChange: l().func,
        current: l().number,
        totalPages: l().number
    };
    function ie(e) {
        var n = e.onChange, t = e.current, l = t === void 0 ? 1 : t, i = e.totalPages, o = re(e, [ "onChange", "current", "totalPages" ]);
        // @docs-props-type PaginatorPageControlPropsBase
                var s = (0, a.useCallback)((function(e, a) {
            var r = a.page;
            // page will always be defined because we're passing "current" to Compact
            // getting TypeScript to understand this seems more trouble than it's worth
            // so we'll validate this behavior through tests
                        n === null || n === void 0 ? void 0 : n(e, {
                page: r
            });
        }), [ n ]);
        var c = (0, a.useCallback)((function(e) {
            var a = e.current, t = e.totalPages;
            if (a != null && t != null) {
                
                return r().createElement(ne, {
                    current: a,
                    onChange: n,
                    totalPages: t
                });
            }
            return null;
        }), [ n ]);
        
        return r().createElement(W, ae({
            current: l,
            renderLabel: c,
            onChange: s,
            totalPages: i
        }, o));
    }
    ie.propTypes = le;
    /* harmony default export */    const oe = ie;
    // CONCATENATED MODULE: external "@splunk/react-icons/DotsThreeHorizontal"
    const se = require("@splunk/react-icons/DotsThreeHorizontal");
    var ce =  e.n(se);
    // CONCATENATED MODULE: ./src/Paginator/PageSeparatorStyles.ts
    var ue = O().span.withConfig({
        displayName: "PageSeparatorStyles__StyledEllipsisWrapper",
        componentId: "j50fqf-0"
    })([ "display:flex;align-items:center;margin:", ";color:", ";" ], (0, v.pick)({
        enterprise: "0px 11px",
        prisma: "0px 20px"
    }), v.variables.contentColorActive);
    // CONCATENATED MODULE: ./src/Paginator/PageSeparator.tsx
    function pe() {
        var e = (0, v.useSplunkTheme)(), n = e.family;
        var a = n === "enterprise" ?  r().createElement("span", {
            role: "separator"
        }, "…") :  r().createElement(ce(), null);
        
        return r().createElement(ue, {
            role: "separator"
        }, a);
    }
    /* harmony default export */    const de = pe;
    // CONCATENATED MODULE: ./src/Paginator/PaginatorStyles.ts
    var ge = O()(m).withConfig({
        displayName: "PaginatorStyles__StyledChevronLeftIcon",
        componentId: "pmabsp-0"
    })([ "margin:", ";" ], (0, v.pick)({
        enterprise: "-1px 3px 0 0",
        prisma: "-2px 6px 0 0"
    }));
    var ve = O()(k).withConfig({
        displayName: "PaginatorStyles__StyledChevronRightIcon",
        componentId: "pmabsp-1"
    })([ "margin:", ";" ], (0, v.pick)({
        enterprise: "-1px 0 0 3px",
        prisma: "-2px 0 0 6px"
    }));
    var fe = O().div.withConfig({
        displayName: "PaginatorStyles__StyledPrevNext",
        componentId: "pmabsp-2"
    })([ "padding:", ";white-space:nowrap;" ], (0, v.pick)({
        enterprise: "0 3px",
        prisma: "0 8px"
    }));
    var me = O()(q).withConfig({
        displayName: "PaginatorStyles__StyledPaginatorButton",
        componentId: "pmabsp-3"
    })([ "&:not([disabled]){", ",", "{", "}}" ], /* sc-sel */ ge, /* sc-sel */ ve, (0, 
    v.pick)({
        prisma: (0, S.css)([ "color:", ";" ], v.variables.contentColorMuted)
    }));
    // CONCATENATED MODULE: ./src/Paginator/Paginator.tsx
    function be() {
        return be = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var a = arguments[n];
                for (var r in a) {
                    ({}).hasOwnProperty.call(a, r) && (e[r] = a[r]);
                }
            }
            return e;
        }, be.apply(null, arguments);
    }
    function ye(e, n) {
        if (null == e) return {};
        var a, r, t = he(e, n);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (r = 0; r < l.length; r++) {
                a = l[r], n.includes(a) || {}.propertyIsEnumerable.call(e, a) && (t[a] = e[a]);
            }
        }
        return t;
    }
    function he(e, n) {
        if (null == e) return {};
        var a = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (n.includes(r)) continue;
                a[r] = e[r];
            }
        }
        return a;
    }
    /** @public */
    /** @public */    var Pe = {
        alwaysShowLastPageLink: l().bool,
        generatePageProps: l().func,
        current: l().number,
        elementRef: l().oneOfType([ l().func, l().object ]),
        numPageLinks: l().number,
        onChange: l().func,
        totalPages: l().number
    };
    function Ce(e) {
        var n = e.alwaysShowLastPageLink, t = n === void 0 ? false : n, l = e.generatePageProps, i = e.current, u = i === void 0 ? 1 : i, p = e.numPageLinks, d = p === void 0 ? 5 : p, g = e.onChange, v = e.totalPages, f = ye(e, [ "alwaysShowLastPageLink", "generatePageProps", "current", "numPageLinks", "onChange", "totalPages" ]);
        // @docs-props-type PaginatorPropsBase
                var m = (0, a.useCallback)((function(e, n) {
            var a = n.page;
            // page will always be defined because we're passing "page" to PaginatorButton
            // getting TypeScript to understand this seems more trouble than it's worth
            // so we'll validate this behavior through tests
                        g === null || g === void 0 ? void 0 : g(e, {
                page: a
            });
        }), [ g ]);
        // Can't show more links than total number of pages.
                var b = Math.min(d, v);
        var y = Math.ceil(b / 2);
        var h = Math.ceil(v - b / 2);
        var P = [];
        if (v <= 1) {
            return null;
        }
        if ((u > v || u < 1) && process.env.NODE_ENV !== "production") {}
        /**
   * Paginator shows numPageLinks page buttons that are nearby the current page.
   * firstPageNearbyCurrent and lastPageNearbyCurrent refer to the page number of the first and last of these buttons.
   */        var C;
        var k;
        if (u <= y + 1) {
            // + 1 to avoid ellipsis between 1 and 2
            C = 1;
            k = C + Math.min(v, b) - 1;
        } else if (u > y && u < h) {
            k = Math.ceil(u + (b - 2) / 2);
            C = k - b + 2;
        } else {
            // include page #1 to avoid ellipsis between 1 and 2
            C = v - b === 1 ? 1 : v - b + 1;
            k = v;
        }
        // bump up the last page to avoid ellipsis
                k = t && k + 1 === v ? v : k;
        var S = u - 1;
        P.push( r().createElement(me, be({}, l === null || l === void 0 ? void 0 : l({
            page: S
        }), {
            "aria-label": (0, c._)("Go to previous page"),
            "data-test": "prev",
            disabled: u === 1,
            key: "prev",
            onClick: m,
            page: S
        }),  r().createElement(fe, null,  r().createElement(ge, null), (0, c._)("Prev"))));
        if (u > y && v > b && C !== 1) {
            var O = 1;
            P.push( r().createElement(me, be({}, l === null || l === void 0 ? void 0 : l({
                page: O
            }), {
                "aria-label": (0, c._)("Go to first page"),
                key: "first",
                onClick: m,
                page: O
            }), "1"));
            P.push( r().createElement(de, {
                "data-test": "page",
                key: "prevEllipsis"
            }));
        }
        for (var w = C; w <= k; w += 1) {
            var x = w === u;
            P.push( r().createElement(me, be({}, l === null || l === void 0 ? void 0 : l({
                page: w
            }), {
                "aria-current": x ? "page" : false,
                "aria-label": (0, s.sprintf)((0, c._)("Page %(pageNumber)d"), {
                    pageNumber: w
                }),
                key: w,
                onClick: m,
                selected: x,
                page: w
            }), String(w)));
        }
        if (u <= h && v > b && k !== v) {
            P.push( r().createElement(de, {
                "data-test": "page",
                key: "nextEllipsis"
            }));
            if (t) {
                P.push( r().createElement(me, be({}, l === null || l === void 0 ? void 0 : l({
                    page: v
                }), {
                    "aria-label": (0, c._)("Go to last page"),
                    "data-test": "last",
                    key: "last",
                    onClick: m,
                    page: v
                }), String(v)));
            }
        }
        var E = u + 1;
        P.push( r().createElement(me, be({}, l === null || l === void 0 ? void 0 : l({
            page: E
        }), {
            "aria-label": (0, c._)("Go to next page"),
            "data-test": "next",
            disabled: u === v,
            key: "next",
            onClick: m,
            page: E
        }),  r().createElement(fe, null, (0, c._)("Next"),  r().createElement(ve, null))));
        
        return r().createElement(o(), be({
            "aria-label": (0, c._)("Pagination"),
            "data-test": "paginator",
            "data-test-current": u,
            flex: true,
            inline: true,
            tag: "nav"
        }, f), P);
    }
    Ce.Compact = W;
    Ce.PageControl = oe;
    Ce.propTypes = Pe;
    /* harmony default export */    const ke = Ce;
    // CONCATENATED MODULE: ./src/Paginator/index.ts
    module.exports = n;
    /******/})();
