// @splunk/react-ui/Progress.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var r = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ r.n = e => {
            /******/ var a = e && e.__esModule ? 
            /******/ () => e["default"]
            /******/ : () => e
            /******/;
            r.d(a, {
                a
            });
            /******/            return a;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ r.d = (e, a) => {
            /******/ for (var t in a) {
                /******/ if (r.o(a, t) && !r.o(e, t)) {
                    /******/ Object.defineProperty(e, t, {
                        enumerable: true,
                        get: a[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ r.o = (r, e) => Object.prototype.hasOwnProperty.call(r, e)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ r.r = r => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(r, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(r, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var e = {};
    // ESM COMPAT FLAG
        r.r(e);
    // EXPORTS
        r.d(e, {
        default: () => /* reexport */ L
    });
    // CONCATENATED MODULE: external "react"
    const a = require("react");
    var t =  r.n(a);
    // CONCATENATED MODULE: external "prop-types"
    const n = require("prop-types");
    var i =  r.n(n);
    // CONCATENATED MODULE: external "lodash/isNumber"
    const o = require("lodash/isNumber");
    var s =  r.n(o);
    // CONCATENATED MODULE: external "@splunk/react-ui/AnimationToggle"
    const l = require("@splunk/react-ui/AnimationToggle");
    // CONCATENATED MODULE: external "styled-components"
    const c = require("styled-components");
    var u =  r.n(c);
    // CONCATENATED MODULE: external "tinycolor2"
    const p = require("tinycolor2");
    var v =  r.n(p);
    // CONCATENATED MODULE: external "@splunk/react-ui/Tooltip"
    const b = require("@splunk/react-ui/Tooltip");
    var f =  r.n(b);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const m = require("@splunk/react-ui/Box");
    var d =  r.n(m);
    // CONCATENATED MODULE: external "@splunk/themes"
    const g = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Progress/ProgressStyles.ts
    var y = u()(d()).withConfig({
        displayName: "ProgressStyles__StyledBox",
        componentId: "csowex-0"
    })([ "background-color:", ";" ], (0, g.pick)({
        enterprise: g.variables.transparent,
        prisma: g.variables.neutral200
    }));
    var C = u()(f()).withConfig({
        displayName: "ProgressStyles__StyledTooltip",
        componentId: "csowex-1"
    })([ "background-color:", ";height:", ";position:relative;overflow:hidden;padding-left:", ";", ";", "" ], (0, 
    g.pickVariant)("$type", {
        info: {
            enterprise: g.variables.accentColorD10,
            prisma: g.variables.interactiveColorPrimary
        },
        success: {
            enterprise: g.variables.successColorD10,
            prisma: g.variables.accentColorPositive
        },
        warning: {
            enterprise: g.variables.accentColorD10,
            // warning not supported in enterprise, fallback to info
            prisma: g.variables.accentColorWarning
        },
        error: {
            enterprise: g.variables.errorColorD10,
            prisma: g.variables.accentColorNegative
        }
    }), (0, g.pick)({
        enterprise: "3px",
        prisma: "4px"
    }), (0, g.pick)({
        enterprise: g.variables.spacingSmall,
        prisma: "10px"
    }), (function(r) {
        var e = r.$animated;
        return e && (0, c.css)([ "transition:width 300ms;" ]);
    }), (0, g.pick)({
        enterprise: (0, c.css)([ "border-right:1px solid ", ";" ], (0, g.pickVariant)("$type", {
            info: {
                enterprise: g.variables.accentColorD10,
                prisma: g.variables.interactiveColorPrimary
            },
            success: {
                enterprise: g.variables.successColorD10,
                prisma: g.variables.accentColorPositive
            },
            warning: {
                enterprise: g.variables.accentColorD10,
                // warning not supported in enterprise, fallback to info
                prisma: g.variables.accentColorWarning
            },
            error: {
                enterprise: g.variables.errorColorD10,
                prisma: g.variables.accentColorNegative
            }
        }))
    }));
    var h = (0, c.keyframes)([ "from{opacity:0;}to{opacity:1;}" ]);
    var k = (0, c.keyframes)([ "from{background-position:130%;}to{background-position:-30%;}" ]);
    var w = u().div.withConfig({
        displayName: "ProgressStyles__StyledPuck",
        componentId: "csowex-2"
    })([ "height:", ";width:", ";position:absolute;right:0;top:0;", ";", "" ], (0, g.pick)({
        enterprise: "3px",
        prisma: "4px"
    }), (0, g.pick)({
        enterprise: "300px",
        prisma: "100%"
    }), (0, g.pick)({
        enterprise: function r() {
            var e = (0, g.pickVariant)("$type", {
                info: g.variables.accentColorD10,
                success: g.variables.successColorD10,
                warning: g.variables.accentColorD10,
                // warning not supported in enterprise, fallback to info
                error: g.variables.errorColorD10
            });
            var a = (0, g.pickVariant)("$type", {
                info: g.variables.accentColorL10,
                success: g.variables.successColorL10,
                warning: g.variables.accentColorL10,
                // warning not supported in enterprise, fallback to info
                error: g.variables.errorColorL10
            });
            var t = (0, g.pickVariant)("$type", {
                info: g.variables.accentColorL30,
                success: g.variables.successColorL30,
                warning: g.variables.accentColorL30,
                // warning not supported in enterprise, fallback to info
                error: g.variables.errorColorL30
            });
            return (0, c.css)([ "background:linear-gradient( 90deg,", ",", ",30%,", ",60%,", ",95%,", ",98%,", " );" ], e, a, t, t, a, e);
        },
        prisma: function r(e) {
            var a = e.$animated;
            var t = (0, g.pickVariant)("$type", {
                info: g.variables.interactiveColorPrimary,
                success: g.variables.accentColorPositive,
                warning: g.variables.accentColorWarning,
                error: g.variables.accentColorNegative
            });
            if (a) {
                var n = (0, g.pickVariant)("$type", {
                    info: g.variables.statusColorInfo,
                    success: g.variables.statusColorNormal,
                    warning: g.variables.statusColorMedium,
                    error: g.variables.statusColorHigh
                });
                return (0, c.css)([ "background:radial-gradient( circle at center,", ",", " 30% );" ], n, t);
            }
            return (0, c.css)([ "background:", ";" ], t);
        }
    }), (function(r) {
        return r.$animated && (0, g.pick)({
            enterprise: (0, c.css)([ "animation-duration:1500ms;animation-name:", ";animation-iteration-count:infinite;animation-direction:alternate;" ], h),
            prisma: function e() {
                var a = (0, g.pickVariant)("$type", {
                    info: g.variables.statusColorInfo,
                    success: g.variables.statusColorNormal,
                    warning: g.variables.statusColorMedium,
                    error: g.variables.statusColorHigh
                })(r);
                return (0, c.css)([ "background-size:200% 100%;animation:", " 1500ms infinite cubic-bezier(0.33,0,0.67,1);&::after{content:'';display:block;width:60px;background:linear-gradient( to right,rgba(255,255,255,0),", " 70%,rgba(255,255,255,0.75) );height:100%;position:absolute;right:0;}" ], k, v()(a(r)).setAlpha(.75).toRgbString());
            }
        });
    }));
    // CONCATENATED MODULE: ./src/Progress/Progress.tsx
    function S() {
        return S = Object.assign ? Object.assign.bind() : function(r) {
            for (var e = 1; e < arguments.length; e++) {
                var a = arguments[e];
                for (var t in a) {
                    ({}).hasOwnProperty.call(a, t) && (r[t] = a[t]);
                }
            }
            return r;
        }, S.apply(null, arguments);
    }
    function O(r, e) {
        return D(r) || j(r, e) || $(r, e) || P();
    }
    function P() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function $(r, e) {
        if (r) {
            if ("string" == typeof r) return x(r, e);
            var a = {}.toString.call(r).slice(8, -1);
            return "Object" === a && r.constructor && (a = r.constructor.name), "Map" === a || "Set" === a ? Array.from(r) : "Arguments" === a || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(a) ? x(r, e) : void 0;
        }
    }
    function x(r, e) {
        (null == e || e > r.length) && (e = r.length);
        for (var a = 0, t = Array(e); a < e; a++) {
            t[a] = r[a];
        }
        return t;
    }
    function j(r, e) {
        var a = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
        if (null != a) {
            var t, n, i, o, s = [], l = !0, c = !1;
            try {
                if (i = (a = a.call(r)).next, 0 === e) {
                    if (Object(a) !== a) return;
                    l = !1;
                } else for (;!(l = (t = i.call(a)).done) && (s.push(t.value), s.length !== e); l = !0) {
                }
            } catch (r) {
                c = !0, n = r;
            } finally {
                try {
                    if (!l && null != a["return"] && (o = a["return"](), Object(o) !== o)) return;
                } finally {
                    if (c) throw n;
                }
            }
            return s;
        }
    }
    function D(r) {
        if (Array.isArray(r)) return r;
    }
    function _(r, e) {
        if (null == r) return {};
        var a, t, n = q(r, e);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(r);
            for (t = 0; t < i.length; t++) {
                a = i[t], e.includes(a) || {}.propertyIsEnumerable.call(r, a) && (n[a] = r[a]);
            }
        }
        return n;
    }
    function q(r, e) {
        if (null == r) return {};
        var a = {};
        for (var t in r) {
            if ({}.hasOwnProperty.call(r, t)) {
                if (e.includes(t)) continue;
                a[t] = r[t];
            }
        }
        return a;
    }
    var A = {
        elementRef: i().oneOfType([ i().func, i().object ]),
        percentage: i().number,
        tooltip: i().node,
        type: i().oneOf([ "info", "success", "warning", "error" ])
    };
    function I(r) {
        var e = r.percentage, n = r.tooltip, i = r.type, o = i === void 0 ? "info" : i, c = _(r, [ "percentage", "tooltip", "type" ]);
        // @docs-props-type ProgressPropsBase
                if (false) {}
        var u = (0, a.useState)(false), p = O(u, 2), v = p[0], b = p[1];
        var f = (0, l.useAnimationToggle)() === "on";
        var m = (0, a.useCallback)((function() {
            b(true);
        }), []);
        var d = (0, a.useCallback)((function() {
            b(false);
        }), []);
        var g = n || "".concat(e, "%");
        
        return t().createElement(y, S({
            $animated: f,
            "data-test": "progress",
            onMouseEnter: m,
            onMouseLeave: d
        }, c), s()(e) &&  t().createElement(C, {
            inline: false,
            open: v,
            content: g,
            style: {
                width: "".concat(e, "%")
            },
            role: "progressbar",
            "aria-valuenow": e,
            "aria-valuemin": 0,
            "aria-valuemax": 100,
            $animated: f,
            $type: o
        },  t().createElement(w, {
            $animated: f,
            $type: o
        })));
    }
    I.propTypes = A;
    /* harmony default export */    const L = I;
    // CONCATENATED MODULE: ./src/Progress/index.ts
    module.exports = e;
    /******/})();
