// @splunk/react-ui/RadioBar.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        Option: () => /* reexport */ P,
        default: () => /* reexport */ Z
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var n =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var o =  e.n(a);
    // CONCATENATED MODULE: external "lodash/has"
    const l = require("lodash/has");
    var i =  e.n(l);
    // CONCATENATED MODULE: external "lodash/omit"
    const u = require("lodash/omit");
    var s =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/ui-utils/keyboard"
    const c = require("@splunk/ui-utils/keyboard");
    // CONCATENATED MODULE: external "@splunk/themes"
    const f = require("@splunk/themes");
    // CONCATENATED MODULE: external "styled-components"
    const d = require("styled-components");
    var p =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const v = require("@splunk/react-ui/Button");
    var b =  e.n(v);
    // CONCATENATED MODULE: ./src/RadioBar/OptionStyles.ts
    var y = p()(b()).withConfig({
        displayName: "OptionStyles__StyledButton",
        componentId: "fyq77p-0"
    })([ "", "" ], (0, f.pick)({
        prisma: (0, d.css)([ "border-radius:3px;font-weight:initial;height:calc(", " - 6px);min-height:calc(", " - 6px);padding:2px 12px;> span{gap:", ";}& + &{margin-left:2px;}", " &:not([disabled],[aria-disabled='true']){", "}" ], f.variables.inputHeight, f.variables.inputHeight, f.variables.spacingXSmall, (function(e) {
            var r = e.selected;
            return r && (0, d.css)([ "box-shadow:", ";&:not([disabled],[aria-disabled='true']){background-color:", ";}&[disabled],&[aria-disabled='true']{background-color:", ";}" ], f.variables.embossShadow, (0, 
            f.pick)({
                dark: f.variables.interactiveColorOverlaySelected,
                light: f.variables.white
            }), f.variables.backgroundColorPage);
        }), (function(e) {
            var r = e.selected;
            return !r && (0, d.css)([ "color:", ";" ], f.variables.contentColorDefault);
        }))
    }));
    var h = p().div.withConfig({
        displayName: "OptionStyles__StyledPrismaIconWrapper",
        componentId: "fyq77p-1"
    })([ "flex:0 0 auto;margin-top:-2px;font-size:15px;" ]);
    var g = p().div.withConfig({
        displayName: "OptionStyles__StyledLabel",
        componentId: "fyq77p-2"
    })([ "", " flex:0 1 auto;", "" ], f.mixins.ellipsis, (0, f.pick)({
        enterprise: (0, d.css)([ "&:not(:first-child){margin-left:", ";}" ], f.variables.spacingXSmall)
    }));
    var m = p().div.withConfig({
        displayName: "OptionStyles__StyledAdornment",
        componentId: "fyq77p-3"
    })([ "flex:0 0 auto;", "" ], (0, f.pick)({
        enterprise: (0, d.css)([ "&:not(:first-child){margin-left:", ";}" ], f.variables.spacingXSmall)
    }));
    // CONCATENATED MODULE: ./src/RadioBar/RadioBarContext.tsx
    var O =  (0, t.createContext)({});
    O.displayName = "RadioBar";
    /* harmony default export */    const w = O;
    // CONCATENATED MODULE: ./src/RadioBar/Option.tsx
    function x() {
        x = Object.assign || function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    if (Object.prototype.hasOwnProperty.call(t, n)) {
                        e[n] = t[n];
                    }
                }
            }
            return e;
        };
        return x.apply(this, arguments);
    }
    function k(e, r) {
        if (e == null) return {};
        var t = S(e, r);
        var n, a;
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (a = 0; a < o.length; a++) {
                n = o[a];
                if (r.indexOf(n) >= 0) continue;
                if (!Object.prototype.propertyIsEnumerable.call(e, n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    function S(e, r) {
        if (e == null) return {};
        var t = {};
        var n = Object.keys(e);
        var a, o;
        for (o = 0; o < n.length; o++) {
            a = n[o];
            if (r.indexOf(a) >= 0) continue;
            t[a] = e[a];
        }
        return t;
    }
    var j = {
        /** @private Set by `RadioBar`. */
        append: o().bool,
        disabled: o().bool,
        startAdornment: o().node,
        endAdornment: o().node,
        icon: o().node,
        label: o().string,
        /** @private Set by `RadioBar`. */
        prepend: o().bool,
        /** @private Set by `RadioBar`. */
        role: o().oneOf([ "radio", "menuitemradio" ]),
        /** @private Set by `Radiobar`. */
        selected: o().bool,
        /** The value of the `Option`. */
        value: o().any.isRequired
    };
    function C(e) {
        var r = e.disabled, a = e.icon, o = e.label, l = e.selected, i = e.startAdornment, u = e.endAdornment, s = e.value, c = e.role, d = c === void 0 ? "radio" : c, p = k(e, [ "disabled", "icon", "label", "selected", "startAdornment", "endAdornment", "value", "role" ]);
        // @docs-props-type OptionPropsBase
                var v = (0, t.useContext)(w), b = v.onClick, O = v.appearance, S = v.error, j = v.onKeyDown, C = v.firstValue, P = v.selectedValue;
        var E = (0, f.useSplunkTheme)(), _ = E.isPrisma;
        if (false) {}
        var T = -1;
        if (P) {
            if (s === P) {
                T = 0;
            }
        } else if (s === C) {
            T = 0;
        }
        var q = T;
        
        return n().createElement(y, x({
            "data-test": "option",
            "data-test-value": s,
            appearance: O,
            disabled: r,
            icon: !_ ? a : null,
            selected: l || false,
            value: s,
            error: S
        }, p, {
            onClick: l ? undefined : function(e) {
                return b === null || b === void 0 ? void 0 : b(e, {
                    value: s,
                    label: o
                });
            },
            role: d,
            "aria-checked": l,
            tabIndex: q,
            onKeyDown: function e(r) {
                return j === null || j === void 0 ? void 0 : j(r, {
                    value: s,
                    label: o
                });
            }
        }), _ && a &&  n().createElement(h, null, a), i &&  n().createElement(m, null, i), o &&  n().createElement(g, null, o), u &&  n().createElement(m, null, u));
    }
    C.propTypes = j;
    /* harmony default export */    const P = C;
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const E = require("@splunk/react-ui/Box");
    var _ =  e.n(E);
    // CONCATENATED MODULE: ./src/RadioBar/RadioBarStyles.ts
    var T = p()(_()).withConfig({
        displayName: "RadioBarStyles__StyledRadioBar",
        componentId: "gg1b79-0"
    })([ "height:", ";", " [data-inline] + &{margin-left:", ";}" ], f.variables.inputHeight, (0, 
    f.pick)({
        prisma: (0, d.css)([ "padding:2px;border-radius:", ";border:1px solid ", ";background:", ";" ], f.variables.borderRadius, (function(e) {
            var r = e.$error;
            return r ? f.variables.accentColorNegative : "transparent";
        }), (function(e) {
            var r = e.$disabled;
            return r ? f.variables.interactiveColorBackgroundDisabled : f.variables.interactiveColorBackground;
        }))
    }), f.variables.spacingSmall);
    // CONCATENATED MODULE: ./src/RadioBar/RadioBar.tsx
    function q(e) {
        "@babel/helpers - typeof";
        if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") {
            q = function e(r) {
                return typeof r;
            };
        } else {
            q = function e(r) {
                return r && typeof Symbol === "function" && r.constructor === Symbol && r !== Symbol.prototype ? "symbol" : typeof r;
            };
        }
        return q(e);
    }
    function R() {
        R = Object.assign || function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    if (Object.prototype.hasOwnProperty.call(t, n)) {
                        e[n] = t[n];
                    }
                }
            }
            return e;
        };
        return R.apply(this, arguments);
    }
    function D(e, r) {
        if (e == null) return {};
        var t = V(e, r);
        var n, a;
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (a = 0; a < o.length; a++) {
                n = o[a];
                if (r.indexOf(n) >= 0) continue;
                if (!Object.prototype.propertyIsEnumerable.call(e, n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    function V(e, r) {
        if (e == null) return {};
        var t = {};
        var n = Object.keys(e);
        var a, o;
        for (o = 0; o < n.length; o++) {
            a = n[o];
            if (r.indexOf(a) >= 0) continue;
            t[a] = e[a];
        }
        return t;
    }
    function B(e, r) {
        var t = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            if (r) n = n.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }));
            t.push.apply(t, n);
        }
        return t;
    }
    function I(e) {
        for (var r = 1; r < arguments.length; r++) {
            var t = arguments[r] != null ? arguments[r] : {};
            if (r % 2) {
                B(Object(t), true).forEach((function(r) {
                    W(e, r, t[r]);
                }));
            } else if (Object.getOwnPropertyDescriptors) {
                Object.defineProperties(e, Object.getOwnPropertyDescriptors(t));
            } else {
                B(Object(t)).forEach((function(r) {
                    Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
                }));
            }
        }
        return e;
    }
    function N(e, r) {
        if (!(e instanceof r)) {
            throw new TypeError("Cannot call a class as a function");
        }
    }
    function A(e, r) {
        for (var t = 0; t < r.length; t++) {
            var n = r[t];
            n.enumerable = n.enumerable || false;
            n.configurable = true;
            if ("value" in n) n.writable = true;
            Object.defineProperty(e, n.key, n);
        }
    }
    function K(e, r, t) {
        if (r) A(e.prototype, r);
        if (t) A(e, t);
        return e;
    }
    function $(e, r) {
        if (typeof r !== "function" && r !== null) {
            throw new TypeError("Super expression must either be null or a function");
        }
        e.prototype = Object.create(r && r.prototype, {
            constructor: {
                value: e,
                writable: true,
                configurable: true
            }
        });
        if (r) H(e, r);
    }
    function H(e, r) {
        H = Object.setPrototypeOf || function e(r, t) {
            r.__proto__ = t;
            return r;
        };
        return H(e, r);
    }
    function M(e) {
        var r = L();
        return function t() {
            var n = U(e), a;
            if (r) {
                var o = U(this).constructor;
                a = Reflect.construct(n, arguments, o);
            } else {
                a = n.apply(this, arguments);
            }
            return X(this, a);
        };
    }
    function X(e, r) {
        if (r && (q(r) === "object" || typeof r === "function")) {
            return r;
        }
        return z(e);
    }
    function z(e) {
        if (e === void 0) {
            throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        }
        return e;
    }
    function L() {
        if (typeof Reflect === "undefined" || !Reflect.construct) return false;
        if (Reflect.construct.sham) return false;
        if (typeof Proxy === "function") return true;
        try {
            Date.prototype.toString.call(Reflect.construct(Date, [], (function() {})));
            return true;
        } catch (e) {
            return false;
        }
    }
    function U(e) {
        U = Object.setPrototypeOf ? Object.getPrototypeOf : function e(r) {
            return r.__proto__ || Object.getPrototypeOf(r);
        };
        return U(e);
    }
    function W(e, r, t) {
        if (r in e) {
            Object.defineProperty(e, r, {
                value: t,
                enumerable: true,
                configurable: true,
                writable: true
            });
        } else {
            e[r] = t;
        }
        return e;
    }
    /** @public */    var F = {
        appearance: o().oneOf([ "default", "pill" ]),
        children: o().node,
        defaultValue: o().any,
        describedBy: o().string,
        elementRef: o().oneOfType([ o().func, o().object ]),
        error: o().bool,
        inline: o().bool,
        labelledBy: o().string,
        name: o().string,
        onChange: o().func,
        /** @private. */
        required: o().bool,
        role: o().oneOf([ "radiogroup", "menubar" ]),
        /** @private */
        splunkTheme: o().object,
        value: o().any
    };
    var G = {
        appearance: "default",
        error: false,
        inline: false,
        role: "radiogroup"
    };
    /**
 * RadioBar is a form control that provides the ability to select one option out of a group.
 */    var J =  function(e) {
        $(a, e);
        var r = M(a);
        // @docs-props-type RadioBarPropsBase
        // eslint-disable-line @typescript-eslint/no-explicit-any
                function a(e) {
            var t;
            N(this, a);
            t = r.call(this, e);
            W(z(t), "controlledExternally", void 0);
            W(z(t), "filteredValues", []);
            W(z(t), "handleClick", (function(e, r) {
                t.handleOnChange(e, r);
            }));
            W(z(t), "handleOnKeyDown", (function(e, r) {
                var n = r.value, a = r.label;
                var o = (0, c.keycode)(e.nativeEvent);
                var l = e.currentTarget;
                var i = null;
                var u = 0;
                var s = t.filteredValues.indexOf(n);
                if (o === "down" || o === "right") {
                    e.preventDefault();
                    var f = t.getNextTarget("forward", l, s), d = f.nextTargetRes, p = f.nextIndexRes;
                    i = d;
                    u = p;
                } else if (o === "up" || o === "left") {
                    e.preventDefault();
                    var v = t.getNextTarget("backward", l, s), b = v.nextTargetRes, y = v.nextIndexRes;
                    i = b;
                    u = y;
                }
                if (i) {
                    var h;
                    (h = i) === null || h === void 0 ? void 0 : h.focus();
                    var g = t.filteredValues[u];
                    t.handleOnChange(e, {
                        value: g,
                        label: a
                    });
                }
            }));
            W(z(t), "handleOnChange", (function(e, r) {
                if (t.props.value !== r.value) {
                    var n, a;
                    (n = (a = t.props).onChange) === null || n === void 0 ? void 0 : n.call(a, e, I({
                        name: t.props.name
                    }, r));
                    if (!t.isControlled()) {
                        t.setState({
                            value: r.value
                        });
                    }
                }
            }));
            t.controlledExternally = i()(e, "value");
            if (!t.isControlled()) {
                t.state = {
                    value: e.defaultValue
                };
            }
            if (false) {}
            return t;
        }
        K(a, [ {
            key: "componentDidUpdate",
            value: function e(r) {
                if (false) {}
                if (false) {}
            }
        }, {
            key: "isControlled",
            value: function e() {
                return this.controlledExternally;
            }
 // TODO: Once converted to a functional component, this function can be removed and RadioBar should use useRovingFocus hook
                }, {
            key: "getNextTarget",
            value: function e(r, t, n) {
                var a = r === "forward" ? 1 : -1;
                var o = r === "forward" ? "nextSibling" : "previousSibling";
                var l = r === "forward" ? "firstElementChild" : "lastElementChild";
                var i = r === "forward" ? 0 : this.filteredValues.length - 1;
                var u = t[o];
                var s = null;
                var c = n + a;
                while (u !== t && s === null) {
                    if (u === null) {
                        var f;
                        u = (f = t.parentElement) === null || f === void 0 ? void 0 : f[l];
                        c = i;
                    }
                    if (u.disabled) {
                        var d = u[o];
                        u = d;
                        c += a;
                    } else {
                        s = u;
                    }
                }
                return {
                    nextTargetRes: s,
                    nextIndexRes: c
                };
            }
        }, {
            key: "render",
            value: function e() {
                var r = this;
                var a = this.props, o = a.appearance, l = a.children, i = a.describedBy, u = a.error, c = a.labelledBy, f = a.splunkTheme, d = a.value, p = a.required, v = a.role, b = D(a, [ "appearance", "children", "describedBy", "error", "labelledBy", "splunkTheme", "value", "required", "role" ]);
                if (false) {}
                var y = this.isControlled() ? d : this.state.value;
                var h = f.isEnterprise;
                var g = true;
                this.filteredValues = [];
                var m = t.Children.toArray(l).filter(t.isValidElement).map((function(e, n, a) {
                    var o = a.length;
                    r.filteredValues.push(e.props.value);
                    if (e.props && !e.props.disabled) {
                        g = false;
                    }
                    
                    return (0, t.cloneElement)(e, {
                        selected: e.props.value === y,
                        key: e.key || n,
                        append: n < o - 1,
                        prepend: n > 0,
                        role: v === "radiogroup" ? "radio" : "menuitemradio"
                    });
                }));
                var O = this.filteredValues[0];
                var x = d == null || this.filteredValues.includes(y);
                if (false) {}
                var k = h && o === "default" ? "default" : "subtle";
                var S = {
                    appearance: k,
                    onClick: this.handleClick,
                    error: h ? u : false,
                    onKeyDown: this.handleOnKeyDown,
                    firstValue: O,
                    selectedValue: x ? y : undefined
                };
                
                return n().createElement(T, R({
                    flex: true,
                    $disabled: g,
                    $error: u,
                    "data-test": "radio-bar",
                    "data-test-value": y,
                    role: v,
                    "aria-labelledby": c,
                    "aria-describedby": i,
                    "aria-required": p
                }, s()(b, "onChange")),  n().createElement(w.Provider, {
                    value: S
                }, m));
            }
        } ]);
        return a;
    }(t.Component);
    W(J, "propTypes", F);
    W(J, "defaultProps", G);
    W(J, "Option", P);
    var Q = (0, f.withSplunkTheme)(J);
    var Y = Q;
    Y.propTypes = J.propTypes;
    Y.Option = P;
    /* harmony default export */    const Z = Y;
    // only for styled components using RadioBar
        // CONCATENATED MODULE: ./src/RadioBar/index.ts
    module.exports = r;
    /******/})();
