// @splunk/react-ui/Resize.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        default: () => /* reexport */ le
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "lodash/keys"
    const a = require("lodash/keys");
    var s =  e.n(a);
    // CONCATENATED MODULE: external "lodash/omit"
    const c = require("lodash/omit");
    var u =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/react-ui/EventListener"
    const l = require("@splunk/react-ui/EventListener");
    var d =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const f = require("@splunk/react-ui/ScreenReaderContent");
    var p =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const v = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/ui-utils/keyboard"
    const h = require("@splunk/ui-utils/keyboard");
    // CONCATENATED MODULE: external "styled-components"
    const y = require("styled-components");
    var m =  e.n(y);
    // CONCATENATED MODULE: external "@splunk/themes"
    const b = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Resize/ResizeStyles.ts
    var w = "9px";
    var g = "14px";
    var S = "3px";
    var E = "11px";
    var k = m().div.withConfig({
        displayName: "ResizeStyles__Styled",
        componentId: "sc-181mwes-0"
    })([ "", ";position:relative;overflow-x:hidden;overflow-y:hidden;" ], b.mixins.reset("block"));
    var z = m().div.withConfig({
        displayName: "ResizeStyles__StyledBorder",
        componentId: "sc-181mwes-1"
    })([ "", ";height:100%;width:100%;border:0 solid ", ";", " ", " ", " ", "" ], b.mixins.reset("block"), (0, 
    b.pick)({
        prisma: b.variables.interactiveColorBackground,
        enterprise: {
            dark: b.variables.gray22,
            light: b.variables.white
        }
    }), (function(e) {
        var t = e.$borderSides;
        return t.n && (0, y.css)([ "border-top-width:", ";" ], w);
    }), (function(e) {
        var t = e.$borderSides;
        return t.e && (0, y.css)([ "border-right-width:", ";" ], w);
    }), (function(e) {
        var t = e.$borderSides;
        return t.s && (0, y.css)([ "border-bottom-width:", ";" ], w);
    }), (function(e) {
        var t = e.$borderSides;
        return t.w && (0, y.css)([ "border-left-width:", ";" ], w);
    }));
    var O = m().svg.withConfig({
        displayName: "ResizeStyles__StyledCornerSvg",
        componentId: "sc-181mwes-2"
    })([ "display:block;stroke:currentColor;" ]);
    var _ = m().path.withConfig({
        displayName: "ResizeStyles__StyledCornerPath",
        componentId: "sc-181mwes-3"
    })([ "stroke-width:1.1px;" ]);
    var x = m().svg.withConfig({
        displayName: "ResizeStyles__StyledSideSvg",
        componentId: "sc-181mwes-4"
    })([ "display:block;fill:currentColor;" ]);
    var R = (0, y.css)([ "", ";color:", ";&:hover,&:focus{color:", ";", "}", "" ], b.mixins.reset("block"), (0, 
    b.pick)({
        prisma: b.variables.contentColorDefault,
        enterprise: {
            dark: b.variables.gray92,
            light: b.variables.borderColor
        }
    }), (0, b.pick)({
        prisma: b.variables.interactiveColorPrimary,
        enterprise: b.variables.linkColor
    }), (function(e) {
        var t = e.$isSeparator, r = e.$cardinalDirection;
        var n = [ "n", "e", "w", "s" ].includes(r);
        return t && n && (0, y.css)([ "background-color:", ";&:active{background-color:", ";}" ], b.variables.borderColorWeak, b.variables.borderColorStrong);
    }), (function(e) {
        var t = e.$showHandles;
        return t === "on-hover" && (0, y.css)([ "opacity:0;transition:opacity 200ms;", ":hover > div > &,&:focus{opacity:1;}" ], /* sc-sel */ k);
    }));
    var D = (0, y.css)([ "", ";width:", ";height:", ";position:absolute;background-clip:content-box;z-index:2;" ], R, E, E);
    var j = (0, y.css)([ "height:100%;" ]);
    var C = (0, y.css)([ "width:100%;display:flex;justify-content:center;" ]);
    var P = (0, y.css)([ "", ";position:absolute;z-index:1;", "" ], R, (function(e) {
        var t = e.$isSeparator;
        return t && (0, b.pickVariant)("$cardinalDirection", {
            n: C,
            s: C,
            e: j,
            w: j
        });
    }));
    var q = function e(t) {
        var r = t.$isSeparator;
        return r ? "0" : "calc(50% - ".concat(g, " / 2)");
    };
    var T = m().button.withConfig({
        displayName: "ResizeStyles__StyledResize",
        componentId: "sc-181mwes-5"
    })([ "", "" ], (0, b.pickVariant)("$cardinalDirection", {
        nw: (0, y.css)([ "", ";top:0;left:0;cursor:nwse-resize;> ", "{transform:rotate(180deg);}" ], D, /* sc-sel */ O),
        n: (0, y.css)([ "", ";cursor:ns-resize;top:0;left:", ";padding:", " 0;" ], P, q, S),
        ne: (0, y.css)([ "", ";right:0;top:0;cursor:nesw-resize;> ", "{transform:rotate(-90deg);}" ], D, /* sc-sel */ O),
        w: (0, y.css)([ "", ";cursor:ew-resize;padding:0 ", ";left:0;top:", ";" ], P, S, q),
        e: (0, y.css)([ "", ";cursor:ew-resize;padding:0 ", ";right:0;top:", ";" ], P, S, q),
        sw: (0, y.css)([ "", ";left:0;bottom:0;cursor:nesw-resize;> ", "{transform:rotate(90deg);}" ], D, /* sc-sel */ O),
        s: (0, y.css)([ "", ";cursor:ns-resize;padding:", " 0;bottom:0;left:", ";" ], P, S, q),
        se: (0, y.css)([ "", ";right:0;bottom:0;cursor:nwse-resize;" ], D)
    }));
    // CONCATENATED MODULE: ./src/utils/ssrDocument.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var $ = {
        body: {
            appendChild: function e() {
                return [];
            }
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        activeElement: {
            blur: function e() {},
            nodeName: ""
        },
        querySelector: function e() {
            return null;
        },
        querySelectorAll: function e() {
            return [];
        },
        getElementById: function e() {
            return null;
        },
        createEvent: function e() {
            return {
                initEvent: function e() {}
            };
        },
        createElement: function e() {
            return {
                children: [],
                childNodes: [],
                style: {},
                setAttribute: function e() {},
                getElementsByTagName: function e() {
                    return [];
                }
            };
        },
        createElementNS: function e() {
            return {};
        },
        importNode: function e() {
            return null;
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        }
    };
    function H() {
        var e = typeof document !== "undefined" ? document : $;
        return e;
    }
    var N = H();
    /* harmony default export */    const I = /* unused pure expression or super */ null && N;
    // CONCATENATED MODULE: ./src/utils/ssrWindow.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var B = {
        document: $,
        navigator: {
            userAgent: ""
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        },
        history: {
            replaceState: function e() {},
            pushState: function e() {},
            go: function e() {},
            back: function e() {}
        },
        CustomEvent: function e() {
            return this;
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        getComputedStyle: function e() {
            return {
                getPropertyValue: function e() {
                    return "";
                }
            };
        },
        Image: function e() {},
        Date: function e() {},
        screen: {},
        setTimeout: function e() {},
        clearTimeout: function e() {},
        matchMedia: function e() {
            return {};
        },
        requestAnimationFrame: function e(t) {
            if (typeof setTimeout === "undefined") {
                t();
                return null;
            }
            return setTimeout(t, 0);
        },
        cancelAnimationFrame: function e(t) {
            if (typeof setTimeout === "undefined") {
                return;
            }
            clearTimeout(t);
        }
    };
    function M() {
        var e = typeof window !== "undefined" ? window : B;
        return e;
    }
    var L = M();
    /* harmony default export */    const A = L;
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function K(e, t) {
        if (e) {
            if (typeof e === "function") {
                e(t);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = t;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Resize/Resize.tsx
    function W(e) {
        "@babel/helpers - typeof";
        return W = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, W(e);
    }
    function X() {
        return X = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, X.apply(null, arguments);
    }
    function Y(e, t) {
        if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function");
    }
    function F(e, t) {
        for (var r = 0; r < t.length; r++) {
            var n = t[r];
            n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), 
            Object.defineProperty(e, ne(n.key), n);
        }
    }
    function V(e, t, r) {
        return t && F(e.prototype, t), r && F(e, r), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function G(e, t) {
        if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), t && J(e, t);
    }
    function J(e, t) {
        return J = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, t) {
            return e.__proto__ = t, e;
        }, J(e, t);
    }
    function Q(e) {
        var t = ee();
        return function() {
            var r, n = te(e);
            if (t) {
                var o = te(this).constructor;
                r = Reflect.construct(n, arguments, o);
            } else r = n.apply(this, arguments);
            return U(this, r);
        };
    }
    function U(e, t) {
        if (t && ("object" == W(t) || "function" == typeof t)) return t;
        if (void 0 !== t) throw new TypeError("Derived constructors may only return object or undefined");
        return Z(e);
    }
    function Z(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function ee() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return (ee = function t() {
            return !!e;
        })();
    }
    function te(e) {
        return te = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, te(e);
    }
    function re(e, t, r) {
        return (t = ne(t)) in e ? Object.defineProperty(e, t, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = r, e;
    }
    function ne(e) {
        var t = oe(e, "string");
        return "symbol" == W(t) ? t : t + "";
    }
    function oe(e, t) {
        if ("object" != W(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var n = r.call(e, t || "default");
            if ("object" != W(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    /** @public */
    /** @public */    var ie = {
        appearance: i().oneOf([ "border", "overlay", "separator" ]),
        children: i().node,
        elementRef: i().oneOfType([ i().func, i().object ]),
        keyIncrement: i().number,
        onRequestResize: i().func.isRequired,
        resizeHandles: i().arrayOf(i().oneOf([ "nw", "n", "ne", "w", "e", "sw", "s", "se" ])).isRequired,
        showHandles: i().oneOf([ "always", "on-hover" ])
    };
    var ae = {
        appearance: "overlay",
        keyIncrement: 10,
        showHandles: "always"
    };
    var se = [ "nw", "n", "ne", "w", "e", "sw", "s", "se" ];
    var ce = {
        nw: (0, v._)("Resize northwest"),
        n: (0, v._)("Resize north"),
        ne: (0, v._)("Resize northeast"),
        w: (0, v._)("Resize west"),
        e: (0, v._)("Resize east"),
        sw: (0, v._)("Resize southwest"),
        s: (0, v._)("Resize south"),
        se: (0, v._)("Resize southeast")
    };
    /**
 * Resize is a utility container with drag handles for resizing.
 */    var ue =  function(e) {
        G(r, e);
        var t = Q(r);
        // @docs-props-type ResizePropsBase
                function r(e) {
            var o;
            Y(this, r);
            o = t.call(this, e);
            re(Z(o), "handleMount", (function(e) {
                o.setState({
                    el: e
                });
                K(o.props.elementRef, e);
            }));
            re(Z(o), "handleDragStart", (function(e, t) {
                if (e.button > 0) {
                    return;
                }
                e.preventDefault();
                e.persist();
                o.setState((function(r) {
                    var n, o;
                    return {
                        dragDirection: t,
                        startWidth: (n = r.el) === null || n === void 0 ? void 0 : n.offsetWidth,
                        startHeight: (o = r.el) === null || o === void 0 ? void 0 : o.offsetHeight,
                        startX: e.clientX,
                        startY: e.clientY
                    };
                }));
            }));
            re(Z(o), "handleDragEnd", (function() {
                o.setState({
                    dragDirection: undefined
                });
            }));
            re(Z(o), "handleDrag", (function(e) {
                var t = o.state, r = t.dragDirection, n = t.startWidth, i = n === void 0 ? 0 : n, a = t.startHeight, s = a === void 0 ? 0 : a, c = t.startX, u = c === void 0 ? 0 : c, l = t.startY, d = l === void 0 ? 0 : l;
                var f = (r === null || r === void 0 ? void 0 : r.match(/(w|e)$/)) ? e.clientX - u : 0;
                var p = (r === null || r === void 0 ? void 0 : r.match(/^(s|n)/)) ? e.clientY - d : 0;
                var v = {
                    width: (r === null || r === void 0 ? void 0 : r.match(/w$/)) ? i - f : i + f,
                    height: (r === null || r === void 0 ? void 0 : r.match(/^n/)) ? s - p : s + p
                };
                o.props.onRequestResize(e, v);
            }));
            re(Z(o), "handleKeyDown", (function(e, t) {
                var r = o.props.keyIncrement;
                var n = (0, h.keycode)(e.nativeEvent);
                var i = t.slice(-1);
                var a = {
                    e: {
                        right: 1,
                        left: -1
                    },
                    w: {
                        right: -1,
                        left: 1
                    }
                };
                var s = (i === "e" || i === "w") && (n === "right" || n === "left") ? a[i][n] : 0;
                var c = t.charAt(0);
                var u = {
                    n: {
                        down: -1,
                        up: 1
                    },
                    s: {
                        down: 1,
                        up: -1
                    }
                };
                var l = (c === "n" || c === "s") && (n === "up" || n === "down") ? u[c][n] : 0;
                if ((s || l) && o.state.el) {
                    e.preventDefault();
                    var d = {
                        width: o.state.el.offsetWidth + s * r,
                        height: o.state.el.offsetHeight + l * r
                    };
                    o.props.onRequestResize(e, d);
                }
            }));
            re(Z(o), "renderHandle", (function(e) {
                var t = o.props, r = t.showHandles, i = t.appearance;
                var a = e.length > 1;
                var s;
                if (a) {
                    s =  n().createElement(O, {
                        viewBox: "0 0 11 11",
                        width: "11px",
                        height: "11px"
                    },  n().createElement(_, {
                        d: "M0 8 L 8 0 M4 8 L 8 4"
                    }));
                } else if (e === "n" || e === "s") {
                    s =  n().createElement(x, {
                        viewBox: "0 0 14 3",
                        width: "14px",
                        height: "3px"
                    },  n().createElement("rect", {
                        width: "14",
                        height: "1",
                        x: "0",
                        y: "0"
                    }),  n().createElement("rect", {
                        width: "14",
                        height: "1",
                        x: "0",
                        y: "2"
                    }));
                } else if (e === "e" || e === "w") {
                    s =  n().createElement(x, {
                        viewBox: "0 0 3 14",
                        width: "3px",
                        height: "14px"
                    },  n().createElement("rect", {
                        width: "1",
                        height: "14",
                        x: "0",
                        y: "0"
                    }),  n().createElement("rect", {
                        width: "1",
                        height: "14",
                        x: "2",
                        y: "0"
                    }));
                }
                
                return n().createElement(T, {
                    key: e,
                    "data-test": "resize-handle-".concat(e),
                    onMouseDown: o["handleDragStart".concat(e)],
                    onKeyDown: o["handleKeyDown".concat(e)],
                    $showHandles: r,
                    $cardinalDirection: e,
                    $isSeparator: i === "separator"
                }, s,  n().createElement(p(), null, ce[e]));
            }));
            o.state = {
                el: null
            };
            if (false) {}
            // Generate handlers for each direction.
                        se.forEach((function(e) {
                // @ts-expect-error - dynamic function creation
                o["handleDragStart".concat(e)] = function(t) {
                    o.handleDragStart(t, e);
                };
                // @ts-expect-error - dynamic function creation
                                o["handleKeyDown".concat(e)] = function(t) {
                    o.handleKeyDown(t, e);
                };
            }));
            return o;
        }
        V(r, [ {
            key: "render",
            value: function e() {
                var t = this.props, o = t.appearance, i = t.children, a = t.resizeHandles;
                var c = this.state, l = c.dragDirection, f = c.el;
 // Handles are inserted before and after children to optimize tab order
                                var p = se.slice(0, 4).filter((function(e) {
                    return a.indexOf(e) > -1;
                })).map(this.renderHandle);
                var v = se.slice(4).filter((function(e) {
                    return a.indexOf(e) > -1;
                })).map(this.renderHandle);
                /* When appearance is border, determine which sides need margin.
       * Converts ['se', 'n'] to { e: true, s: true, n: true}
       */                var h = {};
                if (o === "border") {
                    a.join("").split("").filter((function(e, t, r) {
                        return r.indexOf(e) === t;
                    })).forEach((function(e) {
                        h[e] = true;
                    }));
                }
                
                return n().createElement(k, X({
                    "data-test": "resize",
                    ref: this.handleMount
                }, u()(this.props, s()(r.propTypes))),  n().createElement(z, {
                    $borderSides: h
                }, f && p, i, f && v, l &&  n().createElement(n().Fragment, null,  n().createElement(d(), {
                    target: A,
                    eventType: "mousemove",
                    listener: this.handleDrag
                }),  n().createElement(d(), {
                    target: A,
                    eventType: "mouseup",
                    listener: this.handleDragEnd
                }))));
            }
        } ]);
        return r;
    }(r.Component);
    re(ue, "propTypes", ie);
    re(ue, "defaultProps", ae);
    /* harmony default export */    const le = ue;
    // CONCATENATED MODULE: ./src/Resize/index.ts
    module.exports = t;
    /******/})();
