// @splunk/react-ui/ScreenReaderContent.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/    var e = {
        /***/ 3563: 
        /***/ e => {
            e.exports = require("@splunk/themes");
            /***/        },
        /***/ 23: 
        /***/ e => {
            e.exports = require("prop-types");
            /***/        },
        /***/ 9497: 
        /***/ e => {
            e.exports = require("react");
            /***/        },
        /***/ 232: 
        /***/ e => {
            e.exports = require("styled-components");
            /***/
            /******/        }
    };
    /************************************************************************/
    /******/ // The module cache
    /******/    var r = {};
    /******/
    /******/ // The require function
    /******/    function n(t) {
        /******/ // Check if module is in cache
        /******/ var o = r[t];
        /******/        if (o !== undefined) {
            /******/ return o.exports;
            /******/        }
        /******/ // Create a new module (and put it into the cache)
        /******/        var a = r[t] = {
            /******/ // no module.id needed
            /******/ // no module.loaded needed
            /******/ exports: {}
            /******/        };
        /******/
        /******/ // Execute the module function
        /******/        e[t](a, a.exports, n);
        /******/
        /******/ // Return the exports of the module
        /******/        return a.exports;
        /******/    }
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ n.n = e => {
            /******/ var r = e && e.__esModule ? 
            /******/ () => e["default"]
            /******/ : () => e
            /******/;
            n.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ n.d = (e, r) => {
            /******/ for (var t in r) {
                /******/ if (n.o(r, t) && !n.o(e, t)) {
                    /******/ Object.defineProperty(e, t, {
                        enumerable: true,
                        get: r[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ n.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ n.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // This entry need to be wrapped in an IIFE because it need to be isolated against other modules in the chunk.
        (() => {
        // ESM COMPAT FLAG
        n.r(t);
        // EXPORTS
                n.d(t, {
            default: () => /* reexport */ y
        });
        // EXTERNAL MODULE: external "react"
                var e = n(9497);
        var r =  n.n(e);
        // EXTERNAL MODULE: external "prop-types"
                var o = n(23);
        var a =  n.n(o);
        // EXTERNAL MODULE: external "styled-components"
                var l = n(232);
        var i =  n.n(l);
        // EXTERNAL MODULE: external "@splunk/themes"
                var u = n(3563);
        // CONCATENATED MODULE: ./src/ScreenReaderContent/ScreenReaderContentStyles.ts
        var s = i().span.withConfig({
            displayName: "ScreenReaderContentStyles__Styled",
            componentId: "sc-1lnohwp-0"
        })([ "", ";" ], u.mixins.screenReaderContent());
        // CONCATENATED MODULE: ./src/ScreenReaderContent/ScreenReaderContent.tsx
        function c() {
            return c = Object.assign ? Object.assign.bind() : function(e) {
                for (var r = 1; r < arguments.length; r++) {
                    var n = arguments[r];
                    for (var t in n) {
                        ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                    }
                }
                return e;
            }, c.apply(null, arguments);
        }
        function p(e, r) {
            if (null == e) return {};
            var n, t, o = f(e, r);
            if (Object.getOwnPropertySymbols) {
                var a = Object.getOwnPropertySymbols(e);
                for (t = 0; t < a.length; t++) {
                    n = a[t], r.includes(n) || {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
                }
            }
            return o;
        }
        function f(e, r) {
            if (null == e) return {};
            var n = {};
            for (var t in e) {
                if ({}.hasOwnProperty.call(e, t)) {
                    if (r.includes(t)) continue;
                    n[t] = e[t];
                }
            }
            return n;
        }
        var d = {
            children: a().node.isRequired,
            elementRef: a().oneOfType([ a().func, a().object ])
        };
        /**
 * The screen reader text is used to wrap content that
 * is only accessible through screen readers.
 */        function v(e) {
            var n = e.children, t = e.elementRef, o = p(e, [ "children", "elementRef" ]);
            // @docs-props-type ScreenReaderContentPropsBase
                        
            return r().createElement(s, c({
                "data-test": "screen-reader-content",
                ref: t
            }, o), n);
        }
        v.propTypes = d;
        /* harmony default export */        const y = v;
    }) // CONCATENATED MODULE: ./src/ScreenReaderContent/index.ts
    ();
    module.exports = t;
    /******/})();
