// @splunk/react-ui/Scroll.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/    var e = {
        /***/ 1558: 
        /***/ (e, t, r) => {
            // EXPORTS
            r.d(t, {
                default: () => /* reexport */ l
            });
            // EXTERNAL MODULE: external "react"
                        var n = r(9497);
            // CONCATENATED MODULE: ./src/usePrevious/usePrevious.tsx
            /**
 * This is a private component not intended for use outside @splunk/react-ui
 *
 * Keeps the previous version of a value. Useful for emulating the "prevState"
 * argument to componentDidUpdate() in functional components.
 * See https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state
 *
 * @param value - The value to store
 */
            function o(e) {
                var t = (0, n.useRef)();
                (0, n.useEffect)((function() {
                    t.current = e;
                }), [ e ]);
                return t.current;
            }
            /* harmony default export */            const l = o;
        } // CONCATENATED MODULE: ./src/usePrevious/index.ts
        /***/ ,
        /***/ 9497: 
        /***/ e => {
            e.exports = require("react");
            /***/
            /******/        }
    };
    /************************************************************************/
    /******/ // The module cache
    /******/    var t = {};
    /******/
    /******/ // The require function
    /******/    function r(n) {
        /******/ // Check if module is in cache
        /******/ var o = t[n];
        /******/        if (o !== undefined) {
            /******/ return o.exports;
            /******/        }
        /******/ // Create a new module (and put it into the cache)
        /******/        var l = t[n] = {
            /******/ // no module.id needed
            /******/ // no module.loaded needed
            /******/ exports: {}
            /******/        };
        /******/
        /******/ // Execute the module function
        /******/        e[n](l, l.exports, r);
        /******/
        /******/ // Return the exports of the module
        /******/        return l.exports;
        /******/    }
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ r.n = e => {
            /******/ var t = e && e.__esModule ? 
            /******/ () => e["default"]
            /******/ : () => e
            /******/;
            r.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ r.d = (e, t) => {
            /******/ for (var n in t) {
                /******/ if (r.o(t, n) && !r.o(e, n)) {
                    /******/ Object.defineProperty(e, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ r.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ r.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // This entry need to be wrapped in an IIFE because it need to be isolated against other modules in the chunk.
        (() => {
        // ESM COMPAT FLAG
        r.r(n);
        // EXPORTS
                r.d(n, {
            default: () => /* reexport */ re
        });
        // EXTERNAL MODULE: external "react"
                var e = r(9497);
        var t =  r.n(e);
        // CONCATENATED MODULE: external "prop-types"
        const o = require("prop-types");
        var l =  r.n(o);
        // CONCATENATED MODULE: external "lodash/isFinite"
        const i = require("lodash/isFinite");
        var a =  r.n(i);
        // CONCATENATED MODULE: external "@splunk/react-ui/Animation"
        const u = require("@splunk/react-ui/Animation");
        // CONCATENATED MODULE: external "styled-components"
        const c = require("styled-components");
        var f =  r.n(c);
        // CONCATENATED MODULE: external "@splunk/themes"
        const s = require("@splunk/themes");
        // CONCATENATED MODULE: external "lodash/keys"
        const p = require("lodash/keys");
        var d =  r.n(p);
        // CONCATENATED MODULE: external "lodash/omit"
        const v = require("lodash/omit");
        var y =  r.n(v);
        // CONCATENATED MODULE: external "react-spring"
        const b = require("react-spring");
        // CONCATENATED MODULE: external "@splunk/react-ui/EventListener"
        const m = require("@splunk/react-ui/EventListener");
        var h =  r.n(m);
        // CONCATENATED MODULE: external "@splunk/react-ui/ScrollContainerContext"
        const g = require("@splunk/react-ui/ScrollContainerContext");
        // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
        /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
        function S(e, t) {
            if (e) {
                if (typeof e === "function") {
                    e(t);
                } else {
                    // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                    // the intention here is to signal "we will take care of setting 'current', not you".
                    e.current = t;
 // eslint-disable-line no-param-reassign
                                }
            }
        }
        // EXTERNAL MODULE: ./src/usePrevious/index.ts + 1 modules
                var w = r(1558);
        // CONCATENATED MODULE: ./src/Scroll/Inner.tsx
        function O() {
            return O = Object.assign ? Object.assign.bind() : function(e) {
                for (var t = 1; t < arguments.length; t++) {
                    var r = arguments[t];
                    for (var n in r) {
                        ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                    }
                }
                return e;
            }, O.apply(null, arguments);
        }
        function j(e, t) {
            return C(e) || L(e, t) || T(e, t) || P();
        }
        function P() {
            throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
        }
        function T(e, t) {
            if (e) {
                if ("string" == typeof e) return E(e, t);
                var r = {}.toString.call(e).slice(8, -1);
                return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? E(e, t) : void 0;
            }
        }
        function E(e, t) {
            (null == t || t > e.length) && (t = e.length);
            for (var r = 0, n = Array(t); r < t; r++) {
                n[r] = e[r];
            }
            return n;
        }
        function L(e, t) {
            var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
            if (null != r) {
                var n, o, l, i, a = [], u = !0, c = !1;
                try {
                    if (l = (r = r.call(e)).next, 0 === t) {
                        if (Object(r) !== r) return;
                        u = !1;
                    } else for (;!(u = (n = l.call(r)).done) && (a.push(n.value), a.length !== t); u = !0) {
                    }
                } catch (e) {
                    c = !0, o = e;
                } finally {
                    try {
                        if (!u && null != r["return"] && (i = r["return"](), Object(i) !== i)) return;
                    } finally {
                        if (c) throw o;
                    }
                }
                return a;
            }
        }
        function C(e) {
            if (Array.isArray(e)) return e;
        }
        function R(e, t) {
            if (null == e) return {};
            var r, n, o = _(e, t);
            if (Object.getOwnPropertySymbols) {
                var l = Object.getOwnPropertySymbols(e);
                for (n = 0; n < l.length; n++) {
                    r = l[n], t.includes(r) || {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
                }
            }
            return o;
        }
        function _(e, t) {
            if (null == e) return {};
            var r = {};
            for (var n in e) {
                if ({}.hasOwnProperty.call(e, n)) {
                    if (t.includes(n)) continue;
                    r[n] = e[n];
                }
            }
            return r;
        }
        var k = {
            children: l().node,
            defaultLeft: l().number,
            defaultTop: l().number,
            elementRef: l().oneOfType([ l().func, l().object ]),
            left: l().object,
            onScroll: l().func,
            stopScrollPropagation: l().oneOf([ true, false, "window" ]),
            tagName: l().string,
            top: l().object
        };
        var x = {
            passive: false
        };
        function q(r) {
            var n = r.children, o = r.defaultLeft, l = o === void 0 ? 0 : o, i = r.defaultTop, u = i === void 0 ? 0 : i, c = r.elementRef, f = r.left, s = r.onScroll, p = r.stopScrollPropagation, v = r.tagName, m = v === void 0 ? "div" : v, P = r.top, T = R(r, [ "children", "defaultLeft", "defaultTop", "elementRef", "left", "onScroll", "stopScrollPropagation", "tagName", "top" ]);
            var E = (0, e.useState)("visible"), L = j(E, 2), C = L[0], _ = L[1];
            var q = (0, e.useState)("visible"), I = j(q, 2), M = I[0], A = I[1];
            var N = (0, e.useState)(null), X = j(N, 2), Y = X[0], H = X[1];
            var W = (0, w["default"])(Y);
            (0, e.useEffect)((function() {
                if (Y) {
                    if (!W) {
                        Y.scrollTop = u;
                        Y.scrollLeft = l;
                    }
                    if (a()(P)) {
                        Y.scrollTop = (P === null || P === void 0 ? void 0 : P.get()) || 0;
                    }
                    if (a()(f)) {
                        Y.scrollLeft = (f === null || f === void 0 ? void 0 : f.get()) || 0;
                    }
                }
            }), [ Y, W, u, l, P, f ]);
            var D = (0, e.useCallback)((function(e) {
                var t = Y;
                if (!t) {
                    return;
                }
                // if the element is scrollable in both directions, it's not safe to stop propagation.
                                if (t.clientWidth !== t.scrollWidth && t.clientHeight !== t.scrollHeight) {
                    return;
                }
                if (t.clientWidth !== t.scrollWidth) {
                    if (e.deltaX < 0 && t.scrollLeft < Math.abs(e.deltaX)) {
                        t.scrollLeft = 0;
                        e.stopPropagation();
                        e.preventDefault();
                        return;
                    }
                    var r = t.scrollWidth - t.clientWidth;
                    if (e.deltaX && e.deltaX + t.scrollLeft > r) {
                        t.scrollLeft = r;
                        e.stopPropagation();
                        e.preventDefault();
                    }
                }
                if (t.clientHeight !== t.scrollHeight) {
                    if (e.deltaY < 0 && t.scrollTop < Math.abs(e.deltaY)) {
                        t.scrollTop = 0;
                        e.stopPropagation();
                        e.preventDefault();
                        return;
                    }
                    var n = t.scrollHeight - t.clientHeight;
                    if (e.deltaY && e.deltaY + t.scrollTop > n) {
                        t.scrollTop = n;
                        e.stopPropagation();
                        e.preventDefault();
                    }
                }
            }), [ Y ]);
            var B = (0, e.useCallback)((function(e) {
                H(e);
                S(c, e);
            }), [ c ]);
            var F = (0, e.useCallback)((function() {
                _(document.body.style.overflowX);
                A(document.body.style.overflowY);
                document.body.style.overflowX = "hidden";
                document.body.style.overflowY = "hidden";
            }), []);
            var U = (0, e.useCallback)((function() {
                document.body.style.overflowX = C;
                document.body.style.overflowY = M;
            }), [ C, M ]);
            var $ = b.animated[m];
            
            return t().createElement(t().Fragment, null,  t().createElement($, O({}, y()(T, d()(k)), {
                ref: B,
                onMouseEnter: p === "window" ? F : undefined,
                onMouseLeave: p === "window" ? U : undefined,
                onScroll: s,
                scrollTop: P,
                scrollLeft: f
            }),  t().createElement(g.ScrollContainerProvider, {
                value: Y
            }, n)), p === true && Y &&  t().createElement(h(), {
                target: Y,
                eventType: "wheel",
                listener: D,
                options: x
            }));
        }
        q.propTypes = k;
        /* harmony default export */        const I = q;
        // CONCATENATED MODULE: ./src/Scroll/ScrollStyles.ts
        var M = f()(I).withConfig({
            displayName: "ScrollStyles__StyledInner",
            componentId: "sc-1xspttf-0"
        })([ "", ";overflow:auto;overflow-y:auto;overflow-x:hidden;" ], s.mixins.reset("block"));
        // CONCATENATED MODULE: ./src/Scroll/Scroll.tsx
        function A(e) {
            "@babel/helpers - typeof";
            return A = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
                return typeof e;
            } : function(e) {
                return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
            }, A(e);
        }
        function N() {
            return N = Object.assign ? Object.assign.bind() : function(e) {
                for (var t = 1; t < arguments.length; t++) {
                    var r = arguments[t];
                    for (var n in r) {
                        ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                    }
                }
                return e;
            }, N.apply(null, arguments);
        }
        function X(e, t) {
            if (null == e) return {};
            var r, n, o = Y(e, t);
            if (Object.getOwnPropertySymbols) {
                var l = Object.getOwnPropertySymbols(e);
                for (n = 0; n < l.length; n++) {
                    r = l[n], t.includes(r) || {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
                }
            }
            return o;
        }
        function Y(e, t) {
            if (null == e) return {};
            var r = {};
            for (var n in e) {
                if ({}.hasOwnProperty.call(e, n)) {
                    if (t.includes(n)) continue;
                    r[n] = e[n];
                }
            }
            return r;
        }
        function H(e, t) {
            if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function");
        }
        function W(e, t) {
            for (var r = 0; r < t.length; r++) {
                var n = t[r];
                n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), 
                Object.defineProperty(e, Q(n.key), n);
            }
        }
        function D(e, t, r) {
            return t && W(e.prototype, t), r && W(e, r), Object.defineProperty(e, "prototype", {
                writable: !1
            }), e;
        }
        function B(e, t) {
            if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a function");
            e.prototype = Object.create(t && t.prototype, {
                constructor: {
                    value: e,
                    writable: !0,
                    configurable: !0
                }
            }), Object.defineProperty(e, "prototype", {
                writable: !1
            }), t && F(e, t);
        }
        function F(e, t) {
            return F = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, t) {
                return e.__proto__ = t, e;
            }, F(e, t);
        }
        function U(e) {
            var t = G();
            return function() {
                var r, n = J(e);
                if (t) {
                    var o = J(this).constructor;
                    r = Reflect.construct(n, arguments, o);
                } else r = n.apply(this, arguments);
                return $(this, r);
            };
        }
        function $(e, t) {
            if (t && ("object" == A(t) || "function" == typeof t)) return t;
            if (void 0 !== t) throw new TypeError("Derived constructors may only return object or undefined");
            return z(e);
        }
        function z(e) {
            if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
            return e;
        }
        function G() {
            try {
                var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
            } catch (e) {}
            return (G = function t() {
                return !!e;
            })();
        }
        function J(e) {
            return J = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
                return e.__proto__ || Object.getPrototypeOf(e);
            }, J(e);
        }
        function K(e, t, r) {
            return (t = Q(t)) in e ? Object.defineProperty(e, t, {
                value: r,
                enumerable: !0,
                configurable: !0,
                writable: !0
            }) : e[t] = r, e;
        }
        function Q(e) {
            var t = V(e, "string");
            return "symbol" == A(t) ? t : t + "";
        }
        function V(e, t) {
            if ("object" != A(e) || !e) return e;
            var r = e[Symbol.toPrimitive];
            if (void 0 !== r) {
                var n = r.call(e, t || "default");
                if ("object" != A(n)) return n;
                throw new TypeError("@@toPrimitive must return a primitive value.");
            }
            return ("string" === t ? String : Number)(e);
        }
        var Z = {
            children: l().node,
            elementRef: l().oneOfType([ l().func, l().object ]),
            left: l().number,
            onScroll: l().func,
            onScrollComplete: l().func,
            stopScrollPropagation: l().oneOf([ true, false, "window" ]),
            tagName: l().string,
            top: l().number
        };
        var ee = {
            stopScrollPropagation: false,
            tagName: "div"
        };
        var te =  function(e) {
            B(n, e);
            var r = U(n);
            // @docs-props-type ScrollPropsBase
                        function n(e) {
                var o;
                H(this, n);
                o = r.call(this, e);
                K(z(o), "handleRest", (function() {
                    var e, t;
                    (e = (t = o.props).onScrollComplete) === null || e === void 0 ? void 0 : e.call(t);
                }));
                K(z(o), "handleScroll", (function(e) {
                    var t, r;
                    o.setState({
                        currentLeft: e.currentTarget.scrollLeft,
                        currentTop: e.currentTarget.scrollTop
                    });
                    (t = (r = o.props).onScroll) === null || t === void 0 ? void 0 : t.call(r, e);
                }));
                K(z(o), "renderInner", (function(e) {
                    var r = o.props, n = r.children, l = r.elementRef, i = r.left, u = r.top, c = r.onScrollComplete, f = X(r, [ "children", "elementRef", "left", "top", "onScrollComplete" ]);
                    
                    return t().createElement(M, N({
                        "data-test": "scroll"
                    }, f, {
                        key: "inner",
                        top: a()(u) ? e.top : undefined,
                        left: a()(i) ? e.left : undefined,
                        elementRef: l,
                        onScroll: o.handleScroll
                    }), n);
                }));
                o.state = {
                    currentLeft: 0,
                    currentTop: 0
                };
                return o;
            }
            D(n, [ {
                key: "render",
                value: function e() {
                    var r = this.props, n = r.left, o = r.top;
                    var l = this.state, i = l.currentLeft, c = l.currentTop;
                    var f = {};
                    // When isFinite, the prop is defined and therefore animating.
                                        if (a()(n)) {
                        f.left = n;
                    } else {
                        f.left = i;
                    }
                    if (a()(o)) {
                        f.top = o;
                    } else {
                        f.top = c;
                    }
                    
                    return t().createElement(u.Spring, {
                        from: {
                            left: 0,
                            top: 0
                        },
                        to: f,
                        onRest: this.handleRest,
                        config: {
                            precision: 10
                        }
                    }, this.renderInner);
                }
            } ]);
            return n;
        }(e.Component);
        K(te, "propTypes", Z);
        K(te, "defaultProps", ee);
        /* harmony default export */        const re = te;
    }) // CONCATENATED MODULE: ./src/Scroll/index.ts
    ();
    module.exports = n;
    /******/})();
