// @splunk/react-ui/Search.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var n = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, n) => {
            /******/ for (var t in n) {
                /******/ if (e.o(n, t) && !e.o(r, t)) {
                    /******/ Object.defineProperty(r, t, {
                        enumerable: true,
                        get: n[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        Divider: () => /* reexport */ g.Divider,
        Heading: () => /* reexport */ g.Heading,
        Option: () => /* reexport */ U,
        default: () => /* reexport */ fe
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var t =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "lodash/extend"
    const a = require("lodash/extend");
    var u =  e.n(a);
    // CONCATENATED MODULE: external "lodash/has"
    const l = require("lodash/has");
    var c =  e.n(l);
    // CONCATENATED MODULE: external "lodash/memoize"
    const f = require("lodash/memoize");
    var s =  e.n(f);
    // CONCATENATED MODULE: external "lodash/omit"
    const d = require("lodash/omit");
    var v =  e.n(d);
    // CONCATENATED MODULE: external "lodash/pick"
    const p = require("lodash/pick");
    var b =  e.n(p);
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const y = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const m = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/ui-utils/keyboard"
    const h = require("@splunk/ui-utils/keyboard");
    // CONCATENATED MODULE: external "@splunk/react-ui/Menu"
    const g = require("@splunk/react-ui/Menu");
    // CONCATENATED MODULE: external "@splunk/react-ui/usePrevious"
    const O = require("@splunk/react-ui/usePrevious");
    var j =  e.n(O);
    // CONCATENATED MODULE: external "@splunk/react-ui/Popover"
    const w = require("@splunk/react-ui/Popover");
    var S =  e.n(w);
    // CONCATENATED MODULE: external "@splunk/react-ui/ResultsMenu"
    const P = require("@splunk/react-ui/ResultsMenu");
    var k =  e.n(P);
    // CONCATENATED MODULE: external "@splunk/react-ui/Text"
    const C = require("@splunk/react-ui/Text");
    var R =  e.n(C);
    // CONCATENATED MODULE: ./src/Search/Option.tsx
    function I(e) {
        "@babel/helpers - typeof";
        if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") {
            I = function e(r) {
                return typeof r;
            };
        } else {
            I = function e(r) {
                return r && typeof Symbol === "function" && r.constructor === Symbol && r !== Symbol.prototype ? "symbol" : typeof r;
            };
        }
        return I(e);
    }
    function M() {
        M = Object.assign || function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    if (Object.prototype.hasOwnProperty.call(n, t)) {
                        e[t] = n[t];
                    }
                }
            }
            return e;
        };
        return M.apply(this, arguments);
    }
    function D(e, r) {
        if (e == null) return {};
        var n = q(e, r);
        var t, o;
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (o = 0; o < i.length; o++) {
                t = i[o];
                if (r.indexOf(t) >= 0) continue;
                if (!Object.prototype.propertyIsEnumerable.call(e, t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    function q(e, r) {
        if (e == null) return {};
        var n = {};
        var t = Object.keys(e);
        var o, i;
        for (i = 0; i < t.length; i++) {
            o = t[i];
            if (r.indexOf(o) >= 0) continue;
            n[o] = e[o];
        }
        return n;
    }
    function x(e, r) {
        if (!(e instanceof r)) {
            throw new TypeError("Cannot call a class as a function");
        }
    }
    function E(e, r) {
        for (var n = 0; n < r.length; n++) {
            var t = r[n];
            t.enumerable = t.enumerable || false;
            t.configurable = true;
            if ("value" in t) t.writable = true;
            Object.defineProperty(e, t.key, t);
        }
    }
    function _(e, r, n) {
        if (r) E(e.prototype, r);
        if (n) E(e, n);
        return e;
    }
    function A(e, r) {
        if (typeof r !== "function" && r !== null) {
            throw new TypeError("Super expression must either be null or a function");
        }
        e.prototype = Object.create(r && r.prototype, {
            constructor: {
                value: e,
                writable: true,
                configurable: true
            }
        });
        if (r) B(e, r);
    }
    function B(e, r) {
        B = Object.setPrototypeOf || function e(r, n) {
            r.__proto__ = n;
            return r;
        };
        return B(e, r);
    }
    function T(e) {
        var r = V();
        return function n() {
            var t = K(e), o;
            if (r) {
                var i = K(this).constructor;
                o = Reflect.construct(t, arguments, i);
            } else {
                o = t.apply(this, arguments);
            }
            return H(this, o);
        };
    }
    function H(e, r) {
        if (r && (I(r) === "object" || typeof r === "function")) {
            return r;
        }
        return L(e);
    }
    function L(e) {
        if (e === void 0) {
            throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        }
        return e;
    }
    function V() {
        if (typeof Reflect === "undefined" || !Reflect.construct) return false;
        if (Reflect.construct.sham) return false;
        if (typeof Proxy === "function") return true;
        try {
            Date.prototype.toString.call(Reflect.construct(Date, [], (function() {})));
            return true;
        } catch (e) {
            return false;
        }
    }
    function K(e) {
        K = Object.setPrototypeOf ? Object.getPrototypeOf : function e(r) {
            return r.__proto__ || Object.getPrototypeOf(r);
        };
        return K(e);
    }
    function F(e, r, n) {
        if (r in e) {
            Object.defineProperty(e, r, {
                value: n,
                enumerable: true,
                configurable: true,
                writable: true
            });
        } else {
            e[r] = n;
        }
        return e;
    }
    var W = {
        /** @private */
        active: i().bool,
        description: i().string,
        descriptionPosition: i().oneOf([ "right", "bottom" ]),
        disabled: i().bool,
        icon: i().node,
        label: i().string,
        matchRanges: i().arrayOf(i().shape({
            start: i().number.isRequired,
            end: i().number.isRequired
        })),
        onClick: i().func,
        truncate: i().bool,
        value: i().string.isRequired
    };
    var N = {
        active: false,
        descriptionPosition: "bottom",
        disabled: false,
        truncate: false
    };
    /**
 * An option within a `Search`. This inherits from
 * [PureComponent](https://reactjs.org/docs/react-api.html#reactpurecomponent)
 * so any elements passed to it must also be pure.
 */    var z =  function(e) {
        A(n, e);
        var r = T(n);
        function n() {
            var e;
            x(this, n);
            for (var t = arguments.length, o = new Array(t), i = 0; i < t; i++) {
                o[i] = arguments[i];
            }
            e = r.call.apply(r, [ this ].concat(o));
            F(L(e), "item", null);
            F(L(e), "handleClick", (function(r) {
                var n = e.props, t = n.disabled, o = n.onClick, i = n.to, a = n.value;
                if (!t) {
                    o === null || o === void 0 ? void 0 : o(r, {
                        to: i,
                        value: a
                    });
                }
            }));
            return e;
        }
        _(n, [ {
            key: "scrollIntoViewIfNeeded",
            value: function e() {
                var r;
                (r = this.item) === null || r === void 0 ? void 0 : r.scrollIntoViewIfNeeded();
            }
        }, {
            key: "render",
            value: function e() {
                var r = this;
                var n = this.props, o = n.value, i = n.label, a = D(n, [ "value", "label" ]);
                var u = i === undefined ? o : i;
                
                return t().createElement(g.Item, M({
                    ref: function e(n) {
                        r.item = n;
                    },
                    "data-test": "search-option",
                    "data-test-value": o
                }, a, {
                    onClick: this.handleClick,
                    role: "option"
                }), u);
            }
        } ]);
        return n;
    }(n.PureComponent);
    F(z, "propTypes", W);
    F(z, "defaultProps", N);
    F(z, "type", g.Item);
    /* harmony default export */    const U = z;
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function X(e, r) {
        if (e) {
            if (typeof e === "function") {
                e(r);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = r;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Search/Search.tsx
    function Y() {
        Y = Object.assign || function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    if (Object.prototype.hasOwnProperty.call(n, t)) {
                        e[t] = n[t];
                    }
                }
            }
            return e;
        };
        return Y.apply(this, arguments);
    }
    function $(e, r) {
        var n = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var t = Object.getOwnPropertySymbols(e);
            if (r) t = t.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }));
            n.push.apply(n, t);
        }
        return n;
    }
    function G(e) {
        for (var r = 1; r < arguments.length; r++) {
            var n = arguments[r] != null ? arguments[r] : {};
            if (r % 2) {
                $(Object(n), true).forEach((function(r) {
                    J(e, r, n[r]);
                }));
            } else if (Object.getOwnPropertyDescriptors) {
                Object.defineProperties(e, Object.getOwnPropertyDescriptors(n));
            } else {
                $(Object(n)).forEach((function(r) {
                    Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(n, r));
                }));
            }
        }
        return e;
    }
    function J(e, r, n) {
        if (r in e) {
            Object.defineProperty(e, r, {
                value: n,
                enumerable: true,
                configurable: true,
                writable: true
            });
        } else {
            e[r] = n;
        }
        return e;
    }
    function Q(e, r) {
        return te(e) || ne(e, r) || ee(e, r) || Z();
    }
    function Z() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function ee(e, r) {
        if (!e) return;
        if (typeof e === "string") return re(e, r);
        var n = Object.prototype.toString.call(e).slice(8, -1);
        if (n === "Object" && e.constructor) n = e.constructor.name;
        if (n === "Map" || n === "Set") return Array.from(e);
        if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return re(e, r);
    }
    function re(e, r) {
        if (r == null || r > e.length) r = e.length;
        for (var n = 0, t = new Array(r); n < r; n++) {
            t[n] = e[n];
        }
        return t;
    }
    function ne(e, r) {
        if (typeof Symbol === "undefined" || !(Symbol.iterator in Object(e))) return;
        var n = [];
        var t = true;
        var o = false;
        var i = undefined;
        try {
            for (var a = e[Symbol.iterator](), u; !(t = (u = a.next()).done); t = true) {
                n.push(u.value);
                if (r && n.length === r) break;
            }
        } catch (e) {
            o = true;
            i = e;
        } finally {
            try {
                if (!t && a["return"] != null) a["return"]();
            } finally {
                if (o) throw i;
            }
        }
        return n;
    }
    function te(e) {
        if (Array.isArray(e)) return e;
    }
    function oe(e, r) {
        if (e == null) return {};
        var n = ie(e, r);
        var t, o;
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (o = 0; o < i.length; o++) {
                t = i[o];
                if (r.indexOf(t) >= 0) continue;
                if (!Object.prototype.propertyIsEnumerable.call(e, t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    function ie(e, r) {
        if (e == null) return {};
        var n = {};
        var t = Object.keys(e);
        var o, i;
        for (i = 0; i < t.length; i++) {
            o = t[i];
            if (r.indexOf(o) >= 0) continue;
            n[o] = e[o];
        }
        return n;
    }
    var ae = {
        animateLoading: i().bool,
        children: i().node,
        defaultPlacement: i().oneOf([ "above", "below", "vertical" ]),
        defaultValue: i().string,
        describedBy: i().string,
        disabled: i().bool,
        elementRef: i().oneOfType([ i().func, i().object ]),
        error: i().bool,
        footerMessage: i().node,
        inline: i().bool,
        inputRef: i().oneOfType([ i().func, i().object ]),
        isLoadingOptions: i().bool,
        labelledBy: i().string,
        loadingMessage: i().node,
        menuStyle: i().object,
        name: i().string,
        noOptionsMessage: i().node,
        onBlur: i().func,
        onChange: i().func,
        onClose: i().func,
        onFocus: i().func,
        onKeyDown: i().func,
        onOpen: i().func,
        onScroll: i().func,
        onSelect: i().func,
        placeholder: i().string,
        value: i().string
    };
    function ue(e, r) {
        var n = r.clientX, t = r.clientY;
        var o = e.getBoundingClientRect(), i = o.top, a = o.left, u = o.bottom, l = o.right;
        return n > a && n < l && t > i && t < u;
    }
    var le = s()((function(e) {
        var r = e.anchorWidth, n = e.maxHeight, t = e.menuStyle;
        return u()({
            maxHeight: n,
            overflow: "auto",
            width: Math.max(r !== null && r !== void 0 ? r : 0, 200)
        }, t);
    }));
    function ce(e) {
        var r = e.animateLoading, o = r === void 0 ? false : r, i = e.children, a = e.defaultPlacement, u = e.defaultValue, l = e.describedBy, f = e.disabled, s = f === void 0 ? false : f, d = e.elementRef, p = e.error, g = p === void 0 ? false : p, O = e.footerMessage, w = e.inline, P = w === void 0 ? false : w, C = e.inputRef, I = e.isLoadingOptions, M = I === void 0 ? false : I, D = e.labelledBy, q = e.loadingMessage, x = e.menuStyle, E = x === void 0 ? {} : x, _ = e.name, A = e.noOptionsMessage, B = e.onChange, T = e.onClose, H = e.onFocus, L = e.onKeyDown, V = e.onOpen, K = e.placeholder, F = K === void 0 ? (0, 
        m._)("Search...") : K, W = e.value, N = oe(e, [ "animateLoading", "children", "defaultPlacement", "defaultValue", "describedBy", "disabled", "elementRef", "error", "footerMessage", "inline", "inputRef", "isLoadingOptions", "labelledBy", "loadingMessage", "menuStyle", "name", "noOptionsMessage", "onChange", "onClose", "onFocus", "onKeyDown", "onOpen", "placeholder", "value" ]);
        // @docs-props-type SearchPropsBase
                var z = !!i || M;
 // state
                var U = (0, n.useState)(0), $ = Q(U, 2), J = $[0], Z = $[1];
        var ee = (0, n.useState)(false), re = Q(ee, 2), ne = re[0], te = re[1];
        var ie = (0, n.useState)(undefined), ae = Q(ie, 2), ce = ae[0], fe = ae[1];
        var se = (0, n.useState)(u || ""), de = Q(se, 2), ve = de[0], pe = de[1];
        var be = (0, n.useState)((function() {
            return {
                popoverId: (0, y.createDOMID)("popover"),
                menuId: (0, y.createDOMID)("menu"),
                activeItemId: (0, y.createDOMID)("active-item")
            };
        })), ye = Q(be, 1), me = ye[0], he = me.popoverId, ge = me.menuId, Oe = me.activeItemId;
 // previous state
                var je = j()(J);
 // refs
                var we = (0, n.useRef)(null);
        var Se = (0, n.useRef)(null);
        var Pe = (0, n.useRef)(W !== undefined && W !== null);
        var ke = (0, n.useRef)(false);
        var Ce = (0, n.useRef)([]);
        var Re = (0, n.useRef)(je);
        var Ie = (0, n.useRef)(undefined);
        var Me = function e(r) {
            X(we, r);
            X(d, r);
        };
        var De = function e(r) {
            X(Se, r);
            X(C, r);
        };
        var qe = function e() {
            return Pe.current ? W : ve;
        };
        var xe = function e() {
            var r = qe();
            var t = n.Children.toArray(i).filter(n.isValidElement);
            var o = !ce && t.find((function(e) {
                return e.props.label !== undefined && e.props.value === r;
            }));
            if (o) {
                return o === null || o === void 0 ? void 0 : o.props.label;
            }
            return r;
        };
        var Ee = function e() {
            te(true);
            Z(-1);
            V === null || V === void 0 ? void 0 : V();
        };
        var _e = function e() {
            te(false);
            Re.current = null;
            T === null || T === void 0 ? void 0 : T();
        };
        var Ae = function e() {
            var r;
            (r = Se.current) === null || r === void 0 ? void 0 : r.focus();
        };
        var Be = function e(r, n) {
            /* SUI-930 On IE 11 this handler is essentially deferred after calling
     * this.textInput.focus(). this.focusCalledInternally enables the focus event to be ignored
     * when the menu closes. */
            if (ke.current) {
                ke.current = false;
            } else {
                Ee();
            }
            H === null || H === void 0 ? void 0 : H(r, n);
        };
        var Te = function e(r, n) {
            var t = n.value;
            if (!Pe.current) {
                pe(t);
            }
            Z(-1);
            fe(t);
            B === null || B === void 0 ? void 0 : B(r, {
                value: t,
                name: _
            });
        };
        var He = function e(r, n) {
            Te(r, G({}, n));
            ke.current = true;
            Ae();
            _e();
        };
        var Le = function e(r) {
            var n = Ce.current.length;
            var t = (0, h.keycode)(r.nativeEvent);
            if (ne) {
                switch (t) {
                  case "enter":
                    {
                        var o;
                        // triggering onClick so that links can be activated with the keyboard
                        // since the menu items themselves never receive focus
                        // this will implicitly call handleSelectOption()
                                                (o = document.getElementById(Oe)) === null || o === void 0 ? void 0 : o.click();
                        break;
                    }

                  case "tab":
                    _e();
                    break;

                  case "down":
                    Z(Math.min(J + 1, n - 1));
                    break;

                  case "up":
                    Z(Math.max(J - 1, 0));
                    break;

                  default:
 // do nothing
                                }
            } else if ((0, h.addsCharacter)(r.nativeEvent) !== false || // Safari 9.0 returns undefined
            t === "enter" || t === "backspace" || t === "down" || t === "up") {
                Ee();
            }
            L === null || L === void 0 ? void 0 : L(r);
        };
        var Ve = function e() {
            if (!ne && !s) {
                Ee();
            }
        };
        var Ke = function e(r) {
            var n = r.event, t = r.reason;
            var o = t === "offScreen" || t === "escapeKey" || t === "clickAway" && we.current && !ue(we.current, n);
            if (o) {
                _e();
            }
        };
        var Fe = function e(r) {
            if (Re.current !== J) {
                r === null || r === void 0 ? void 0 : r.scrollIntoViewIfNeeded();
            }
        };
        var We = function e(r) {
            var a = r.anchorWidth, u = r.maxHeight;
            var l = n.Children.toArray(i).filter(n.isValidElement);
 // Highlight Active
                        Ie.current = undefined;
 // Only update the menu options if they are valid and the dropdown is open.
            // This is necessary to handle the outro animation of the dropdown.
                        if (z && ne) {
                Ce.current = l.map((function(e, r) {
                    var t = function r(n, t) {
                        var o, i;
                        var a = (o = (i = e.props).onClick) === null || o === void 0 ? void 0 : o.call(i, n, t);
                        if (a !== false) {
                            He(n, t);
                        }
                    };
                    if (!c()(e.props, "active")) {
                        // ignore Headings and Dividers
                        return e;
                    }
                    if (r === J) {
                        if (!e.props.disabled) {
                            Ie.current = e.props.value;
                        }
                        
                        return (0, n.cloneElement)(e, {
                            ref: Fe,
                            id: Oe,
                            onClick: t,
                            active: true
                        });
                    }
                    
                    return (0, n.cloneElement)(e, {
                        onClick: t
                    });
                }));
            }
            var f = le({
                anchorWidth: a,
                maxHeight: u,
                menuStyle: E
            });
            
            return t().createElement(k(), Y({
                style: f,
                focusMode: "never",
                isLoading: M
            }, b()(N, "onScroll"), {
                noOptionsMessage: A,
                footerMessage: O,
                animateLoading: o,
                loadingMessage: q,
                menuId: ge
            }), Ce.current);
        };
        var Ne = qe();
        var ze = xe();
        var Ue = ne && !!we.current && z;
        var Xe = z ? "combobox" : "searchbox";
        var Ye = {
            "aria-activedescendant": Ue && J > -1 ? Oe : undefined,
            "aria-expanded": z ? ne && !!we.current : undefined,
            "aria-controls": Ue ? ge : undefined
        };
        
        return t().createElement(R(), Y({
            "data-test": "search"
        }, v()(N, "onScroll"), {
            appearance: "search",
            autoCapitalize: "off",
            autoComplete: "off",
            autoCorrect: "off",
            "data-test-popover-id": he,
            "data-test-label": ze,
            "data-test-value": Ne,
            "data-test-open": Ue,
            describedBy: l,
            disabled: s,
            elementRef: Me,
            error: g,
            inline: P,
            inputRef: De,
            labelledBy: D,
            name: _,
            onFocus: Be,
            onClick: Ve,
            onChange: Te,
            onKeyDown: Le,
            placeholder: F,
            spellCheck: false,
            value: ze,
            role: Xe
        }, Ye),  t().createElement(S(), {
            anchor: function() {
                return we.current;
            }(),
            autoCloseWhenOffScreen: true,
            canCoverAnchor: false,
            defaultPlacement: a,
            id: he,
            onRequestClose: Ke,
            open: Ue,
            repositionMode: "flip"
        }, We));
    }
    ce.propTypes = ae;
    ce.Option = U;
    ce.Divider = g.Divider;
    ce.Heading = g.Heading;
    /* harmony default export */    const fe = ce;
    // CONCATENATED MODULE: ./src/Search/index.ts
    module.exports = r;
    /******/})();
