// @splunk/react-ui/SidePanel.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        default: () => /* reexport */ V
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "lodash/keys"
    const a = require("lodash/keys");
    var l =  e.n(a);
    // CONCATENATED MODULE: external "lodash/omit"
    const u = require("lodash/omit");
    var c =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/react-ui/ModalLayer"
    const s = require("@splunk/react-ui/ModalLayer");
    var p =  e.n(s);
    // CONCATENATED MODULE: external "styled-components"
    const f = require("styled-components");
    var y =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/react-ui/TransitionOpen"
    const d = require("@splunk/react-ui/TransitionOpen");
    var b =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/themes"
    const m = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/SidePanel/SidePanelStyles.ts
    var v = (0, f.css)([ "", " position:fixed;flex-direction:column;box-shadow:", ";background-color:", ";z-index:", ";" ], m.mixins.reset("flex"), (0, 
    m.pick)({
        enterprise: "0 0 12px rgba(0, 0, 0, 0.3)",
        prisma: m.variables.overlayShadow
    }), m.variables.backgroundColorSidebar, m.variables.zindexModal);
    var h = y()(b()).withConfig({
        displayName: "SidePanelStyles__StyledLeftPanelTransitionOpen",
        componentId: "mtbcuz-0"
    })([ "", ";top:0;left:0;height:100vh;" ], v);
    var g = y()(b()).withConfig({
        displayName: "SidePanelStyles__StyledRightPanelTransitionOpen",
        componentId: "mtbcuz-1"
    })([ "", ";right:0;top:0;height:100vh;" ], v);
    var O = y()(b()).withConfig({
        displayName: "SidePanelStyles__StyledTopPanelTransitionOpen",
        componentId: "mtbcuz-2"
    })([ "", ";top:0;left:0;width:100vw;" ], v);
    var S = y()(b()).withConfig({
        displayName: "SidePanelStyles__StyledBottomPanelTransitionOpen",
        componentId: "mtbcuz-3"
    })([ "", ";left:0;bottom:0;width:100vw;" ], v);
    // CONCATENATED MODULE: ./src/SidePanel/SidePanel.tsx
    function w(e) {
        "@babel/helpers - typeof";
        return w = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, w(e);
    }
    function P() {
        return P = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, P.apply(null, arguments);
    }
    function j(e) {
        return k(e) || _(e) || C(e) || A();
    }
    function A() {
        throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function C(e, t) {
        if (e) {
            if ("string" == typeof e) return E(e, t);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? E(e, t) : void 0;
        }
    }
    function _(e) {
        if ("undefined" != typeof Symbol && null != e[Symbol.iterator] || null != e["@@iterator"]) return Array.from(e);
    }
    function k(e) {
        if (Array.isArray(e)) return E(e);
    }
    function E(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var r = 0, n = Array(t); r < t; r++) {
            n[r] = e[r];
        }
        return n;
    }
    function T(e, t) {
        var r = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            t && (n = n.filter((function(t) {
                return Object.getOwnPropertyDescriptor(e, t).enumerable;
            }))), r.push.apply(r, n);
        }
        return r;
    }
    function q(e) {
        for (var t = 1; t < arguments.length; t++) {
            var r = null != arguments[t] ? arguments[t] : {};
            t % 2 ? T(Object(r), !0).forEach((function(t) {
                U(e, t, r[t]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : T(Object(r)).forEach((function(t) {
                Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
            }));
        }
        return e;
    }
    function x(e, t) {
        if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function");
    }
    function N(e, t) {
        for (var r = 0; r < t.length; r++) {
            var n = t[r];
            n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), 
            Object.defineProperty(e, $(n.key), n);
        }
    }
    function F(e, t, r) {
        return t && N(e.prototype, t), r && N(e, r), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function M(e, t) {
        if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), t && I(e, t);
    }
    function I(e, t) {
        return I = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, t) {
            return e.__proto__ = t, e;
        }, I(e, t);
    }
    function R(e) {
        var t = L();
        return function() {
            var r, n = B(e);
            if (t) {
                var o = B(this).constructor;
                r = Reflect.construct(n, arguments, o);
            } else r = n.apply(this, arguments);
            return z(this, r);
        };
    }
    function z(e, t) {
        if (t && ("object" == w(t) || "function" == typeof t)) return t;
        if (void 0 !== t) throw new TypeError("Derived constructors may only return object or undefined");
        return D(e);
    }
    function D(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function L() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return (L = function t() {
            return !!e;
        })();
    }
    function B(e) {
        return B = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, B(e);
    }
    function U(e, t, r) {
        return (t = $(t)) in e ? Object.defineProperty(e, t, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = r, e;
    }
    function $(e) {
        var t = G(e, "string");
        return "symbol" == w(t) ? t : t + "";
    }
    function G(e, t) {
        if ("object" != w(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var n = r.call(e, t || "default");
            if ("object" != w(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    /** @public */    var H = {
        children: i().node,
        dockPosition: i().oneOf([ "top", "bottom", "left", "right" ]),
        innerClassName: i().string,
        innerStyle: i().object,
        onAnimationEnd: i().func,
        onRequestClose: i().func,
        open: i().bool,
        outerClassName: i().string,
        outerStyle: i().object,
        useLayerForClickAway: i().bool
    };
    var J = {
        dockPosition: "right",
        open: false,
        useLayerForClickAway: true
    };
    var K = {
        left: "slideFromLeft",
        right: "slideFromRight",
        top: "slideFromTop",
        bottom: "slideFromBottom"
    };
    var Q =  function(e) {
        M(r, e);
        var t = R(r);
        F(r, null, [ {
            key: "getDerivedStateFromProps",
            // @docs-props-type SidePanelPropsBase
            value: function e(t, r) {
                if (t.open !== r.prevOpen) {
                    return {
                        panelAnimating: true,
                        prevOpen: t.open
                    };
                }
                return null;
            }
        } ]);
        function r(e) {
            var o;
            x(this, r);
            o = t.call(this, e);
            U(D(o), "handlePanelAnimationEnd", (function() {
                var e, t;
                o.setState({
                    panelAnimating: false
                });
                (e = (t = o.props).onAnimationEnd) === null || e === void 0 ? void 0 : e.call(t);
            }));
            U(D(o), "renderModal", (function() {
                var e = o.props, t = e.children, i = e.dockPosition, a = e.innerClassName, u = e.innerStyle, s = e.outerClassName, p = e.outerStyle;
                var f = {
                    left: h,
                    right: g,
                    top: O,
                    bottom: S
                }[i];
                var y = i === "left" || i === "right" ? {
                    height: "100%"
                } : {
                    width: "100%"
                };
                
                return n().createElement(f, P({
                    animateOnMount: true,
                    animation: K[i],
                    "data-test": "side-panel",
                    innerClassName: a,
                    innerStyle: q(q({}, y), u),
                    onAnimationEnd: o.handlePanelAnimationEnd,
                    open: o.props.open,
                    outerClassName: s,
                    outerStyle: p,
                    retainFocus: true,
                    role: "dialog",
                    tabIndex: -1,
                    takeFocus: true
                }, c()(o.props, [].concat(j(l()(r.propTypes)), [ "onAnimationEnd" ]))), t);
            }));
            o.state = {
                panelAnimating: false,
                prevOpen: e.open
            };
            return o;
        }
        F(r, [ {
            key: "render",
            value: function e() {
                var t = this.props, r = t.open, o = t.onRequestClose, i = t.useLayerForClickAway;
                var a = this.state.panelAnimating;
                var l = i ? "interactive" : "hidden";
                
                return n().createElement(p(), {
                    childrenInAnimation: a,
                    onRequestClose: o,
                    open: r,
                    renderModal: this.renderModal,
                    scrim: l
                });
            }
        } ]);
        return r;
    }(r.Component);
    U(Q, "propTypes", H);
    U(Q, "defaultProps", J);
    /* harmony default export */    const V = Q;
    // CONCATENATED MODULE: ./src/SidePanel/index.ts
    module.exports = t;
    /******/})();
