// @splunk/react-ui/Slider.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        default: () => /* reexport */ Z
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const i = require("prop-types");
    var a =  e.n(i);
    // CONCATENATED MODULE: external "decimal.js-light"
    const o = require("decimal.js-light");
    var s =  e.n(o);
    // CONCATENATED MODULE: external "lodash/has"
    const l = require("lodash/has");
    var u =  e.n(l);
    // CONCATENATED MODULE: external "lodash/keys"
    const c = require("lodash/keys");
    var p =  e.n(c);
    // CONCATENATED MODULE: external "lodash/omit"
    const d = require("lodash/omit");
    var f =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/ui-utils/keyboard"
    const v = require("@splunk/ui-utils/keyboard");
    // CONCATENATED MODULE: external "@splunk/themes"
    const h = require("@splunk/themes");
    // CONCATENATED MODULE: external "@splunk/react-ui/EventListener"
    const m = require("@splunk/react-ui/EventListener");
    var b =  e.n(m);
    // CONCATENATED MODULE: external "@splunk/react-ui/Popover"
    const y = require("@splunk/react-ui/Popover");
    var g =  e.n(y);
    // CONCATENATED MODULE: external "styled-components"
    const k = require("styled-components");
    var x =  e.n(k);
    // CONCATENATED MODULE: external "tinycolor2"
    const w = require("tinycolor2");
    var S =  e.n(w);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const C = require("@splunk/react-ui/Box");
    var T =  e.n(C);
    // CONCATENATED MODULE: ./src/Slider/SliderStyles.ts
    var V = x()(T()).withConfig({
        displayName: "SliderStyles__StyledBox",
        componentId: "sc-16ooit4-0"
    })([ "", ";flex:1 1 auto;&[data-inline]{width:300px;}" ], h.mixins.reset("block"));
    var E = x().div.withConfig({
        displayName: "SliderStyles__StyledInput",
        componentId: "sc-16ooit4-1"
    })([ "flex:1 0 0;vertical-align:middle;position:relative;cursor:default;height:", ";max-width:100%;" ], h.variables.inputHeight);
    var M = "1px";
    var O = x().div.withConfig({
        displayName: "SliderStyles__StyledSliderBar",
        componentId: "sc-16ooit4-2"
    })([ "position:absolute;top:", ";left:0;height:", ";width:100%;border-radius:2.5px;", "" ], (0, 
    h.pick)({
        enterprise: "12px",
        prisma: "18px"
    }), (0, h.pick)({
        enterprise: "5px",
        prisma: "4px"
    }), (function(e) {
        var t = e.$disabled, r = e.$error, n = e.$position, i = e.$stepMarksWidth;
        if (t) {
            return (0, k.css)([ "background-color:", ";" ], (0, h.pick)({
                enterprise: h.variables.gray92,
                prisma: h.variables.contentColorDisabled
            }));
        }
        var a = r ? h.variables.accentColorNegative : (0, h.pick)({
            prisma: h.variables.contentColorActive,
            enterprise: {
                light: h.variables.gray22,
                dark: h.variables.white
            }
        });
        var o = (0, h.pick)({
            enterprise: {
                dark: h.variables.gray20,
                light: h.variables.gray96
            },
            prisma: h.variables.neutral100
        });
        var s = (0, h.pick)({
            enterprise: h.variables.gray60,
            prisma: h.variables.interactiveColorBorder
        });
        return i !== undefined ? (0, k.css)([ "background-image:linear-gradient( to right,", ",", " ", ",transparent ", ",transparent 100% ),repeating-linear-gradient( to right,", ",", " ", ",transparent ", ",transparent ", "% ),linear-gradient( to right,", ",", " ", "%,", " ", "%,", " );" ], a, a, M, M, o, o, M, M, i, a, a, n, s, n, s) : (0, 
        k.css)([ "background-image:linear-gradient( to right,", ",", " ", "%,", " ", "%,", " );" ], a, a, n, s, n, s);
    }));
    var B = (0, h.pick)({
        enterprise: 18,
        prisma: 20
    });
    var _ = x().button.withConfig({
        displayName: "SliderStyles__StyledSliderThumb",
        componentId: "sc-16ooit4-3"
    })([ "", ";display:block;position:absolute;cursor:pointer;width:", "px;height:", "px;border-radius:", ";border-width:0;background-color:", ";margin-left:-8px;top:", ";left:", "%;&:hover:not([disabled]){outline:0;z-index:1;", "}&:focus:not([disabled]),&:active{outline:0;box-shadow:", ";z-index:1;", "}&[disabled]{cursor:not-allowed;background-color:", ";}", "" ], h.mixins.reset("inline"), B, B, (0, 
    h.pick)({
        enterprise: "9px",
        prisma: "10px"
    }), (0, h.pick)({
        enterprise: {
            light: h.variables.gray22,
            dark: h.variables.white
        },
        prisma: h.variables.contentColorActive
    }), (0, h.pick)({
        enterprise: "6px",
        prisma: "10px"
    }), (function(e) {
        var t = e.$position;
        return t;
    }), (0, h.pick)({
        enterprise: (0, k.css)([ "box-shadow:", ";" ], h.variables.focusShadow)
    }), h.variables.focusShadow, (0, h.pick)({
        prisma: (0, k.css)([ "background-color:", ";" ], h.mixins.overlayColors(h.variables.contentColorActive, h.variables.interactiveColorOverlayActive)),
        enterprise: (0, k.css)([ "border-color:", ";" ], (function(e) {
            return S()(h.variables.focusColor(e)).setAlpha(.8).toRgbString();
        }))
    }), (0, h.pick)({
        enterprise: h.variables.gray92,
        prisma: h.variables.neutral400
    }), (function(e) {
        var t = e.$error;
        return t && (0, k.css)([ "background-color:", ";&:focus:not([disabled]),&:active{background-color:", ";}" ], h.variables.accentColorNegative, (0, 
        h.pick)({
            enterprise: h.variables.errorColorD40,
            prisma: h.mixins.overlayColors(h.variables.accentColorNegative, h.variables.interactiveColorOverlayActive)
        }));
    }));
    var P = x().div.withConfig({
        displayName: "SliderStyles__StyledPopoverContent",
        componentId: "sc-16ooit4-4"
    })([ "padding:", ";font-size:", ";line-height:", ";" ], (0, h.pick)({
        enterprise: h.variables.spacingSmall,
        prisma: "12px 16px"
    }), (0, h.pick)({
        enterprise: h.variables.fontSizeSmall,
        prisma: "14px"
    }), (0, h.pick)({
        enterprise: "18px",
        prisma: "24px"
    }));
    var j = (0, k.css)([ "flex:0 0 auto;line-height:", ";", "" ], h.variables.inputHeight, (function(e) {
        var t = e.$disabled;
        return t && (0, k.css)([ "color:", ";" ], (0, h.pick)({
            enterprise: h.variables.textDisabledColor,
            prisma: h.variables.contentColorDisabled
        }));
    }));
    var D = x()(T()).withConfig({
        displayName: "SliderStyles__StyledMinLabelBox",
        componentId: "sc-16ooit4-5"
    })([ "", ";margin-right:", ";text-align:right;" ], j, (0, h.pick)({
        enterprise: "15px",
        prisma: "16px"
    }));
    var q = x()(T()).withConfig({
        displayName: "SliderStyles__StyledMaxLabelBox",
        componentId: "sc-16ooit4-6"
    })([ "", ";margin-left:", ";text-align:left;" ], j, (0, h.pick)({
        enterprise: "15px",
        prisma: "16px"
    }));
    // CONCATENATED MODULE: ./src/Slider/Slider.tsx
    function $(e) {
        "@babel/helpers - typeof";
        if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") {
            $ = function e(t) {
                return typeof t;
            };
        } else {
            $ = function e(t) {
                return t && typeof Symbol === "function" && t.constructor === Symbol && t !== Symbol.prototype ? "symbol" : typeof t;
            };
        }
        return $(e);
    }
    function N() {
        N = Object.assign || function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    if (Object.prototype.hasOwnProperty.call(r, n)) {
                        e[n] = r[n];
                    }
                }
            }
            return e;
        };
        return N.apply(this, arguments);
    }
    function R(e, t) {
        if (!(e instanceof t)) {
            throw new TypeError("Cannot call a class as a function");
        }
    }
    function F(e, t) {
        for (var r = 0; r < t.length; r++) {
            var n = t[r];
            n.enumerable = n.enumerable || false;
            n.configurable = true;
            if ("value" in n) n.writable = true;
            Object.defineProperty(e, n.key, n);
        }
    }
    function I(e, t, r) {
        if (t) F(e.prototype, t);
        if (r) F(e, r);
        return e;
    }
    function L(e, t) {
        if (typeof t !== "function" && t !== null) {
            throw new TypeError("Super expression must either be null or a function");
        }
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: true,
                configurable: true
            }
        });
        if (t) A(e, t);
    }
    function A(e, t) {
        A = Object.setPrototypeOf || function e(t, r) {
            t.__proto__ = r;
            return t;
        };
        return A(e, t);
    }
    function W(e) {
        var t = U();
        return function r() {
            var n = H(e), i;
            if (t) {
                var a = H(this).constructor;
                i = Reflect.construct(n, arguments, a);
            } else {
                i = n.apply(this, arguments);
            }
            return z(this, i);
        };
    }
    function z(e, t) {
        if (t && ($(t) === "object" || typeof t === "function")) {
            return t;
        }
        return K(e);
    }
    function K(e) {
        if (e === void 0) {
            throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        }
        return e;
    }
    function U() {
        if (typeof Reflect === "undefined" || !Reflect.construct) return false;
        if (Reflect.construct.sham) return false;
        if (typeof Proxy === "function") return true;
        try {
            Date.prototype.toString.call(Reflect.construct(Date, [], (function() {})));
            return true;
        } catch (e) {
            return false;
        }
    }
    function H(e) {
        H = Object.setPrototypeOf ? Object.getPrototypeOf : function e(t) {
            return t.__proto__ || Object.getPrototypeOf(t);
        };
        return H(e);
    }
    function X(e, t, r) {
        if (t in e) {
            Object.defineProperty(e, t, {
                value: r,
                enumerable: true,
                configurable: true,
                writable: true
            });
        } else {
            e[t] = r;
        }
        return e;
    }
    var G = {
        defaultValue: a().number,
        describedBy: a().string,
        disabled: a().bool,
        displayValue: a().string,
        elementRef: a().oneOfType([ a().func, a().object ]),
        error: a().bool,
        inline: a().bool,
        labelledBy: a().string,
        max: a().number,
        maxLabel: a().node,
        min: a().number,
        minLabel: a().node,
        name: a().string,
        onChange: a().func,
        /** @private. */
        required: a().bool,
        step: a().number,
        stepMarks: a().oneOf([ "focus", "always", "never" ]),
        /** @private */
        splunkTheme: a().object,
        value: a().number
    };
    var J = {
        disabled: false,
        error: false,
        inline: false,
        min: 1,
        max: 5,
        step: 1,
        stepMarks: "focus"
    };
    var Q =  function(e) {
        L(r, e);
        var t = W(r);
        // @docs-props-type SliderPropsBase
                function r(e) {
            var n;
            R(this, r);
            n = t.call(this, e);
            X(K(n), "controlledExternally", void 0);
            X(K(n), "sliderThumb", null);
            X(K(n), "checkPositionBounds", (function(e) {
                if (e > 100) {
                    return 100;
                }
                if (e < 0) {
                    return 0;
                }
                return e;
            }));
            X(K(n), "checkValueBounds", (function(e) {
                if (e > n.props.max) {
                    return n.props.max;
                }
                if (e < n.props.min) {
                    return n.props.min;
                }
                return e;
            }));
            X(K(n), "handleBlur", (function() {
                n.setState({
                    isFocused: false,
                    showTooltip: false
                });
            }));
            X(K(n), "handleClick", (function(e) {
                var t = n.percentageFromEvent(e);
                if (t !== undefined) {
                    var r = n.roundValueToStep(n.positionToValue(t));
                    n.setValue(e, {
                        value: r
                    });
                }
            }));
            X(K(n), "handleFocus", (function() {
                n.setState({
                    isFocused: true,
                    showTooltip: true
                });
            }));
            X(K(n), "getValueOffset", (function(e, t) {
                var r = n.props, i = r.min, a = r.step;
                var o = new (s())(e).minus(i).modulo(a).toNumber();
                if (o > 0) {
                    return t === "forward" ? a - o : o;
                }
                return a;
            }));
            X(K(n), "handleKeyDown", (function(e) {
                var t = n.getValue();
                if (t !== undefined) {
                    if ((0, v.keycode)(e.nativeEvent) === "right") {
                        t = n.checkValueBounds(t + n.getValueOffset(t, "forward"));
                    } else if ((0, v.keycode)(e.nativeEvent) === "left") {
                        t = n.checkValueBounds(t - n.getValueOffset(t, "backward"));
                    } else if ((0, v.keycode)(e.nativeEvent) === "up") {
                        e.preventDefault();
                        t = n.checkValueBounds(t + n.getValueOffset(t, "forward"));
                    } else if ((0, v.keycode)(e.nativeEvent) === "down") {
                        e.preventDefault();
                        t = n.checkValueBounds(t - n.getValueOffset(t, "backward"));
                    } else {
                        return;
                    }
                    t = n.roundValueToStep(t);
                    n.setValue(e, {
                        value: t
                    });
                }
            }));
            X(K(n), "handleMount", (function(e) {
                n.setState({
                    sliderBar: e
                });
            }));
            X(K(n), "handleMouseDown", (function(e) {
                e.preventDefault();
                n.focus();
                n.setState({
                    selected: true,
                    showTooltip: true
                });
            }));
            X(K(n), "handleMouseEnter", (function() {
                n.setState({
                    showTooltip: true
                });
            }));
            X(K(n), "handleMouseLeave", (function() {
                if (!n.state.selected) {
                    n.setState({
                        showTooltip: false
                    });
                }
            }));
            X(K(n), "handleMouseMove", (function(e) {
                if (n.state.selected) {
                    var t = n.percentageFromEvent(e);
                    if (t !== undefined) {
                        var r = n.roundValueToStep(n.positionToValue(t));
                        n.setValue(e, {
                            value: r
                        });
                    }
                }
            }));
            X(K(n), "handleMouseUp", (function() {
                n.setState({
                    selected: false,
                    showTooltip: false
                });
            }));
            X(K(n), "percentageFromEvent", (function(e) {
                if (n.state.sliderBar === null) {
                    return undefined;
                }
                var t = n.state.sliderBar.getBoundingClientRect();
                var r = e.clientX - t.left;
                return n.checkPositionBounds(r / t.width * 100);
            }));
            X(K(n), "positionToValue", (function(e) {
                var t = n.props.max - n.props.min;
                return e / 100 * t + n.props.min;
            }));
            X(K(n), "roundValueToStep", (function(e) {
                var t = n.props, r = t.step, i = t.min, a = t.max;
 // Handles if the max is in between steps but val is larger than the halfway point between last step mark and max
                                if (e > a - (a - i) % r / 2) {
                    return a;
                }
                return new (s())(e).minus(i).div(r).todp(0).mul(r).add(i).toNumber();
            }));
            X(K(n), "valueToPosition", (function(e) {
                if (e === undefined) {
                    return undefined;
                }
                var t = n.props.max - n.props.min;
                return (e - n.props.min) / t * 100;
            }));
            n.controlledExternally = u()(e, "value");
            var i = u()(e, "defaultValue") ? e.defaultValue : n.roundValueToStep((e.max - e.min) / 2);
            n.state = {
                isFocused: false,
                selected: false,
                showTooltip: false,
                sliderBar: null,
                value: n.isControlled() ? undefined : i
            };
            if (false) {}
            if (e.min >= e.max && process.env.NODE_ENV !== "production") {}
            return n;
        }
        I(r, [ {
            key: "componentDidUpdate",
            value: function e(t) {
                if (false) {}
                if (false) {}
            }
        }, {
            key: "getValue",
            value: function e() {
                return this.isControlled() ? this.props.value : this.state.value;
            }
        }, {
            key: "setValue",
            value: function e(t, r) {
                var n = r.value;
                var i = this.getValue();
                var a = this.props.name;
                if (i !== n) {
                    var o, s;
                    (o = (s = this.props).onChange) === null || o === void 0 ? void 0 : o.call(s, t, {
                        value: n,
                        name: a
                    });
                    if (!this.isControlled()) {
                        this.setState({
                            value: n
                        });
                    }
                }
            }
        }, {
            key: "focus",
            /**
     * Places focus on the slider.
     */
            value: function e() {
                var t;
                (t = this.sliderThumb) === null || t === void 0 ? void 0 : t.focus();
            }
        }, {
            key: "stepWidthInPercentage",
            value: function e() {
                return this.props.step * 100 / (this.props.max - this.props.min);
            }
        }, {
            key: "stepMarksWidth",
            value: function e() {
                // don't render stepMarks if they are too small i.e. stepWidth is less than SliderThumb divided by 2
                // in that case, undefined is returned
                var t = this.props, r = t.max, n = t.min, i = t.step, a = t.splunkTheme;
                var o = this.state.sliderBar;
                if (o === null) {
                    return undefined;
                }
                var s = Math.round((r - n) / i);
                var l = o.getBoundingClientRect();
                var u = B(a);
                return l.width / s > u / 2 ? this.stepWidthInPercentage() : undefined;
            }
        }, {
            key: "isControlled",
            value: function e() {
                return this.controlledExternally;
            }
        }, {
            key: "render",
            value: function e() {
                var t = this;
                var i = this.props, a = i.describedBy, o = i.disabled, s = i.displayValue, l = i.elementRef, u = i.error, c = i.inline, d = i.labelledBy, v = i.max, h = i.maxLabel, m = i.min, y = i.minLabel, k = i.required, x = i.stepMarks;
                var w = this.state, S = w.isFocused, C = w.selected, T = w.sliderBar;
                var M = this.getValue();
                var B = this.valueToPosition(M);
                var j = x === "focus" && S || x === "always";
                var $ = j && T ? this.stepMarksWidth() : undefined;
                var R = f()(this.props, p()(r.propTypes));
                /* eslint-disable jsx-a11y/no-static-element-interactions */                
                return n().createElement(V, N({
                    "data-test": "slider",
                    "data-test-value": M,
                    elementRef: l,
                    flex: true,
                    inline: c
                }, R), C &&  n().createElement(n().Fragment, null,  n().createElement(b(), {
                    target: window,
                    eventType: "mouseup",
                    listener: this.handleMouseUp
                }),  n().createElement(b(), {
                    target: window,
                    eventType: "mousemove",
                    listener: this.handleMouseMove
                })), y !== null &&  n().createElement(D, {
                    "data-test": "min-label",
                    $disabled: o
                }, y || m),  n().createElement(E, {
                    onClick: o ? undefined : this.handleClick
                },  n().createElement(O, {
                    "aria-required": k,
                    "data-test": "bar",
                    $disabled: o,
                    ref: this.handleMount,
                    $error: u,
                    $position: B,
                    $stepMarksWidth: $
                }),  n().createElement(_, {
                    "aria-describedby": a,
                    "aria-labelledby": d,
                    "aria-invalid": u,
                    "aria-valuemax": v,
                    "aria-valuemin": m,
                    "aria-valuenow": M,
                    "data-test": "handle",
                    onBlur: this.handleBlur,
                    onMouseEnter: this.handleMouseEnter,
                    onMouseLeave: this.handleMouseLeave,
                    onKeyDown: o ? undefined : this.handleKeyDown,
                    onMouseDown: o ? undefined : this.handleMouseDown,
                    onFocus: this.handleFocus,
                    ref: function e(r) {
                        t.sliderThumb = r;
                    },
                    role: "slider",
                    disabled: o,
                    $error: u,
                    $position: B
                }),  n().createElement(g(), {
                    open: this.state.showTooltip,
                    anchor: this.sliderThumb,
                    appearance: "inverted",
                    align: "center"
                },  n().createElement(P, null, s || M))), h !== null &&  n().createElement(q, {
                    "data-test": "max-label",
                    $disabled: o
                }, h || v));
                /* eslint-enable jsx-a11y/no-static-element-interactions */            }
        } ]);
        return r;
    }(r.Component);
    X(Q, "propTypes", G);
    X(Q, "defaultProps", J);
    var Y = (0, h.withSplunkTheme)(Q);
    Y.propTypes = Q.propTypes;
    /* harmony default export */    const Z = Y;
    // CONCATENATED MODULE: ./src/Slider/index.ts
    module.exports = t;
    /******/})();
