// @splunk/react-ui/SplitButton.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        Item: () => /* reexport */ J,
        default: () => /* reexport */ X
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var n =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const i = require("prop-types");
    var o =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/ChevronDown"
    const a = require("@splunk/react-icons/enterprise/ChevronDown");
    var l =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/react-ui/Dropdown"
    const c = require("@splunk/react-ui/Dropdown");
    var u =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/react-ui/Menu"
    const s = require("@splunk/react-ui/Menu");
    var p =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const f = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "styled-components"
    const d = require("styled-components");
    var b =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const v = require("@splunk/react-ui/Button");
    var y =  e.n(v);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const m = require("@splunk/react-ui/Box");
    var h =  e.n(m);
    // CONCATENATED MODULE: external "@splunk/themes"
    const g = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/SplitButton/SplitButtonStyles.ts
    var O = b()(y()).withConfig({
        displayName: "SplitButtonStyles__StyledButton",
        componentId: "sc-9igsyh-0"
    })([ "", "" ], (0, g.pick)({
        prisma: function e(r) {
            var t = r.appearance;
            return t === "default" && (0, d.css)([ "&:first-child{border-right:1px solid ", ";}" ], g.variables.interactiveColorBorder);
        }
    }));
    var w = b()(h()).withConfig({
        displayName: "SplitButtonStyles__StyledSeparator",
        componentId: "sc-9igsyh-1"
    })([ "width:", ";height:", ";background-color:", ";" ], (0, g.pick)({
        prisma: (0, g.pickVariant)("$appearance", {
            primary: "2px",
            destructive: "2px"
        }),
        enterprise: "1px"
    }), g.variables.inputHeight, (0, g.pickVariant)("$appearance", {
        primary: {
            enterprise: g.variables.white,
            prisma: {
                light: g.variables.white,
                dark: g.variables.black
            }
        },
        destructive: {
            enterprise: {
                light: g.variables.white,
                dark: g.variables.black
            },
            prisma: {
                light: g.variables.white,
                dark: g.variables.black
            }
        }
    }));
    var k = b()(h()).withConfig({
        displayName: "SplitButtonStyles__StyledBox",
        componentId: "sc-9igsyh-2"
    })([ "", " [data-inline] + &{margin-left:", ";}" ], (function(e) {
        var r = e.$inline;
        return !r && (0, d.css)([ "width:100%;" ]);
    }), g.variables.spacingSmall);
    // CONCATENATED MODULE: external "lodash/omit"
    const S = require("lodash/omit");
    var j =  e.n(S);
    // CONCATENATED MODULE: ./src/SplitButton/Item.tsx
    function P(e) {
        "@babel/helpers - typeof";
        return P = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, P(e);
    }
    function _() {
        return _ = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, _.apply(null, arguments);
    }
    function C(e, r) {
        if (null == e) return {};
        var t, n, i = E(e, r);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (n = 0; n < o.length; n++) {
                t = o[n], r.includes(t) || {}.propertyIsEnumerable.call(e, t) && (i[t] = e[t]);
            }
        }
        return i;
    }
    function E(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (r.includes(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    function I(e, r) {
        if (!(e instanceof r)) throw new TypeError("Cannot call a class as a function");
    }
    function q(e, r) {
        for (var t = 0; t < r.length; t++) {
            var n = r[t];
            n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), 
            Object.defineProperty(e, A(n.key), n);
        }
    }
    function B(e, r, t) {
        return r && q(e.prototype, r), t && q(e, t), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function x(e, r) {
        if ("function" != typeof r && null !== r) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(r && r.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), r && M(e, r);
    }
    function M(e, r) {
        return M = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, r) {
            return e.__proto__ = r, e;
        }, M(e, r);
    }
    function T(e) {
        var r = V();
        return function() {
            var t, n = $(e);
            if (r) {
                var i = $(this).constructor;
                t = Reflect.construct(n, arguments, i);
            } else t = n.apply(this, arguments);
            return R(this, t);
        };
    }
    function R(e, r) {
        if (r && ("object" == P(r) || "function" == typeof r)) return r;
        if (void 0 !== r) throw new TypeError("Derived constructors may only return object or undefined");
        return N(e);
    }
    function N(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function V() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return (V = function r() {
            return !!e;
        })();
    }
    function $(e) {
        return $ = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, $(e);
    }
    function D(e, r, t) {
        return (r = A(r)) in e ? Object.defineProperty(e, r, {
            value: t,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = t, e;
    }
    function A(e) {
        var r = H(e, "string");
        return "symbol" == P(r) ? r : r + "";
    }
    function H(e, r) {
        if ("object" != P(e) || !e) return e;
        var t = e[Symbol.toPrimitive];
        if (void 0 !== t) {
            var n = t.call(e, r || "default");
            if ("object" != P(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    var z = {
        appearance: o().oneOf([ "default", "primary", "destructive" ]),
        children: o().node,
        disabled: o().bool,
        isMain: o().bool,
        onClick: o().func
    };
    var F = {
        appearance: "default"
    };
    /**
 * An item within a `SplitButton`. This inherits from
 * [PureComponent](https://reactjs.org/docs/react-api.html#reactpurecomponent)
 * so any elements passed to it must also be pure.
 */    var G =  function(e) {
        x(t, e);
        var r = T(t);
        function t() {
            var e;
            I(this, t);
            for (var n = arguments.length, i = new Array(n), o = 0; o < n; o++) {
                i[o] = arguments[o];
            }
            e = r.call.apply(r, [ this ].concat(i));
            D(N(e), "item", null);
            D(N(e), "handleClick", (function(r) {
                var t = e.props, n = t.disabled, i = t.onClick;
                if (!n) {
                    i === null || i === void 0 ? void 0 : i(r);
                }
            }));
            return e;
        }
 // @docs-props-type ItemPropsBase
                B(t, [ {
            key: "scrollIntoViewIfNeeded",
            // eslint-disable-next-line react/no-unused-class-component-methods
            value: function e() {
                var r;
                (r = this.item) === null || r === void 0 ? void 0 : r.scrollIntoViewIfNeeded();
            }
            /**
     * Place focus on the item.
     */
            // eslint-disable-next-line react/no-unused-class-component-methods
                }, {
            key: "focus",
            value: function e() {
                var r;
                (r = this.item) === null || r === void 0 ? void 0 : r.focus();
            }
        }, {
            key: "render",
            value: function e() {
                var r = this;
                var t = this.props, i = t.disabled, o = t.isMain, a = t.children, l = t.appearance, c = C(t, [ "disabled", "isMain", "children", "appearance" ]);
                if (o) {
                    
                    return n().createElement(O, _({
                        onClick: this.handleClick,
                        disabled: i,
                        append: true,
                        appearance: l
                    }, c), a);
                }
                
                return n().createElement(p().Item, _({
                    ref: function e(t) {
                        r.item = t;
                    },
                    onClick: this.handleClick,
                    disabled: i
                }, j()(c, [ "role" ])), a);
            }
        } ]);
        return t;
    }(t.PureComponent);
    D(G, "propTypes", z);
    D(G, "defaultProps", F);
    /* harmony default export */    const J = G;
    // CONCATENATED MODULE: ./src/SplitButton/SplitButton.tsx
    function K() {
        return K = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, K.apply(null, arguments);
    }
    function L(e, r) {
        if (null == e) return {};
        var t, n, i = Q(e, r);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (n = 0; n < o.length; n++) {
                t = o[n], r.includes(t) || {}.propertyIsEnumerable.call(e, t) && (i[t] = e[t]);
            }
        }
        return i;
    }
    function Q(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (r.includes(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    var U = {
        appearance: o().oneOf([ "default", "primary", "destructive" ]),
        children: o().node,
        disabled: o().bool,
        elementRef: o().oneOfType([ o().func, o().object ]),
        onClick: o().func,
        inline: o().bool
    };
    function W(e) {
        var r = e.appearance, i = r === void 0 ? "default" : r, o = e.children, a = e.disabled, c = e.elementRef, s = e.inline, d = s === void 0 ? true : s, b = e.onClick, v = L(e, [ "appearance", "children", "disabled", "elementRef", "inline", "onClick" ]);
 // @docs-props-type SplitButtonPropsBase
                var y = t.Children.toArray(o).filter(t.isValidElement);
        var m = 0;
        var h = 0;
        y.forEach((function(e, r) {
            var t = e.props, n = t.disabled, i = t.isMain;
            if (i) {
                m = r;
            }
            if (n) {
                h += 1;
            }
        }));
        var g =  (0, t.cloneElement)(y[m], {
            isMain: true,
            appearance: i,
            disabled: a || y[m].props.disabled
        });
        // Remove mainButton from array of dropdown items
                y.splice(m, 1);
        var S = y;
        var j =  n().createElement(O, {
            prepend: true,
            disabled: a || h === S.length,
            appearance: i,
            "data-test": "split-button-toggle",
            "aria-label": (0, f._)("Open more options"),
            icon:  n().createElement(l(), null)
        });
        
        return n().createElement(k, K({
            "aria-label": (0, f._)("Split button"),
            "data-inline": d || null,
            "data-test": "split-button-container",
            elementRef: c,
            $inline: d,
            inline: true,
            flex: true,
            role: "group",
            onClick: a ? undefined : b
        }, v), g, i !== "default" &&  n().createElement(w, {
            inline: true,
            $appearance: i
        }),  n().createElement("div", null,  n().createElement(u(), {
            toggle: j
        },  n().createElement(p(), null, S))));
    }
    W.propTypes = U;
    W.Item = J;
    /* harmony default export */    const X = W;
    // CONCATENATED MODULE: ./src/SplitButton/index.ts
    module.exports = r;
    /******/})();
