// @splunk/react-ui/StaticContent.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var n = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, n) => {
            /******/ for (var t in n) {
                /******/ if (e.o(n, t) && !e.o(r, t)) {
                    /******/ Object.defineProperty(r, t, {
                        enumerable: true,
                        get: n[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ v
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var t =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const i = require("prop-types");
    var a =  e.n(i);
    // CONCATENATED MODULE: external "styled-components"
    const l = require("styled-components");
    var o =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const s = require("@splunk/react-ui/Box");
    var c =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/themes"
    const p = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/StaticContent/StaticContentStyles.ts
    var u = o()(c()).withConfig({
        displayName: "StaticContentStyles__StyledBox",
        componentId: "sxqyrp-0"
    })([ "", ";position:relative;margin:0;font-weight:", ";word-wrap:break-word;min-height:", ";padding:", ";[data-inline] + &[data-inline]{margin-left:", ";}" ], p.mixins.reset("flex"), p.variables.fontWeightSemiBold, p.variables.inputHeight, (0, 
    p.pick)({
        enterprise: {
            comfortable: "6px 7px",
            compact: "4px 5px"
        },
        prisma: {
            comfortable: "10px 0",
            compact: "6px 0"
        }
    }), (0, p.pick)({
        enterprise: p.variables.spacingXSmall,
        prisma: p.variables.spacingLarge
    }));
    // CONCATENATED MODULE: ./src/StaticContent/StaticContent.tsx
    function f() {
        return f = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, f.apply(null, arguments);
    }
    function d(e, r) {
        if (null == e) return {};
        var n, t, i = b(e, r);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                n = a[t], r.includes(n) || {}.propertyIsEnumerable.call(e, n) && (i[n] = e[n]);
            }
        }
        return i;
    }
    function b(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (r.includes(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    var m = {
        children: a().node,
        elementRef: a().oneOfType([ a().func, a().object ]),
        inline: a().bool,
        size: a().oneOf([ "small", "medium" ]),
        labelledBy: a().string
    };
    /**
 * This component is intended for use in a control group, either to display a static value or
 * between two controls.
 */    function y(e) {
        var r = e.children, i = e.inline, a = i === void 0 ? false : i, l = e.labelledBy, o = e.size, s = d(e, [ "children", "inline", "labelledBy", "size" ]);
        // @docs-props-type StaticContentPropsBase
                (0, n.useEffect)((function() {
            if (false) {}
        }), [ o ]);
        
        return t().createElement(u, f({
            "aria-labelledby": l,
            "data-test": "static-content",
            inline: a
        }, s), r);
    }
    y.propTypes = m;
    /* harmony default export */    const v = y;
    // CONCATENATED MODULE: ./src/StaticContent/index.ts
    module.exports = r;
    /******/})();
