// @splunk/react-ui/TabLayout.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var n = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, n) => {
            /******/ for (var r in n) {
                /******/ if (e.o(n, r) && !e.o(t, r)) {
                    /******/ Object.defineProperty(t, r, {
                        enumerable: true,
                        get: n[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        Panel: () => /* reexport */ S,
        default: () => /* reexport */ V
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var r =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var a =  e.n(o);
    // CONCATENATED MODULE: external "lodash/has"
    const i = require("lodash/has");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "lodash/keys"
    const c = require("lodash/keys");
    var u =  e.n(c);
    // CONCATENATED MODULE: external "lodash/omit"
    const s = require("lodash/omit");
    var p =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const f = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "@splunk/react-ui/TabBar"
    const d = require("@splunk/react-ui/TabBar");
    var y =  e.n(d);
    // CONCATENATED MODULE: external "styled-components"
    const b = require("styled-components");
    var v =  e.n(b);
    // CONCATENATED MODULE: external "@splunk/themes"
    const h = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/TabLayout/TabLayoutStyles.ts
    var m = v().div.withConfig({
        displayName: "TabLayoutStyles__StyledPanel",
        componentId: "lf0quy-0"
    })([ "", "" ], h.mixins.reset("block"));
    var g = v().div.withConfig({
        displayName: "TabLayoutStyles__Styled",
        componentId: "lf0quy-1"
    })([ "text-align:center;margin-top:", ";", "" ], h.variables.spacingLarge, (function(e) {
        var t = e.$layout;
        return t === "vertical" && (0, b.css)([ "display:flex;& > [role='tablist']{flex:0 0 auto;}& > ", "{width:100%;}" ], /* sc-sel */ m);
    }));
    // CONCATENATED MODULE: ./src/TabLayout/Panel.tsx
    function O() {
        return O = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var n = arguments[t];
                for (var r in n) {
                    ({}).hasOwnProperty.call(n, r) && (e[r] = n[r]);
                }
            }
            return e;
        }, O.apply(null, arguments);
    }
    function P(e, t) {
        if (null == e) return {};
        var n, r, o = j(e, t);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (r = 0; r < a.length; r++) {
                n = a[r], t.includes(n) || {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function j(e, t) {
        if (null == e) return {};
        var n = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (t.includes(r)) continue;
                n[r] = e[r];
            }
        }
        return n;
    }
    var w = {
        children: a().node,
        count: a().number,
        disabled: a().bool,
        elementRef: a().oneOfType([ a().func, a().object ]),
        icon: a().node,
        label: a().oneOfType([ a().string, a().element ]),
        panelId: a().string.isRequired,
        tooltip: a().node
    };
    function I(e) {
        var t = e.children, n = e.disabled, o = n === void 0 ? false : n, a = e.elementRef, i = e.panelId, l = P(e, [ "children", "disabled", "elementRef", "panelId" ]);
        // @docs-props-type PanelPropsBase
                
        return r().createElement(m, O({
            "data-test-panel-id": i,
            "data-test": "panel",
            ref: a,
            role: "tabpanel"
        }, p()(l, [ "icon", "label", "tooltip" ])), t);
    }
    I.propTypes = w;
    /* harmony default export */    const S = I;
    // CONCATENATED MODULE: ./src/TabLayout/TabLayout.tsx
    function T(e) {
        "@babel/helpers - typeof";
        return T = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, T(e);
    }
    function _() {
        return _ = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var n = arguments[t];
                for (var r in n) {
                    ({}).hasOwnProperty.call(n, r) && (e[r] = n[r]);
                }
            }
            return e;
        }, _.apply(null, arguments);
    }
    function C(e, t) {
        if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function");
    }
    function E(e, t) {
        for (var n = 0; n < t.length; n++) {
            var r = t[n];
            r.enumerable = r.enumerable || !1, r.configurable = !0, "value" in r && (r.writable = !0), 
            Object.defineProperty(e, L(r.key), r);
        }
    }
    function q(e, t, n) {
        return t && E(e.prototype, t), n && E(e, n), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function k(e, t) {
        if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), t && x(e, t);
    }
    function x(e, t) {
        return x = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, t) {
            return e.__proto__ = t, e;
        }, x(e, t);
    }
    function R(e) {
        var t = D();
        return function() {
            var n, r = M(e);
            if (t) {
                var o = M(this).constructor;
                n = Reflect.construct(r, arguments, o);
            } else n = r.apply(this, arguments);
            return A(this, n);
        };
    }
    function A(e, t) {
        if (t && ("object" == T(t) || "function" == typeof t)) return t;
        if (void 0 !== t) throw new TypeError("Derived constructors may only return object or undefined");
        return z(e);
    }
    function z(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function D() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return (D = function t() {
            return !!e;
        })();
    }
    function M(e) {
        return M = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, M(e);
    }
    function B(e, t, n) {
        return (t = L(t)) in e ? Object.defineProperty(e, t, {
            value: n,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = n, e;
    }
    function L(e) {
        var t = N(e, "string");
        return "symbol" == T(t) ? t : t + "";
    }
    function N(e, t) {
        if ("object" != T(e) || !e) return e;
        var n = e[Symbol.toPrimitive];
        if (void 0 !== n) {
            var r = n.call(e, t || "default");
            if ("object" != T(r)) return r;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    /** @public */    var W = {
        autoActivate: a().bool,
        activePanelId: a().string,
        appearance: a().oneOf([ "navigation", "context" ]),
        children: a().node,
        count: a().number,
        defaultActivePanelId: a().string,
        elementRef: a().oneOfType([ a().func, a().object ]),
        iconSize: a().oneOf([ "inline", "small", "large" ]),
        layout: a().oneOf([ "horizontal", "vertical" ]),
        onChange: a().func,
        tabWidth: a().number
    };
    var $ = {
        appearance: "navigation",
        layout: "horizontal"
    };
    /**
 * The `TabLayout` is a group of managed `Panels`. Only one panel can be open at a time.
 * `TabLayout` supports both the controlled and uncontrolled patterns.
 */    var U =  function(e) {
        k(o, e);
        var t = R(o);
        // @docs-props-type TabLayoutPropsBase
                function o(e) {
            var n;
            C(this, o);
            n = t.call(this, e);
            B(z(n), "controlledExternally", void 0);
            B(z(n), "guid", void 0);
            B(z(n), "handleChange", (function(e, t) {
                var r, o;
                var a = t.selectedTabId;
                if (!n.isControlled()) {
                    n.setState({
                        activePanelId: a
                    });
                }
                (r = (o = n.props).onChange) === null || r === void 0 ? void 0 : r.call(o, e, {
                    activePanelId: a
                });
            }));
            n.controlledExternally = l()(e, "activePanelId");
            n.guid = (0, f.createDOMID)();
            if (!n.isControlled()) {
                n.state = {
                    activePanelId: e.defaultActivePanelId
                };
            }
            if (false) {}
            return n;
        }
        q(o, [ {
            key: "componentDidUpdate",
            value: function e(t) {
                if (false) {}
                if (false) {}
            }
        }, {
            key: "isControlled",
            value: function e() {
                return this.controlledExternally;
            }
        }, {
            key: "render",
            value: function e() {
                var t = this;
                var a = this.props, i = a.children, l = a.layout;
                var c = this.isControlled() ? this.props.activePanelId : this.state.activePanelId;
                var s;
                var f = n.Children.toArray(i).filter(n.isValidElement).map((function(e) {
                    var o = e.props;
                    var a = "".concat(t.guid, "-").concat(o.panelId);
                    var i = "".concat(t.guid, "-").concat(o.panelId, "-tab");
                    if (o.panelId === c) {
                        s =  (0, n.cloneElement)(e, {
                            "aria-labelledby": i,
                            id: a
                        });
                    }
                    var l = {};
                    // tell TS that PanelProps can contain `data-title`
                                        var u = o["data-title"];
                    if (u != null) {
                        l["data-title"] = u;
                    }
                    
                    return r().createElement(y().Tab, _({
                        count: o.count,
                        icon: o.icon,
                        key: o.panelId,
                        label: o.label,
                        tabId: o.panelId,
                        id: i,
                        ariaControls: a,
                        tooltip: o.tooltip,
                        disabled: o.disabled
                    }, l));
                }));
                if (false) {}
                
                return r().createElement(g, _({
                    "data-test-active-panel-id": c,
                    "data-test": "tab-layout",
                    ref: this.props.elementRef,
                    $layout: l
                }, p()(this.props, u()(o.propTypes))),  r().createElement(y(), {
                    autoActivate: this.props.autoActivate,
                    activeTabId: c,
                    appearance: this.props.appearance,
                    iconPosition: this.props.iconPosition,
                    iconSize: this.props.iconSize,
                    layout: this.props.layout,
                    onChange: this.handleChange,
                    tabWidth: this.props.tabWidth
                }, f), s);
            }
        } ]);
        return o;
    }(n.Component);
    B(U, "propTypes", W);
    B(U, "defaultProps", $);
    B(U, "Panel", S);
    /* harmony default export */    const V = U;
    // CONCATENATED MODULE: ./src/TabLayout/index.ts
    module.exports = t;
    /******/})();
