// @splunk/react-ui/Text.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        default: () => /* reexport */ Be
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const i = require("prop-types");
    var a =  e.n(i);
    // CONCATENATED MODULE: external "lodash/has"
    const o = require("lodash/has");
    var l =  e.n(o);
    // CONCATENATED MODULE: external "lodash/keys"
    const s = require("lodash/keys");
    var c =  e.n(s);
    // CONCATENATED MODULE: external "lodash/omit"
    const p = require("lodash/omit");
    var d =  e.n(p);
    // CONCATENATED MODULE: external "lodash/pickBy"
    const u = require("lodash/pickBy");
    var m =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/react-ui/ControlGroup"
    const f = require("@splunk/react-ui/ControlGroup");
    // CONCATENATED MODULE: external "@splunk/react-icons/Cross"
    const v = require("@splunk/react-icons/Cross");
    var h =  e.n(v);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/Close"
    const b = require("@splunk/react-icons/enterprise/Close");
    var y =  e.n(b);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/Search"
    const g = require("@splunk/react-icons/enterprise/Search");
    var x =  e.n(g);
    // CONCATENATED MODULE: external "@splunk/react-icons/Magnifier"
    const C = require("@splunk/react-icons/Magnifier");
    var w =  e.n(C);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const k = require("@splunk/react-ui/ScreenReaderContent");
    var S =  e.n(k);
    // CONCATENATED MODULE: external "@splunk/react-ui/TextArea"
    const A = require("@splunk/react-ui/TextArea");
    var E =  e.n(A);
    // CONCATENATED MODULE: external "@splunk/themes"
    const O = require("@splunk/themes");
    // CONCATENATED MODULE: external "@splunk/ui-utils/style"
    const T = require("@splunk/ui-utils/style");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const I = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/react-icons/EyeSlashed"
    const $ = require("@splunk/react-icons/EyeSlashed");
    var j =  e.n($);
    // CONCATENATED MODULE: ./src/Text/IconOutlinedHide.tsx
    function P() {
        var e = (0, O.useSplunkTheme)(), t = e.family, r = e.density;
        var i = r === "compact" ? "20" : "24";
        var a = "1.2em";
        var o = t === "prisma" ? i : a;
        
        return n().createElement(j(), {
            width: o,
            height: o
        });
    }
    /* harmony default export */    const _ = P;
    // CONCATENATED MODULE: external "@splunk/react-icons/Eye"
    const D = require("@splunk/react-icons/Eye");
    var B =  e.n(D);
    // CONCATENATED MODULE: ./src/Text/IconOutlinedView.tsx
    function N() {
        var e = (0, O.useSplunkTheme)(), t = e.family, r = e.density;
        var i = r === "compact" ? "20" : "24";
        var a = "1.2em";
        var o = t === "prisma" ? i : a;
        
        return n().createElement(B(), {
            width: o,
            height: o
        });
    }
    /* harmony default export */    const q = N;
    // CONCATENATED MODULE: external "styled-components"
    const M = require("styled-components");
    var R =  e.n(M);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const W = require("@splunk/react-ui/Box");
    var V =  e.n(W);
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const z = require("@splunk/react-ui/Button");
    var F =  e.n(z);
    // CONCATENATED MODULE: ./src/Text/TextStyles.ts
    var H = (0, M.css)([ "&,&[type]{cursor:not-allowed;color:", ";&::placeholder{color:", ";}}" ], (0, 
    O.pick)({
        enterprise: O.variables.textDisabledColor,
        prisma: O.variables.contentColorDisabled
    }), (0, O.pick)({
        enterprise: O.variables.textDisabledColor,
        prisma: O.variables.contentColorDisabled
    }));
    var K = (0, M.css)([ "border-color:", ";box-shadow:inset 0 1px 2px rgba(0,0,0,0.025);", "" ], O.variables.interactiveColorBorderDisabled, (0, 
    O.pick)({
        enterprise: (0, M.css)([ "background-color:", ";" ], O.variables.interactiveColorBackgroundDisabled)
    }));
    var L = (0, M.css)([ "border-radius:", ";cursor:pointer;font-size:0.83333em;flex-grow:0;", ";&:not([disabled]){color:", ";}" ], (0, 
    O.pick)({
        enterprise: O.variables.borderRadius,
        prisma: "50%"
    }), (0, O.pick)({
        enterprise: (0, M.css)([ "width:", ";height:", ";" ], O.variables.inputHeight, O.variables.inputHeight),
        prisma: {
            comfortable: (0, M.css)([ "width:26px;min-width:26px;min-height:26px;margin:8px;padding:0;" ]),
            compact: (0, M.css)([ "width:22px;min-width:22px;min-height:22px;margin:8px;padding:0;" ])
        }
    }), (0, O.pick)({
        enterprise: {
            light: O.variables.gray60,
            dark: O.variables.white
        },
        prisma: O.variables.contentColorMuted
    }));
    var G = R()(F()).withConfig({
        displayName: "TextStyles__StyledClearButton",
        componentId: "eg7n6t-0"
    })([ "display:none;visibility:hidden;", "" ], L);
    var U = R().span.withConfig({
        displayName: "TextStyles__StyledSearchIconWrapper",
        componentId: "eg7n6t-1"
    })([ "", ";color:", ";pointer-events:none;padding:", ";", "" ], O.mixins.reset("inline-block"), (0, 
    O.pick)({
        enterprise: {
            light: O.variables.gray60,
            dark: O.variables.white
        },
        prisma: O.variables.contentColorMuted
    }), (0, O.pick)({
        comfortable: "0 8px",
        compact: "0 6px"
    }), (function(e) {
        var t = e.$disabled;
        return t && (0, M.css)([ "", "" ], (0, O.pick)({
            enterprise: (0, M.css)([ "color:", ";" ], O.variables.textDisabledColor),
            prisma: (0, M.css)([ "color:", ";" ], O.variables.contentColorDisabled)
        }));
    }));
    var J = R()(V()).withConfig({
        displayName: "TextStyles__StyledBox",
        componentId: "eg7n6t-2"
    })([ "display:flex;justify-content:space-between;flex-grow:1;flex-shrink:1;position:relative;&[data-inline]{width:230px;flex-basis:230px;[data-inline] + &{margin-left:", ";}}&:focus-within:not([disabled]){", "{", "}", "{visibility:visible;display:block;}}", "" ], O.variables.spacingSmall, /* sc-sel */ U, (0, 
    O.pick)({
        enterprise: (0, M.css)([ "display:none;" ])
    }), /* sc-sel */ G, (function(e) {
        var t = e.$isTimeInput;
        return t && (0, M.css)([ "@media screen and (min--moz-device-pixel-ratio:0){", "{display:none;}}" ], /* sc-sel */ G);
    }));
    var Q = R().span.withConfig({
        displayName: "TextStyles__StyledInputWrapper",
        componentId: "eg7n6t-3"
    })([ "", ";cursor:text;flex-grow:1;min-height:", ";position:relative;overflow:hidden;border:1px solid ", ";border-radius:", ";box-sizing:border-box;background-color:", ";box-shadow:inset 0 1px 1px rgba(0,0,0,0.075);padding:", ";align-items:center;", " ", " ", " ", " ", " ", " ", "" ], O.mixins.reset("inline-flex"), O.variables.inputHeight, (0, 
    O.pick)({
        prisma: O.variables.interactiveColorBorder,
        enterprise: {
            light: O.variables.interactiveColorBorder,
            dark: O.variables.borderColorWeak
        }
    }), O.variables.borderRadius, (0, O.pick)({
        enterprise: {
            light: O.variables.white,
            dark: O.variables.gray22
        },
        prisma: O.variables.transparent
    }), (0, O.pick)({
        enterprise: "0 10px",
        prisma: "0 12px"
    }), (function(e) {
        var t = e.$hasEndAdornment;
        return t && (0, M.css)([ "padding-right:0;" ]);
    }), (function(e) {
        var t = e.$hasStartAdornment;
        return t && (0, M.css)([ "padding-left:0;" ]);
    }), (function(e) {
        var t = e.$isDisabled;
        return !t && (0, M.css)([ "&:focus-within{box-shadow:", ";color:", ";}" ], O.variables.focusShadow, O.variables.contentColorActive);
    }), (function(e) {
        var t = e.$isDisabled, r = e.$isDimmed;
        return t || r ? K : (0, M.css)([ "&:hover{border-color:", ";}" ], (0, O.pick)({
            enterprise: {
                light: O.variables.interactiveColorBorderHover,
                dark: O.variables.borderColorWeak
            },
            prisma: O.variables.interactiveColorBorderHover
        }));
    }), (function(e) {
        var t = e.$error;
        return t && (0, M.css)([ "&,&:hover{border-color:", ";}" ], O.variables.accentColorNegative);
    }), (function(e) {
        var t = e.$append;
        return t && (0, M.css)([ "margin-right:-1px;border-top-right-radius:0;border-bottom-right-radius:0;border-right:none;" ]);
    }), (function(e) {
        var t = e.$prepend;
        return t && (0, M.css)([ "border-top-left-radius:0;border-bottom-left-radius:0;" ]);
    }));
    /* Some of these need greater specificity than input[type=text] */    var X = R().input.withConfig({
        displayName: "TextStyles__StyledInput",
        componentId: "eg7n6t-4"
    })([ "", ";&,&[type]{outline-style:none;border:0;background:transparent;color:", ";flex-grow:1;font-family:", ";font-size:inherit;line-height:inherit;&[type='time']{&::-webkit-calendar-picker-indicator{display:none;}height:", ";@media screen and (min--moz-device-pixel-ratio:0){padding-right:", ";}}position:relative;&::placeholder{color:", ";opacity:1;}", " &[disabled],&[aria-disabled='true']{", "}}" ], O.mixins.reset("inline-flex"), (0, 
    O.pick)({
        enterprise: O.variables.textColor,
        prisma: O.variables.contentColorActive
    }), O.variables.sansFontFamily, (0, O.pick)({
        enterprise: "20px",
        prisma: {
            comfortable: "24px",
            compact: "20px"
        }
    }), (0, O.pick)({
        enterprise: O.variables.spacingSmall,
        prisma: "14px"
    }), O.variables.contentColorMuted, (function(e) {
        var t = e.$error;
        return t && (0, M.css)([ "&,&:hover{color:", ";}" ], (0, O.pick)({
            enterprise: {
                light: O.variables.errorColorD10,
                dark: O.variables.errorColorL20
            },
            prisma: O.variables.contentColorActive
        }));
    }), H);
    var Y = R()(F()).withConfig({
        displayName: "TextStyles__StyledVisibilityToggle",
        componentId: "eg7n6t-5"
    })([ "", "" ], L);
    var Z = R().span.withConfig({
        displayName: "TextStyles__StyledPlaceholder",
        componentId: "eg7n6t-6"
    })([ "pointer-events:none;color:", ";position:absolute;overflow:hidden;font-size:inherit;line-height:inherit;height:", ";margin-right:", ";", ";", ";", "" ], (0, 
    O.pick)({
        enterprise: O.variables.textGray,
        prisma: O.variables.contentColorMuted
    }), O.variables.lineHeight, (0, O.pick)({
        enterprise: "10px",
        prisma: "14px"
    }), (function(e) {
        var t = e.$hasStartAdornment, r = e.$startAdornmentWidth;
        return t && (r ? (0, M.css)([ "margin-left:", "px;" ], r) : (0, M.css)([ "margin-left:", ";" ], (0, 
        O.pick)({
            enterprise: {
                comfortable: "30px",
                compact: "26px"
            },
            prisma: {
                comfortable: "38px",
                compact: "30px"
            }
        })));
    }), (function(e) {
        var t = e.$hasEndAdornment, r = e.$endAdornmentWidth;
        return t && (r ? (0, M.css)([ "max-width:calc( 100% - ", " - ", "px );" ], (0, O.pick)({
            enterprise: "10px",
            prisma: "14px"
        }), r) : (0, M.css)([ "max-width:calc( 100% - ", " - ", " );" ], (0, O.pick)({
            enterprise: "10px",
            prisma: "14px"
        }), (0, O.pick)({
            enterprise: {
                comfortable: "30px",
                compact: "26px"
            },
            prisma: {
                comfortable: "38px",
                compact: "30px"
            }
        })));
    }), (function(e) {
        var t = e.$hasBothAdornment, r = e.$endAdornmentWidth, n = e.$startAdornmentWidth;
        return t && (0, M.css)([ "max-width:calc(100% - ", "px - ", "px);" ], n, r);
    }));
    var ee = (0, M.css)([ "display:inline-flex;align-items:center;justify-content:center;height:", ";pointer-events:none;" ], (0, 
    O.pick)({
        enterprise: {
            comfortable: "30px",
            compact: "26px"
        },
        prisma: {
            comfortable: "38px",
            compact: "30px"
        }
    }));
    var te = R().div.withConfig({
        displayName: "TextStyles__StyledStartAdornmentHolder",
        componentId: "eg7n6t-7"
    })([ "", " ", ";" ], ee, (function(e) {
        var t = e.$width;
        return t ? (0, M.css)([ "min-width:", "px;" ], t) : (0, M.css)([ "min-width:", ";" ], (0, 
        O.pick)({
            enterprise: {
                comfortable: "30px",
                compact: "26px"
            },
            prisma: {
                comfortable: "38px",
                compact: "30px"
            }
        }));
    }));
    var re = R().div.withConfig({
        displayName: "TextStyles__StyledEndAdornmentHolder",
        componentId: "eg7n6t-8"
    })([ "", " ", ";" ], ee, (function(e) {
        var t = e.$width;
        return t ? (0, M.css)([ "min-width:", "px;" ], t) : (0, M.css)([ "min-width:", ";" ], (0, 
        O.pick)({
            enterprise: {
                comfortable: "30px",
                compact: "26px"
            },
            prisma: {
                comfortable: "38px",
                compact: "30px"
            }
        }));
    }));
    var ne = R().div.withConfig({
        displayName: "TextStyles__StyledAdornment",
        componentId: "eg7n6t-9"
    })([ "display:inline-flex;align-items:center;justify-content:center;position:absolute;pointer-events:none;z-index:1;height:", ";color:", ";", ";", ";", "" ], (0, 
    O.pick)({
        enterprise: {
            comfortable: "30px",
            compact: "26px"
        },
        prisma: {
            comfortable: "38px",
            compact: "30px"
        }
    }), (0, O.pick)({
        enterprise: {
            light: O.variables.gray60,
            dark: O.variables.white
        },
        prisma: O.variables.contentColorMuted
    }), (function(e) {
        var t = e.$position;
        return t === "start" ? (0, M.css)([ "top:1px;left:1px;" ]) : (0, M.css)([ "top:1px;right:1px;" ]);
    }), (function(e) {
        var t = e.disabled;
        return t && H;
    }), (0, O.pick)({
        enterprise: (0, M.css)([ "margin-left:-1px;margin-right:-1px;" ])
    }));
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function ie(e, t) {
        if (e) {
            if (typeof e === "function") {
                e(t);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = t;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Text/Text.tsx
    function ae(e) {
        "@babel/helpers - typeof";
        return ae = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, ae(e);
    }
    function oe(e) {
        return pe(e) || ce(e) || se(e) || le();
    }
    function le() {
        throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function se(e, t) {
        if (e) {
            if ("string" == typeof e) return de(e, t);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? de(e, t) : void 0;
        }
    }
    function ce(e) {
        if ("undefined" != typeof Symbol && null != e[Symbol.iterator] || null != e["@@iterator"]) return Array.from(e);
    }
    function pe(e) {
        if (Array.isArray(e)) return de(e);
    }
    function de(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var r = 0, n = Array(t); r < t; r++) {
            n[r] = e[r];
        }
        return n;
    }
    function ue(e, t) {
        var r = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            t && (n = n.filter((function(t) {
                return Object.getOwnPropertyDescriptor(e, t).enumerable;
            }))), r.push.apply(r, n);
        }
        return r;
    }
    function me(e) {
        for (var t = 1; t < arguments.length; t++) {
            var r = null != arguments[t] ? arguments[t] : {};
            t % 2 ? ue(Object(r), !0).forEach((function(t) {
                Oe(e, t, r[t]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : ue(Object(r)).forEach((function(t) {
                Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
            }));
        }
        return e;
    }
    function fe(e, t) {
        if (null == e) return {};
        var r, n, i = ve(e, t);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (n = 0; n < a.length; n++) {
                r = a[n], t.includes(r) || {}.propertyIsEnumerable.call(e, r) && (i[r] = e[r]);
            }
        }
        return i;
    }
    function ve(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (t.includes(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    function he() {
        return he = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, he.apply(null, arguments);
    }
    function be(e, t) {
        if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function");
    }
    function ye(e, t) {
        for (var r = 0; r < t.length; r++) {
            var n = t[r];
            n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), 
            Object.defineProperty(e, Te(n.key), n);
        }
    }
    function ge(e, t, r) {
        return t && ye(e.prototype, t), r && ye(e, r), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function xe(e, t) {
        if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), t && Ce(e, t);
    }
    function Ce(e, t) {
        return Ce = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, t) {
            return e.__proto__ = t, e;
        }, Ce(e, t);
    }
    function we(e) {
        var t = Ae();
        return function() {
            var r, n = Ee(e);
            if (t) {
                var i = Ee(this).constructor;
                r = Reflect.construct(n, arguments, i);
            } else r = n.apply(this, arguments);
            return ke(this, r);
        };
    }
    function ke(e, t) {
        if (t && ("object" == ae(t) || "function" == typeof t)) return t;
        if (void 0 !== t) throw new TypeError("Derived constructors may only return object or undefined");
        return Se(e);
    }
    function Se(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function Ae() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return (Ae = function t() {
            return !!e;
        })();
    }
    function Ee(e) {
        return Ee = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, Ee(e);
    }
    function Oe(e, t, r) {
        return (t = Te(t)) in e ? Object.defineProperty(e, t, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = r, e;
    }
    function Te(e) {
        var t = Ie(e, "string");
        return "symbol" == ae(t) ? t : t + "";
    }
    function Ie(e, t) {
        if ("object" != ae(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var n = r.call(e, t || "default");
            if ("object" != ae(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    /** @public */
    /** @public */
    /** @public */    var $e = {
        appearance: a().oneOf([ "default", "search" ]),
        append: a().bool,
        autoCapitalize: a().string,
        autoComplete: a().string,
        autoCorrect: a().string,
        autoFocus: a().bool,
        canClear: a().bool,
        children: a().node,
        /** @private. */
        classNamePrivate: a().string,
        defaultValue: a().string,
        describedBy: a().string,
        disabled: a().oneOfType([ a().bool, a().oneOf([ "dimmed" ]) ]),
        elementRef: a().oneOfType([ a().func, a().object ]),
        endAdornment: a().node,
        error: a().bool,
        inline: a().bool,
        /** @private. */
        inputClassName: a().string,
        inputId: a().string,
        inputRef: a().oneOfType([ a().func, a().object ]),
        labelledBy: a().string,
        maxLength: a().number,
        multiline: a().bool,
        name: a().string,
        onBlur: a().func,
        onChange: a().func,
        onFocus: a().func,
        onKeyDown: a().func,
        onSelect: a().func,
        onInputClick: a().func,
        placeholder: a().string,
        prepend: a().bool,
        /** @private. */
        required: a().bool,
        rowsMax: a().number,
        rowsMin: a().number,
        spellCheck: a().bool,
        tabIndex: a().number,
        startAdornment: a().node,
        /** @private */
        splunkTheme: a().object,
        type: a().string,
        passwordVisibilityToggle: a().bool,
        useSyntheticPlaceholder: a().bool,
        value: a().string
    };
    var je = {
        appearance: "default",
        append: false,
        autoFocus: false,
        canClear: false,
        disabled: false,
        error: false,
        inline: false,
        multiline: false,
        placeholder: "",
        prepend: false,
        rowsMax: 8,
        rowsMin: 2,
        tabIndex: 0,
        type: "text",
        passwordVisibilityToggle: false
    };
    /** Note: Text places role and aria props onto the input. All other props are placed on the wrapper. */    var Pe =  function(e) {
        xe(r, e);
        var t = we(r);
        ge(r, null, [ {
            key: "validateRows",
            // @docs-props-type TextPropsBase
            value: function e(t) {
                var r = t.rowsMin, n = t.rowsMax;
                if (false) {}
            }
        } ]);
        function r(e) {
            var i;
            be(this, r);
            i = t.call(this, e);
            Oe(Se(i), "controlledExternally", void 0);
            Oe(Se(i), "input", null);
            Oe(Se(i), "startAdornment", null);
            Oe(Se(i), "endAdornment", null);
            Oe(Se(i), "getAdornmentWidth", (function() {
                var e = i.startAdornment ? i.startAdornment.getBoundingClientRect() : undefined;
                var t = e && Math.round(e.width);
                if (i.state.startAdornmentWidth !== t) {
                    i.setState({
                        startAdornmentWidth: t
                    });
                }
                var r = i.endAdornment ? i.endAdornment.getBoundingClientRect() : undefined;
                var n = r && Math.round(r.width);
                if (i.state.endAdornmentWidth !== n) {
                    i.setState({
                        endAdornmentWidth: n
                    });
                }
            }));
            Oe(Se(i), "handleInputMount", (function(e) {
                i.input = e;
                ie(i.props.inputRef, e);
            }));
            Oe(Se(i), "handleInputChange", (function(e) {
                var t, r;
                var n = e.target.value;
                var a = i.props.name;
                if (!i.isControlled()) {
                    i.setState({
                        value: n
                    });
                }
                (t = (r = i.props).onChange) === null || t === void 0 ? void 0 : t.call(r, e, {
                    value: n,
                    name: a
                });
            }));
            Oe(Se(i), "handleInputKeyDown", (function(e) {
                var t, r;
                (t = (r = i.props).onKeyDown) === null || t === void 0 ? void 0 : t.call(r, e);
            }));
            Oe(Se(i), "handleInputSelect", (function(e) {
                var t, r;
                (t = (r = i.props).onSelect) === null || t === void 0 ? void 0 : t.call(r, e);
            }));
            Oe(Se(i), "handleInputClick", (function(e) {
                var t, r;
                (t = (r = i.props).onInputClick) === null || t === void 0 ? void 0 : t.call(r, e);
            }));
            Oe(Se(i), "handleBoxClick", (function(e) {
                var t, r;
                if (document.activeElement !== i.input) {
                    var n;
                    (n = i.input) === null || n === void 0 ? void 0 : n.focus();
                }
                (t = (r = i.props).onClick) === null || t === void 0 ? void 0 : t.call(r, e);
            }));
            Oe(Se(i), "handleAdornmentClick", (function(e) {
                e.stopPropagation();
            }));
            Oe(Se(i), "handleInputFocus", (function(e) {
                var t, r;
                var n = e.target.value;
                var a = i.props.name;
                (t = (r = i.props).onFocus) === null || t === void 0 ? void 0 : t.call(r, e, {
                    value: n,
                    name: a
                });
            }));
            Oe(Se(i), "handleInputBlur", (function(e) {
                var t, r;
                var n = e.target.value;
                var a = i.props.name;
                (t = (r = i.props).onBlur) === null || t === void 0 ? void 0 : t.call(r, e, {
                    value: n,
                    name: a
                });
            }));
            Oe(Se(i), "handleClear", (function(e) {
                var t, r;
                e.preventDefault();
                var n = "";
                var a = i.props.name;
                if (!i.isControlled()) {
                    i.setState({
                        value: n
                    });
                }
                i.focus();
                (t = (r = i.props).onChange) === null || t === void 0 ? void 0 : t.call(r, e, {
                    value: n,
                    name: a
                });
            }));
            Oe(Se(i), "handleVisibilityToggle", (function() {
                i.setState((function(e) {
                    return {
                        hideVisibility: !e.hideVisibility
                    };
                }));
            }));
            Oe(Se(i), "renderEndAdornment", (function() {
                var e;
                var t = i.props, r = t.appearance, a = t.endAdornment, o = t.passwordVisibilityToggle, l = t.disabled, s = t.canClear, c = t.splunkTheme;
                var p = i.state.hideVisibility;
                var d = c.isPrisma, u = c.isCompact, m = c.isEnterprise;
                var f = u ? "20px" : "24px";
                var v = {
                    ref: function e(t) {
                        i.endAdornment = t;
                    },
                    disabled: !!l,
                    $position: "end"
                };
                if (a) {
                    
                    return n().createElement(ne, he({
                        "data-test": "end-adornment",
                        onClick: i.handleAdornmentClick
                    }, v), a);
                }
                if (o && !l) {
                    var b = p ?  n().createElement(q, null) :  n().createElement(_, null);
                    var g = p ? (0, I._)("Show password") : (0, I._)("Hide password");
                    
                    return n().createElement(ne, he({
                        onClick: i.handleAdornmentClick
                    }, v),  n().createElement(Y, {
                        appearance: "subtle",
                        "data-test": "password-toggle",
                        inline: false,
                        onClick: i.handleVisibilityToggle,
                        type: "button",
                        icon: b,
                        "aria-label": g
                    }));
                }
                var C = i.context;
                var w = (e = C.labelAttrs) === null || e === void 0 ? void 0 : e.text;
                var k = w ? (0, I._)("Clear ".concat(w, " text field")) : (0, I._)("Clear text field");
                var S = i.getDisplayValue();
                if (m && r === "search") {
                    if (!S) {
                        
                        return n().createElement(ne, v,  n().createElement(U, {
                            $disabled: !!l,
                            "data-test": "search-icon"
                        },  n().createElement(x(), {
                            screenReaderText: null,
                            hideDefaultTooltip: true,
                            size: "16px",
                            inline: false
                        })));
                    }
                    
                    return n().createElement(ne, v, !l &&  n().createElement(G, {
                        appearance: "subtle",
                        "data-test": "clear",
                        inline: false,
                        "aria-label": k,
                        icon:  n().createElement(y(), {
                            size: 1,
                            screenReaderText: null
                        }),
                        onClick: i.handleClear
                    }),  n().createElement(U, {
                        "data-test": "search-icon",
                        $disabled: !!l
                    },  n().createElement(x(), {
                        screenReaderText: null,
                        hideDefaultTooltip: true,
                        size: "16px",
                        inline: false
                    })));
                }
                if (!!S && (d && r === "search" || s)) {
                    
                    return n().createElement(ne, v, !l &&  n().createElement(G, {
                        "data-test": "clear",
                        appearance: "subtle",
                        "aria-label": k,
                        onClick: i.handleClear,
                        icon: m ?  n().createElement(y(), {
                            inline: false,
                            screenReaderText: null
                        }) :  n().createElement(h(), {
                            height: f,
                            inline: false,
                            width: f
                        })
                    }));
                }
                return undefined;
            }));
            Oe(Se(i), "renderStartAdornment", (function() {
                var e = i.props, t = e.appearance, r = e.startAdornment, a = e.splunkTheme, o = e.disabled;
                var l = a.isPrisma, s = a.isCompact;
                var c = s ? "20px" : "24px";
                var p = {
                    ref: function e(t) {
                        i.startAdornment = t;
                    },
                    disabled: !!o,
                    $position: "start"
                };
                if (r) {
                    
                    return n().createElement(ne, he({
                        "data-test": "start-adornment",
                        onClick: i.handleAdornmentClick
                    }, p), r);
                }
                if (l && t === "search") {
                    
                    return n().createElement(ne, p,  n().createElement(U, {
                        "data-test": "search-icon",
                        $disabled: !!o
                    },  n().createElement(w(), {
                        height: c,
                        width: c
                    })));
                }
                return undefined;
            }));
            Oe(Se(i), "getDisplayValue", (function() {
                var e = i.isControlled() ? i.props.value : i.state.value;
                return e;
            }));
            i.controlledExternally = l()(e, "value");
            i.state = {
                value: e.defaultValue || "",
                hideVisibility: true
            };
            if (false) {}
            if (false) {}
            if (l()(e, "useSyntheticPlaceholder")) {
                // eslint-disable-next-line no-console
                console.warn("The 'Text' prop 'useSyntheticPlaceholder' has been marked for deprecation.");
            }
            if (false) {}
            if (false) {}
            if (false) {}
            r.validateRows(e);
            return i;
        }
        ge(r, [ {
            key: "componentDidMount",
            value: function e() {
                this.getAdornmentWidth();
            }
        }, {
            key: "componentDidUpdate",
            value: function e(t) {
                if (false) {}
                if (false) {}
                r.validateRows(this.props);
                this.getAdornmentWidth();
            }
        }, {
            key: "isControlled",
            value: function e() {
                return this.controlledExternally;
            }
        }, {
            key: "blur",
            // eslint-disable-next-line react/no-unused-class-component-methods
            value: function e() {
                var t;
                (t = this.input) === null || t === void 0 ? void 0 : t.blur();
            }
            /**
     * Place focus on the input.
     */        }, {
            key: "focus",
            value: function e(t) {
                var r;
                (r = this.input) === null || r === void 0 ? void 0 : r.focus(t);
            }
 // eslint-disable-next-line react/no-unused-class-component-methods
                }, {
            key: "select",
            value: function e() {
                var t;
                (t = this.input) === null || t === void 0 ? void 0 : t.select();
            }
        }, {
            key: "render",
            value: function e() {
                var t = this.props, r = t.append, i = t.autoCapitalize, a = t.autoComplete, o = t.autoCorrect, l = t.autoFocus, s = t.children, p = t.className, u = t.classNamePrivate, f = t.disabled, v = t.describedBy, h = t.elementRef, b = t.error, y = t.inline, g = t.inputClassName, x = t.inputId, C = t.labelledBy, w = t.multiline, k = t.maxLength, A = t.name, O = t.passwordVisibilityToggle, I = t.placeholder, $ = t.prepend, j = t.required, P = t.spellCheck, _ = t.tabIndex, D = t.title, B = t.type, N = t.useSyntheticPlaceholder, q = fe(t, [ "append", "autoCapitalize", "autoComplete", "autoCorrect", "autoFocus", "children", "className", "classNamePrivate", "disabled", "describedBy", "elementRef", "error", "inline", "inputClassName", "inputId", "labelledBy", "multiline", "maxLength", "name", "passwordVisibilityToggle", "placeholder", "prepend", "required", "spellCheck", "tabIndex", "title", "type", "useSyntheticPlaceholder" ]);
                var M = me(me({
                    role: "textbox"
                }, m()(q, (function(e, t) {
                    return t === "role" || t.indexOf("aria-") === 0;
                }))), {}, {
                    "aria-describedby": v,
                    "aria-labelledby": C,
                    "aria-invalid": b || undefined,
                    "aria-disabled": f === "dimmed"
                });
                var R = d()(q, [ "inputRef", "onBlur", "onChange", "onClick", "onFocus", "onKeyDown", "onSelect", "onInputClick", "rowsMax", "rowsMin" ].concat(oe(c()(M))));
                var W = this.getDisplayValue();
                var V = {
                    className: (0, T.toClassName)(p, g),
                    $append: r || undefined,
                    $error: b,
                    $prepend: $ || undefined
                };
                var z = this.state.hideVisibility ? "password" : "text";
                var F = O ? z : B;
                var H = me(me({}, M), {}, {
                    "data-test": "textbox",
                    autoCapitalize: i,
                    autoComplete: a,
                    autoCorrect: o,
                    autoFocus: l,
                    id: x,
                    maxLength: k,
                    placeholder: I && !N ? I : undefined,
                    name: A,
                    onChange: this.handleInputChange,
                    onKeyDown: this.handleInputKeyDown,
                    onSelect: this.handleInputSelect,
                    onClick: this.handleInputClick,
                    onFocus: this.handleInputFocus,
                    onBlur: this.handleInputBlur,
                    ref: this.handleInputMount,
                    required: j,
                    spellCheck: P,
                    style: {
                        height: this.state.height
                    },
                    title: D,
                    tabIndex: _,
                    type: F,
                    value: W,
                    $error: b
                });
                var K = N && I && !W;
                // Do not render value in DOM when type is password, or passwordVisibilityToggle is enabled
                                var L = B === "password" || O ? undefined : W;
                // Firefox hack (SUI-2716). Remove when fixed: https://bugzilla.mozilla.org/show_bug.cgi?id=1427173
                                var G = B === "time";
                var U = this.renderStartAdornment();
                var Y = this.renderEndAdornment();
                var ee = !!U;
                var ne = !!Y;
                if (w) {
                    
                    return n().createElement(E(), this.props);
                }
                
                return n().createElement(J, he({
                    tabIndex: -1,
                    $isTimeInput: G,
                    className: (0, T.toClassName)(p, u),
                    "data-test": "text",
                    "data-test-value": L,
                    elementRef: h,
                    flex: true,
                    inline: y,
                    onClick: this.handleBoxClick
                }, R), U,  n().createElement(Q, he({
                    $hasEndAdornment: ne,
                    $hasStartAdornment: ee,
                    $isDisabled: f === true,
                    $isDimmed: f === "dimmed"
                }, V), ee &&  n().createElement(te, {
                    $width: this.state.startAdornmentWidth
                }), f ?  n().createElement(X, he({
                    className: (0, T.toClassName)(p, u),
                    "data-test": "textbox",
                    disabled: f === true,
                    readOnly: true,
                    type: F,
                    value: W,
                    placeholder: I && !N ? I : undefined
                }, M)) :  n().createElement(X, he({}, H, {
                    onClick: this.handleInputClick
                })), K &&  n().createElement(S(), null, I), K &&  n().createElement(Z, {
                    $endAdornmentWidth: this.state.endAdornmentWidth,
                    $hasBothAdornment: ee && ne,
                    $hasEndAdornment: ne,
                    $hasStartAdornment: ee,
                    $startAdornmentWidth: this.state.startAdornmentWidth,
                    "aria-hidden": true,
                    "data-role": "placeholder"
                }, I), s, ne &&  n().createElement(re, {
                    $width: this.state.endAdornmentWidth
                })), Y);
            }
        } ]);
        return r;
    }(r.Component);
    Oe(Pe, "propTypes", $e);
    Oe(Pe, "defaultProps", je);
    Oe(Pe, "componentType", "Text");
    Oe(Pe, "contextType", f.ControlGroupContext);
    var _e = (0, O.withSplunkTheme)(Pe);
    var De = _e;
    De.propTypes = Pe.propTypes;
    De.componentType = Pe.componentType;
    /* harmony default export */    const Be = De;
    // only for styled components using Text
        // CONCATENATED MODULE: ./src/Text/index.ts
    module.exports = t;
    /******/})();
