// @splunk/react-ui/TextArea.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        default: () => /* reexport */ Ve
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var o =  e.n(a);
    // CONCATENATED MODULE: external "lodash/defer"
    const i = require("lodash/defer");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "lodash/has"
    const s = require("lodash/has");
    var c =  e.n(s);
    // CONCATENATED MODULE: external "lodash/keys"
    const u = require("lodash/keys");
    var d =  e.n(u);
    // CONCATENATED MODULE: external "lodash/noop"
    const p = require("lodash/noop");
    var f =  e.n(p);
    // CONCATENATED MODULE: external "lodash/omit"
    const h = require("lodash/omit");
    var v =  e.n(h);
    // CONCATENATED MODULE: external "lodash/pickBy"
    const m = require("lodash/pickBy");
    var b =  e.n(m);
    // CONCATENATED MODULE: external "lodash/throttle"
    const y = require("lodash/throttle");
    var g =  e.n(y);
    // CONCATENATED MODULE: external "@splunk/react-ui/ControlGroup"
    const x = require("@splunk/react-ui/ControlGroup");
    // CONCATENATED MODULE: external "@splunk/react-ui/EventListener"
    const w = require("@splunk/react-ui/EventListener");
    var C =  e.n(w);
    // CONCATENATED MODULE: external "@splunk/themes"
    const S = require("@splunk/themes");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const k = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/ui-utils/style"
    const A = require("@splunk/ui-utils/style");
    // CONCATENATED MODULE: external "@splunk/react-icons/Cross"
    const O = require("@splunk/react-icons/Cross");
    var E =  e.n(O);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/Close"
    const I = require("@splunk/react-icons/enterprise/Close");
    var T =  e.n(I);
    // CONCATENATED MODULE: external "styled-components"
    const j = require("styled-components");
    var P =  e.n(j);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const M = require("@splunk/react-ui/Box");
    var $ =  e.n(M);
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const _ = require("@splunk/react-ui/Button");
    var N =  e.n(_);
    // CONCATENATED MODULE: ./src/TextArea/TextAreaStyles.ts
    var q = P()(N()).withConfig({
        displayName: "TextAreaStyles__StyledClearButton",
        componentId: "gfy8yp-0"
    })([ "display:none;visibility:hidden;border-radius:", ";cursor:pointer;font-size:0.83333em;flex-grow:0;", ";&:not([disabled]){color:", ";}" ], (0, 
    S.pick)({
        enterprise: S.variables.borderRadius,
        prisma: "50%"
    }), (0, S.pick)({
        enterprise: (0, j.css)([ "width:", ";height:", ";" ], S.variables.inputHeight, S.variables.inputHeight),
        prisma: {
            comfortable: (0, j.css)([ "width:26px;min-width:26px;min-height:26px;margin:8px;padding:0;" ]),
            compact: (0, j.css)([ "width:22px;min-width:22px;min-height:22px;margin:8px;padding:0;" ])
        }
    }), (0, S.pick)({
        enterprise: {
            light: S.variables.gray60,
            dark: S.variables.white
        },
        prisma: S.variables.contentColorMuted
    }));
    var W = P().span.withConfig({
        displayName: "TextAreaStyles__StyledSearchIconWrapper",
        componentId: "gfy8yp-1"
    })([ "", ";color:", ";pointer-events:none;padding:", ";", "" ], S.mixins.reset("inline-block"), (0, 
    S.pick)({
        enterprise: {
            light: S.variables.gray60,
            dark: S.variables.white
        },
        prisma: S.variables.contentColorMuted
    }), (0, S.pick)({
        comfortable: "0 8px",
        compact: "0 6px"
    }), (function(e) {
        var t = e.$disabled;
        return t && (0, j.css)([ "", "" ], (0, S.pick)({
            enterprise: (0, j.css)([ "color:", ";" ], S.variables.textDisabledColor),
            prisma: (0, j.css)([ "color:", ";" ], S.variables.contentColorDisabled)
        }));
    }));
    var B = "230px";
    var D = P()($()).withConfig({
        displayName: "TextAreaStyles__StyledBox",
        componentId: "gfy8yp-2"
    })([ "display:flex;justify-content:space-between;flex-grow:1;flex-shrink:1;position:relative;&[data-inline]{width:", ";flex-basis:", ";[data-inline] + &{margin-left:", ";}}&:focus-within:not([disabled]){", " ", "{visibility:visible;display:block;}}" ], B, B, S.variables.spacingSmall, (0, 
    S.pick)({
        enterprise: (0, j.css)([ "", "{display:none;}" ], /* sc-sel */ W)
    }), /* sc-sel */ q);
    var R = (0, S.pick)({
        prisma: "12px",
        enterprise: "10px"
    });
    var z = (0, S.pick)({
        prisma: "12px",
        enterprise: "0"
    });
    var F = P().span.withConfig({
        displayName: "TextAreaStyles__StyledTextAreaWrapper",
        componentId: "gfy8yp-3"
    })([ "", " flex-grow:1;min-height:", ";position:relative;overflow:hidden;border:1px solid ", ";border-radius:", ";box-sizing:border-box;background-color:", ";box-shadow:inset 0 1px 1px rgba(0,0,0,0.075);align-items:center;padding:0 ", " 0 ", ";", " ", " &:hover:not([disabled]){border-color:", ";}&:focus-within:not([disabled]){box-shadow:", ";color:", ";}", " ", " ", " ", "" ], S.mixins.reset("inline-flex"), S.variables.inputHeight, (0, 
    S.pick)({
        enterprise: {
            light: S.variables.borderColorStrong,
            dark: S.variables.borderColorWeak
        },
        prisma: S.variables.interactiveColorBorder
    }), S.variables.borderRadius, (0, S.pick)({
        enterprise: {
            light: S.variables.white,
            dark: S.variables.gray22
        },
        prisma: S.variables.transparent
    }), z, R, (function(e) {
        var t = e.$hasEndAdornment;
        return t && (0, j.css)([ "padding-right:0;" ]);
    }), (function(e) {
        var t = e.$hasStartAdornment;
        return t && (0, j.css)([ "padding-left:0;" ]);
    }), (0, S.pick)({
        enterprise: {
            light: S.variables.borderColorStrong,
            dark: S.variables.borderColorWeak
        },
        prisma: S.variables.interactiveColorBorderHover
    }), S.variables.focusShadow, S.variables.contentColorActive, (function(e) {
        var t = e.$error;
        return t && (0, j.css)([ "&,&:hover:not([disabled]){border-color:", ";}&:focus-within:not([disabled]){box-shadow:", ";color:", ";}" ], S.variables.accentColorNegative, S.variables.focusShadow, S.variables.contentColorActive);
    }), (function(e) {
        var t = e.$append;
        return t && (0, j.css)([ "margin-right:-1px;border-top-right-radius:0;border-bottom-right-radius:0;border-right:none;" ]);
    }), (function(e) {
        var t = e.$prepend;
        return t && (0, j.css)([ "border-top-left-radius:0;border-bottom-left-radius:0;" ]);
    }), (function(e) {
        var t = e.disabled;
        return t && (0, j.css)([ "", " box-shadow:inset 0 1px 2px rgba(0,0,0,0.025);" ], (0, 
        S.pick)({
            enterprise: {
                light: (0, j.css)([ "background-color:", ";border-color:", ";" ], S.variables.gray96, S.variables.gray92),
                dark: (0, j.css)([ "background-color:", ";border-color:", ";" ], S.variables.gray22, S.variables.gray30)
            },
            prisma: (0, j.css)([ "border-color:", ";" ], S.variables.interactiveColorBorderDisabled)
        }));
    }));
    var H = P().textarea.withConfig({
        displayName: "TextAreaStyles__StyledTextArea",
        componentId: "gfy8yp-4"
    })([ "", ";outline-style:none;border:0;background:transparent;color:", ";flex-grow:1;font-family:", ";font-size:inherit;line-height:inherit;padding-top:", ";padding-bottom:", ";padding-right:", ";position:relative;resize:none;overflow:auto;white-space:pre-wrap;", " &::placeholder{color:", ";opacity:1;}", " ", "" ], S.mixins.reset("inline-flex"), S.variables.contentColorActive, S.variables.sansFontFamily, (0, 
    S.pick)({
        enterprise: {
            comfortable: S.variables.spacingXSmall,
            compact: "3px"
        },
        prisma: {
            comfortable: S.variables.spacingSmall,
            compact: "5px"
        }
    }), (0, S.pick)({
        enterprise: {
            comfortable: S.variables.spacingXSmall,
            compact: "3px"
        },
        prisma: {
            comfortable: S.variables.spacingSmall,
            compact: "5px"
        }
    }), (0, S.pick)({
        enterprise: "10px",
        prisma: "14px"
    }), (function(e) {
        var t = e.$rowsMin;
        return t && (0, j.css)([ "height:calc( ", " + (", " * ", ") + ", " );" ], S.variables.inputHeight, t - 1, S.variables.lineHeight, t > 1 ? "-3px" : "-2px");
    }), S.variables.contentColorMuted, (function(e) {
        var t = e.$error;
        return t && (0, j.css)([ "color:", ";" ], (0, S.pick)({
            enterprise: {
                light: S.variables.errorColorD10,
                dark: S.variables.errorColorL20
            },
            prisma: S.variables.contentColorActive
        }));
    }), (function(e) {
        var t = e.disabled;
        return t && (0, j.css)([ "cursor:not-allowed;color:", ";&::placeholder{color:", ";}" ], S.variables.contentColorDisabled, S.variables.contentColorDisabled);
    }));
    var V = P()(H).withConfig({
        displayName: "TextAreaStyles__StyledTextAreaShadow",
        componentId: "gfy8yp-5"
    })([ "position:absolute;overflow:hidden;left:-10000px;top:-10000px;visibility:hidden;pointer-events:none;", "" ], (function(e) {
        var t = e.$startAdornmentWidth, r = e.$endAdornmentWidth;
        var n = t ? "".concat(t, "px") : R;
        var a = r ? "".concat(r, "px") : z;
        return (0, j.css)([ "width:calc(100% - ", " - ", ");" ], n, a);
    }));
    var L = (0, S.pick)({
        enterprise: {
            comfortable: "30px",
            compact: "26px"
        },
        prisma: {
            comfortable: "38px",
            compact: "30px"
        }
    });
    var K = (0, S.pick)({
        enterprise: "10px",
        prisma: "14px"
    });
    var G = P().span.withConfig({
        displayName: "TextAreaStyles__StyledPlaceholder",
        componentId: "gfy8yp-6"
    })([ "pointer-events:none;color:", ";position:absolute;overflow:hidden;height:", ";margin-right:", ";", ";", ";", "" ], (0, 
    S.pick)({
        enterprise: S.variables.textGray,
        prisma: S.variables.contentColorMuted
    }), S.variables.lineHeight, K, (function(e) {
        var t = e.$hasStartAdornment, r = e.$startAdornmentWidth;
        return t && (r ? (0, j.css)([ "margin-left:", "px;" ], r) : (0, j.css)([ "margin-left:", ";" ], L));
    }), (function(e) {
        var t = e.$hasEndAdornment, r = e.$endAdornmentWidth;
        return t && (r ? (0, j.css)([ "max-width:calc(100% - ", " - ", "px);" ], K, r) : (0, 
        j.css)([ "max-width:calc(100% - ", " - ", ");" ], K, L));
    }), (function(e) {
        var t = e.$hasBothAdornment, r = e.$endAdornmentWidth, n = e.$startAdornmentWidth;
        return t && (0, j.css)([ "max-width:calc(100% - ", "px - ", "px);" ], n, r);
    }));
    var U = P().div.withConfig({
        displayName: "TextAreaStyles__StyledAdornmentSpacer",
        componentId: "gfy8yp-7"
    })([ "display:inline-flex;align-items:center;justify-content:center;height:", ";pointer-events:none;", ";" ], L, (function(e) {
        var t = e.$width;
        return t ? (0, j.css)([ "min-width:", "px;" ], t) : (0, j.css)([ "min-width:", ";" ], L);
    }));
    var X = P().div.withConfig({
        displayName: "TextAreaStyles__StyledAdornment",
        componentId: "gfy8yp-8"
    })([ "display:inline-flex;align-items:center;justify-content:center;position:absolute;pointer-events:none;z-index:1;height:", ";color:", ";", " ", ";", "" ], L, (0, 
    S.pick)({
        enterprise: {
            light: S.variables.gray60,
            dark: S.variables.white
        },
        prisma: S.variables.contentColorMuted
    }), (0, S.pickVariant)("$position", {
        start: (0, j.css)([ "top:1px;left:1px;" ]),
        end: (0, j.css)([ "top:1px;right:1px;" ])
    }), (function(e) {
        var t = e.disabled;
        return t && (0, j.css)([ "cursor:not-allowed;color:", ";" ], S.variables.contentColorDisabled);
    }), (0, S.pick)({
        enterprise: (0, j.css)([ "margin-left:-1px;margin-right:-1px;" ])
    }));
    // CONCATENATED MODULE: ./src/TextArea/icons/ClearButton.tsx
    function J() {
        return J = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, J.apply(null, arguments);
    }
    function Q(e) {
        var t = (0, S.useSplunkTheme)(), r = t.isEnterprise, a = t.isCompact;
        var o = r ? "pill" : "secondary";
        var i = a ? "20px" : "24px";
        var l = r ?  n().createElement(T(), {
            size: 1
        }) :  n().createElement(E(), {
            width: i,
            height: i
        });
        
        return n().createElement(q, J({
            "data-test": "clear",
            icon: l,
            appearance: o,
            inline: false
        }, e));
    }
    // CONCATENATED MODULE: external "@splunk/react-icons/Magnifier"
    const Y = require("@splunk/react-icons/Magnifier");
    var Z =  e.n(Y);
    // CONCATENATED MODULE: external "@splunk/react-icons/enterprise/Search"
    const ee = require("@splunk/react-icons/enterprise/Search");
    var te =  e.n(ee);
    // CONCATENATED MODULE: ./src/TextArea/icons/SearchIcon.tsx
    function re() {
        var e = (0, S.useSplunkTheme)(), t = e.isCompact, r = e.isEnterprise;
        if (r) {
            
            return n().createElement(te(), {
                hideDefaultTooltip: true,
                inline: false,
                screenReaderText: "Search",
                size: "16px"
            });
        }
        var a = t ? "20px" : "24px";
        
        return n().createElement(Z(), {
            "aria-label": "Search",
            height: a,
            width: a
        });
    }
    // CONCATENATED MODULE: ./src/TextArea/syncHeightWithShadow.ts
    // This file was adapted from [Call-Em-All's material-ui](https://github.com/mui-org/material-ui),
    // which is MIT-licensed:
    // The MIT License (MIT)
    // Copyright (c) 2014 Call-Em-All
    // Permission is hereby granted, free of charge, to any person obtaining a copy
    // of this software and associated documentation files (the "Software"), to deal
    // in the Software without restriction, including without limitation the rights
    // to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
    // copies of the Software, and to permit persons to whom the Software is
    // furnished to do so, subject to the following conditions:
    // The above copyright notice and this permission notice shall be included in all
    // copies or substantial portions of the Software.
    // THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
    // IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
    // FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
    // AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
    // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
    // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
    // SOFTWARE.
    // End of MIT license text.
    function ne(e) {
        var t = e.height, r = e.rowsMax, n = e.rowsMin, a = e.shadow;
        if (a) {
            var o = window.getComputedStyle(a);
            var i = o.getPropertyValue("line-height");
            // jsdom does not properly inherit/re-calculate the line-height property so we skip in these cases SUI-1943, SUI-2428
                        if (i === "inherit" || i === "normal") {
                return false;
            }
            var l = parseInt(i, 10);
            var s = parseInt(o.getPropertyValue("padding-top"), 10);
            var c = parseInt(o.getPropertyValue("padding-bottom"), 10);
            var u = parseInt(o.getPropertyValue("border-top-width"), 10);
            var d = parseInt(o.getPropertyValue("border-bottom-width"), 10);
            var p = a.scrollHeight + u + d + 1;
 // need one extra.
            /* eslint-disable no-restricted-globals */
            // Leverage the global `isNaN` here for IE support, rather than Number.isNaN.
            // We know newHeight is numeric because we parseInt() above, so `isNaN` should be reliable.
                        if (p === undefined || isNaN(p)) {
                return false;
            }
            /* eslint-enable no-restricted-globals */            if (r >= n) {
                var f = r * l + s + c + u + d + 1;
 // need one extra.
                                p = Math.min(f, p);
            }
            var h = n * l + s + c + u + d + 1;
 // need one extra.
                        p = Math.max(h, p);
            if (t !== p) {
                return p;
            }
        }
        return false;
    }
    // CONCATENATED MODULE: ./src/utils/ssrDocument.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var ae = {
        body: {
            appendChild: function e() {
                return [];
            }
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        activeElement: {
            blur: function e() {},
            nodeName: ""
        },
        querySelector: function e() {
            return null;
        },
        querySelectorAll: function e() {
            return [];
        },
        getElementById: function e() {
            return null;
        },
        createEvent: function e() {
            return {
                initEvent: function e() {}
            };
        },
        createElement: function e() {
            return {
                children: [],
                childNodes: [],
                style: {},
                setAttribute: function e() {},
                getElementsByTagName: function e() {
                    return [];
                }
            };
        },
        createElementNS: function e() {
            return {};
        },
        importNode: function e() {
            return null;
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        }
    };
    function oe() {
        var e = typeof document !== "undefined" ? document : ae;
        return e;
    }
    var ie = oe();
    /* harmony default export */    const le = /* unused pure expression or super */ null && ie;
    // CONCATENATED MODULE: ./src/utils/ssrWindow.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var se = {
        document: ae,
        navigator: {
            userAgent: ""
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        },
        history: {
            replaceState: function e() {},
            pushState: function e() {},
            go: function e() {},
            back: function e() {}
        },
        CustomEvent: function e() {
            return this;
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        getComputedStyle: function e() {
            return {
                getPropertyValue: function e() {
                    return "";
                }
            };
        },
        Image: function e() {},
        Date: function e() {},
        screen: {},
        setTimeout: function e() {},
        clearTimeout: function e() {},
        matchMedia: function e() {
            return {};
        },
        requestAnimationFrame: function e(t) {
            if (typeof setTimeout === "undefined") {
                t();
                return null;
            }
            return setTimeout(t, 0);
        },
        cancelAnimationFrame: function e(t) {
            if (typeof setTimeout === "undefined") {
                return;
            }
            clearTimeout(t);
        }
    };
    function ce() {
        var e = typeof window !== "undefined" ? window : se;
        return e;
    }
    var ue = ce();
    /* harmony default export */    const de = ue;
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function pe(e, t) {
        if (e) {
            if (typeof e === "function") {
                e(t);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = t;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/TextArea/TextArea.tsx
    function fe(e) {
        "@babel/helpers - typeof";
        return fe = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, fe(e);
    }
    function he() {
        return he = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, he.apply(null, arguments);
    }
    function ve(e) {
        return ge(e) || ye(e) || be(e) || me();
    }
    function me() {
        throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function be(e, t) {
        if (e) {
            if ("string" == typeof e) return xe(e, t);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? xe(e, t) : void 0;
        }
    }
    function ye(e) {
        if ("undefined" != typeof Symbol && null != e[Symbol.iterator] || null != e["@@iterator"]) return Array.from(e);
    }
    function ge(e) {
        if (Array.isArray(e)) return xe(e);
    }
    function xe(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var r = 0, n = Array(t); r < t; r++) {
            n[r] = e[r];
        }
        return n;
    }
    function we(e, t) {
        var r = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            t && (n = n.filter((function(t) {
                return Object.getOwnPropertyDescriptor(e, t).enumerable;
            }))), r.push.apply(r, n);
        }
        return r;
    }
    function Ce(e) {
        for (var t = 1; t < arguments.length; t++) {
            var r = null != arguments[t] ? arguments[t] : {};
            t % 2 ? we(Object(r), !0).forEach((function(t) {
                Ne(e, t, r[t]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : we(Object(r)).forEach((function(t) {
                Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
            }));
        }
        return e;
    }
    function Se(e, t) {
        if (null == e) return {};
        var r, n, a = ke(e, t);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (n = 0; n < o.length; n++) {
                r = o[n], t.includes(r) || {}.propertyIsEnumerable.call(e, r) && (a[r] = e[r]);
            }
        }
        return a;
    }
    function ke(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (t.includes(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    function Ae(e, t) {
        if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function");
    }
    function Oe(e, t) {
        for (var r = 0; r < t.length; r++) {
            var n = t[r];
            n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), 
            Object.defineProperty(e, qe(n.key), n);
        }
    }
    function Ee(e, t, r) {
        return t && Oe(e.prototype, t), r && Oe(e, r), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function Ie(e, t) {
        if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a function");
        e.prototype = Object.create(t && t.prototype, {
            constructor: {
                value: e,
                writable: !0,
                configurable: !0
            }
        }), Object.defineProperty(e, "prototype", {
            writable: !1
        }), t && Te(e, t);
    }
    function Te(e, t) {
        return Te = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(e, t) {
            return e.__proto__ = t, e;
        }, Te(e, t);
    }
    function je(e) {
        var t = $e();
        return function() {
            var r, n = _e(e);
            if (t) {
                var a = _e(this).constructor;
                r = Reflect.construct(n, arguments, a);
            } else r = n.apply(this, arguments);
            return Pe(this, r);
        };
    }
    function Pe(e, t) {
        if (t && ("object" == fe(t) || "function" == typeof t)) return t;
        if (void 0 !== t) throw new TypeError("Derived constructors may only return object or undefined");
        return Me(e);
    }
    function Me(e) {
        if (void 0 === e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        return e;
    }
    function $e() {
        try {
            var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], (function() {})));
        } catch (e) {}
        return ($e = function t() {
            return !!e;
        })();
    }
    function _e(e) {
        return _e = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
            return e.__proto__ || Object.getPrototypeOf(e);
        }, _e(e);
    }
    function Ne(e, t, r) {
        return (t = qe(t)) in e ? Object.defineProperty(e, t, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = r, e;
    }
    function qe(e) {
        var t = We(e, "string");
        return "symbol" == fe(t) ? t : t + "";
    }
    function We(e, t) {
        if ("object" != fe(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var n = r.call(e, t || "default");
            if ("object" != fe(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    /** @public */
    /** @public */
    /** @public */    var Be = {
        appearance: o().oneOf([ "default", "search" ]),
        append: o().bool,
        autoCapitalize: o().string,
        autoComplete: o().string,
        autoCorrect: o().string,
        autoFocus: o().bool,
        canClear: o().bool,
        children: o().node,
        /** @private. */
        classNamePrivate: o().string,
        defaultValue: o().string,
        describedBy: o().string,
        disabled: o().bool,
        elementRef: o().oneOfType([ o().func, o().object ]),
        endAdornment: o().node,
        error: o().bool,
        inline: o().bool,
        /** @private. */
        inputClassName: o().string,
        inputId: o().string,
        inputRef: o().oneOfType([ o().func, o().object ]),
        labelledBy: o().string,
        maxLength: o().number,
        name: o().string,
        onBlur: o().func,
        onChange: o().func,
        onFocus: o().func,
        onKeyDown: o().func,
        onSelect: o().func,
        onInputClick: o().func,
        placeholder: o().string,
        prepend: o().bool,
        /** @private. */
        required: o().bool,
        rowsMax: o().number,
        rowsMin: o().number,
        spellCheck: o().bool,
        tabIndex: o().number,
        startAdornment: o().node,
        /** @private */
        splunkTheme: o().object,
        value: o().string
    };
    var De = {
        appearance: "default",
        append: false,
        autoFocus: false,
        canClear: false,
        disabled: false,
        error: false,
        inline: false,
        placeholder: "",
        prepend: false,
        rowsMax: 8,
        rowsMin: 2,
        tabIndex: 0
    };
    var Re = function e(t) {
        var r = t ? t.getBoundingClientRect() : undefined;
        var n = r && Math.round(r.width);
        return n;
    };
    /** Note: TextArea places role and aria props onto the input. All other props are placed on the wrapper. */    var ze =  function(e) {
        Ie(r, e);
        var t = je(r);
        Ee(r, null, [ {
            key: "validateRows",
            // @docs-props-type TextAreaPropsBase
            value: function e(t) {
                var r = t.rowsMin, n = t.rowsMax;
                if (false) {}
            }
        } ]);
        function r(e) {
            var a;
            Ae(this, r);
            a = t.call(this, e);
            Ne(Me(a), "controlledExternally", void 0);
            Ne(Me(a), "input", null);
            Ne(Me(a), "shadow", null);
            Ne(Me(a), "startAdornment", null);
            Ne(Me(a), "endAdornment", null);
            Ne(Me(a), "getAdornmentWidth", (function() {
                var e = Re(a.startAdornment);
                if (a.state.startAdornmentWidth !== e) {
                    a.setState({
                        startAdornmentWidth: e
                    });
                }
                var t = Re(a.endAdornment);
                if (a.state.endAdornmentWidth !== t) {
                    a.setState({
                        endAdornmentWidth: t
                    });
                }
            }));
            Ne(Me(a), "handleResize", (function() {
                a.syncHeightWithShadow();
            }));
            Ne(Me(a), "handleInputMount", (function(e) {
                a.input = e;
                pe(a.props.inputRef, e);
            }));
            Ne(Me(a), "handleShadowMount", (function(e) {
                a.shadow = e;
            }));
            Ne(Me(a), "handleInputChange", (function(e) {
                var t, r;
                var n = e.target.value;
                var o = a.props.name;
                if (!a.isControlled()) {
                    a.setState({
                        value: n
                    });
                }
                (t = (r = a.props).onChange) === null || t === void 0 ? void 0 : t.call(r, e, {
                    value: n,
                    name: o
                });
            }));
            Ne(Me(a), "handleInputKeyDown", (function(e) {
                var t, r;
                (t = (r = a.props).onKeyDown) === null || t === void 0 ? void 0 : t.call(r, e);
            }));
            Ne(Me(a), "handleInputSelect", (function(e) {
                var t, r;
                (t = (r = a.props).onSelect) === null || t === void 0 ? void 0 : t.call(r, e);
            }));
            Ne(Me(a), "handleInputClick", (function(e) {
                var t, r;
                (t = (r = a.props).onInputClick) === null || t === void 0 ? void 0 : t.call(r, e);
            }));
            Ne(Me(a), "handleInputFocus", (function(e) {
                var t, r;
                var n = e.target.value;
                var o = a.props.name;
                (t = (r = a.props).onFocus) === null || t === void 0 ? void 0 : t.call(r, e, {
                    value: n,
                    name: o
                });
            }));
            Ne(Me(a), "handleInputBlur", (function(e) {
                var t, r;
                var n = e.target.value;
                var o = a.props.name;
                (t = (r = a.props).onBlur) === null || t === void 0 ? void 0 : t.call(r, e, {
                    value: n,
                    name: o
                });
            }));
            Ne(Me(a), "handleClear", (function(e) {
                var t, r;
                e.preventDefault();
                var n = "";
                var o = a.props.name;
                if (!a.isControlled()) {
                    a.setState({
                        value: n
                    });
                }
                a.focus();
                (t = (r = a.props).onChange) === null || t === void 0 ? void 0 : t.call(r, e, {
                    value: n,
                    name: o
                });
            }));
            Ne(Me(a), "syncHeightWithShadow", (function() {
                var e = ne({
                    height: a.state.height,
                    rowsMax: a.props.rowsMax,
                    rowsMin: a.props.rowsMin,
                    shadow: a.shadow
                });
                if (e !== false) {
                    a.setState({
                        height: e
                    });
                }
            }));
            Ne(Me(a), "renderAdornment", (function(e, t) {
                var r = a.props.disabled;
                var o = t === "start" ? function(e) {
                    a.startAdornment = e;
                } : function(e) {
                    a.endAdornment = e;
                };
                var i = {
                    ref: o,
                    disabled: r,
                    $position: t
                };
                
                return n().createElement(X, i, e);
            }));
            Ne(Me(a), "renderEndAdornment", (function() {
                var e;
                var t = a.props, r = t.appearance, o = t.endAdornment, i = t.disabled, l = t.canClear, s = t.splunkTheme;
                var c = s.isPrisma, u = s.isEnterprise;
                if (o) {
                    return a.renderAdornment(o, "end");
                }
                var d = a.context;
                var p = (e = d.labelAttrs) === null || e === void 0 ? void 0 : e.text;
                var f = p ? (0, k._)("Clear ".concat(p, " text field")) : (0, k._)("Clear text field");
                if (u && r === "search") {
                    /* Both icons are rendered, and StyledBox toggles visibility based on :focus-within */
                    return a.renderAdornment( n().createElement(n().Fragment, null,  n().createElement(Q, {
                        "aria-label": f,
                        onClick: a.handleClear
                    }),  n().createElement(W, {
                        $disabled: i
                    },  n().createElement(re, null))), "end");
                }
                var h = !!a.getDisplayValue();
                if (h && (c && r === "search" || l)) {
                    return a.renderAdornment(!i &&  n().createElement(Q, {
                        "aria-label": f,
                        onClick: a.handleClear
                    }), "end");
                }
                return undefined;
            }));
            Ne(Me(a), "renderStartAdornment", (function() {
                var e = a.props, t = e.appearance, r = e.startAdornment, o = e.splunkTheme, i = e.disabled;
                var l = o.isPrisma;
                if (r) {
                    return a.renderAdornment(r, "start");
                }
                if (l && t === "search") {
                    return a.renderAdornment( n().createElement(W, {
                        $disabled: i
                    },  n().createElement(re, null)), "start");
                }
                return undefined;
            }));
            Ne(Me(a), "getDisplayValue", (function() {
                var e = a.isControlled() ? a.props.value : a.state.value;
                return e;
            }));
            a.controlledExternally = c()(e, "value");
            a.state = {
                value: e.defaultValue || ""
            };
            if (false) {}
            if (false) {}
            if (false) {}
            r.validateRows(e);
            a.handleResize = g()(a.handleResize, 100);
            return a;
        }
        Ee(r, [ {
            key: "componentDidMount",
            value: function e() {
                l()(this.syncHeightWithShadow);
 // wait for styles to load
                                this.getAdornmentWidth();
            }
        }, {
            key: "componentDidUpdate",
            value: function e(t) {
                if (false) {}
                if (false) {}
                this.syncHeightWithShadow();
                r.validateRows(this.props);
                this.getAdornmentWidth();
            }
        }, {
            key: "isControlled",
            value: function e() {
                return this.controlledExternally;
            }
        }, {
            key: "blur",
            // eslint-disable-next-line react/no-unused-class-component-methods
            value: function e() {
                var t;
                (t = this.input) === null || t === void 0 ? void 0 : t.blur();
            }
            /**
     * Place focus on the input.
     */        }, {
            key: "focus",
            value: function e() {
                var t;
                (t = this.input) === null || t === void 0 ? void 0 : t.focus();
            }
 // eslint-disable-next-line react/no-unused-class-component-methods
                }, {
            key: "select",
            value: function e() {
                var t;
                (t = this.input) === null || t === void 0 ? void 0 : t.select();
            }
        }, {
            key: "render",
            value: function e() {
                var t = this.props, r = t.append, a = t.autoCapitalize, o = t.autoComplete, i = t.autoCorrect, l = t.autoFocus, s = t.children, c = t.className, u = t.classNamePrivate, p = t.disabled, h = t.describedBy, m = t.elementRef, y = t.error, g = t.inline, x = t.inputClassName, w = t.inputId, S = t.labelledBy, k = t.maxLength, O = t.name, E = t.placeholder, I = t.prepend, T = t.required, j = t.rowsMin, P = t.spellCheck, M = t.tabIndex, $ = t.title, _ = Se(t, [ "append", "autoCapitalize", "autoComplete", "autoCorrect", "autoFocus", "children", "className", "classNamePrivate", "disabled", "describedBy", "elementRef", "error", "inline", "inputClassName", "inputId", "labelledBy", "maxLength", "name", "placeholder", "prepend", "required", "rowsMin", "spellCheck", "tabIndex", "title" ]);
                var N = Ce(Ce({}, b()(_, (function(e, t) {
                    return t === "role" || t.indexOf("aria-") === 0;
                }))), {}, {
                    "aria-describedby": h,
                    "aria-labelledby": S,
                    "aria-invalid": y || undefined
                });
                var q = v()(_, [ "inputRef", "onBlur", "onChange", "onFocus", "onKeyDown", "onSelect", "onInputClick" ].concat(ve(d()(N))));
                var W = this.getDisplayValue();
                var B = {
                    className: (0, A.toClassName)(c, x),
                    $append: r || undefined,
                    $error: y,
                    $prepend: I || undefined
                };
                var R = Ce(Ce({}, N), {}, {
                    "data-test": "textbox",
                    autoCapitalize: a,
                    autoComplete: o,
                    autoCorrect: i,
                    autoFocus: l,
                    disabled: p,
                    id: w,
                    maxLength: k,
                    placeholder: E,
                    name: O,
                    onChange: this.handleInputChange,
                    onKeyDown: this.handleInputKeyDown,
                    onSelect: this.handleInputSelect,
                    onClick: this.handleInputClick,
                    onFocus: this.handleInputFocus,
                    onBlur: this.handleInputBlur,
                    ref: this.handleInputMount,
                    required: T,
                    spellCheck: P,
                    style: {
                        height: this.state.height
                    },
                    tabIndex: M,
                    title: $,
                    value: W,
                    $error: y
                });
                var z = this.renderStartAdornment();
                var L = this.renderEndAdornment();
                var K = !!z;
                var G = !!L;
                
                return n().createElement(D, he({
                    tabIndex: -1,
                    className: (0, A.toClassName)(c, u),
                    "data-test": "text-area",
                    "data-test-value": W,
                    elementRef: m,
                    flex: true,
                    inline: g
                }, q), z,  n().createElement(F, he({
                    $hasEndAdornment: G,
                    $hasStartAdornment: K,
                    disabled: p
                }, B), K &&  n().createElement(U, {
                    $width: this.state.startAdornmentWidth
                }),  n().createElement(V, {
                    "aria-hidden": "true",
                    onChange: f(),
                    $startAdornmentWidth: this.state.startAdornmentWidth,
                    $endAdornmentWidth: this.state.endAdornmentWidth,
                    tabIndex: -1,
                    rows: this.props.rowsMin,
                    ref: this.handleShadowMount,
                    value: E && !W ? E : W
                }),  n().createElement(H, he({
                    $rowsMin: j
                }, R, {
                    onClick: this.handleInputClick
                })),  n().createElement(C(), {
                    target: de,
                    eventType: "resize",
                    listener: this.handleResize
                }), s, G &&  n().createElement(U, {
                    $width: this.state.endAdornmentWidth
                })), L);
            }
        } ]);
        return r;
    }(r.Component);
    Ne(ze, "propTypes", Be);
    Ne(ze, "defaultProps", De);
    Ne(ze, "componentType", "TextArea");
    Ne(ze, "contextType", x.ControlGroupContext);
    var Fe = (0, S.withSplunkTheme)(ze);
    var He = Fe;
    He.propTypes = ze.propTypes;
    He.componentType = ze.componentType;
    /* harmony default export */    const Ve = He;
    // only for styled components using TextArea
        // CONCATENATED MODULE: ./src/TextArea/index.ts
    module.exports = t;
    /******/})();
