// @splunk/react-ui/useResizeObserver.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/    var e = {
        /***/ 9497: 
        /***/ e => {
            e.exports = require("react");
            /***/
            /******/        }
    };
    /************************************************************************/
    /******/ // The module cache
    /******/    var r = {};
    /******/
    /******/ // The require function
    /******/    function t(n) {
        /******/ // Check if module is in cache
        /******/ var o = r[n];
        /******/        if (o !== undefined) {
            /******/ return o.exports;
            /******/        }
        /******/ // Create a new module (and put it into the cache)
        /******/        var i = r[n] = {
            /******/ // no module.id needed
            /******/ // no module.loaded needed
            /******/ exports: {}
            /******/        };
        /******/
        /******/ // Execute the module function
        /******/        e[n](i, i.exports, t);
        /******/
        /******/ // Return the exports of the module
        /******/        return i.exports;
        /******/    }
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ t.d = (e, r) => {
            /******/ for (var n in r) {
                /******/ if (t.o(r, n) && !t.o(e, n)) {
                    /******/ Object.defineProperty(e, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ t.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ t.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // This entry need to be wrapped in an IIFE because it need to be isolated against other modules in the chunk.
        (() => {
        // ESM COMPAT FLAG
        t.r(n);
        // EXPORTS
                t.d(n, {
            default: () => /* reexport */ c
        });
        // EXTERNAL MODULE: external "react"
                var e = t(9497);
        // CONCATENATED MODULE: ./src/useResizeObserver/useResizeObserver.ts
        function r(e, r) {
            return a(e) || l(e, r) || i(e, r) || o();
        }
        function o() {
            throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
        }
        function i(e, r) {
            if (e) {
                if ("string" == typeof e) return u(e, r);
                var t = {}.toString.call(e).slice(8, -1);
                return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? u(e, r) : void 0;
            }
        }
        function u(e, r) {
            (null == r || r > e.length) && (r = e.length);
            for (var t = 0, n = Array(r); t < r; t++) {
                n[t] = e[t];
            }
            return n;
        }
        function l(e, r) {
            var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
            if (null != t) {
                var n, o, i, u, l = [], a = !0, f = !1;
                try {
                    if (i = (t = t.call(e)).next, 0 === r) {
                        if (Object(t) !== t) return;
                        a = !1;
                    } else for (;!(a = (n = i.call(t)).done) && (l.push(n.value), l.length !== r); a = !0) {
                    }
                } catch (e) {
                    f = !0, o = e;
                } finally {
                    try {
                        if (!a && null != t["return"] && (u = t["return"](), Object(u) !== u)) return;
                    } finally {
                        if (f) throw o;
                    }
                }
                return l;
            }
        }
        function a(e) {
            if (Array.isArray(e)) return e;
        }
        /**
 * Watches a React ref element for changes in size.
 *
 * @param ref - The React ref to observe size changes on
 */        var f = function t(n) {
            var o = (0, e.useState)({
                width: 0,
                height: 0
            }), i = r(o, 2), u = i[0], l = i[1];
            (0, e.useEffect)((function() {
                var e = n.current;
                if (typeof window !== "undefined" && window.ResizeObserver) {
                    var r = new ResizeObserver((function(e) {
                        if (e[0]) {
                            var r, t;
                            l({
                                width: (n === null || n === void 0 ? void 0 : (r = n.current) === null || r === void 0 ? void 0 : r.getBoundingClientRect().width) || 0,
                                height: (n === null || n === void 0 ? void 0 : (t = n.current) === null || t === void 0 ? void 0 : t.getBoundingClientRect().height) || 0
                            });
                        }
                    }));
                    if (e) {
                        r.observe(e);
                    }
                    return function() {
                        if (e) {
                            r.unobserve(e);
                        }
                    };
                }
                // Return a no-op function to satisfy consistent-return rule
                                return function() {};
            }), [ n ]);
            return u;
        };
        /* harmony default export */        const c = f;
    }) // CONCATENATED MODULE: ./src/useResizeObserver/index.ts
    ();
    module.exports = n;
    /******/})();
