// @splunk/react-ui/WaitSpinner.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var i in t) {
                /******/ if (e.o(t, i) && !e.o(r, i)) {
                    /******/ Object.defineProperty(r, i, {
                        enumerable: true,
                        get: t[i]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ _
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var i =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const n = require("prop-types");
    var a =  e.n(n);
    // CONCATENATED MODULE: external "@splunk/react-ui/AnimationToggle"
    const s = require("@splunk/react-ui/AnimationToggle");
    // CONCATENATED MODULE: external "@splunk/themes"
    const o = require("@splunk/themes");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const l = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "styled-components"
    const c = require("styled-components");
    var m =  e.n(c);
    // CONCATENATED MODULE: ./src/WaitSpinner/WaitSpinnerStyles.ts
    var f = m().div.withConfig({
        displayName: "WaitSpinnerStyles__Styled",
        componentId: "sc-1nu971z-0"
    })([ "", ";align-items:center;" ], o.mixins.reset("inline-flex"));
    var p = (0, c.keyframes)([ "100%{transform:rotate(360deg);}" ]);
    var d = (0, c.keyframes)([ "0%{transform:scale(0);opacity:0;}100%{transform:scale(1);opacity:1;}" ]);
    var u = "2.07s";
    var y = m().svg.withConfig({
        displayName: "WaitSpinnerStyles__StyledSvg",
        componentId: "sc-1nu971z-1"
    })([ "", " ", "" ], (0, o.pickVariant)("$size", {
        small: {
            enterprise: (0, c.css)([ "width:14px;height:14px;" ]),
            prisma: (0, c.css)([ "width:16px;height:16px;" ])
        },
        medium: {
            enterprise: (0, c.css)([ "width:19px;height:19px;" ]),
            prisma: (0, c.css)([ "width:24px;height:24px;" ])
        },
        large: {
            enterprise: (0, c.css)([ "width:28px;height:28px;" ]),
            prisma: (0, c.css)([ "width:40px;height:40px;" ])
        }
    }), (function(e) {
        var r = e.$animated;
        return r && (0, o.pick)({
            enterprise: (0, c.css)([ "transform-origin:center;animation:", " 1.2s steps(64) infinite;" ], p),
            prisma: (0, c.css)([ "animation:", " ", " infinite linear,", " 500ms cubic-bezier(0.01,0,0,1);" ], p, u, d)
        });
    }));
    var h = (0, c.keyframes)([ "0%{stroke-dasharray:110 10;stroke-dashoffset:-5.5;animation-timing-function:cubic-bezier(0.8,0,0.83,1);}50%{stroke-dasharray:26 94;stroke-dashoffset:-152;}100%{stroke-dasharray:110 10;stroke-dashoffset:-246.5;animation-timing-function:cubic-bezier(0.33,0,0.67,1);}" ]);
    var v = (0, c.keyframes)([ "0%{stroke-dasharray:7 113;stroke-dashoffset:3;animation-timing-function:cubic-bezier(0.8,0,0.83,1);}50%{stroke-dasharray:90 30;stroke-dashoffset:-60;}100%{stroke-dasharray:7 113;stroke-dashoffset:-238;animation-timing-function:cubic-bezier(0.33,0,0.67,1);}" ]);
    var g = m().circle.withConfig({
        displayName: "WaitSpinnerStyles__StyledPrismaBasePath",
        componentId: "sc-1nu971z-2"
    })([ "stroke:", ";stroke-width:2;fill:transparent;", " stroke-dasharray:110 10;stroke-dashoffset:-5.5;" ], o.variables.contentColorDisabled, (function(e) {
        var r = e.$animated;
        return r && (0, c.css)([ "animation:", " ", " infinite;animation-fill-mode:backwards;" ], h, u);
    }));
    var k = m().circle.withConfig({
        displayName: "WaitSpinnerStyles__StyledPrismaFillPath",
        componentId: "sc-1nu971z-3"
    })([ "fill:transparent;stroke-width:2;stroke:", ";stroke-dasharray:7 113;stroke-dashoffset:3;", "" ], (0, 
    o.pickVariant)("$size", {
        small: o.variables.contentColorDefault,
        medium: o.variables.contentColorDefault,
        large: o.variables.interactiveColorPrimary
    }), (function(e) {
        var r = e.$animated;
        return r && (0, c.css)([ "animation:", " ", " infinite;" ], v, u);
    }));
    var b = m().circle.withConfig({
        displayName: "WaitSpinnerStyles__StyledEnterpriseCircle",
        componentId: "sc-1nu971z-4"
    })([ "fill:transparent;stroke:", ";stroke-width:2px;", "" ], (0, o.pick)({
        enterprise: {
            dark: o.variables.white,
            light: o.variables.gray60
        },
        prisma: o.variables.contentColorMuted
    }), (0, o.pick)({
        enterprise: (0, c.css)([ "stroke-dasharray:34 19;" ])
    }));
    // CONCATENATED MODULE: ./src/WaitSpinner/WaitSpinner.tsx
    function w() {
        return w = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var i in t) {
                    ({}).hasOwnProperty.call(t, i) && (e[i] = t[i]);
                }
            }
            return e;
        }, w.apply(null, arguments);
    }
    function x(e, r) {
        if (null == e) return {};
        var t, i, n = S(e, r);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (i = 0; i < a.length; i++) {
                t = a[i], r.includes(t) || {}.propertyIsEnumerable.call(e, t) && (n[t] = e[t]);
            }
        }
        return n;
    }
    function S(e, r) {
        if (null == e) return {};
        var t = {};
        for (var i in e) {
            if ({}.hasOwnProperty.call(e, i)) {
                if (r.includes(i)) continue;
                t[i] = e[i];
            }
        }
        return t;
    }
    var z = {
        elementRef: a().oneOfType([ a().func, a().object ]),
        screenReaderText: a().oneOfType([ a().string, a().oneOf([ "null" ]) ]),
        size: a().oneOf([ "small", "medium", "large" ])
    };
    function O(e) {
        var r = e.elementRef, t = e.screenReaderText, n = t === void 0 ? (0, l._)("Waiting") : t, a = e.size, c = a === void 0 ? "small" : a, m = x(e, [ "elementRef", "screenReaderText", "size" ]);
        // @docs-props-type WaitSpinnerPropsBase
                if (false) {}
        var p = (0, s.useAnimationToggle)() === "on";
        var d = (0, o.useSplunkTheme)(), u = d.isPrisma;
        var h = u ? "0 0 40 40" : "0 0 19 19";
        
        return i().createElement(f, w({
            "data-test": "wait-spinner"
        }, m),  i().createElement(y, {
            viewBox: h,
            version: "1.1",
            xmlns: "http://www.w3.org/2000/svg",
            ref: r,
            $animated: p,
            $size: c
        }, n &&  i().createElement("title", null, n),  i().createElement("g", null, u ?  i().createElement(i().Fragment, null,  i().createElement(g, {
            r: "19",
            cx: "20",
            cy: "20",
            $animated: p,
            $size: c
        }),  i().createElement(k, {
            r: "19",
            cx: "20",
            cy: "20",
            $animated: p,
            $size: c
        })) :  i().createElement(b, {
            cx: "9.5",
            cy: "9.5",
            r: "8.5"
        }))));
    }
    O.propTypes = z;
    /* harmony default export */    const _ = O;
    // CONCATENATED MODULE: ./src/WaitSpinner/index.ts
    module.exports = r;
    /******/})();
